---
title: "Super_2023"
author: "GrGladkov"
date: "2023-11-30"
output: html_document
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/home/gladkov/storage/bags_2023/")

```


```{r}

library(dada2)
library(ShortRead)
library(Biostrings)
library(seqinr)
library(tidyverse)
library(fuzzyjoin)
require(dada2)
require(Biostrings)
require(DECIPHER)
require(phyloseq)
require(seqinr)
require(data.table)
require(metagMisc)
require(tibble)
library(phyloseq)
library(tidyverse)
library(vegan)

```

#16S run

```{r}

path <- "16S/raw"
fnFs <- sort(list.files(path, pattern = "R1_001.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern = "R2_001.fastq.gz", full.names = TRUE))
list.files(path)
list.files()
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)

sample.names %>% 
  naturalsort::naturalsort() %>% 
  as.data.frame() %>% 
  write_tsv('sample_names_16S.tsv')

```


```{r}

fnFs <- sort(list.files(path, pattern = "R1_001.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern = "R2_001.fastq.gz", full.names = TRUE))

```



```{r}

fnFs
plotQualityProfile(fnFs, aggregate = TRUE)

```




```{r}

plotQualityProfile(fnRs, aggregate = TRUE)

```


```{r}

?dada2::dada()

```



```{r}

path <- "its2/raw"
fnFs <- sort(list.files(path, pattern = "R1_001.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern = "R2_001.fastq.gz", full.names = TRUE))

sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
sample.names

sample.names %>% 
  naturalsort::naturalsort() %>% 
  as.data.frame() %>% 
  write_tsv('sample_names_its2.tsv')

srea


```
### functions

```{r}

plot_rich_reads_samlenames_lm <- function(physeq, group = "Site", label = "Repeat"){
  rish <- estimate_richness(physeq, measures = "Observed")
  reads.sum <- as.data.frame(sample_sums(physeq))
  reads.summary <- cbind(rish, reads.sum)
  colnames(reads.summary) <- c("otus","reads")
  reads.summary["Repeat"] <-unlist(purrr::map(stringr::str_split(rownames(physeq@sam_data), "\\.", 2), function(x) x[[2]]))
  reads.summary["Site"] <- physeq@sam_data[[group]]
  library(ggrepel)
  require(ggforce)
  p1 <- ggplot(data=reads.summary) + 
    geom_point(aes(y=otus, x=log2(reads), color=Site),size=3) + 
    geom_text_repel(aes(y=otus, x=log2(reads), label=paste0(Repeat))) + 
    theme_bw() +
    geom_smooth(aes(y=otus, x=log2(reads), fill=Site, color=Site),method=lm, se=FALSE, ymin = 1) + 
    scale_x_continuous(sec.axis = sec_axis(sec.axis ~ 2**.)) 

  return(p1)
}

plot_alpha_w_toc_mod <- function(ps, group, metric) {
  
  require(phyloseq)
  require(ggplot2)
  
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  
  er <- estimate_richness(ps_a)
  df_er <- cbind(ps_a@sam_data, er)
  df_er <- df_er %>% select(c(group, metric))
  stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
    rstatix::tukey_hsd() %>%
    filter(p.adj.signif != "ns")
  y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))

  plot_richness(ps_a, x=group, measures=metric, color=group) + 
    geom_boxplot() +
    geom_point(size=1.2, alpha=0.3) +
    ggpubr::stat_pvalue_manual(
      stat.test,
      label = "p.adj.signif",
      y.position = y,
      tip.length = 0.005,
      bracket.nudge.y = 1.5,
      vjust = 0.6,
      size = 3) +
    theme_light() + 
        scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8) +
    theme(axis.text.x = element_text(size=12, angle = 45, hjust=1),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),
          legend.position = "None") +
    labs(y=paste(metric, "index")) 
}


beta_custom_norm_NMDS_elli_w <- function(ps, seed = 7888, normtype="vst", Color="What", Group="Repeat"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  library(ggforce)
  
  ps@otu_table[ps@otu_table < 0] <- 0
  ordination.b <- ordinate(ps, "NMDS", "bray")
  mds <- as.data.frame(ordination.b$points)
  p  <-  plot_ordination(ps,
                         ordination.b,
                         type="sample",
                         color = Color,
                         title="NMDS - Bray-Curtis",
                         # title=NULL,
                         axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    annotate("text",
    x=min(mds$MDS1) + abs(min(mds$MDS1))/4,
    y=max(mds$MDS2),
    label=paste0("Stress -- ", round(ordination.b$stress, 3))) +
    geom_mark_ellipse(aes_string(group = Group, label = Group),
                      label.fontsize = 8,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                      con.colour='gray') +
    theme(legend.position = "none") +
    scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8)
  
  return(p)
}

plot_alpha_w_toc <- function(ps, group, metric) {
  
  require(phyloseq)
  require(ggplot2)
  
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  
  er <- estimate_richness(ps_a)
  df_er <- cbind(ps_a@sam_data, er)
  df_er <- df_er %>% select(c(group, metric))
  stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
    rstatix::tukey_hsd()
  y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))

  plot_richness(ps_a, x=group, measures=metric) + 
    geom_boxplot() +
    geom_point(size=1.2, alpha=0.3) +
    ggpubr::stat_pvalue_manual(
      stat.test, 
      label = "p.adj.signif", 
      y.position = y) +
    theme_light() + 
    scale_color_brewer(palette="Dark2") +
    theme(axis.text.x = element_text(angle = 90),
          axis.title.x=element_blank()) +
    labs(y=paste(metric, "index")) 
}

phyloseq_to_ampvis2 <- function(physeq) {
  #check object for class
  if(!any(class(physeq) %in% "phyloseq"))
    stop("physeq object must be of class \"phyloseq\"", call. = FALSE)
  
  #ampvis2 requires taxonomy and abundance table, phyloseq checks for the latter
  if(is.null(physeq@tax_table))
    stop("No taxonomy found in the phyloseq object and is required for ampvis2", call. = FALSE)
  
  #OTUs must be in rows, not columns
  if(phyloseq::taxa_are_rows(physeq))
    abund <- as.data.frame(phyloseq::otu_table(physeq)@.Data)
  else
    abund <- as.data.frame(t(phyloseq::otu_table(physeq)@.Data))
  
  #tax_table is assumed to have OTUs in rows too
  tax <- phyloseq::tax_table(physeq)@.Data
  
  #merge by rownames (OTUs)
  otutable <- merge(
    abund,
    tax,
    by = 0,
    all.x = TRUE,
    all.y = FALSE,
    sort = FALSE
  )
  colnames(otutable)[1] <- "OTU"
  
  #extract sample_data (metadata)
  if(!is.null(physeq@sam_data)) {
    metadata <- data.frame(
      phyloseq::sample_data(physeq),
      row.names = phyloseq::sample_names(physeq), 
      stringsAsFactors = FALSE, 
      check.names = FALSE
    )
    
    #check if any columns match exactly with rownames
    #if none matched assume row names are sample identifiers
    samplesCol <- unlist(lapply(metadata, function(x) {
      identical(x, rownames(metadata))}))
    
    if(any(samplesCol)) {
      #error if a column matched and it's not the first
      if(!samplesCol[[1]])
        stop("Sample ID's must be in the first column in the sample metadata, please reorder", call. = FALSE)
    } else {
      #assume rownames are sample identifiers, merge at the end with name "SampleID"
      if(any(colnames(metadata) %in% "SampleID"))
        stop("A column in the sample metadata is already named \"SampleID\" but does not seem to contain sample ID's", call. = FALSE)
      metadata$SampleID <- rownames(metadata)
      
      #reorder columns so SampleID is the first
      metadata <- metadata[, c(which(colnames(metadata) %in% "SampleID"), 1:(ncol(metadata)-1L)), drop = FALSE]
    }
  } else
    metadata <- NULL
  
  #extract phylogenetic tree, assumed to be of class "phylo"
  if(!is.null(physeq@phy_tree)) {
    tree <- phyloseq::phy_tree(physeq)
  } else
    tree <- NULL
  
  #extract OTU DNA sequences, assumed to be of class "XStringSet"
  if(!is.null(physeq@refseq)) {
    #convert XStringSet to DNAbin using a temporary file (easiest)
    fastaTempFile <- tempfile(pattern = "ampvis2_", fileext = ".fa")
    Biostrings::writeXStringSet(physeq@refseq, filepath = fastaTempFile)
  } else
    fastaTempFile <- NULL
  
  #load as normally with amp_load
  ampvis2::amp_load(
    otutable = otutable,
    metadata = metadata,
    tree = tree,
    fasta = fastaTempFile
  )
}

detachAllPackages <- function() {

  basic.packages <- c("package:stats","package:graphics","package:grDevices","package:utils","package:datasets","package:methods","package:base")

  package.list <- search()[ifelse(unlist(gregexpr("package:",search()))==1,TRUE,FALSE)]

  package.list <- setdiff(package.list,basic.packages)

  if (length(package.list)>0)  for (package in package.list) detach(package, character.only=TRUE, force = TRUE)

}

ps_vst <- function(ps, factor){
  diagdds = phyloseq::phyloseq_to_deseq2(ps, as.formula(paste( "~", factor)))              
  diagdds = DESeq2::estimateSizeFactors(diagdds, type="poscounts")
  diagdds = DESeq2::estimateDispersions(diagdds, fitType = "local") 
  pst <- DESeq2::varianceStabilizingTransformation(diagdds)
  pst.dimmed <- t(as.matrix(SummarizedExperiment::assay(pst))) 
  # pst.dimmed[pst.dimmed < 0.0] <- 0.0
  ps.varstab <- ps
  otu_table(ps.varstab) <- otu_table(pst.dimmed, taxa_are_rows = FALSE) 
  return(ps.varstab)
} 

filter_chrmitnas <- function(physeq){
  ps_object <- physeq 
  ps_object <- subset_taxa(ps_object, Phylum != "NA")
  
  ps_object@tax_table[is.na(ps_object@tax_table)] <- ps@tax_table@.Data
  ps_object <- subset_taxa(ps_object,
                           !(Family  == "Mitochondria" |
                               Class   == "Chloroplast" |
                               Order   == "Chloroplast"))
  ps_object@tax_table <- dplyr::na_if(ps_object@tax_table, TRUE)
  ps.f <- ps_object
  
  out_old <- capture.output(physeq)[4] %>% 
    str_extract(regex("([1-9]).*(?= by)"))
  out_new <- capture.output(ps.f)[4] %>% 
    str_extract(regex("([1-9]).*(?= by)"))
  message(paste0("old = ", out_old))
  message(paste0("filtered = ", out_new))
  
  return(ps.f)
  
}

beta_custom_norm_PCA_elli_w <- function(ps, seed = 7888, normtype="vst", Color="What", Group="Repeat"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  library(ggforce)
  
  ps@otu_table[ps@otu_table < 0] <- 0
  ordination.b <- ordinate(ps, "PCoA", "euclidean")
  mds <- as.data.frame(ordination.b$points)
  p  <-  plot_ordination(ps,
                         ordination.b,
                         type="sample",
                         color = Color,
                         title="PCA - clr-transformed",
                         # title=NULL,
                         axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes_string(group = Group, label = Group),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                      con.colour='gray') +
    theme(legend.position = "none") +
    scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8)
  
  return(p)
}

filter_chrmitnas <- function(physeq){
  ps_object <- physeq 
  ps_object <- subset_taxa(ps_object, Phylum != "NA")
  
  ps_object@tax_table <- ps_object@tax_table %>% 
    as.data.frame() %>% 
    mutate_all(~ replace_na(., "unknown")) %>% 
    as.matrix() %>% 
    tax_table()
  
  ps_object <- subset_taxa(ps_object,
                           !(Family  == "Mitochondria" | 
                               Class   == "Chloroplast" | 
                               Order   == "Chloroplast"))

  ps_object@tax_table <- ps_object@tax_table %>% 
    as.data.frame() %>% 
    mutate_all(~ na_if(., "unknown")) %>% 
    as.matrix() %>% 
    tax_table()
  
  ps.f <- ps_object
  
  out_old <- capture.output(physeq)[4] %>%
    str_extract(regex("([1-9]).*(?= by)"))
  out_new <- capture.output(ps.f)[4] %>%
    str_extract(regex("([1-9]).*(?= by)"))
  message(paste0("old = ", out_old))
  message(paste0("filtered = ", out_new))

  return(ps.f)
  
}

get_list_from_ps <- function(physeq, amaz_fac) {
  my_keys <- levels(sample_data(physeq)[[amaz_fac]])
  mylist <- vector(mode="list", length=length(my_keys))
  names(mylist) <- my_keys
  for (i in levels(sample_data(physeq)[[amaz_fac]])) { 
    x <- prune_samples(sample_data(physeq)[[amaz_fac]] %in% i, physeq)
    x <- prune_taxa(taxa_sums(x) > 0, x)
    mylist[[i]] <- taxa_names(x)
  }
  return(mylist)
}


beta_custom_norm_PCoA_elli_w <- function(ps, seed = 7888, normtype="vst", Color="What", Group="Repeat"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  library(ggforce)
  
  ps@otu_table[ps@otu_table < 0] <- 0
  ordination.b <- ordinate(ps, "PCoA", "bray")
  mds <- as.data.frame(ordination.b$points)
  p  <-  plot_ordination(ps,
                         ordination.b,
                         type="sample",
                         color = Color,
                         title="PCoA - bray - rarefied",
                         # title=NULL,
                         axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes_string(group = Group, label = Group),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                      con.colour='gray') +
    theme(legend.position = "none") +
    scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8)
  
  return(p)
}

plot_alpha_w_toc <- function(ps, group, metric) {
  
  require(phyloseq)
  require(ggplot2)
  
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  
  er <- estimate_richness(ps_a)
  df_er <- cbind(ps_a@sam_data, er)
  df_er <- df_er %>% select(c(group, metric))
  stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
    rstatix::tukey_hsd()
  y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))

  plot_richness(ps_a, x=group, measures=metric) + 
    geom_boxplot() +
    geom_point(size=1.2, alpha=0.3) +
    ggpubr::stat_pvalue_manual(
      stat.test, 
      label = "p.adj.signif", 
      y.position = y) +
    theme_light() + 
    scale_color_brewer(palette="Dark2") +
    theme(axis.text.x = element_text(angle = 90),
          axis.title.x=element_blank()) +
    labs(y=paste(metric, "index")) 
}


lsf.str()

```
## import 16S

### metadata for 16S and ITS 

```{r}

setwd("~/storage/bags_2023/")

library(readxl)
library(tidyverse)
library(phyloseq)

list.files()
read_excel("bags_illumina2023_map.xlsx")
path <- "bags_illumina2023_map.xlsx"

mad <- path %>%
        excel_sheets() %>%
        set_names() %>%
       purrr::map_df(read_excel,
           path = path) 

otu_table <- read_tsv("16S/otu_table.txt") %>% 
  column_to_rownames("...1") %>% 
  t() %>% 
  as.matrix()


taxa <- read_tsv("16S/taxa.txt") %>% 
  column_to_rownames("...1") %>% 
  as.matrix()

otu_table %>% 
  colnames()


map <- mad %>% 
  mutate(ID_16S = str_replace_all(ID_16S, '-', '.')) %>% 
  mutate(Group = paste0(Type, '_', Substrate, '_', Period)) %>% 
  group_by(Group) %>%  
  mutate(across(where(is.numeric), ~replace_na(., median(., na.rm=TRUE)))) %>% 
  column_to_rownames('ID_16S')
  
map

ps <- phyloseq(otu_table(otu_table, taxa_are_rows=FALSE), 
               sample_data(map),
               tax_table(taxa))

ps

```


```{r}

ps.f <- filter_chrmitnas(ps)


ps.f@tax_table <- ps.f@tax_table@.Data %>% 
  data.frame() %>% 
  mutate(Phylum = str_replace_all(Phylum, c("Firmicutes" = "Bacillota",
                                            "Chloroflexi" = "Chloroflexota",
                                            "Proteobacteria" = "Pseudomonadota",
                                            "Desulfobacterota" = "Thermodesulfobacteriota",
                                            "Crenarchaeota" = "Thermoproteota",
                                            "Cyanobacteria" = "Cyanobacteriota"))) %>% 
  as.matrix() %>% 
  tax_table()


ps.f@sam_data

```



```{r, fig.width=14, fig.height=8}

plot_rich_reads_samlenames_lm(ps.f, group = "Group", label = "Group")  
ps.f <- prune_samples(sample_names(ps.f) != 'Andronov.SEQ282.16s.148', ps.f)
ps.f <- prune_samples(sample_names(ps.f) != 'Andronov.SEQ282.16s.153', ps.f)
ps.f <- prune_samples(sample_names(ps.f) != 'Andronov.SEQ282.16s.145', ps.f)
plot_rich_reads_samlenames_lm(ps.f, group = "Group", label = "Group")  

```


```{r, fig.width=9, fig.height=9}

beta_custom_norm_NMDS_elli_w(ps.f, Color = 'Substrate', Group = 'Group')

```

```{r, fig.width=9, fig.height=9}

library(dplyr)


ps.f.clr <- ps.f
ps.otu.clr <- as.data.frame(t(ps.f@otu_table)) %>% 
  compositions::clr() %>% 
  t()

ps.f.clr@otu_table <-  otu_table(ps.otu.clr, taxa_are_rows = FALSE)
ps.f.r <- rarefy_even_depth(ps.f, rngseed = 1488)

```


```{r, fig.width=9, fig.height=9}

beta_custom_norm_PCA_elli_w(ps.f.clr, Color = 'Group', Group = 'Group')

```

```{r, fig.width=9, fig.height=9}

beta_custom_norm_NMDS_elli_w(ps.f.clr, Color = 'Group', Group = 'Group')

```


```{r, fig.width=9, fig.height=9}

beta_custom_norm_PCoA_elli_w(ps.f.r, Color = 'Group', Group = 'Group')

```




```{r}

ps.bag <- prune_samples((sample_data(ps.f)$Substrate %in% c('linden_leaves', 'oat_straw')) & (sample_data(ps.f)$Type == 'BAGS'), ps.f)
ps.bag  <- prune_taxa(taxa_sums(ps.bag) > 0, ps.bag)

```

```{r, fig.width=7, fig.height=7}

ps.bag.r <- rarefy_even_depth(ps.bag, rngseed = 1488)
beta_custom_norm_NMDS_elli_w(ps.bag.r, Color = 'Substrate', Group = 'Group')

```



```{r}

ps.cor <- prune_samples(!is.na(sample_data(ps.f)$ID_ITS), ps.f)
ps.cor  <- prune_taxa(taxa_sums(ps.cor) > 0, ps.cor)

ps.cor@sam_data <- ps.cor@sam_data %>% 
  data.frame() %>% 
  mutate(Group2 = ifelse(Period %in% c('gen1', 'gen2'), 'инокулюм', 
                         ifelse(Type %in% c('peat_bag1', 'peat_bag2', 'peat_bag3'), 'торф', 
                                ifelse(Type %in% c('linden_leaves', 'oat_straw'), 'субстрат', Type)))) %>% 
  sample_data()

ps.cor <- prune_samples(!is.na(sample_data(ps.f)$ID_ITS), ps.f)
ps.cor  <- prune_taxa(taxa_sums(ps.cor) > 0, ps.cor)

ps.cor <- prune_samples(!is.na(sample_data(ps.f)$ID_ITS), ps.f)
ps.cor  <- prune_taxa(taxa_sums(ps.cor) > 0, ps.cor)


taxa <- ps.cor@tax_table
taxa[taxa == "Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium"] <- "Allorhizobium"
ps.cor@tax_table <- taxa

ps.cor.bags <- prune_samples(sample_data(ps.cor)$Type == 'BAGS', ps.cor)
ps.cor.bags  <- prune_taxa(taxa_sums(ps.cor.bags) > 0, ps.cor.bags)

ps.cor@sam_data 

```

```{r, fig.width=8, fig.height=5}

ps.cor.r <- rarefy_even_depth(ps.cor, rngseed = 1488)
beta_custom_norm_PCoA_elli_w(ps.cor.r, Color = 'Group', Group = 'Group')

ps.cor.r@sam_data

```

## its

```{r}


otu_table <- read_tsv("its2/otu_table.txt") %>% 
  column_to_rownames("...1") %>% 
  t() %>% 
  as.matrix()


taxa <- read_tsv("its2/taxa.txt") %>% 
  column_to_rownames("...1") %>% 
  select(-c('...9')) %>% 
  as.matrix()


map <- mad %>% 
  mutate(ID_ITS = str_replace_all(ID_ITS, '-', '.')) %>% 
  mutate(Group = paste0(Type, '_', Substrate, '_', Period)) %>% 
  group_by(Group) %>%  
  mutate(across(where(is.numeric), ~replace_na(., median(., na.rm=TRUE)))) %>% 
  filter(!is.na(ID_ITS)) %>% 
  column_to_rownames('ID_ITS')


psits <- phyloseq(otu_table(otu_table, taxa_are_rows=FALSE), 
               sample_data(map),
               tax_table(taxa))

psits@sam_data <- psits@sam_data %>% 
  data.frame() %>% 
  mutate(Group2 = ifelse(Period %in% c('gen1', 'gen2'), 'инокулюм', 
                         ifelse(Type %in% c('peat_bag1', 'peat_bag2', 'peat_bag3'), 'торф', 
                                ifelse(Type %in% c('linden_leaves', 'oat_straw'), 'субстрат', Type)))) %>% 
  sample_data()


psits <- filter_chrmitnas(psits)




psits.bags <- prune_samples(sample_data(psits)$Type == 'BAGS', psits)
psits.bags  <- prune_taxa(taxa_sums(psits.bags) > 0, psits.bags)

```



```{r, fig.width=14, fig.height=8}

plot_rich_reads_samlenames_lm(psits, group = "Group", label = "Group")  

```

```{r, fig.height=10, fig.width=8}

psits@sam_data <- psits@sam_data %>% 
  data.frame() %>% 
  mutate(Group = as.factor(Group)) %>% 
  mutate(Group = recode(Group,
                        BAGS_birch_leaves_5 = "БАГС_ЛБ",
                        BAGS_linden_leaves_5 = "БАГС_ЛЛ",
                        BAGS_oat_straw_5 = "БАГС_СО",
                        BAGS_oat_straw_gen1 = "БАГС_2018",
                        BAGS_oat_straw_gen2 = "БАГС_2021",
                        BAGS_wheat_straw_5 = "БАГС_СП",
                        linden_leaves_linden_leaves_NA = "ЛЛ",
                        oat_straw_oat_straw_NA = "СО",
                        peat_bag1_peat_NA = "торф",
                        peat_bag2_peat_NA = "торф",
                        peat_bag3_peat_NA = "торф")
  ) %>% sample_data()


ps.cor@sam_data <- ps.cor@sam_data %>% 
  data.frame() %>% 
  mutate(Group = as.factor(Group)) %>% 
  mutate(Group = recode(Group,
                        BAGS_birch_leaves_5 = "БАГС_ЛБ",
                        BAGS_linden_leaves_5 = "БАГС_ЛЛ",
                        BAGS_oat_straw_5 = "БАГС_СО",
                        BAGS_oat_straw_gen1 = "БАГС_2018",
                        BAGS_oat_straw_gen2 = "БАГС_2021",
                        BAGS_wheat_straw_5 = "БАГС_СП",
                        linden_leaves_linden_leaves_NA = "ЛЛ",
                        oat_straw_oat_straw_NA = "СО",
                        peat_bag1_peat_NA = "торф",
                        peat_bag2_peat_NA = "торф",
                        peat_bag3_peat_NA = "торф")
  ) %>% sample_data()

psits.r <- rarefy_even_depth(psits, rngseed = 1488)

psits.r@sam_data <- psits.r@sam_data %>% 
  data.frame() %>% 
  mutate(Group = as.factor(Group)) %>% 
  mutate(Group = recode(Group,
                        BAGS_birch_leaves_5 = "БАГС_ЛБ",
                        BAGS_linden_leaves_5 = "БАГС_ЛЛ",
                        BAGS_oat_straw_5 = "БАГС_СО",
                        BAGS_oat_straw_gen1 = "БАГС_2018",
                        BAGS_oat_straw_gen2 = "БАГС_2021",
                        BAGS_wheat_straw_5 = "БАГС_СП",
                        linden_leaves_linden_leaves_NA = "ЛЛ",
                        oat_straw_oat_straw_NA = "СО",
                        peat_bag1_peat_NA = "торф",
                        peat_bag2_peat_NA = "торф",
                        peat_bag3_peat_NA = "торф")
  ) %>% sample_data()
         
         

psits.r@sam_data

ps.cor.r@sam_data <- ps.cor.r@sam_data %>% 
  data.frame() %>% 
  mutate(Group = as.factor(Group)) %>% 
  mutate(Group = recode(Group,
                        BAGS_birch_leaves_5 = "БАГС_ЛБ",
                        BAGS_linden_leaves_5 = "БАГС_ЛЛ",
                        BAGS_oat_straw_5 = "БАГС_СО",
                        BAGS_oat_straw_gen1 = "БАГС_2018",
                        BAGS_oat_straw_gen2 = "БАГС_2021",
                        BAGS_wheat_straw_5 = "БАГС_СП",
                        linden_leaves_linden_leaves_NA = "ЛЛ",
                        oat_straw_oat_straw_NA = "СО",
                        peat_bag1_peat_NA = "торф",
                        peat_bag2_peat_NA = "торф",
                        peat_bag3_peat_NA = "торф")
  ) %>% sample_data()


b_core_its
b_core_its <- beta_custom_norm_PCoA_elli_w(psits.r, Color = 'Group2', Group = 'Group')
b_core_16S <- beta_custom_norm_PCoA_elli_w(ps.cor.r, Color = 'Group2', Group = 'Group')

ggpubr::ggarrange(b_core_16S, b_core_its, nrow = 2, labels = c('16S', 'ITS2'), hjust = 0)


```









```{r}

ps.cor@sam_data

```



```{r, fig.width=12, fig.height=20}


amp.cor <- phyloseq_to_ampvis2(ps.cor)

amp_heatmap(amp.cor,
            tax_show = 140,
            tax_aggregate = "Genus",
            tax_add = "Phylum",
            group_by = "Group",
            plot_values_size = 2.8, 
            facet_by = "Group2",
            showRemainingTaxa = TRUE) + theme(axis.text.y=element_text(size=rel(1.2)))


ps.gra@tax_table
amp.gra$metadata %>% 
  as.data.frame() %>% 
  filter(soil_type == 'field') 

```
### alpha core


```{r, fig.width=10, fig.height=8}

ps.cor.r <- rarefy_even_depth(ps.cor, rngseed = 1488)
psits.r <- rarefy_even_depth(psits, rngseed = 1488)

psits.r@sam_data <- psits.r@sam_data %>% 
  data.frame() %>% 
  mutate(Group2 = ifelse(Period %in% c('gen1', 'gen2'), 'инокулюм', 
                         ifelse(Type %in% c('peat_bag1', 'peat_bag2', 'peat_bag3'), 'торф', 
                                ifelse(Type %in% c('linden_leaves', 'oat_straw'), 'субстрат', Type)))) %>% 
  mutate(Group2 = ifelse(Group %in% c('БАГС_СО', 'БАГС_ЛЛ'), 'БАГС', Group2)) %>% 
  sample_data()

ps.cor.r@sam_data <- ps.cor.r@sam_data %>% 
  data.frame() %>% 
  mutate(Group2 = ifelse(Period %in% c('gen1', 'gen2'), 'инокулюм', 
                         ifelse(Type %in% c('peat_bag1', 'peat_bag2', 'peat_bag3'), 'торф', 
                                ifelse(Type %in% c('linden_leaves', 'oat_straw'), 'субстрат', Type)))) %>% 
  mutate(Group2 = ifelse(Group %in% c('БАГС_СО', 'БАГС_ЛЛ'), 'БАГС', Group2)) %>% 
  sample_data()

psits.r <- prune_samples(sample_data(psits.r)$Group2 != 'BAGS', psits.r)
psits.r  <- prune_taxa(taxa_sums(psits.r) > 0, psits.r)

ps.cor.r <- prune_samples(sample_data(ps.cor.r)$Group2 != 'BAGS', ps.cor.r)
ps.cor.r  <- prune_taxa(taxa_sums(ps.cor.r) > 0, ps.cor.r)

pi_a1 <-  plot_alpha_w_toc_mod(psits.r, group="Group2", metric="Observed") 
pi_a2 <- plot_alpha_w_toc_mod(psits.r, group="Group2", metric="Shannon") 
pi_a3 <- plot_alpha_w_toc_mod(psits.r, group="Group2", metric="InvSimpson") 

p_a1 <-  plot_alpha_w_toc_mod(ps.cor.r, group="Group2", metric="Observed") 
p_a2 <- plot_alpha_w_toc_mod(ps.cor.r, group="Group2", metric="Shannon") 
p_a3 <- plot_alpha_w_toc_mod(ps.cor.r, group="Group2", metric="InvSimpson") 

a <- ggpubr::ggarrange(p_a1, p_a2, p_a3, pi_a1, pi_a2, pi_a3,nrow = 2, ncol = 3, labels = c('16S', '','','ITS2','',''), hjust = -1)
a

psits.r@sam_data
```

```{r}

ps.cor@sam_data

```



```{r, fig.width=12, fig.height=15}


amp.its <- phyloseq_to_ampvis2(psits)
# amp.its$metadata
amp_heatmap(amp.its,
            tax_show = 60,
            tax_aggregate = "OTU",
            tax_add = "Genus",
            group_by = "Group",
            plot_values_size = 3, 
            facet_by = "Group2",
            showRemainingTaxa = TRUE) + theme(axis.text.y=element_text(size=rel(2)))

```


```{r, fig.width=9, fig.height=9}

library(ggVennDiagram)

plot_vienn <- function(ps, group){
  physeq <- prune_taxa(taxa_sums(ps) > 0, ps)
  groups <- levels(sample_data(physeq)[[group]] %>% as.factor())
  data <- merge_samples(physeq, group) %>% 
    psmelt() %>% 
    group_by(Sample, OTU) %>% 
    summarise(ASVs_abund = list(paste(OTU, 1:sum(Abundance))), Abund = sum(Abundance), .groups='keep') %>% 
    filter(Abund > 0)
  
  asvs <- data %>% select(Sample, OTU) %>% group_by(Sample) %>% summarise(ASVs = list(OTU)) %>% as.list()
  d1 <- asvs[[2]]
  names(d1) <- asvs[[1]]
  d1
  
  weighted.asvs <- data %>% select(Sample, ASVs_abund) %>% group_by(Sample) %>% summarise(ASVs = list(unlist(ASVs_abund)))
  d2 <- weighted.asvs[[2]]
  names(d2) <- weighted.asvs[[1]]
  d2
  
  
  list(ggVennDiagram(d1) + 
         # ggtitle("ASVs") + 
         scale_fill_distiller(palette = "OrRd", trans = "reverse"),
       ggVennDiagram(d2) + 
         # ggtitle("Reads") + 
         scale_fill_distiller(palette = "OrRd", trans = "reverse"))
}


```

```{r, cache=TRUE}

ancom_core <-  ANCOMBC::ancombc2(data = ps.cor.bags, 
  fix_formula = "Group",
  rand_formula = NULL,
  p_adj_method = "fdr",
  pseudo = 0, 
  pseudo_sens = TRUE,
  prv_cut = 0.10,
  s0_perc = 0.05,
  group = "Group",
  struc_zero = TRUE,
  neg_lb = TRUE,
  alpha = 0.05,
  n_cl = 15, 
  tax_level = NULL,
  verbose = TRUE,
  global = FALSE, 
  pairwise = FALSE, 
  dunnet = FALSE, 
  trend = FALSE,
  iter_control = list(tol = 1e-2, max_iter = 20, verbose = TRUE),
  em_control = list(tol = 1e-5, max_iter = 100),
  lme_control = lme4::lmerControl(),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
    nrow = 2,
    byrow = TRUE),
    matrix(c(-1, 0, 1, -1),
    nrow = 2,
    byrow = TRUE)),
   node = list(2, 2),
   solver = "ECOS",
  B = 1)
  )

ancom_coreits <-  ANCOMBC::ancombc2(data = psits.bags, 
  fix_formula = "Group",
  rand_formula = NULL,
  p_adj_method = "fdr",
  pseudo = 0, 
  pseudo_sens = TRUE,
  prv_cut = 0.10,
  s0_perc = 0.05,
  group = "Group",
  struc_zero = TRUE,
  neg_lb = TRUE,
  alpha = 0.05,
  n_cl = 15, 
  tax_level = NULL,
  verbose = TRUE,
  global = FALSE, 
  pairwise = FALSE, 
  dunnet = FALSE, 
  trend = FALSE,
  iter_control = list(tol = 1e-2, max_iter = 20, verbose = TRUE),
  em_control = list(tol = 1e-5, max_iter = 100),
  lme_control = lme4::lmerControl(),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
    nrow = 2,
    byrow = TRUE),
    matrix(c(-1, 0, 1, -1),
    nrow = 2,
    byrow = TRUE)),
   node = list(2, 2),
   solver = "ECOS",
  B = 1)
  )

saveRDS(ancom_core, 'ancom_core')

```

```{r}

ancom_core <- readRDS('ancom_core')

non_str_otu <- ancom_core$res$taxon

zeros <- ancom_core$zero_ind %>% 
  column_to_rownames("taxon") %>% 
  rowSums() > 0L

zeros.all <- zeros[zeros == FALSE]
zeros.all <- names(zeros.all)

zeros.all %>% 
  length()

amp.cor.ancm



amp.cor.ancm$tax
```





```{r, fig.height=20, fig.width=8}

amp.cor <- phyloseq_to_ampvis2(ps.cor.bags)
amp.cor.ancm <- ampvis2::amp_filter_taxa(amp.cor, zeros.all, normalise = TRUE)

p_core_16 <- amp_heatmap(amp.cor.ancm,
            tax_show = 30,
            tax_aggregate = "Genus",
            # tax_add = "Phylum",
            group_by = "Group",
            plot_values_size = 3.5, 
            facet_by = "Group2",
            normalise = FALSE,
            showRemainingTaxa = TRUE) + theme(axis.text.y=element_text(size=rel(2)), axis.text.x=element_text(size=rel(2)))

p_core_16

amp.cor.ancm$abund %>% 
  colSums() %>% 
  mean()

amp_heatmap(ampits.cor.ancm,
            tax_show = 100,
            tax_aggregate = "OTU",
            tax_add = "Genus",
            group_by = "Group",
            plot_values_size = 2.5, 
            facet_by = "Group2",
            normalise = FALSE,
            showRemainingTaxa = TRUE) + theme(axis.text.y=element_text(size=rel(1.5)), axis.text.x=element_text(size=rel(2)))


```




```{r}

psits.bags@sam_data <- psits.bags@sam_data %>% 
  data.frame() %>% 
  mutate(Group = as.factor(Group)) %>% 
  mutate(Group = recode(Group,
                        BAGS_birch_leaves_5 = "БАГС_листья_березы",
                        BAGS_linden_leaves_5 = "БАГС_листья_липы",
                        BAGS_oat_straw_5 = "БАГС_солома_овса",
                        BAGS_oat_straw_gen1 = "маточная_культура_2018",
                        BAGS_oat_straw_gen2 = "маточная_культура_2021",
                        BAGS_wheat_straw_5 = "БАГС_солома_пшеницы",
                        linden_leaves_linden_leaves_NA = "листья_липы",
                        oat_straw_oat_straw_NA = "солома_овса",
                        peat_bag1_peat_NA = "торф",
                        peat_bag2_peat_NA = "торф",
                        peat_bag3_peat_NA = "торф")
  ) %>% sample_data()
         
         

psits.r@sam_data

ps.cor.bags@sam_data <- ps.cor.bags@sam_data %>% 
  data.frame() %>% 
  mutate(Group = as.factor(Group)) %>% 
  mutate(Group = recode(Group,
                        BAGS_birch_leaves_5 = "БАГС_листья_березы",
                        BAGS_linden_leaves_5 = "БАГС_листья_липы",
                        BAGS_oat_straw_5 = "БАГС_солома_овса",
                        BAGS_oat_straw_gen1 = "маточная_культура_2018",
                        BAGS_oat_straw_gen2 = "маточная_культура_2021",
                        BAGS_wheat_straw_5 = "БАГС_солома_пшеницы",
                        linden_leaves_linden_leaves_NA = "листья_липы",
                        oat_straw_oat_straw_NA = "солома_овса",
                        peat_bag1_peat_NA = "торф",
                        peat_bag2_peat_NA = "торф",
                        peat_bag3_peat_NA = "торф")
  ) %>% sample_data()

ps.cor.bags@sam_data <- ps.cor.bags@sam_data %>% 
  data.frame() %>% 
  mutate(Group2 = ifelse(Period %in% c('gen1', 'gen2'), 'old_bags', 
                         ifelse(Type %in% c('peat_bag1', 'peat_bag2', 'peat_bag3'), 'peat', 
                                ifelse(Type %in% c('linden_leaves', 'oat_straw'), 'cell_substrate', Type)))) %>% 
  sample_data()


psits.bags@sam_data <- psits.bags@sam_data %>% 
  data.frame() %>% 
  mutate(Group2 = ifelse(Period %in% c('gen1', 'gen2'), 'old_bags', 
                         ifelse(Type %in% c('peat_bag1', 'peat_bag2', 'peat_bag3'), 'peat', 
                                ifelse(Type %in% c('linden_leaves', 'oat_straw'), 'cell_substrate', Type)))) %>% 
  mutate(Group2 = ifelse(Group %in% c('БАГС_СО', 'БАГС_ЛЛ'), 'БАГС', 'BAGS')) %>% 
  sample_data()







psits.bags@sam_data

```


```{r, fig.height=8, fig.width=8}

zeros <- ancom_coreits$zero_ind %>% 
  column_to_rownames("taxon") %>% 
  rowSums() > 0L

zeros.all <- zeros[zeros == FALSE]
zeros.all <- names(zeros.all)

ampits.cor <- phyloseq_to_ampvis2(psits.bags)
ampits.cor.ancm <- ampvis2::amp_filter_taxa(ampits.cor, zeros.all, normalise = TRUE)

p_core_i <- amp_heatmap(ampits.cor.ancm,
            tax_show = 30,
            tax_aggregate = "Genus",
            # tax_add = "Phylum",
            group_by = "Group",
            plot_values_size = 3.5, 
            facet_by = "Group2",
            normalise = FALSE,
            showRemainingTaxa = TRUE) + theme(axis.text.y=element_text(size=rel(2)), axis.text.x=element_text(size=rel(2)))

p_core_i


ampits.cor.ancm$abund %>% 
  colSums() %>% 
  mean()




# amp_heatmap(amp.cor.ancm,
#             tax_show = 60,
#             tax_aggregate = "Genus",
#             tax_add = "Phylum",
#             group_by = "Group",
#             plot_values_size = 2.8, 
#             facet_by = "Group2", 
#             normalise = FALSE,
#             showRemainingTaxa = TRUE) + theme(axis.text.y=element_text(size=rel(1.5)))


```
```{r, fig.height=10, fig.width=14}

ggpubr::ggarrange(p_core_16, p_core_i, ncol = 2, labels = c('16S', 'ITS2'), hjust = -0.5)

p_core_16

```

## source

```{r}

ps.core.oat<- prune_samples((sample_data(ps.cor)$Group %in% c('BAGS_oat_straw_gen1', 'BAGS_oat_straw_5', 'oat_straw_oat_straw_NA')) | 
                              (sample_data(ps.cor)$Group2 == 'peat'), ps.cor)
ps.core.oat  <- prune_taxa(taxa_sums(ps.core.oat) > 0, ps.core.oat)

p.core.oat <- plot_vienn(ps.core.oat, "Group2")

p.core.oat[[1]]

```




```{r}

3022/(3022 + 980 + 412 + 663 + 51 + 41 + 14 + 22)

```

```{r}


p.core.oat <- plot_vienn(ps.core.oat, "Group2")

psits.oat<- prune_samples((sample_data(psits)$Group %in% c('BAGS_oat_straw_gen1', 'BAGS_oat_straw_5', 'oat_straw_oat_straw_NA')) | 
                              (sample_data(psits)$Group2 == 'peat'), psits)
psits.oat  <- prune_taxa(taxa_sums(psits.oat) > 0, psits.oat)

plot_vienn(psits.oat, "Group2")

```
  
  
  
  mutate(Group2 = ifelse(Period %in% c('gen1', 'gen2'), 'инокулюм', 
                         ifelse(Type %in% c('peat_bag1', 'peat_bag2', 'peat_bag3'), 'торф', 
                                ifelse(Type %in% c('linden_leaves', 'oat_straw'), 'субстрат', Type)))) %>% 

                        BAGS_birch_leaves_5 = "БАГС_ЛБ",
                        BAGS_linden_leaves_5 = "БАГС_ЛЛ",
                        BAGS_oat_straw_5 = "БАГС_СО",
                        BAGS_oat_straw_gen1 = "БАГС_2018",
                        BAGS_oat_straw_gen2 = "БАГС_2021",
                        BAGS_wheat_straw_5 = "БАГС_СП",
                        linden_leaves_linden_leaves_NA = "ЛЛ",
                        oat_straw_oat_straw_NA = "СО",
                        peat_bag1_peat_NA = "торф",
                        peat_bag2_peat_NA = "торф",
                        peat_bag3_peat_NA = "торф")
                        
                        
                        
```{r}

ps.cor@sam_data

```
                        
```{r}

ps.core.oat<- prune_samples((sample_data(ps.cor)$Group %in% c('БАГС_2018', 'БАГС_СО', 'СО')) | 
                              (sample_data(ps.cor)$Group2 == 'торф'), ps.cor)
ps.core.oat  <- prune_taxa(taxa_sums(ps.core.oat) > 0, ps.core.oat)

p.core.oat <- plot_vienn(ps.core.oat, "Group2")

ps.core.lin <- prune_samples((sample_data(ps.cor)$Group %in% c('БАГС_2018', 'БАГС_ЛЛ', 'ЛЛ')) | 
                              (sample_data(ps.cor)$Group2 == 'торф'), ps.cor)
ps.core.lin  <- prune_taxa(taxa_sums(ps.core.lin) > 0, ps.core.lin)

p.core.lin <- plot_vienn(ps.core.lin, "Group2")
  
psits.lin <- prune_samples((sample_data(psits)$Group %in% c('БАГС_2018', 'БАГС_ЛЛ', 'ЛЛ')) | 
                              (sample_data(psits)$Group2 == 'торф'), psits)
psits.lin  <- prune_taxa(taxa_sums(psits.lin) > 0, psits.lin)

p.core.lin.its <- plot_vienn(psits.lin, "Group2")

psits.oat <- prune_samples((sample_data(psits)$Group %in% c('БАГС_2018', 'БАГС_СО', 'СО')) | 
                              (sample_data(psits)$Group2 == 'торф'), psits)
psits.oat  <- prune_taxa(taxa_sums(psits.oat) > 0, psits.oat)

p.core.oat.its <- plot_vienn(psits.lin, "Group2")


```
```{r, fig.width=10, fig.height=10}

ggpubr::ggarrange(p.core.oat[[1]], p.core.oat.its[[1]], p.core.lin[[1]], p.core.lin.its[[1]], nrow = 2, ncol =2, legend = 'none', labels = c('16S - СО', 'ITS2 - СО', '16S - ЛЛ', 'ITS2 - ЛЛ'))

```


```{r}

plot_vienn(ps.core.oat, "Group2")

```

```{r}

plot_rich_reads_samlenames_lm(ps.core.oat, group = "Group2", label = "Group2") 
plot_rich_reads_samlenames_lm(ps.core.lin, group = "Group2", label = "Group2") 

```



```{r}

plot_rich_reads_samlenames_lm(psits.oat, group = "Group2", label = "Group2") 
plot_rich_reads_samlenames_lm(psits.lin, group = "Group2", label = "Group2") 


```




## Chrono


```{r, fig.height=12, fig.width=12}

ps.chr <- prune_samples(!sample_data(ps.bag)$Period %in% c('gen1', 'gen2'), ps.bag)
ps.chr  <- prune_taxa(taxa_sums(ps.chr) > 0, ps.chr)


beta_custom_norm_NMDS_elli_w(ps.chr, Color = 'Substrate', Group = 'Group' )

ps_vst <- function(ps, factor){
  diagdds = phyloseq::phyloseq_to_deseq2(ps, as.formula(paste( "~", factor)))              
  diagdds = DESeq2::estimateSizeFactors(diagdds, type="poscounts")
  diagdds = DESeq2::estimateDispersions(diagdds, fitType = "local") 
  pst <- DESeq2::varianceStabilizingTransformation(diagdds)
  pst.dimmed <- t(as.matrix(SummarizedExperiment::assay(pst))) 
  # pst.dimmed[pst.dimmed < 0.0] <- 0.0
  ps.varstab <- ps
  otu_table(ps.varstab) <- otu_table(pst.dimmed, taxa_are_rows = FALSE) 
  return(ps.varstab)
} 
  
ps.chr.vst <- ps_vst(ps = ps.chr, 'Group')
beta_custom_norm_NMDS_elli_w(ps.chr.vst, Color = 'Substrate', Group = 'Group' )


amp.cr <- phyloseq_to_ampvis2(ps.chr)
amp.cr

amp_heatmap(amp.chr,
            tax_show = 60,
            tax_aggregate = "Phylum",
            tax_add = "Kingdom",
            group_by = "Group",
            plot_values_size = 2.5, 
            facet_by = "Substrate",
            normalise = TRUE,
            showRemainingTaxa = TRUE) + theme(axis.text.y=element_text(size=rel(1.5)), axis.text.x=element_text(size=rel(2)))


```

### alpha

```{r, fig.height=8, fig.width=7}


er <- estimate_richness(ps.chr.r)
df_er <- cbind(ps.chr.r@sam_data, er)

df_er

df_er %>% 
  rownames_to_column("ID") %>% 
  select(c(Substrate, Period, Observed, Observed, Shannon, InvSimpson))%>% 
  group_by(Substrate, Period) %>%
  pivot_longer(c("Observed", "Shannon", "InvSimpson")) %>% 
  mutate(metric = factor(name, levels = c("Observed", "Shannon", "InvSimpson"))) %>% 
  mutate(Субстрат = as.factor(Substrate) %>% recode(oat_straw = "БАГС_СО", linden_leaves = "БАГС_ЛЛ")) %>% 
  # mutate(Group = recode(Group, 
  #                       cPrimer = "priming",
  #                       lPrimer = "priming",
  #                       l1 = "early",
  #                       c1 = "early",
  #                       l2 = "middle",
  #                       c2 = "middle",
  #                       c3 = "late",
  #                       l3 = "late")) %>% 
  ggplot(aes(y = value, x = Period, group = Субстрат)) +
  geom_point(aes(color = Субстрат)) +
  stat_summary(aes(y = value, color = Субстрат), fun.y=mean, geom="line") +
    labs(x = "Временные точки",
       y = "Индекс") +
  theme_bw() +
  facet_wrap(~metric, scales = "free", ncol = 1) +
  theme(legend.position="bottom") +
      scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8)


```

```{r}

ps.chr.clr <- ps.chr
ps.otu.clr <- as.data.frame(t(ps.f@otu_table)) %>% 
  compositions::clr() %>% 
  t()
ps.chr.clr@otu_table <-  otu_table(ps.otu.clr, taxa_are_rows = FALSE)


beta_custom_norm_NMDS_elli_w(ps.chr.clr, Color = 'Substrate', Group = 'Group')



```



```{r, fig.height=5, fig.width=8}

ps.chr.r <- rarefy_even_depth(ps.chr, rngseed = 1488)
beta_custom_norm_PCoA_elli_w(ps.chr.r, Color = 'Substrate', Group = 'Group')

```



```{r}

avg <- ps.chr.r@otu_table %>% 
  as.data.frame() %>%
  vegan::avgdist(10)

# avg %>%
#   as.matrix() %>%
#   as_tibble(rownames= "sample") %>%
#   pivot_longer(-sample) %>%
#   filter(sample < name) %>%
#   mutate(repeat_a = str_replace(sample, ".*-", ""),
#          repeat_b = str_replace(name, ".*-", ""), 
#          day_a = as.numeric(str_replace(sapply(strsplit(sample, "-"), `[`, 3), "D", "")),
#           day_b = as.numeric(str_replace(sapply(strsplit(name, "-"), `[`, 3), "D", "")),
#          diff = abs(day_a - day_b),
#          early = day_a < 10) %>% 
#   filter(repeat_a == repeat_b & diff < 10) %>%
#   group_by(diff, repeat_a, early) %>%
#   summarize(median = median(value)) %>%
#   ungroup() %>%
#   ggplot(aes(x=diff, y=median, color=early, group=paste0(repeat_a, early))) +
#   geom_line(size=0.25) +
#   geom_smooth(aes(group=early), se=FALSE, size=4) +
#   labs(x="Distance between time points",
#        y="Median Bray-Curtis distance") +
#   scale_x_continuous(breaks=1:9) +
#   scale_color_manual(name=NULL,
#                      breaks=c(TRUE, FALSE),
#                      values=c("blue", "red"),
#                      labels=c("Early", "Late")) +
#   guides(color = guide_legend(override.aes = list(size=1))) +
#   theme_classic()

avg %>%
  as.matrix() %>%
  as_tibble(rownames= "sample") %>%
  pivot_longer(-sample) %>%
  filter(sample < name) 



  mutate(repeat_a = str_replace(sample, ".*-", ""),
         repeat_b = str_replace(name, ".*-", ""), 
         day_a = as.numeric(str_replace(sapply(strsplit(sample, "\\."), `[`, 3), "D", "")),
         day_b = as.numeric(str_replace(sapply(strsplit(name, "\\."), `[`, 3), "D", ""))) %>% 
  mutate(day_a = as.numeric(as.factor(day_a) %>% forcats::fct_recode("2" = "3", "3" = "5", "4" = "7", "5" = "8", "6" = "10", "7" = "13", "8" = "14", "9" = "14", "10" = "15")),
         day_b = as.numeric(as.factor(day_b) %>% forcats::fct_recode("2" = "3", "3" = "5", "4" = "7", "5" = "8", "6" = "10", "7" = "12", "8" = "13", "9" = "14", "10" = "15")),
         diff = abs(day_a - day_b)) %>% 
  filter(diff == 1) %>% 
  mutate(day_b = as.factor(day_b) %>% forcats::fct_recode("14" = "2", "28" = "3","49" = "4",  "63" = "5", "91" = "6", "161" = "9", "140" = "8", "182" = "10", "119" = "7")) %>% 
  ggplot(aes(x=day_b, y=value)) +
    geom_boxplot() +
  theme_bw() +
  labs(x="Days",
       y="Bray-Curtis distance")


  ps.chr.r@sam_data
  
```



```{r}



ps.chr@sam_data  <- ps.chr@sam_data %>% 
  data.frame() %>% 
  mutate(Group = paste0(str_split_i(Group, '_' , 3), '_', Period)) %>% 
  sample_data()

ps.chr.f
ps.chr

```




```{r}

ps.chr.f <- phyloseq::filter_taxa(ps.chr, function(x) sum(x > 10) > (0.1*length(x)), TRUE)


out.f <-  ANCOMBC::ancombc2(data=ps.chr.f, 
  fix_formula = "Group",
  prv_cut = 0,
  p_adj_method = "BH",
  tax_level = NULL,
  rand_formula= NULL,
  pseudo = 0, 
  pseudo_sens = FALSE,
  s0_perc = 0.05,
  group = "Group",
  struc_zero = FALSE,
  neg_lb = TRUE,
  alpha = 0.05,
  n_cl = 20, 
  verbose = TRUE,
  global = FALSE, 
  pairwise = FALSE, 
  dunnet = FALSE, 
  trend = FALSE
  )
  
out.f

unregister <- function() {
  env <- foreach:::.foreachGlobals
  rm(list=ls(name=env), pos=env)
}

saveRDS(out.f, "out.f")
out.f <- readRDS()
list.files()
# anc <- readRDS("")
out.f

unregister()

```



```{r}

ps.anc.f <- ps.chr.f

samp_frac <-  out.f$samp_frac
samp_frac[is.na(samp_frac)] <-  0 
# Add pesudo-count (1) to avoid taking the log of 0
log_obs_abn = log(out.f$feature_table + 1)
# Adjust the log observed abundances
log_corr_abn = t(t(log_obs_abn) - samp_frac)
otu_table(ps.anc.f) <- otu_table(t(log_corr_abn), taxa_are_rows = FALSE)

beta_custom_norm_PCA_elli_w(ps.anc.f, Color = "Substrate", Group = "Group")

```



```{r}

ps.anc.f@sam_data$Substrate

ps.chr.oat <- prune_samples(sample_data(ps.anc.f)$Substrate == 'oat_straw', ps.anc.f)
ps.chr.oat  <- prune_taxa(taxa_sums(ps.chr.oat) > 0, ps.chr.oat)

ps.chr.lin <- prune_samples(sample_data(ps.anc.f)$Substrate == 'linden_leaves', ps.anc.f)
ps.chr.lin  <- prune_taxa(taxa_sums(ps.chr.lin) > 0, ps.chr.lin)

ps.chr.lin
ps.chr.oat

```


## oat

```{r}

library(dplyr)
library(phyloseq)

ps.vst.mod <- ps.chr.oat

ps.vst.mod@sam_data <- ps.vst.mod@sam_data %>% 
  data.frame() %>% 
  mutate(Repeat = paste0(Group, "_", Sample_No)
                         ) %>% 
  sample_data()

data3 <- ps.vst.mod@otu_table@.Data %>% 
  as.data.frame()

library(doParallel)
#Find out how many cores are available (if you don't already know)
cores<-detectCores()
#Create cluster with desired number of cores, leave one open for the machine         
#core processes
cl <- makeCluster(cores[1]-1)
#Register cluster
registerDoParallel(cl)

rownames(data3) <- as.character(ps.vst.mod@sam_data$Repeat)
powers <-  c(seq(from = 1, to=10, by=0.5), seq(from = 11, to=20, by=1))
sft3 <-  pickSoftThreshold(data3, powerVector = powers, verbose = 5, networkType = "signed hybrid")

plot(sft3$fitIndices[,1], -sign(sft3$fitIndices[,3])*sft3$fitIndices[,2], xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n", main = paste("Scale independence"))
text(sft3$fitIndices[,1], -sign(sft3$fitIndices[,3])*sft3$fitIndices[,2], labels=powers,cex=0.9,col="red")
abline(h=0.9,col="salmon")


```

```{r, fig.width=6, fig.height=6}

detachAllPackages()
library(WGCNA)


net3 <- WGCNA::blockwiseModules(data3,
                          power=16,
                          TOMType="signed",
                          networkType="signed hybrid",
                          nThreads=15)

library(phyloseq)
library(tidyverse)
library(ggpubr)
library(ampvis2)
library(heatmaply)
library(WGCNA)
library(phyloseq)
library(ggtree)
library(tidyverse)
library(KneeArrower)

net3$colors %>% 
  unique()

# mergedColors2 <-  WGCNA::labels2colors(net3$colors, colorSeq = c(net3$colors %>% 
  # unique()))

mergedColors2 <- net3$colors

plotDendroAndColors(
  net3$dendrograms[[1]],
  mergedColors2[net3$blockGenes[[1]]],
  "Module colors",
  dendroLabels = FALSE,
  hang = 0.03,
  addGuide = TRUE,
  guideHang = 0.05)

```


```{r, fig.height=7, fig.width=8}

modules_of_interest = mergedColors2 %>% 
  unique()

module_df <- data.frame(
  asv = names(net3$colors),
  colors = mergedColors2
)

# module_df[module_df == "yellow"] <- "salmon"

submod <-  module_df %>%
  subset(colors %in% modules_of_interest)

row.names(module_df) = module_df$asv

subexpr = as.data.frame(t(data3))[submod$asv,]

submod_df <- data.frame(subexpr) %>%
  mutate(
    asv = row.names(.)
  ) %>%
  pivot_longer(-asv) %>%
  mutate(
    module = module_df[asv,]$colors
  )
submod_df
submod_df <- submod_df %>% 
  mutate(name = str_split_i(name, '_' ,2)) %>% 
  group_by(name, asv) %>% 
  summarise(value = mean(value), asv = asv, module = module) %>% 
  relocate(c(asv, name, value, module)) %>% 
  ungroup() %>% 
  mutate(module = as.factor(module))

p <- submod_df %>% 
  ggplot(., aes(x=name, y=value, group=asv)) +
  geom_line(aes(color = module),
            alpha = 0.2) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90),
    legend.position = "none") +
  facet_grid(rows = vars(module)) +
  labs(x = "day",
       y = "normalized abundance")

p
p_wgscna_oat <- p + scale_color_manual(values = levels(submod_df$module))

#   scale_x_discrete(labels=(c("D01"="3",
#                                 "D03"="14",
#                                 "D05"="28",
#                                 "D07"="49" ,
#                                 "D08"="63",
#                                 "D10"="91",
#                                 "D12"="119",
#                                 "D13"="140",
#                                 "D14"="161",
#                                 "D15"="182")
#          ))

p_wgscna_oat


```

```{r, fig.height=7, fig.width=10}

ids <- ps.vst.mod@sam_data %>% 
  data.frame() %>%
  rownames_to_column("ID") %>% 
  # filter(Region == "Yakutsk") %>%
  pull(ID)
  

agra_data <- ps.vst.mod@sam_data %>% 
  data.frame() %>% 
  select_if(., is.double) %>% 
  select( -c("Sample_No", "Year", "Age_days")) 
  


# mergedColors2 %>% unique()
nOTUs <- ncol(data3)
nSamples <- nrow(data3)

# Recalculate MEs with color labels
MEs0 <- moduleEigengenes(data3, mergedColors2)$eigengenes
MEs <- orderMEs(MEs0)
MEs



names(MEs) <- substring(names(MEs), 3)
names(MEs) <- c("black", "blue","brown", "green", "grey", "red", "turq", "yellow")  

moduleTraitCor <- cor(MEs, agra_data, use = "p")
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples)

# PLOT
sizeGrWindow(10,6)
textMatrix <- paste(signif(moduleTraitCor, 2), "\n(", format.pval(moduleTraitPvalue, digits = 2, eps = 0.001, nsmall = 3), ")", sep = "")
dim(textMatrix) <- dim(moduleTraitCor)
par(mar = c(4, 10, 2, 2))

# Display the correlation values within a heatmap
labeledHeatmap(Matrix = moduleTraitCor, 
               xLabels = names(agra_data),
               yLabels = names(MEs), 
               ySymbols = names(MEs), 
               font.lab.y = 2,
               font.lab.x = 2,
               # yLabelsPosition = "right",
               cex.lab.y = 0.85,
               colorLabels = FALSE, 
               colors = blueWhiteRed(50),
               textMatrix = textMatrix, 
               setStdMargins = FALSE,
               cex.text = 0.8, 
               zlim = c(-1,1), 
               plotLegend = FALSE)
               # main = paste("Currelation of the WGCNA clusters with soil characteristics"))



```

"brown"     "blue"      "turquoise" "grey"      "green"     "yellow"    "black"     "red" 

```{r}

clust.turq <- names(net3$colors[net3$colors %in% c("turquoise")])

clust.bb <- names(net3$colors[net3$colors %in% c("blue", "black")])

```


```{r, fig.width=7, fig.height=7}

amp.chr <- phyloseq_to_ampvis2(ps.chr)
amp.chr.oat <- ampvis2::amp_filter_samples(amp.chr, Substrate=='oat_straw', normalise = TRUE)
amp.chr.oat <- ampvis2::amp_filter_taxa(amp.chr.oat, clust.bb, normalise = FALSE)

amp.chr.oat
oat_cell <- amp_heatmap(amp.chr.oat,
            tax_show = 30,
            tax_aggregate = "Genus",
            tax_add = "Phylum",
            group_by = "Group",
            plot_values_size = 2.8, 
            facet_by = "Substrate",
            normalise = FALSE,
            showRemainingTaxa = TRUE) + theme(axis.text.y=element_text(size=rel(1.2)))

oat_cell

```


## leaves

```{r}

library(dplyr)
library(phyloseq)

ps.vst.mod <- ps.chr.lin

ps.vst.mod@sam_data <- ps.vst.mod@sam_data %>% 
  data.frame() %>% 
  mutate(Repeat = paste0(Group, "_", Sample_No)
                         ) %>% 
  sample_data()

data3 <- ps.vst.mod@otu_table@.Data %>% 
  as.data.frame()

library(doParallel)
#Find out how many cores are available (if you don't already know)
cores<-detectCores()
#Create cluster with desired number of cores, leave one open for the machine         
#core processes
cl <- makeCluster(cores[1]-1)
#Register cluster
registerDoParallel(cl)

rownames(data3) <- as.character(ps.vst.mod@sam_data$Repeat)
powers <-  c(seq(from = 1, to=10, by=0.5), seq(from = 11, to=20, by=1))
sft3 <-  pickSoftThreshold(data3, powerVector = powers, verbose = 5, networkType = "signed hybrid")

plot(sft3$fitIndices[,1], -sign(sft3$fitIndices[,3])*sft3$fitIndices[,2], xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n", main = paste("Scale independence"))
text(sft3$fitIndices[,1], -sign(sft3$fitIndices[,3])*sft3$fitIndices[,2], labels=powers,cex=0.9,col="red")
abline(h=0.9,col="salmon")


```



```{r}

detachAllPackages()
library(WGCNA)


net3 <- WGCNA::blockwiseModules(data3,
                          power=5.5,
                          TOMType="signed",
                          networkType="signed hybrid",
                          nThreads=15)

library(phyloseq)
library(tidyverse)
library(ggpubr)
library(ampvis2)
library(heatmaply)
library(WGCNA)
library(phyloseq)
library(ggtree)
library(tidyverse)
library(KneeArrower)

net3$colors %>% 
  unique()

# mergedColors2 <-  WGCNA::labels2colors(net3$colors, colorSeq = c(net3$colors %>% 
  # unique()))

mergedColors2 <- net3$colors

plotDendroAndColors(
  net3$dendrograms[[1]],
  mergedColors2[net3$blockGenes[[1]]],
  "Module colors",
  dendroLabels = FALSE,
  hang = 0.03,
  addGuide = TRUE,
  guideHang = 0.05)

```

```{r, fig.height=7, fig.width=8}

modules_of_interest = mergedColors2 %>% 
  unique()

module_df <- data.frame(
  asv = names(net3$colors),
  colors = mergedColors2
)

# module_df[module_df == "yellow"] <- "salmon"

submod <-  module_df %>%
  subset(colors %in% modules_of_interest)

row.names(module_df) = module_df$asv

subexpr = as.data.frame(t(data3))[submod$asv,]

submod_df <- data.frame(subexpr) %>%
  mutate(
    asv = row.names(.)
  ) %>%
  pivot_longer(-asv) %>%
  mutate(
    module = module_df[asv,]$colors
  )
submod_df
submod_df <- submod_df %>% 
  mutate(name = str_split_i(name, '_' ,2)) %>% 
  group_by(name, asv) %>% 
  summarise(value = mean(value), asv = asv, module = module) %>% 
  relocate(c(asv, name, value, module)) %>% 
  ungroup() %>% 
  mutate(module = as.factor(module))

p <- submod_df %>% 
  ggplot(., aes(x=name, y=value, group=asv)) +
  geom_line(aes(color = module),
            alpha = 0.2) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90),
    legend.position = "none") +
  facet_grid(rows = vars(module)) +
  labs(x = "day",
       y = "normalized abundance")

p
p + scale_color_manual(values = levels(submod_df$module))

+
#   scale_x_discrete(labels=(c("D01"="3",
#                                 "D03"="14",
#                                 "D05"="28",
#                                 "D07"="49" ,
#                                 "D08"="63",
#                                 "D10"="91",
#                                 "D12"="119",
#                                 "D13"="140",
#                                 "D14"="161",
#                                 "D15"="182")
#          ))


```

```{r, fig.height=7, fig.width=10}

ids <- ps.vst.mod@sam_data %>% 
  data.frame() %>%
  rownames_to_column("ID") %>% 
  # filter(Region == "Yakutsk") %>%
  pull(ID)
  

agra_data <- ps.vst.mod@sam_data %>% 
  data.frame() %>% 
  select_if(., is.double) %>% 
  select( -c("Sample_No", "Year", "Age_days")) 
  
  # %>% 
  # relocate(pH, P2O5) %>% 
  # relocate(Ni, .after = N.NO3)

# mergedColors2 %>% unique()
nOTUs <- ncol(data3)
nSamples <- nrow(data3)

# Recalculate MEs with color labels
MEs0 <- moduleEigengenes(data3, mergedColors2)$eigengenes
MEs <- orderMEs(MEs0)

MEs

# names(MEs) <- substring(names(MEs), 3)
names(MEs) <- c("green", "blue","yellow", "turq.", "black", "pink", "brown", "red", "grey")  

moduleTraitCor <- cor(MEs, agra_data, use = "p")
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples)

# PLOT
sizeGrWindow(10,6)
textMatrix <- paste(signif(moduleTraitCor, 2), "\n(", format.pval(moduleTraitPvalue, digits = 2, eps = 0.001, nsmall = 3), ")", sep = "")
dim(textMatrix) <- dim(moduleTraitCor)
par(mar = c(4, 10, 2, 2))

# Display the correlation values within a heatmap
heat_elements <- labeledHeatmap(Matrix = moduleTraitCor, 
               xLabels = names(agra_data),
               yLabels = names(MEs), 
               ySymbols = names(MEs), 
               font.lab.y = 2,
               font.lab.x = 2,
               # yLabelsPosition = "right",
               cex.lab.y = 0.85,
               colorLabels = FALSE, 
               colors = blueWhiteRed(50),
               textMatrix = textMatrix, 
               setStdMargins = FALSE,
               cex.text = 0.8, 
               zlim = c(-1,1), 
               plotLegend = FALSE,
               main = paste("Currelation of the WGCNA clusters with soil characteristics"))

```

```{r}

clust.dd <- names(net3$colors[net3$colors %in% c("green", "blue", "yellow")])




```


```{r, fig.width=7, fig.height=7}

amp.chr <- phyloseq_to_ampvis2(ps.chr)

# amp.chr.lin <- ampvis2::amp_filter_samples(amp.chr, Substrate=='linden_leaves', normalise = TRUE)
amp.chr.lin <- ampvis2::amp_filter_taxa(amp.chr.lin, clust.dd, normalise = FALSE)


lin_cell <- amp_heatmap(amp.chr.lin,
            tax_show = 30,
            tax_aggregate = "Genus",
            # tax_add = "Phylum",
            group_by = "Group",
            plot_values_size = 3.5, 
            # facet_by = "Group2",
            normalise = FALSE,
            showRemainingTaxa = TRUE) + theme(axis.text.y=element_text(size=rel(2)), axis.text.x=element_text(size=rel(2)))

oat_cell <- amp_heatmap(amp.chr.oat,
            tax_show = 30,
            tax_aggregate = "Genus",
            # tax_add = "Phylum",
            group_by = "Group",
            plot_values_size = 3.5, 
            # facet_by = "Group2",
            normalise = FALSE,
            showRemainingTaxa = TRUE) + theme(axis.text.y=element_text(size=rel(2)), axis.text.x=element_text(size=rel(2)))

lin_cell

```



```{r, fig.width=14}

ggpubr::ggarrange(oat_cell, lin_cell, labels = c('БАГС_CО','БАГС_ЛЛ'), vjust = 1.2, hjust = -0.002, font.label = list(size = 15, font="regular"))

```








