{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %config IPCompleter.greedy=False\n",
    "%config Completer.use_jedi = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "find: ‘bwa-mem2’: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "!find bwa-mem2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/gladkov2/anaconda3/bin/metabat\n",
      "/home/gladkov2/anaconda3/pkgs/metabat2-2.15-h137b6e9_0/bin/metabat\n",
      "/home/gladkov2/anaconda3/envs/metabat\n",
      "/home/gladkov2/anaconda3/envs/metabat/bin/metabat\n"
     ]
    }
   ],
   "source": [
    "!find  /home/gladkov2/anaconda3/ -name \"metabat\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/gladkov2/storage/temp/serg/checkm\n"
     ]
    }
   ],
   "source": [
    "!find  /home/gladkov2/storage/ -name \"checkm\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### vamb env"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/storage/metagenome/bact_4x\n",
      "/mnt/storage/metagenome/bact_4x/binning\n",
      "total 26G\n",
      "-rw-rw-r-- 1 gladkov2 3.8G Oct  8 13:01 26_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 3.7G Oct  8 13:09 46_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 3.0G Oct  8 13:00 26_ont.fasta\n",
      "-rw-rw-r-- 1 gladkov2 2.9G Oct  8 13:09 46_ont.fasta\n",
      "-rw-rw-r-- 1 gladkov2 2.6G Oct  8 12:57 14_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 2.2G Oct  8 13:06 29_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 2.1G Oct  8 12:56 14_ont.fasta\n",
      "-rw-rw-r-- 1 gladkov2 1.8G Oct  8 13:06 29_ont.fasta\n",
      "-rw-rw-r-- 1 gladkov2 1.3G Oct  8 13:06 26_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 1.3G Oct  8 13:14 46_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 842M Oct  8 13:00 14_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 712M Oct  8 13:09 29_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 109M Oct  8 12:55 46.fq\n",
      "-rw-rw-r-- 1 gladkov2  95M Oct  8 12:55 26.fq\n",
      "-rw-rw-r-- 1 gladkov2  89M Oct  8 12:55 14.fq\n",
      "-rw-rw-r-- 1 gladkov2  88M Oct  8 12:55 29.fq\n"
     ]
    }
   ],
   "source": [
    "%cd ~/storage/metagenome/bact_4x/\n",
    "# %mkdir binning\n",
    "%cd binning\n",
    "%ll -hS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'14': ['14.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode01/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/14_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/14_R_wa.fq'],\n",
       " '26': ['26.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode02/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/26_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/26_R_wa.fq'],\n",
       " '29': ['29.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode03/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/29_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/29_R_wa.fq'],\n",
       " '46': ['46.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode04/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/46_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/46_R_wa.fq']}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = {\"14\":[\"14.fq\",\n",
    "                    \"/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode01/all.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/14_F_wa.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/14_R_wa.fq\"],\n",
    "     \"26\":[\"26.fq\",\n",
    "                    \"/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode02/all.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/26_F_wa.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/26_R_wa.fq\"],\n",
    "     \"29\":[\"29.fq\",\n",
    "                    \"/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode03/all.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/29_F_wa.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/29_R_wa.fq\"],\n",
    "     \"46\":[\"46.fq\",\n",
    "                    \"/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode04/all.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/46_F_wa.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/46_R_wa.fq\"],\n",
    "    }\n",
    "\n",
    "d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: /home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode01/all.fq: Permission denied\n"
     ]
    }
   ],
   "source": [
    "!$ont"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!ont = d['14'][1]\n",
    "# !head $ont\n",
    "!sed -n '1~4s/^@/>/p;2~4p' $ont | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "merge not a valid action\n",
      "\n",
      "    available actions:\n",
      "        `extract`: extract sequences from a fasta file\n",
      "        `info`: show info about the fasta file and exit.\n",
      "        `split`: split a large fasta file into separate files\n",
      "                 and/or into K-mers.\n",
      "        `flatten`: flatten a fasta file inplace so that later\n",
      "                   command-line (and programmattic) access via\n",
      "                   pyfasta will use the inplace flattened version\n",
      "                   rather than creating another .flat copy of the\n",
      "                   sequence.\n",
      "\n",
      "    to view the help for a particular action, use:\n",
      "        pyfasta [action] --help\n",
      "    e.g.:\n",
      "        pyfasta extract --help\n",
      "    \n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "%rm *\n",
    "!cp /home/gladkov2/storage/metagenome/bact_4x/new_con/diamond/con/14.faa 14.fq\n",
    "!cp /home/gladkov2/storage/metagenome/bact_4x/new_con/diamond/con/26.faa 26.fq\n",
    "!cp /home/gladkov2/storage/metagenome/bact_4x/new_con/diamond/con/29.faa 29.fq\n",
    "!cp /home/gladkov2/storage/metagenome/bact_4x/new_con/diamond/con/46.faa 46.fq\n",
    "# !sed -n '1~4s/^@/>/p;2~4p' $ont | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'd' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-3-5fe1ba162f8e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mont\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0md\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msystem\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'head $ont'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'd' is not defined"
     ]
    }
   ],
   "source": [
    "ont = d[f][1]\n",
    "!head $ont"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!bwa-mem2 mem -t 50 14.fq 14_ont.fasta > test.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 5 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "!samtools sort test.sam > test.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::2.191*1.30] collected minimizers\n",
      "[M::mm_idx_gen::4.686*2.25] sorted minimizers\n",
      "[M::main::4.687*2.25] loaded/built the index for 1903 target sequence(s)\n",
      "[M::mm_mapopt_update::4.904*2.20] mid_occ = 32\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1903\n",
      "[M::mm_idx_stat::5.012*2.17] distinct minimizers: 11752665 (78.56% are singletons); average occurrences: 1.458; average spacing: 5.359\n",
      "[M::worker_pipeline::13.696*27.45] mapped 165675 sequences\n",
      "[M::worker_pipeline::18.378*33.18] mapped 167791 sequences\n",
      "[M::worker_pipeline::21.642*34.93] mapped 168228 sequences\n",
      "[M::worker_pipeline::25.580*33.65] mapped 167010 sequences\n",
      "[M::worker_pipeline::25.887*33.27] mapped 18490 sequences\n",
      "[M::worker_pipeline::26.324*32.93] mapped 170049 sequences\n",
      "[M::worker_pipeline::26.784*32.60] mapped 170049 sequences\n",
      "[M::main] Version: 2.17-r941\n",
      "[M::main] CMD: minimap2 -ax map-ont -t 50 14.fq 14_ont.fasta /home/gladkov2/storage/metagenome/bact_4x/new_con/14_F_wa.fq /home/gladkov2/storage/metagenome/bact_4x/new_con/14_R_wa.fq\n",
      "[M::main] Real time: 26.891 sec; CPU: 873.219 sec; Peak RSS: 4.738 GB\n",
      "[bam_sort_core] merging from 4 files and 1 in-memory blocks...\n",
      "[M::mm_idx_gen::2.414*1.29] collected minimizers\n",
      "[M::mm_idx_gen::3.128*2.84] sorted minimizers\n",
      "[M::main::3.128*2.84] loaded/built the index for 1589 target sequence(s)\n",
      "[M::mm_mapopt_update::3.356*2.72] mid_occ = 32\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1589\n",
      "[M::mm_idx_stat::3.477*2.66] distinct minimizers: 12434700 (78.28% are singletons); average occurrences: 1.464; average spacing: 5.358\n",
      "[M::worker_pipeline::13.829*32.26] mapped 166217 sequences\n",
      "[M::worker_pipeline::17.164*33.63] mapped 165938 sequences\n",
      "[M::worker_pipeline::21.409*36.89] mapped 168032 sequences\n",
      "[M::worker_pipeline::25.183*38.69] mapped 168103 sequences\n",
      "[M::worker_pipeline::29.703*40.31] mapped 168850 sequences\n",
      "[M::worker_pipeline::32.491*37.89] mapped 170567 sequences\n",
      "[M::worker_pipeline::32.674*37.69] mapped 5047 sequences\n",
      "[M::worker_pipeline::33.075*37.40] mapped 144038 sequences\n",
      "[M::worker_pipeline::33.553*37.06] mapped 144038 sequences\n",
      "[M::main] Version: 2.17-r941\n",
      "[M::main] CMD: minimap2 -ax map-ont -t 50 26.fq 26_ont.fasta /home/gladkov2/storage/metagenome/bact_4x/new_con/26_F_wa.fq /home/gladkov2/storage/metagenome/bact_4x/new_con/26_R_wa.fq\n",
      "[M::main] Real time: 33.668 sec; CPU: 1243.424 sec; Peak RSS: 4.539 GB\n",
      "[bam_sort_core] merging from 7 files and 1 in-memory blocks...\n",
      "[M::mm_idx_gen::2.145*1.31] collected minimizers\n",
      "[M::mm_idx_gen::2.781*2.91] sorted minimizers\n",
      "[M::main::2.782*2.91] loaded/built the index for 1519 target sequence(s)\n",
      "[M::mm_mapopt_update::3.018*2.76] mid_occ = 25\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1519\n",
      "[M::mm_idx_stat::3.156*2.68] distinct minimizers: 12578691 (81.83% are singletons); average occurrences: 1.352; average spacing: 5.357\n",
      "[M::worker_pipeline::11.500*31.29] mapped 174887 sequences\n",
      "[M::worker_pipeline::16.058*36.55] mapped 177473 sequences\n",
      "[M::worker_pipeline::18.880*36.80] mapped 175827 sequences\n",
      "[M::worker_pipeline::19.886*34.99] mapped 79661 sequences\n",
      "[M::worker_pipeline::20.407*34.36] mapped 150978 sequences\n",
      "[M::worker_pipeline::20.884*33.85] mapped 150978 sequences\n",
      "[M::main] Version: 2.17-r941\n",
      "[M::main] CMD: minimap2 -ax map-ont -t 50 29.fq 29_ont.fasta /home/gladkov2/storage/metagenome/bact_4x/new_con/29_F_wa.fq /home/gladkov2/storage/metagenome/bact_4x/new_con/29_R_wa.fq\n",
      "[M::main] Real time: 20.981 sec; CPU: 707.131 sec; Peak RSS: 4.574 GB\n",
      "[bam_sort_core] merging from 4 files and 1 in-memory blocks...\n",
      "[M::mm_idx_gen::2.595*1.38] collected minimizers\n",
      "[M::mm_idx_gen::3.369*3.17] sorted minimizers\n",
      "[M::main::3.369*3.17] loaded/built the index for 1682 target sequence(s)\n",
      "[M::mm_mapopt_update::3.692*2.98] mid_occ = 30\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1682\n",
      "[M::mm_idx_stat::3.891*2.88] distinct minimizers: 14347284 (78.10% are singletons); average occurrences: 1.458; average spacing: 5.359\n",
      "[M::worker_pipeline::12.932*29.76] mapped 161990 sequences\n",
      "[M::worker_pipeline::17.104*34.42] mapped 161556 sequences\n",
      "[M::worker_pipeline::20.421*35.94] mapped 160818 sequences\n",
      "[M::worker_pipeline::24.741*38.41] mapped 161469 sequences\n",
      "[M::worker_pipeline::28.280*39.87] mapped 163187 sequences\n",
      "[M::worker_pipeline::30.945*37.78] mapped 141280 sequences\n",
      "[M::worker_pipeline::31.843*37.07] mapped 325975 sequences\n",
      "[M::worker_pipeline::32.761*36.38] mapped 325975 sequences\n",
      "[M::main] Version: 2.17-r941\n",
      "[M::main] CMD: minimap2 -ax map-ont -t 50 46.fq 46_ont.fasta /home/gladkov2/storage/metagenome/bact_4x/new_con/46_F_wa.fq /home/gladkov2/storage/metagenome/bact_4x/new_con/46_R_wa.fq\n",
      "[M::main] Real time: 32.871 sec; CPU: 1192.083 sec; Peak RSS: 4.743 GB\n",
      "[bam_sort_core] merging from 7 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "for f in d:\n",
    "    con = d[f][0]\n",
    "    mi_f = d[f][2]\n",
    "    mi_r = d[f][3]\n",
    "    ont = d[f][1]\n",
    "    !sed -n '1~4s/^@/>/p;2~4p' $ont > $f\\_ont.fasta\n",
    "    !minimap2 -ax map-ont $con  $f\\_ont.fasta $mi_f $mi_r -t 50  > $f\\_bwa.sam\n",
    "    !samtools sort $f\\_bwa.sam > $f\\_bwa_sorted.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1084633 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "29790 + 0 secondary\n",
      "27551 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "802588 + 0 mapped (74.00% : N/A)\n",
      "0 + 0 paired in sequencing\n",
      "0 + 0 read1\n",
      "0 + 0 read2\n",
      "0 + 0 properly paired (N/A : N/A)\n",
      "0 + 0 with itself and mate mapped\n",
      "0 + 0 singletons (N/A : N/A)\n",
      "0 + 0 with mate mapped to a different chr\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::2.160*1.32] collected minimizers\n",
      "[M::mm_idx_gen::2.632*2.70] sorted minimizers\n",
      "[M::main::2.633*2.70] loaded/built the index for 1903 target sequence(s)\n",
      "[M::mm_mapopt_update::2.837*2.58] mid_occ = 32\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1903\n",
      "[M::mm_idx_stat::2.940*2.53] distinct minimizers: 11752665 (78.56% are singletons); average occurrences: 1.458; average spacing: 5.359\n",
      "[M::worker_pipeline::12.510*32.57] mapped 165675 sequences\n",
      "[M::worker_pipeline::16.321*36.56] mapped 167791 sequences\n",
      "[M::worker_pipeline::20.448*39.13] mapped 168228 sequences\n",
      "[M::worker_pipeline::23.738*36.84] mapped 167010 sequences\n",
      "[M::worker_pipeline::24.031*36.40] mapped 18490 sequences\n",
      "[M::worker_pipeline::24.477*35.97] mapped 170049 sequences\n",
      "[M::worker_pipeline::24.939*35.55] mapped 170049 sequences\n",
      "[M::main] Version: 2.17-r941\n",
      "[M::main] CMD: minimap2 -ax map-ont -t 50 14.fq 14_ont.fasta /home/gladkov2/storage/metagenome/bact_4x/new_con/14_F_wa.fq /home/gladkov2/storage/metagenome/bact_4x/new_con/14_R_wa.fq\n",
      "[M::main] Real time: 25.041 sec; CPU: 886.746 sec; Peak RSS: 4.688 GB\n",
      "[bam_sort_core] merging from 4 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "minimap2 -ax map-ont 14.fq  14_ont.fasta /home/gladkov2/storage/metagenome/bact_4x/new_con/14_F_wa.fq  /home/gladkov2/storage/metagenome/bact_4x/new_con/14_R_wa.fq -t 50 > test5.sam \n",
    "samtools sort test5.sam > test5.bam\n",
    "samtools flagstat test5.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "rm test4.sam test4.bam\n",
    "bwa-mem2 mem -t 50 14.fq 14_temp_f.fq /home/gladkov2/storage/metagenome/bact_4x/new_con/14_R_wa.fq  > test4.sam\n",
    "samtools sort test4.sam > test4.bam\n",
    "samtools flagstat test4.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "14\n",
      "1084633 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "29790 + 0 secondary\n",
      "27551 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "802588 + 0 mapped (74.00% : N/A)\n",
      "0 + 0 paired in sequencing\n",
      "0 + 0 read1\n",
      "0 + 0 read2\n",
      "0 + 0 properly paired (N/A : N/A)\n",
      "0 + 0 with itself and mate mapped\n",
      "0 + 0 singletons (N/A : N/A)\n",
      "0 + 0 with mate mapped to a different chr\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "\n",
      "26\n",
      "1360549 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "27936 + 0 secondary\n",
      "31783 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "1209509 + 0 mapped (88.90% : N/A)\n",
      "0 + 0 paired in sequencing\n",
      "0 + 0 read1\n",
      "0 + 0 read2\n",
      "0 + 0 properly paired (N/A : N/A)\n",
      "0 + 0 with itself and mate mapped\n",
      "0 + 0 singletons (N/A : N/A)\n",
      "0 + 0 with mate mapped to a different chr\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "\n",
      "29\n",
      "955338 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "25870 + 0 secondary\n",
      "19664 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "740807 + 0 mapped (77.54% : N/A)\n",
      "0 + 0 paired in sequencing\n",
      "0 + 0 read1\n",
      "0 + 0 read2\n",
      "0 + 0 properly paired (N/A : N/A)\n",
      "0 + 0 with itself and mate mapped\n",
      "0 + 0 singletons (N/A : N/A)\n",
      "0 + 0 with mate mapped to a different chr\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "\n",
      "46\n",
      "1671656 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "41636 + 0 secondary\n",
      "27770 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "1400623 + 0 mapped (83.79% : N/A)\n",
      "0 + 0 paired in sequencing\n",
      "0 + 0 read1\n",
      "0 + 0 read2\n",
      "0 + 0 properly paired (N/A : N/A)\n",
      "0 + 0 with itself and mate mapped\n",
      "0 + 0 singletons (N/A : N/A)\n",
      "0 + 0 with mate mapped to a different chr\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\n"
     ]
    }
   ],
   "source": [
    "for f in d:\n",
    "    print(f'\\n{f}')\n",
    "    !samtools flagstat $f\\_bwa_sorted.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### change env to metabat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 26G\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2  89M Oct  8 12:55 14.fq\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 2.6G Oct  8 12:57 14_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 842M Oct  8 13:00 14_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 2.1G Oct  8 12:56 14_ont.fasta\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2  95M Oct  8 12:55 26.fq\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 3.8G Oct  8 13:01 26_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 1.3G Oct  8 13:06 26_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 3.0G Oct  8 13:00 26_ont.fasta\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2  88M Oct  8 12:55 29.fq\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 2.2G Oct  8 13:06 29_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 712M Oct  8 13:09 29_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 1.8G Oct  8 13:06 29_ont.fasta\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 109M Oct  8 12:55 46.fq\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 3.7G Oct  8 13:09 46_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 1.3G Oct  8 13:14 46_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 2.9G Oct  8 13:09 46_ont.fasta\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'14': ['14.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode01/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/14_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/14_R_wa.fq'],\n",
       " '26': ['26.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode02/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/26_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/26_R_wa.fq'],\n",
       " '29': ['29.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode03/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/29_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/29_R_wa.fq'],\n",
       " '46': ['46.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode04/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/46_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/46_R_wa.fq']}"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!ls -lh\n",
    "d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output depth matrix to 14_depth.txt\n",
      "Reference fasta file 14.fq\n",
      "Output matrix to 14_depth.txt\n",
      "Reading reference fasta file: 14.fq\n",
      "... 1903 sequences\n",
      "Opening 1 bams\n",
      "Consolidating headers\n",
      "Processing bam files\n",
      "Thread 0 finished: 14_bwa_sorted.bam with 1084633 reads and 169783 readsWellMapped\n",
      "Creating depth matrix file: 14_depth.txt\n",
      "Closing most bam files\n",
      "Closing last bam file\n",
      "Finished\n",
      "Output depth matrix to 26_depth.txt\n",
      "Reference fasta file 26.fq\n",
      "Output matrix to 26_depth.txt\n",
      "Reading reference fasta file: 26.fq\n",
      "... 1589 sequences\n",
      "Opening 1 bams\n",
      "Consolidating headers\n",
      "Processing bam files\n",
      "Thread 0 finished: 26_bwa_sorted.bam with 1360549 reads and 191790 readsWellMapped\n",
      "Creating depth matrix file: 26_depth.txt\n",
      "Closing most bam files\n",
      "Closing last bam file\n",
      "Finished\n",
      "Output depth matrix to 29_depth.txt\n",
      "Reference fasta file 29.fq\n",
      "Output matrix to 29_depth.txt\n",
      "Reading reference fasta file: 29.fq\n",
      "... 1519 sequences\n",
      "Opening 1 bams\n",
      "Consolidating headers\n",
      "Processing bam files\n",
      "Thread 0 finished: 29_bwa_sorted.bam with 955338 reads and 166009 readsWellMapped\n",
      "Creating depth matrix file: 29_depth.txt\n",
      "Closing most bam files\n",
      "Closing last bam file\n",
      "Finished\n",
      "Output depth matrix to 46_depth.txt\n",
      "Reference fasta file 46.fq\n",
      "Output matrix to 46_depth.txt\n",
      "Reading reference fasta file: 46.fq\n",
      "... 1682 sequences\n",
      "Opening 1 bams\n",
      "Consolidating headers\n",
      "Processing bam files\n",
      "Thread 0 finished: 46_bwa_sorted.bam with 1671656 reads and 419608 readsWellMapped\n",
      "Creating depth matrix file: 46_depth.txt\n",
      "Closing most bam files\n",
      "Closing last bam file\n",
      "Finished\n"
     ]
    }
   ],
   "source": [
    "\n",
    "for f in d:\n",
    "    !jgi_summarize_bam_contig_depths --outputDepth $f\\_depth.txt --referenceFasta $f\\.fq  $f\\_bwa_sorted.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "2 bins (4266870 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "22 bins (88398034 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "0 bins (0 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "19 bins (95396352 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "0 bins (0 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "22 bins (88789908 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "3 bins (12204608 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "32 bins (109310530 bases in total) formed.\n",
      "[2021-10-08 14:32:18] INFO: CheckM v1.1.3\n",
      "[2021-10-08 14:32:18] INFO: checkm lineage_wf -x fa -f checkm/N2_MAGs_checkm.tab -t 50 mags/ checkm/\n",
      "[2021-10-08 14:32:18] INFO: [CheckM - tree] Placing bins in reference genome tree.\n",
      "[2021-10-08 14:32:21] INFO: Identifying marker genes in 100 bins with 50 threads:\n",
      "    Finished processing 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:34:56] INFO: Saving HMM info to file.\n",
      "[2021-10-08 14:34:56] INFO: Calculating genome statistics for 100 bins with 50 threads:\n",
      "    Finished processing 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:34:57] INFO: Extracting marker genes to align.\n",
      "[2021-10-08 14:34:57] INFO: Parsing HMM hits to marker genes:\n",
      "    Finished parsing hits for 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:35:02] INFO: Extracting 43 HMMs with 50 threads:\n",
      "    Finished extracting 43 of 43 (100.00%) HMMs.\n",
      "[2021-10-08 14:35:02] INFO: Aligning 43 marker genes with 50 threads:\n",
      "    Finished aligning 43 of 43 (100.00%) marker genes.\n",
      "[2021-10-08 14:35:04] INFO: Reading marker alignment files.\n",
      "[2021-10-08 14:35:04] INFO: Concatenating alignments.\n",
      "[2021-10-08 14:35:04] INFO: Placing 100 bins into the genome tree with pplacer (be patient).\n",
      "[2021-10-08 14:45:23] INFO: { Current stage: 0:13:04.949 || Total: 0:13:04.949 }\n",
      "[2021-10-08 14:45:23] INFO: [CheckM - lineage_set] Inferring lineage-specific marker sets.\n",
      "[2021-10-08 14:45:23] INFO: Reading HMM info from file.\n",
      "[2021-10-08 14:45:23] INFO: Parsing HMM hits to marker genes:\n",
      "    Finished parsing hits for 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:45:27] INFO: Determining marker sets for each genome bin.\n",
      "    Finished processing 100 of 100 (100.00%) bins (current: 14_gen_cov.1). \n",
      "[2021-10-08 14:45:36] INFO: Marker set written to: checkm/lineage.ms\n",
      "[2021-10-08 14:45:36] INFO: { Current stage: 0:00:12.574 || Total: 0:13:17.524 }\n",
      "[2021-10-08 14:45:36] INFO: [CheckM - analyze] Identifying marker genes in bins.\n",
      "[2021-10-08 14:45:38] INFO: Identifying marker genes in 100 bins with 50 threads:\n",
      "    Finished processing 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:51:23] INFO: Saving HMM info to file.\n",
      "[2021-10-08 14:51:26] INFO: { Current stage: 0:05:50.556 || Total: 0:19:08.080 }\n",
      "[2021-10-08 14:51:26] INFO: Parsing HMM hits to marker genes:\n",
      "    Finished parsing hits for 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:51:58] INFO: Aligning marker genes with multiple hits in a single bin:\n",
      "    Finished processing 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:54:34] INFO: { Current stage: 0:03:07.884 || Total: 0:22:15.964 }\n",
      "[2021-10-08 14:54:34] INFO: Calculating genome statistics for 100 bins with 50 threads:\n",
      "    Finished processing 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:54:35] INFO: { Current stage: 0:00:01.030 || Total: 0:22:16.995 }\n",
      "[2021-10-08 14:54:35] INFO: [CheckM - qa] Tabulating genome statistics.\n",
      "[2021-10-08 14:54:35] INFO: Calculating AAI between multi-copy marker genes.\n",
      "[2021-10-08 14:54:36] INFO: Reading HMM info from file.\n",
      "[2021-10-08 14:54:38] INFO: Parsing HMM hits to marker genes:\n",
      "    Finished parsing hits for 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:55:10] INFO: QA information written to: checkm/N2_MAGs_checkm.tab\n",
      "[2021-10-08 14:55:10] INFO: { Current stage: 0:00:34.806 || Total: 0:22:51.802 }\n",
      "[2021-10-08 14:55:11] INFO: CheckM v1.1.3\n",
      "[2021-10-08 14:55:11] INFO: checkm qa ./checkm/lineage.ms ./checkm -o 2 -f res --tab_table -t 50\n",
      "[2021-10-08 14:55:11] INFO: [CheckM - qa] Tabulating genome statistics.\n",
      "[2021-10-08 14:55:11] INFO: Calculating AAI between multi-copy marker genes.\n",
      "[2021-10-08 14:55:11] INFO: Reading HMM info from file.\n",
      "[2021-10-08 14:55:13] INFO: Parsing HMM hits to marker genes:\n",
      "    Finished parsing hits for 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:55:46] INFO: QA information written to: res\n",
      "[2021-10-08 14:55:46] INFO: { Current stage: 0:00:34.753 || Total: 0:00:34.753 }\n",
      "/bin/bash: mags/*.fa: ambiguous redirect\n"
     ]
    }
   ],
   "source": [
    "for f in d:\n",
    "    !metabat2 -i $f\\.fq -o mags/$f\\_gen_cov -a  $f\\_depth.txt\n",
    "    !metabat2 -i $f\\.fq -o mags/$f\\_gen_w_cov\n",
    "!checkm lineage_wf -x fa  -f checkm/N2_MAGs_checkm.tab -t 50 mags/ checkm/\n",
    "!checkm qa ./checkm/lineage.ms ./checkm -o 2 -f res --tab_table -t 50\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/St9fMgSzcS\n",
      "[barrnap] Scanning /tmp/St9fMgSzcS for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/St9fMgSzcS'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_cov.1.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_cov.1.fa_rrna.fa' -fi '/tmp/St9fMgSzcS' -bed '/tmp/urFodpi61E'\n",
      "index file /tmp/St9fMgSzcS.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/XGFrtgtipT\n",
      "[barrnap] Scanning /tmp/XGFrtgtipT for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/XGFrtgtipT'\n",
      "[barrnap] Found: 16S_rRNA c109 L=1541/1585 2329..3869 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c1812 L=1541/1585 25672..27212 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c550 L=1454/1585 32..1485 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c1812 L=2875/3232 22250..25124 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c550 L=2875/3232 2034..4908 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c109 L=1781/3232 1..1781 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c1812 L=103/119 21997..22099 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c550 L=103/119 5059..5161 + 5S ribosomal RNA\n",
      "[barrnap] Found 8 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_cov.2.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_cov.2.fa_rrna.fa' -fi '/tmp/XGFrtgtipT' -bed '/tmp/yux0X5lbpL'\n",
      "index file /tmp/XGFrtgtipT.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/I9_aCZGnKc\n",
      "[barrnap] Scanning /tmp/I9_aCZGnKc for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/I9_aCZGnKc'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.1.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.1.fa_rrna.fa' -fi '/tmp/I9_aCZGnKc' -bed '/tmp/G9SvbS1_Rc'\n",
      "index file /tmp/I9_aCZGnKc.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/gyxBzGLHDu\n",
      "[barrnap] Scanning /tmp/gyxBzGLHDu for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/gyxBzGLHDu'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.10.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.10.fa_rrna.fa' -fi '/tmp/gyxBzGLHDu' -bed '/tmp/fPYexBX9Sb'\n",
      "index file /tmp/gyxBzGLHDu.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/k6fzDGCkz1\n",
      "[barrnap] Scanning /tmp/k6fzDGCkz1 for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/k6fzDGCkz1'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.11.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.11.fa_rrna.fa' -fi '/tmp/k6fzDGCkz1' -bed '/tmp/CZ3TyeNsAD'\n",
      "index file /tmp/k6fzDGCkz1.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/SB0b_b6lPR\n",
      "[barrnap] Scanning /tmp/SB0b_b6lPR for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/SB0b_b6lPR'\n",
      "[barrnap] Found: 16S_rRNA c38 L=1566/1585 55174..56739 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c38 L=2899/3232 57226..60124 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c38 L=111/119 60227..60337 + 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.12.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.12.fa_rrna.fa' -fi '/tmp/SB0b_b6lPR' -bed '/tmp/ep9JI_6vKv'\n",
      "index file /tmp/SB0b_b6lPR.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/4P3HduZwid\n",
      "[barrnap] Scanning /tmp/4P3HduZwid for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/4P3HduZwid'\n",
      "[barrnap] Found: 5S_rRNA c857 L=108/119 11..118 + 5S ribosomal RNA\n",
      "[barrnap] Found 1 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.13.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.13.fa_rrna.fa' -fi '/tmp/4P3HduZwid' -bed '/tmp/y1HFbeNYit'\n",
      "index file /tmp/4P3HduZwid.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/OpIhHnmCRm\n",
      "[barrnap] Scanning /tmp/OpIhHnmCRm for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/OpIhHnmCRm'\n",
      "[barrnap] Found: 16S_rRNA c44 L=1481/1585 763710..765190 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c1263 L=1483/1585 39170..40652 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c1753 L=1482/1585 3146..4627 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c349 L=1486/1585 222214..223699 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c1263 L=2840/3232 41654..44493 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c349 L=3029/3232 218519..221547 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c1753 L=2517/3232 1..2517 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c1798 L=1999/3232 31524..33522 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c44 L=1032/3232 766192..767223 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c349 L=110/119 218238..218347 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c1394 L=110/119 122059..122168 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c1909 L=110/119 350..459 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c1798 L=108/119 31328..31435 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c57 L=80/119 109187..109266 - 5S ribosomal RNA (partial)\n",
      "[barrnap] Found 14 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.14.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.14.fa_rrna.fa' -fi '/tmp/OpIhHnmCRm' -bed '/tmp/rscHX1IQ9s'\n",
      "index file /tmp/OpIhHnmCRm.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/qt3krgs7QS\n",
      "[barrnap] Scanning /tmp/qt3krgs7QS for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/qt3krgs7QS'\n",
      "[barrnap] Found: 16S_rRNA c1772 L=1481/1585 1147..2627 - 16S ribosomal RNA\n",
      "[barrnap] Rejecting short 235 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found 1 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.15.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.15.fa_rrna.fa' -fi '/tmp/qt3krgs7QS' -bed '/tmp/piqRophjLY'\n",
      "index file /tmp/qt3krgs7QS.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/ukd9vRynOl\n",
      "[barrnap] Scanning /tmp/ukd9vRynOl for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/ukd9vRynOl'\n",
      "[barrnap] Found: 16S_rRNA c1335 L=1532/1585 23622..25153 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c1335 L=2885/3232 20051..22935 - 23S ribosomal RNA\n",
      "[barrnap] Found 2 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.16.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.16.fa_rrna.fa' -fi '/tmp/ukd9vRynOl' -bed '/tmp/YESGGviwts'\n",
      "index file /tmp/ukd9vRynOl.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/FxWP7iaaPa\n",
      "[barrnap] Scanning /tmp/FxWP7iaaPa for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/FxWP7iaaPa'\n",
      "[barrnap] Found: 16S_rRNA c109 L=1541/1585 2329..3869 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c1812 L=1541/1585 25672..27212 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c550 L=1454/1585 32..1485 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c1812 L=2875/3232 22250..25124 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c550 L=2875/3232 2034..4908 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c109 L=1781/3232 1..1781 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c1812 L=103/119 21997..22099 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c550 L=103/119 5059..5161 + 5S ribosomal RNA\n",
      "[barrnap] Found 8 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.17.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.17.fa_rrna.fa' -fi '/tmp/FxWP7iaaPa' -bed '/tmp/Lv6wUhaiuP'\n",
      "index file /tmp/FxWP7iaaPa.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/gtC_l_wMFJ\n",
      "[barrnap] Scanning /tmp/gtC_l_wMFJ for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/gtC_l_wMFJ'\n",
      "[barrnap] Found: 16S_rRNA c24 L=1538/1585 1953..3490 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c88 L=1535/1585 818..2352 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c551 L=1306/1585 1..1306 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c551 L=2928/3232 1926..4853 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c88 L=1266/3232 843728..844993 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c24 L=1478/3232 5..1482 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Rejecting short 336 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 5S_rRNA c551 L=108/119 4977..5084 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c88 L=108/119 843498..843605 - 5S ribosomal RNA\n",
      "[barrnap] Found 8 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.18.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.18.fa_rrna.fa' -fi '/tmp/gtC_l_wMFJ' -bed '/tmp/CUJx9UJDIA'\n",
      "index file /tmp/gtC_l_wMFJ.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/aKDnqTCBcS\n",
      "[barrnap] Scanning /tmp/aKDnqTCBcS for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/aKDnqTCBcS'\n",
      "[barrnap] Found: 16S_rRNA c273 L=1541/1585 723391..724931 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c273 L=2863/3232 720050..722912 - 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c273 L=107/119 719805..719911 - 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.19.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.19.fa_rrna.fa' -fi '/tmp/aKDnqTCBcS' -bed '/tmp/SvuynJdKL2'\n",
      "index file /tmp/aKDnqTCBcS.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/KgkDZ3QrlP\n",
      "[barrnap] Scanning /tmp/KgkDZ3QrlP for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/KgkDZ3QrlP'\n",
      "[barrnap] Found: 16S_rRNA c3 L=1516/1585 498..2013 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c3 L=1515/1585 3258199..3259713 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c3 L=3103/3232 2457..5559 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c578 L=2502/3232 352388..354889 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c3 L=2849/3232 3260157..3263005 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c3 L=110/119 5665..5774 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c578 L=110/119 352173..352282 - 5S ribosomal RNA\n",
      "[barrnap] Found 7 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.2.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.2.fa_rrna.fa' -fi '/tmp/KgkDZ3QrlP' -bed '/tmp/6hF0yy7dSe'\n",
      "index file /tmp/KgkDZ3QrlP.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/RhL_3Qogbx\n",
      "[barrnap] Scanning /tmp/RhL_3Qogbx for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/RhL_3Qogbx'\n",
      "[barrnap] Found: 16S_rRNA c12 L=1457/1585 35785..37241 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c1433 L=1164/3232 2078..3241 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c12 L=952/3232 37946..38897 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c1433 L=109/119 3321..3429 + 5S ribosomal RNA\n",
      "[barrnap] Found 4 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.20.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.20.fa_rrna.fa' -fi '/tmp/RhL_3Qogbx' -bed '/tmp/AnQXsDRQfy'\n",
      "index file /tmp/RhL_3Qogbx.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/jDTAPwvyYN\n",
      "[barrnap] Scanning /tmp/jDTAPwvyYN for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/jDTAPwvyYN'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.21.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.21.fa_rrna.fa' -fi '/tmp/jDTAPwvyYN' -bed '/tmp/9nQLrwjqxG'\n",
      "index file /tmp/jDTAPwvyYN.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/q8zRoX5rcr\n",
      "[barrnap] Scanning /tmp/q8zRoX5rcr for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/q8zRoX5rcr'\n",
      "[barrnap] Rejecting short 124 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 5S_rRNA c1233 L=99/119 290..388 + 5S ribosomal RNA\n",
      "[barrnap] Found 1 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.22.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.22.fa_rrna.fa' -fi '/tmp/q8zRoX5rcr' -bed '/tmp/zoQ8HQP2mb'\n",
      "index file /tmp/q8zRoX5rcr.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/WAtVQWM_hj\n",
      "[barrnap] Scanning /tmp/WAtVQWM_hj for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/WAtVQWM_hj'\n",
      "[barrnap] Found: 16S_rRNA c1202 L=910/1585 1..910 - 16S ribosomal RNA (partial)\n",
      "[barrnap] Found 1 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.3.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.3.fa_rrna.fa' -fi '/tmp/WAtVQWM_hj' -bed '/tmp/0tyCKg_Q5B'\n",
      "index file /tmp/WAtVQWM_hj.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/3W2t4GpGqu\n",
      "[barrnap] Scanning /tmp/3W2t4GpGqu for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/3W2t4GpGqu'\n",
      "[barrnap] Found: 16S_rRNA c1379 L=1552/1585 16577..18128 + 16S ribosomal RNA\n",
      "[barrnap] Rejecting short 508 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found 1 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.4.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.4.fa_rrna.fa' -fi '/tmp/3W2t4GpGqu' -bed '/tmp/QurpnTHLcE'\n",
      "index file /tmp/3W2t4GpGqu.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/BFteze9YqY\n",
      "[barrnap] Scanning /tmp/BFteze9YqY for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/BFteze9YqY'\n",
      "[barrnap] Found: 16S_rRNA c15 L=1479/1585 3497..4975 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c140 L=472/1585 1..472 - 16S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c15 L=2714/3232 10..2723 - 23S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.5.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.5.fa_rrna.fa' -fi '/tmp/BFteze9YqY' -bed '/tmp/kXu89OnCQS'\n",
      "index file /tmp/BFteze9YqY.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/6ThO0MGvYr\n",
      "[barrnap] Scanning /tmp/6ThO0MGvYr for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/6ThO0MGvYr'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.6.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.6.fa_rrna.fa' -fi '/tmp/6ThO0MGvYr' -bed '/tmp/ajg9jTotVb'\n",
      "index file /tmp/6ThO0MGvYr.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/1IdDX6En7n\n",
      "[barrnap] Scanning /tmp/1IdDX6En7n for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/1IdDX6En7n'\n",
      "[barrnap] Rejecting short 587 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 5S_rRNA c1886 L=107/119 756..862 + 5S ribosomal RNA\n",
      "[barrnap] Found 1 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.7.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.7.fa_rrna.fa' -fi '/tmp/1IdDX6En7n' -bed '/tmp/lJxY1sDO2U'\n",
      "index file /tmp/1IdDX6En7n.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/qD_mIJYdwI\n",
      "[barrnap] Scanning /tmp/qD_mIJYdwI for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/qD_mIJYdwI'\n",
      "[barrnap] Found: 16S_rRNA c1203 L=1526/1585 606..2131 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c1409 L=1523/1585 3088..4610 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c1887 L=1236/1585 2..1237 + 16S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c1887 L=2874/3232 1894..4767 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c1200 L=2416/3232 51512..53927 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c1409 L=2667/3232 2..2668 - 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c1887 L=112/119 4941..5052 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c1200 L=107/119 51240..51346 - 5S ribosomal RNA\n",
      "[barrnap] Found 8 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.8.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.8.fa_rrna.fa' -fi '/tmp/qD_mIJYdwI' -bed '/tmp/pMMsBYLoKc'\n",
      "index file /tmp/qD_mIJYdwI.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/4oyzx9bZhQ\n",
      "[barrnap] Scanning /tmp/4oyzx9bZhQ for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/4oyzx9bZhQ'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/14_gen_w_cov.9.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/14_gen_w_cov.9.fa_rrna.fa' -fi '/tmp/4oyzx9bZhQ' -bed '/tmp/__ddIAeODe'\n",
      "index file /tmp/4oyzx9bZhQ.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/iUyBlCX0NO\n",
      "[barrnap] Scanning /tmp/iUyBlCX0NO for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/iUyBlCX0NO'\n",
      "[barrnap] Found: 16S_rRNA c426 L=1552/1585 1001180..1002731 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c426 L=2995/3232 1003395..1006389 + 23S ribosomal RNA\n",
      "[barrnap] Rejecting short 156 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 5S_rRNA c426 L=110/119 1006561..1006670 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c614 L=110/119 329..438 + 5S ribosomal RNA\n",
      "[barrnap] Found 4 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.1.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.1.fa_rrna.fa' -fi '/tmp/iUyBlCX0NO' -bed '/tmp/d2mfO3gghE'\n",
      "index file /tmp/iUyBlCX0NO.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/_yRbEphFIw\n",
      "[barrnap] Scanning /tmp/_yRbEphFIw for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/_yRbEphFIw'\n",
      "[barrnap] Found: 16S_rRNA c1382 L=1480/1585 225..1704 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c1382 L=2712/3232 2384..5095 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c1381 L=1092/3232 1..1092 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.10.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.10.fa_rrna.fa' -fi '/tmp/_yRbEphFIw' -bed '/tmp/vtN0fREydx'\n",
      "index file /tmp/_yRbEphFIw.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/ITbR_s6Hwd\n",
      "[barrnap] Scanning /tmp/ITbR_s6Hwd for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/ITbR_s6Hwd'\n",
      "[barrnap] Found: 16S_rRNA c121 L=1527/1585 672735..674261 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c1146 L=1526/1585 243..1768 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c121 L=2873/3232 668336..671208 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c182 L=2873/3232 470..3342 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c113 L=2526/3232 7..2532 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c113 L=111/119 2745..2855 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c121 L=111/119 668013..668123 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c182 L=111/119 3555..3665 + 5S ribosomal RNA\n",
      "[barrnap] Found 8 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.11.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.11.fa_rrna.fa' -fi '/tmp/ITbR_s6Hwd' -bed '/tmp/1YLpdQG1CH'\n",
      "index file /tmp/ITbR_s6Hwd.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/DE_hKA4h9P\n",
      "[barrnap] Scanning /tmp/DE_hKA4h9P for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/DE_hKA4h9P'\n",
      "[barrnap] Found: 16S_rRNA c558 L=1478/1585 124237..125714 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c203 L=1486/1585 123721..125206 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c558 L=2721/3232 126388..129108 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c109 L=1062/3232 1..1062 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Rejecting short 142 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 5S_rRNA c558 L=110/119 129240..129349 + 5S ribosomal RNA\n",
      "[barrnap] Found 5 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.12.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.12.fa_rrna.fa' -fi '/tmp/DE_hKA4h9P' -bed '/tmp/YT7yB_zjve'\n",
      "index file /tmp/DE_hKA4h9P.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/ryO8REVJIr\n",
      "[barrnap] Scanning /tmp/ryO8REVJIr for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/ryO8REVJIr'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.13.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.13.fa_rrna.fa' -fi '/tmp/ryO8REVJIr' -bed '/tmp/AbzMzZYZLV'\n",
      "index file /tmp/ryO8REVJIr.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/vYhJIAgGSC\n",
      "[barrnap] Scanning /tmp/vYhJIAgGSC for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/vYhJIAgGSC'\n",
      "[barrnap] Found: 16S_rRNA c18 L=1515/1585 728764..730278 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c18 L=1516/1585 148596..150111 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c18 L=3103/3232 150555..153657 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c18 L=3103/3232 730722..733824 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c18 L=110/119 153763..153872 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c18 L=110/119 733930..734039 + 5S ribosomal RNA\n",
      "[barrnap] Found 6 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.14.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.14.fa_rrna.fa' -fi '/tmp/vYhJIAgGSC' -bed '/tmp/r0hmwpxQ0S'\n",
      "index file /tmp/vYhJIAgGSC.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/6Mv4qEKUnC\n",
      "[barrnap] Scanning /tmp/6Mv4qEKUnC for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/6Mv4qEKUnC'\n",
      "[barrnap] Found: 16S_rRNA c71 L=1572/1585 1610506..1612077 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c71 L=2892/3232 1612556..1615447 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c71 L=111/119 1615547..1615657 + 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.15.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.15.fa_rrna.fa' -fi '/tmp/6Mv4qEKUnC' -bed '/tmp/NmRH2yqzZu'\n",
      "index file /tmp/6Mv4qEKUnC.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/AuUhc0LK2V\n",
      "[barrnap] Scanning /tmp/AuUhc0LK2V for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/AuUhc0LK2V'\n",
      "[barrnap] Found: 16S_rRNA c1465 L=1542/1585 3184..4725 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c1465 L=2566/3232 1..2566 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Rejecting short 324 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 5S_rRNA c358 L=98/119 16255..16352 - 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.16.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.16.fa_rrna.fa' -fi '/tmp/AuUhc0LK2V' -bed '/tmp/UeiV6FcS8I'\n",
      "index file /tmp/AuUhc0LK2V.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/MJiu2unIQ4\n",
      "[barrnap] Scanning /tmp/MJiu2unIQ4 for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/MJiu2unIQ4'\n",
      "[barrnap] Found: 23S_rRNA c106 L=2884/3232 136692..139575 - 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c80 L=106/119 252..357 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c106 L=108/119 136461..136568 - 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.17.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.17.fa_rrna.fa' -fi '/tmp/MJiu2unIQ4' -bed '/tmp/bMiyRdyI1M'\n",
      "index file /tmp/MJiu2unIQ4.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/0dajl91BQm\n",
      "[barrnap] Scanning /tmp/0dajl91BQm for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/0dajl91BQm'\n",
      "[barrnap] Found: 16S_rRNA s27 L=1460/1585 57559..59018 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA s27 L=2876/3232 59724..62599 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c114 L=2780/3232 572..3351 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c114 L=110/119 3471..3580 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA s27 L=109/119 62699..62807 + 5S ribosomal RNA\n",
      "[barrnap] Found 5 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.18.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.18.fa_rrna.fa' -fi '/tmp/0dajl91BQm' -bed '/tmp/HgSphMwqJj'\n",
      "index file /tmp/0dajl91BQm.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/FNrG4hV80r\n",
      "[barrnap] Scanning /tmp/FNrG4hV80r for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/FNrG4hV80r'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.19.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.19.fa_rrna.fa' -fi '/tmp/FNrG4hV80r' -bed '/tmp/WXZVntEOYH'\n",
      "index file /tmp/FNrG4hV80r.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/9DI7GsPOKi\n",
      "[barrnap] Scanning /tmp/9DI7GsPOKi for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/9DI7GsPOKi'\n",
      "[barrnap] Found: 16S_rRNA c697 L=1485/1585 303..1787 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c1383 L=2396/3232 18..2413 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c110 L=106/119 141..246 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c1383 L=106/119 2498..2603 + 5S ribosomal RNA\n",
      "[barrnap] Found 4 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.2.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.2.fa_rrna.fa' -fi '/tmp/9DI7GsPOKi' -bed '/tmp/Rihkr80rdx'\n",
      "index file /tmp/9DI7GsPOKi.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/J74iIvU8te\n",
      "[barrnap] Scanning /tmp/J74iIvU8te for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/J74iIvU8te'\n",
      "[barrnap] Found: 16S_rRNA c56 L=1481/1585 891665..893145 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c29 L=1241/1585 746454..747694 - 16S ribosomal RNA (partial)\n",
      "[barrnap] Rejecting short 361 nt predicted 16S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 23S_rRNA c56 L=2786/3232 887894..890679 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c29 L=2786/3232 742633..745418 - 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c29 L=110/119 742295..742404 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c56 L=110/119 887556..887665 - 5S ribosomal RNA\n",
      "[barrnap] Found 6 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.3.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.3.fa_rrna.fa' -fi '/tmp/J74iIvU8te' -bed '/tmp/mWIB3Ykbji'\n",
      "index file /tmp/J74iIvU8te.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/9ZRelPJqPi\n",
      "[barrnap] Scanning /tmp/9ZRelPJqPi for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/9ZRelPJqPi'\n",
      "[barrnap] Found: 16S_rRNA c263 L=1025/1585 204213..205237 - 16S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c263 L=2721/3232 200836..203556 - 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c263 L=110/119 200627..200736 - 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.4.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.4.fa_rrna.fa' -fi '/tmp/9ZRelPJqPi' -bed '/tmp/8sgCxk4LlR'\n",
      "index file /tmp/9ZRelPJqPi.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/Wn2hu9HUzK\n",
      "[barrnap] Scanning /tmp/Wn2hu9HUzK for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/Wn2hu9HUzK'\n",
      "[barrnap] Found: 16S_rRNA c522 L=1512/1585 105278..106789 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c522 L=2866/3232 101981..104846 - 23S ribosomal RNA\n",
      "[barrnap] Rejecting short 312 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 5S_rRNA c522 L=105/119 101737..101841 - 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.5.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.5.fa_rrna.fa' -fi '/tmp/Wn2hu9HUzK' -bed '/tmp/6wbacG9lOr'\n",
      "index file /tmp/Wn2hu9HUzK.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/3sO4SHfqmB\n",
      "[barrnap] Scanning /tmp/3sO4SHfqmB for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/3sO4SHfqmB'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.6.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.6.fa_rrna.fa' -fi '/tmp/3sO4SHfqmB' -bed '/tmp/P74VrlGZxt'\n",
      "index file /tmp/3sO4SHfqmB.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/ZWTItXD9ID\n",
      "[barrnap] Scanning /tmp/ZWTItXD9ID for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/ZWTItXD9ID'\n",
      "[barrnap] Found: 16S_rRNA c1259 L=1543/1585 3282..4824 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c77 L=1541/1585 3352..4892 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c77 L=2853/3232 2..2854 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c1259 L=2804/3232 3..2806 - 23S ribosomal RNA\n",
      "[barrnap] Rejecting short 96 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 5S_rRNA c81 L=108/119 175..282 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c1369 L=105/119 242..346 + 5S ribosomal RNA\n",
      "[barrnap] Found 6 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.7.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.7.fa_rrna.fa' -fi '/tmp/ZWTItXD9ID' -bed '/tmp/imi5hOGW9h'\n",
      "index file /tmp/ZWTItXD9ID.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/cb0nfd_K5h\n",
      "[barrnap] Scanning /tmp/cb0nfd_K5h for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/cb0nfd_K5h'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.8.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.8.fa_rrna.fa' -fi '/tmp/cb0nfd_K5h' -bed '/tmp/XMOZXxRd2G'\n",
      "index file /tmp/cb0nfd_K5h.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/QZpOklshzI\n",
      "[barrnap] Scanning /tmp/QZpOklshzI for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/QZpOklshzI'\n",
      "[barrnap] Found: 16S_rRNA c43 L=1525/1585 2083387..2084911 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c82 L=1525/1585 952520..954044 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c83 L=1525/1585 3341..4865 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c1297 L=1526/1585 1965..3490 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c83 L=2736/3232 1..2736 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c82 L=2440/3232 954649..957088 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c1297 L=1261/3232 2..1262 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c43 L=1400/3232 2085516..2086915 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Rejecting short 231 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Rejecting short 228 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 5S_rRNA c269 L=107/119 359..465 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c82 L=107/119 363..469 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c450 L=108/119 208..315 + 5S ribosomal RNA\n",
      "[barrnap] Found 11 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/26_gen_w_cov.9.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/26_gen_w_cov.9.fa_rrna.fa' -fi '/tmp/QZpOklshzI' -bed '/tmp/wqGmuj7FE7'\n",
      "index file /tmp/QZpOklshzI.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/5SwXjccF7W\n",
      "[barrnap] Scanning /tmp/5SwXjccF7W for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/5SwXjccF7W'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.1.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.1.fa_rrna.fa' -fi '/tmp/5SwXjccF7W' -bed '/tmp/2eeONP1jCO'\n",
      "index file /tmp/5SwXjccF7W.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/ymNecr8Y5m\n",
      "[barrnap] Scanning /tmp/ymNecr8Y5m for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/ymNecr8Y5m'\n",
      "[barrnap] Found: 16S_rRNA c33 L=1456/1585 1869464..1870919 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c33 L=1487/1585 47569..49055 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c33 L=2780/3232 49848..52627 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c33 L=2853/3232 1865826..1868678 - 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c33 L=110/119 52704..52813 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c33 L=110/119 1865640..1865749 - 5S ribosomal RNA\n",
      "[barrnap] Found 6 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.10.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.10.fa_rrna.fa' -fi '/tmp/ymNecr8Y5m' -bed '/tmp/5u17ORqGEP'\n",
      "index file /tmp/ymNecr8Y5m.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/yQB5Tj9sxS\n",
      "[barrnap] Scanning /tmp/yQB5Tj9sxS for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/yQB5Tj9sxS'\n",
      "[barrnap] Found: 16S_rRNA c636 L=1485/1585 3627..5111 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c636 L=2801/3232 194..2994 - 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c636 L=108/119 4..111 - 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.11.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.11.fa_rrna.fa' -fi '/tmp/yQB5Tj9sxS' -bed '/tmp/2QrKxllIma'\n",
      "index file /tmp/yQB5Tj9sxS.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/7kMEJcw_hX\n",
      "[barrnap] Scanning /tmp/7kMEJcw_hX for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/7kMEJcw_hX'\n",
      "[barrnap] Found: 16S_rRNA c96 L=1526/1585 59134..60659 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c96 L=3109/3232 55558..58666 - 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c96 L=110/119 55331..55440 - 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.12.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.12.fa_rrna.fa' -fi '/tmp/7kMEJcw_hX' -bed '/tmp/tk6GjulOQm'\n",
      "index file /tmp/7kMEJcw_hX.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/MVxYCcBwkP\n",
      "[barrnap] Scanning /tmp/MVxYCcBwkP for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/MVxYCcBwkP'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.13.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.13.fa_rrna.fa' -fi '/tmp/MVxYCcBwkP' -bed '/tmp/eiDU_wg319'\n",
      "index file /tmp/MVxYCcBwkP.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/NSDBo9DfV1\n",
      "[barrnap] Scanning /tmp/NSDBo9DfV1 for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/NSDBo9DfV1'\n",
      "[barrnap] Found: 23S_rRNA c755 L=2903/3232 121047..123949 + 23S ribosomal RNA\n",
      "[barrnap] Found 1 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.14.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.14.fa_rrna.fa' -fi '/tmp/NSDBo9DfV1' -bed '/tmp/Z0U48cCAyp'\n",
      "index file /tmp/NSDBo9DfV1.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/EYRvJYFFsm\n",
      "[barrnap] Scanning /tmp/EYRvJYFFsm for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/EYRvJYFFsm'\n",
      "[barrnap] Found: 16S_rRNA c250 L=1479/1585 3182..4660 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c197 L=1486/1585 69201..70686 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c197 L=2721/3232 65786..68506 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c250 L=2487/3232 2..2488 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c197 L=109/119 65566..65674 - 5S ribosomal RNA\n",
      "[barrnap] Found 5 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.15.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.15.fa_rrna.fa' -fi '/tmp/EYRvJYFFsm' -bed '/tmp/4_unqQHNtW'\n",
      "index file /tmp/EYRvJYFFsm.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/pwqeW3OsdZ\n",
      "[barrnap] Scanning /tmp/pwqeW3OsdZ for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/pwqeW3OsdZ'\n",
      "[barrnap] Found: 16S_rRNA c851 L=1560/1585 2788..4347 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c851 L=2896/3232 4845..7740 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c851 L=111/119 7950..8060 + 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.16.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.16.fa_rrna.fa' -fi '/tmp/pwqeW3OsdZ' -bed '/tmp/_VCkhewphb'\n",
      "index file /tmp/pwqeW3OsdZ.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/43AoSxhbX8\n",
      "[barrnap] Scanning /tmp/43AoSxhbX8 for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/43AoSxhbX8'\n",
      "[barrnap] Found: 16S_rRNA c894 L=1527/1585 231..1757 - 16S ribosomal RNA\n",
      "[barrnap] Found 1 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.17.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.17.fa_rrna.fa' -fi '/tmp/43AoSxhbX8' -bed '/tmp/Mp55WTYLnj'\n",
      "index file /tmp/43AoSxhbX8.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/h8LcUD6FwG\n",
      "[barrnap] Scanning /tmp/h8LcUD6FwG for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/h8LcUD6FwG'\n",
      "[barrnap] Found: 16S_rRNA c49 L=1516/1585 31015..32530 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c49 L=3109/3232 32922..36030 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c49 L=111/119 36167..36277 + 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.18.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.18.fa_rrna.fa' -fi '/tmp/h8LcUD6FwG' -bed '/tmp/6a2VFBJjz5'\n",
      "index file /tmp/h8LcUD6FwG.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/YIuACjXKVy\n",
      "[barrnap] Scanning /tmp/YIuACjXKVy for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/YIuACjXKVy'\n",
      "[barrnap] Found: 16S_rRNA c97 L=1521/1585 40..1560 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c97 L=3094/3232 2534809..2537902 - 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c97 L=108/119 2534614..2534721 - 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.19.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.19.fa_rrna.fa' -fi '/tmp/YIuACjXKVy' -bed '/tmp/my8BB0S4D4'\n",
      "index file /tmp/YIuACjXKVy.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/qQnwKoNFRA\n",
      "[barrnap] Scanning /tmp/qQnwKoNFRA for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/qQnwKoNFRA'\n",
      "[barrnap] Found: 16S_rRNA c759 L=1557/1585 26103..27659 + 16S ribosomal RNA\n",
      "[barrnap] Rejecting short 245 nt predicted 16S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 23S_rRNA c759 L=2935/3232 28199..31133 + 23S ribosomal RNA\n",
      "[barrnap] Found 2 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.2.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.2.fa_rrna.fa' -fi '/tmp/qQnwKoNFRA' -bed '/tmp/hFz2BD6PyP'\n",
      "index file /tmp/qQnwKoNFRA.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/Nrn1HdWJFl\n",
      "[barrnap] Scanning /tmp/Nrn1HdWJFl for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/Nrn1HdWJFl'\n",
      "[barrnap] Found: 16S_rRNA c66 L=1539/1585 2414..3952 - 16S ribosomal RNA\n",
      "[barrnap] Rejecting short 205 nt predicted 16S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 23S_rRNA c66 L=2882/3232 854188..857069 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c105 L=2889/3232 2108732..2111620 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c66 L=1946/3232 1..1946 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c66 L=108/119 853958..854065 - 5S ribosomal RNA\n",
      "[barrnap] Found 5 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.20.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.20.fa_rrna.fa' -fi '/tmp/Nrn1HdWJFl' -bed '/tmp/mUk4weDxqj'\n",
      "index file /tmp/Nrn1HdWJFl.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/ZnsuHrijQ7\n",
      "[barrnap] Scanning /tmp/ZnsuHrijQ7 for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/ZnsuHrijQ7'\n",
      "[barrnap] Found: 16S_rRNA c802 L=1481/1585 408076..409556 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c254 L=1481/1585 1790..3270 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c637 L=2816/3232 805..3620 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c802 L=2724/3232 410259..412982 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c254 L=1104/3232 1..1104 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c637 L=111/119 3704..3814 + 5S ribosomal RNA\n",
      "[barrnap] Found 6 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.21.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.21.fa_rrna.fa' -fi '/tmp/ZnsuHrijQ7' -bed '/tmp/vHOTDh1Tws'\n",
      "index file /tmp/ZnsuHrijQ7.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/EDAeJ9afKe\n",
      "[barrnap] Scanning /tmp/EDAeJ9afKe for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/EDAeJ9afKe'\n",
      "[barrnap] Found: 16S_rRNA c769 L=1482/1585 3230..4711 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c768 L=1484/1585 43600..45083 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c768 L=2789/3232 46123..48911 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c769 L=2188/3232 2..2189 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found 4 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.22.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.22.fa_rrna.fa' -fi '/tmp/EDAeJ9afKe' -bed '/tmp/CgGxEymABn'\n",
      "index file /tmp/EDAeJ9afKe.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/O7rPl4ZCMp\n",
      "[barrnap] Scanning /tmp/O7rPl4ZCMp for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/O7rPl4ZCMp'\n",
      "[barrnap] Found: 16S_rRNA c77 L=1541/1585 3481..5021 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c77 L=2876/3232 5519..8394 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c77 L=2844/3232 3..2846 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c77 L=105/119 2980..3084 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c77 L=105/119 8528..8632 + 5S ribosomal RNA\n",
      "[barrnap] Found 5 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.3.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.3.fa_rrna.fa' -fi '/tmp/O7rPl4ZCMp' -bed '/tmp/bQvClPu0PQ'\n",
      "index file /tmp/O7rPl4ZCMp.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/6wm3OEfhHV\n",
      "[barrnap] Scanning /tmp/6wm3OEfhHV for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/6wm3OEfhHV'\n",
      "[barrnap] Found: 16S_rRNA s893 L=1522/1585 2233..3754 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c414 L=1522/1585 4306..5827 - 16S ribosomal RNA\n",
      "[barrnap] Rejecting short 156 nt predicted 16S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 23S_rRNA c414 L=2873/3232 855..3727 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA s70 L=2873/3232 1225508..1228380 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA s893 L=1653/3232 2..1654 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c414 L=107/119 638..744 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA s70 L=107/119 1225291..1225397 - 5S ribosomal RNA\n",
      "[barrnap] Found 7 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.4.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.4.fa_rrna.fa' -fi '/tmp/6wm3OEfhHV' -bed '/tmp/Oegmqxejzv'\n",
      "index file /tmp/6wm3OEfhHV.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/04ew6P3Mck\n",
      "[barrnap] Scanning /tmp/04ew6P3Mck for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/04ew6P3Mck'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.5.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.5.fa_rrna.fa' -fi '/tmp/04ew6P3Mck' -bed '/tmp/xZOSCnYPU2'\n",
      "index file /tmp/04ew6P3Mck.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/JPzokVnfuK\n",
      "[barrnap] Scanning /tmp/JPzokVnfuK for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/JPzokVnfuK'\n",
      "[barrnap] Found: 16S_rRNA c635 L=1488/1585 4626..6113 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c635 L=2814/3232 978..3791 - 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c635 L=110/119 609..718 - 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.6.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.6.fa_rrna.fa' -fi '/tmp/JPzokVnfuK' -bed '/tmp/LCOIm396DC'\n",
      "index file /tmp/JPzokVnfuK.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/MIr2YLNsuf\n",
      "[barrnap] Scanning /tmp/MIr2YLNsuf for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/MIr2YLNsuf'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.7.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.7.fa_rrna.fa' -fi '/tmp/MIr2YLNsuf' -bed '/tmp/tOidk2pVBm'\n",
      "index file /tmp/MIr2YLNsuf.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/ihhBcbJ7Qm\n",
      "[barrnap] Scanning /tmp/ihhBcbJ7Qm for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/ihhBcbJ7Qm'\n",
      "[barrnap] Found: 16S_rRNA c410 L=1490/1585 20122..21611 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c410 L=2786/3232 22328..25113 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c410 L=110/119 25235..25344 + 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.8.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.8.fa_rrna.fa' -fi '/tmp/ihhBcbJ7Qm' -bed '/tmp/ewu5KAMlnP'\n",
      "index file /tmp/ihhBcbJ7Qm.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/dG4DMHCvOM\n",
      "[barrnap] Scanning /tmp/dG4DMHCvOM for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/dG4DMHCvOM'\n",
      "[barrnap] Found: 16S_rRNA c99 L=1526/1585 49355..50880 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c99 L=3100/3232 51242..54341 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c99 L=109/119 54477..54585 + 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/29_gen_w_cov.9.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/29_gen_w_cov.9.fa_rrna.fa' -fi '/tmp/dG4DMHCvOM' -bed '/tmp/L1uAW15awv'\n",
      "index file /tmp/dG4DMHCvOM.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/pFDp9dNIIX\n",
      "[barrnap] Scanning /tmp/pFDp9dNIIX for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/pFDp9dNIIX'\n",
      "[barrnap] Found: 16S_rRNA c12 L=1515/1585 2305542..2307056 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c12 L=1516/1585 1716909..1718424 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c12 L=3103/3232 1718850..1721952 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c12 L=3103/3232 2307482..2310584 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c12 L=110/119 1722047..1722156 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c12 L=110/119 2310678..2310787 + 5S ribosomal RNA\n",
      "[barrnap] Found 6 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_cov.1.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_cov.1.fa_rrna.fa' -fi '/tmp/pFDp9dNIIX' -bed '/tmp/k2DvuyE7EA'\n",
      "index file /tmp/pFDp9dNIIX.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/UiKH6eOAtt\n",
      "[barrnap] Scanning /tmp/UiKH6eOAtt for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/UiKH6eOAtt'\n",
      "[barrnap] Found: 16S_rRNA c85 L=907/1585 3..909 + 16S ribosomal RNA (partial)\n",
      "[barrnap] Rejecting short 173 nt predicted 16S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Rejecting short 184 nt predicted 16S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 23S_rRNA c85 L=2883/3232 1494..4376 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c150 L=2885/3232 1845842..1848726 - 23S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_cov.2.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_cov.2.fa_rrna.fa' -fi '/tmp/UiKH6eOAtt' -bed '/tmp/hH5_wqy5EE'\n",
      "index file /tmp/UiKH6eOAtt.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/ybPCsGzf4B\n",
      "[barrnap] Scanning /tmp/ybPCsGzf4B for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/ybPCsGzf4B'\n",
      "[barrnap] Found: 16S_rRNA c117 L=1479/1585 146137..147615 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c703 L=1476/1585 11..1486 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c117 L=2710/3232 142659..145368 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c703 L=2712/3232 2255..4966 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c117 L=91/119 142454..142544 - 5S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c703 L=91/119 5081..5171 + 5S ribosomal RNA (partial)\n",
      "[barrnap] Found 6 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_cov.3.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_cov.3.fa_rrna.fa' -fi '/tmp/ybPCsGzf4B' -bed '/tmp/KgwNYTpXVT'\n",
      "index file /tmp/ybPCsGzf4B.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/bGTStLxYs2\n",
      "[barrnap] Scanning /tmp/bGTStLxYs2 for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/bGTStLxYs2'\n",
      "[barrnap] Found: 16S_rRNA c71 L=1376/1585 3..1378 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c71 L=2727/3232 2035..4761 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c587 L=2727/3232 250524..253250 - 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c71 L=110/119 4851..4960 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c587 L=111/119 250324..250434 - 5S ribosomal RNA\n",
      "[barrnap] Found 5 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.1.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.1.fa_rrna.fa' -fi '/tmp/bGTStLxYs2' -bed '/tmp/DrWi9baKde'\n",
      "index file /tmp/bGTStLxYs2.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/43imO1dzqK\n",
      "[barrnap] Scanning /tmp/43imO1dzqK for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/43imO1dzqK'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.10.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.10.fa_rrna.fa' -fi '/tmp/43imO1dzqK' -bed '/tmp/MXVoaF66Gl'\n",
      "index file /tmp/43imO1dzqK.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/9A0vo6hDn0\n",
      "[barrnap] Scanning /tmp/9A0vo6hDn0 for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/9A0vo6hDn0'\n",
      "[barrnap] Found: 23S_rRNA c552 L=1294/3232 72..1365 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c552 L=85/119 1470..1554 + 5S ribosomal RNA (partial)\n",
      "[barrnap] Found 2 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.11.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.11.fa_rrna.fa' -fi '/tmp/9A0vo6hDn0' -bed '/tmp/EEMuxnaZ1c'\n",
      "index file /tmp/9A0vo6hDn0.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/EVxDmWADBB\n",
      "[barrnap] Scanning /tmp/EVxDmWADBB for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/EVxDmWADBB'\n",
      "[barrnap] Found: 16S_rRNA c790 L=668/1585 20452..21119 - 16S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c790 L=2940/3232 16642..19581 - 23S ribosomal RNA\n",
      "[barrnap] Found 2 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.12.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.12.fa_rrna.fa' -fi '/tmp/EVxDmWADBB' -bed '/tmp/tim3MVlhoM'\n",
      "index file /tmp/EVxDmWADBB.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/VcxvdyPEtu\n",
      "[barrnap] Scanning /tmp/VcxvdyPEtu for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/VcxvdyPEtu'\n",
      "[barrnap] Found: 16S_rRNA c981 L=1554/1585 96526..98079 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c1244 L=1555/1585 3023..4577 - 16S ribosomal RNA\n",
      "[barrnap] Rejecting short 68 nt predicted 16S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 23S_rRNA c1244 L=2807/3232 1..2807 - 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c252 L=112/119 140..251 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c981 L=112/119 98265..98376 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c249 L=113/119 131..243 + 5S ribosomal RNA\n",
      "[barrnap] Found 6 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.13.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.13.fa_rrna.fa' -fi '/tmp/VcxvdyPEtu' -bed '/tmp/dH6G8A7b34'\n",
      "index file /tmp/VcxvdyPEtu.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/EOqylKhlO4\n",
      "[barrnap] Scanning /tmp/EOqylKhlO4 for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/EOqylKhlO4'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.14.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.14.fa_rrna.fa' -fi '/tmp/EOqylKhlO4' -bed '/tmp/gHc0G_jfSt'\n",
      "index file /tmp/EOqylKhlO4.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/UjUoLCUtZj\n",
      "[barrnap] Scanning /tmp/UjUoLCUtZj for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/UjUoLCUtZj'\n",
      "[barrnap] Found: 16S_rRNA c85 L=907/1585 3..909 + 16S ribosomal RNA (partial)\n",
      "[barrnap] Rejecting short 173 nt predicted 16S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Rejecting short 184 nt predicted 16S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 23S_rRNA c85 L=2883/3232 1494..4376 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c150 L=2885/3232 1845842..1848726 - 23S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.15.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.15.fa_rrna.fa' -fi '/tmp/UjUoLCUtZj' -bed '/tmp/A_fmO_Vxbv'\n",
      "index file /tmp/UjUoLCUtZj.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/a_fqUSpTfg\n",
      "[barrnap] Scanning /tmp/a_fqUSpTfg for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/a_fqUSpTfg'\n",
      "[barrnap] Found: 16S_rRNA c92 L=1558/1585 3225..4782 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c238 L=1543/1585 2042..3584 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c333 L=1543/1585 2913..4455 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c333 L=1545/1585 534966..536510 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c239 L=1545/1585 1760..3304 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c236 L=1544/1585 1924..3467 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c92 L=2914/3232 99..3012 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c333 L=2706/3232 2..2707 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c333 L=2069/3232 537092..539160 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c238 L=1837/3232 2..1838 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c236 L=1717/3232 2..1718 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c239 L=1308/3232 1..1308 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c239 L=112/119 1435..1546 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c333 L=112/119 536844..536955 + 5S ribosomal RNA\n",
      "[barrnap] Found 14 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.16.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.16.fa_rrna.fa' -fi '/tmp/a_fqUSpTfg' -bed '/tmp/EuCCdljx7G'\n",
      "index file /tmp/a_fqUSpTfg.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/3xJCtXRTSi\n",
      "[barrnap] Scanning /tmp/3xJCtXRTSi for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/3xJCtXRTSi'\n",
      "[barrnap] Found: 16S_rRNA c31 L=1541/1585 2189482..2191022 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c31 L=2877/3232 2186113..2188989 - 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c31 L=103/119 2185878..2185980 - 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.17.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.17.fa_rrna.fa' -fi '/tmp/3xJCtXRTSi' -bed '/tmp/cx4jnOc_A6'\n",
      "index file /tmp/3xJCtXRTSi.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/5sySXAomTK\n",
      "[barrnap] Scanning /tmp/5sySXAomTK for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/5sySXAomTK'\n",
      "[barrnap] Found: 5S_rRNA c1039 L=110/119 29479..29588 - 5S ribosomal RNA\n",
      "[barrnap] Found 1 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.18.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.18.fa_rrna.fa' -fi '/tmp/5sySXAomTK' -bed '/tmp/4N51ZmpgTi'\n",
      "index file /tmp/5sySXAomTK.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/hVbbcCQiaF\n",
      "[barrnap] Scanning /tmp/hVbbcCQiaF for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/hVbbcCQiaF'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.19.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.19.fa_rrna.fa' -fi '/tmp/hVbbcCQiaF' -bed '/tmp/iNPcLjx08D'\n",
      "index file /tmp/hVbbcCQiaF.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/juN4yCw8ez\n",
      "[barrnap] Scanning /tmp/juN4yCw8ez for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/juN4yCw8ez'\n",
      "[barrnap] Found: 23S_rRNA c1300 L=2880/3232 14411..17290 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c1542 L=2155/3232 1..2155 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c1300 L=108/119 14100..14207 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c1542 L=108/119 2357..2464 + 5S ribosomal RNA\n",
      "[barrnap] Found 4 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.2.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.2.fa_rrna.fa' -fi '/tmp/juN4yCw8ez' -bed '/tmp/BDG9G8Iibq'\n",
      "index file /tmp/juN4yCw8ez.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/wnJgkCnXDP\n",
      "[barrnap] Scanning /tmp/wnJgkCnXDP for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/wnJgkCnXDP'\n",
      "[barrnap] Found: 16S_rRNA c234 L=1541/1585 1105..2645 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c235 L=1541/1585 1341..2881 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c197 L=1541/1585 128755..130295 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c3 L=1546/1585 389647..391192 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c197 L=1681/3232 130580..132260 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Rejecting short 733 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Rejecting short 469 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Rejecting short 265 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 5S_rRNA c234 L=112/119 900..1011 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c235 L=90/119 1150..1239 - 5S ribosomal RNA (partial)\n",
      "[barrnap] Found 7 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.20.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.20.fa_rrna.fa' -fi '/tmp/wnJgkCnXDP' -bed '/tmp/T3LuWnnQnz'\n",
      "index file /tmp/wnJgkCnXDP.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/fODOneXIWK\n",
      "[barrnap] Scanning /tmp/fODOneXIWK for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/fODOneXIWK'\n",
      "[barrnap] Found: 16S_rRNA c615 L=1512/1585 532..2043 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c1625 L=1511/1585 22610..24120 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c614 L=1283/1585 3..1285 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c1625 L=2869/3232 18950..21818 - 23S ribosomal RNA\n",
      "[barrnap] Found 4 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.21.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.21.fa_rrna.fa' -fi '/tmp/fODOneXIWK' -bed '/tmp/CJ3Pi7utSy'\n",
      "index file /tmp/fODOneXIWK.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/NX7uDC3LDY\n",
      "[barrnap] Scanning /tmp/NX7uDC3LDY for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/NX7uDC3LDY'\n",
      "[barrnap] Found: 16S_rRNA c578 L=1635/1585 46515..48149 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c578 L=3044/3232 48994..52037 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c578 L=109/119 52323..52431 + 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.22.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.22.fa_rrna.fa' -fi '/tmp/NX7uDC3LDY' -bed '/tmp/uRUpUHf9qq'\n",
      "index file /tmp/NX7uDC3LDY.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/qxO4_nZqNn\n",
      "[barrnap] Scanning /tmp/qxO4_nZqNn for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/qxO4_nZqNn'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.23.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.23.fa_rrna.fa' -fi '/tmp/qxO4_nZqNn' -bed '/tmp/Qjw_11tj6Y'\n",
      "index file /tmp/qxO4_nZqNn.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/ndykQfXtv6\n",
      "[barrnap] Scanning /tmp/ndykQfXtv6 for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/ndykQfXtv6'\n",
      "[barrnap] Found: 16S_rRNA c73 L=1541/1585 483109..484649 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c60 L=1480/1585 429..1908 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c1554 L=1484/1585 1018..2501 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c73 L=2821/3232 484932..487752 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c60 L=2724/3232 2580..5303 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c14 L=1377/3232 110914..112290 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c551 L=1296/3232 85..1380 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Rejecting short 366 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 5S_rRNA c14 L=110/119 110592..110701 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c551 L=107/119 1467..1573 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c60 L=110/119 5439..5548 + 5S ribosomal RNA\n",
      "[barrnap] Found 10 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.24.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.24.fa_rrna.fa' -fi '/tmp/ndykQfXtv6' -bed '/tmp/T54Wfq4tcF'\n",
      "index file /tmp/ndykQfXtv6.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/lm0B6l64XZ\n",
      "[barrnap] Scanning /tmp/lm0B6l64XZ for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/lm0B6l64XZ'\n",
      "[barrnap] Found: 16S_rRNA c117 L=1479/1585 146137..147615 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c703 L=1476/1585 11..1486 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c117 L=2710/3232 142659..145368 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c703 L=2712/3232 2255..4966 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c117 L=91/119 142454..142544 - 5S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c703 L=91/119 5081..5171 + 5S ribosomal RNA (partial)\n",
      "[barrnap] Found 6 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.25.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.25.fa_rrna.fa' -fi '/tmp/lm0B6l64XZ' -bed '/tmp/hdhVd67qWo'\n",
      "index file /tmp/lm0B6l64XZ.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/9ONVflqysI\n",
      "[barrnap] Scanning /tmp/9ONVflqysI for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/9ONVflqysI'\n",
      "[barrnap] Found: 16S_rRNA c84 L=1515/1585 3587707..3589221 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c84 L=3087/3232 3589552..3592638 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c84 L=111/119 3592733..3592843 + 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.26.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.26.fa_rrna.fa' -fi '/tmp/9ONVflqysI' -bed '/tmp/ODcRT9uJYF'\n",
      "index file /tmp/9ONVflqysI.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/TUXn_43B3D\n",
      "[barrnap] Scanning /tmp/TUXn_43B3D for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/TUXn_43B3D'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.27.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.27.fa_rrna.fa' -fi '/tmp/TUXn_43B3D' -bed '/tmp/UeQMaxCbjS'\n",
      "index file /tmp/TUXn_43B3D.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/mIXykrtgl5\n",
      "[barrnap] Scanning /tmp/mIXykrtgl5 for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/mIXykrtgl5'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.28.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.28.fa_rrna.fa' -fi '/tmp/mIXykrtgl5' -bed '/tmp/Od1e_izHvW'\n",
      "index file /tmp/mIXykrtgl5.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/gGw7XBFXfk\n",
      "[barrnap] Scanning /tmp/gGw7XBFXfk for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/gGw7XBFXfk'\n",
      "[barrnap] Found: 16S_rRNA c1056 L=1452/1585 14300..15751 + 16S ribosomal RNA\n",
      "[barrnap] Found 1 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.29.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.29.fa_rrna.fa' -fi '/tmp/gGw7XBFXfk' -bed '/tmp/wKxDet0Ogb'\n",
      "index file /tmp/gGw7XBFXfk.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/UpGcCHlX7o\n",
      "[barrnap] Scanning /tmp/UpGcCHlX7o for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/UpGcCHlX7o'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.3.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.3.fa_rrna.fa' -fi '/tmp/UpGcCHlX7o' -bed '/tmp/DT5w8GUQmi'\n",
      "index file /tmp/UpGcCHlX7o.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/CkaHrjaWZH\n",
      "[barrnap] Scanning /tmp/CkaHrjaWZH for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/CkaHrjaWZH'\n",
      "[barrnap] Found: 16S_rRNA c208 L=1476/1585 578129..579604 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c208 L=2889/3232 574216..577104 - 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c208 L=110/119 574023..574132 - 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.30.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.30.fa_rrna.fa' -fi '/tmp/CkaHrjaWZH' -bed '/tmp/nnfB3EcyxU'\n",
      "index file /tmp/CkaHrjaWZH.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/mCi_N8nSHG\n",
      "[barrnap] Scanning /tmp/mCi_N8nSHG for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/mCi_N8nSHG'\n",
      "[barrnap] Found: 16S_rRNA c12 L=1515/1585 2305542..2307056 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c12 L=1516/1585 1716909..1718424 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c12 L=3103/3232 1718850..1721952 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c12 L=3103/3232 2307482..2310584 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c12 L=110/119 1722047..1722156 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c12 L=110/119 2310678..2310787 + 5S ribosomal RNA\n",
      "[barrnap] Found 6 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.31.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.31.fa_rrna.fa' -fi '/tmp/mCi_N8nSHG' -bed '/tmp/8HgamIsFQz'\n",
      "index file /tmp/mCi_N8nSHG.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/Tob7sHZWmy\n",
      "[barrnap] Scanning /tmp/Tob7sHZWmy for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/Tob7sHZWmy'\n",
      "[barrnap] Found: 16S_rRNA c1206 L=1482/1585 116335..117816 + 16S ribosomal RNA\n",
      "[barrnap] Found 1 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.32.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.32.fa_rrna.fa' -fi '/tmp/Tob7sHZWmy' -bed '/tmp/mgAax5dTWN'\n",
      "index file /tmp/Tob7sHZWmy.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/GYywLXeFjr\n",
      "[barrnap] Scanning /tmp/GYywLXeFjr for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/GYywLXeFjr'\n",
      "[barrnap] Found: 16S_rRNA c241 L=1543/1585 1283..2825 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c233 L=1544/1585 2317..3860 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c233 L=2045/3232 1..2045 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c332 L=1599/3232 2..1600 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c241 L=942/3232 5..946 - 23S ribosomal RNA (partial)\n",
      "[barrnap] Found 5 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.4.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.4.fa_rrna.fa' -fi '/tmp/GYywLXeFjr' -bed '/tmp/EPzbfsIkkc'\n",
      "index file /tmp/GYywLXeFjr.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/8liMEIvuBm\n",
      "[barrnap] Scanning /tmp/8liMEIvuBm for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/8liMEIvuBm'\n",
      "[barrnap] Found: 16S_rRNA c222 L=1481/1585 509960..511440 - 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c222 L=2713/3232 506563..509275 - 23S ribosomal RNA\n",
      "[barrnap] Found 2 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.5.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.5.fa_rrna.fa' -fi '/tmp/8liMEIvuBm' -bed '/tmp/8W8WuDMjeY'\n",
      "index file /tmp/8liMEIvuBm.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/9ZszkEBdoB\n",
      "[barrnap] Scanning /tmp/9ZszkEBdoB for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/9ZszkEBdoB'\n",
      "[barrnap] Found: 16S_rRNA c59 L=1486/1585 275..1760 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c1186 L=1493/1585 55333..56825 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA c57 L=1404/1585 6..1409 + 16S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c57 L=2737/3232 1977..4713 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c59 L=2748/3232 2329..5076 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c1186 L=2734/3232 57105..59838 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c55 L=2741/3232 361..3101 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c58 L=2672/3232 3..2674 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA c70 L=1905/3232 1..1905 + 23S ribosomal RNA (partial)\n",
      "[barrnap] Found: 5S_rRNA c57 L=110/119 4848..4957 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c58 L=110/119 2808..2917 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c59 L=110/119 5210..5319 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c70 L=110/119 2039..2148 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c55 L=111/119 3239..3349 + 5S ribosomal RNA\n",
      "[barrnap] Found 14 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.6.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.6.fa_rrna.fa' -fi '/tmp/9ZszkEBdoB' -bed '/tmp/CiXAn23m6l'\n",
      "index file /tmp/9ZszkEBdoB.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/UlsvzwqBEe\n",
      "[barrnap] Scanning /tmp/UlsvzwqBEe for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/UlsvzwqBEe'\n",
      "[barrnap] Found: 16S_rRNA c1525 L=570/1585 1..570 + 16S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA c1525 L=2727/3232 1223..3949 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA c1525 L=110/119 4039..4148 + 5S ribosomal RNA\n",
      "[barrnap] Found 3 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.7.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.7.fa_rrna.fa' -fi '/tmp/UlsvzwqBEe' -bed '/tmp/5kpOGcU6K8'\n",
      "index file /tmp/UlsvzwqBEe.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/4UTqY2TU9E\n",
      "[barrnap] Scanning /tmp/4UTqY2TU9E for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/4UTqY2TU9E'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.8.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.8.fa_rrna.fa' -fi '/tmp/4UTqY2TU9E' -bed '/tmp/s2xZvzszqo'\n",
      "index file /tmp/4UTqY2TU9E.fai not found, generating...\n",
      "[barrnap] Done.\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/gladkov2/anaconda3/envs/metabat/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/gladkov2/anaconda3/envs/metabat/bin/bedtools\n",
      "[barrnap] Will use 1 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/glpXIXNE1p\n",
      "[barrnap] Scanning /tmp/glpXIXNE1p for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 1 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/gladkov2/anaconda3/envs/metabat/lib/barrnap/bin/../db/bac.hmm' '/tmp/glpXIXNE1p'\n",
      "[barrnap] Found 0 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: mags/46_gen_w_cov.9.fa_rrna.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo 'mags/46_gen_w_cov.9.fa_rrna.fa' -fi '/tmp/glpXIXNE1p' -bed '/tmp/ablTVnGfhy'\n",
      "index file /tmp/glpXIXNE1p.fai not found, generating...\n",
      "[barrnap] Done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "FILES=\"mags/*\"\n",
    "for f in $FILES\n",
    "do\n",
    "    barrnap -o $f\\_rrna.fa < $f > $f\\_rrna.gff\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "!rm -rf mags"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "2 bins (4266870 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "0 bins (0 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "0 bins (0 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "3 bins (12204608 bases in total) formed.\n"
     ]
    }
   ],
   "source": [
    "for f in d:\n",
    "    !metabat2 -i $f\\.fq -o mags/$f\\_gen_cov -a  $f\\_depth.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "checkm lineage_wf -x fa  -f checkm/N2_MAGs_checkm.tab -t 50 metabat2_bins_3/ out/\n",
    "checkm qa ./out/lineage.ms ./out -o 2 -f res --tab_table -t 20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: checkm lineage_wf [-h] [-r] [--ali] [--nt] [-g] [-u UNIQUE] [-m MULTI]\n",
      "                         [--force_domain] [--no_refinement]\n",
      "                         [--individual_markers] [--skip_adj_correction]\n",
      "                         [--skip_pseudogene_correction]\n",
      "                         [--aai_strain AAI_STRAIN] [-a ALIGNMENT_FILE]\n",
      "                         [--ignore_thresholds] [-e E_VALUE] [-l LENGTH]\n",
      "                         [-f FILE] [--tab_table] [-x EXTENSION] [-t THREADS]\n",
      "                         [--pplacer_threads PPLACER_THREADS] [-q]\n",
      "                         [--tmpdir TMPDIR]\n",
      "                         bin_dir output_dir\n",
      "\n",
      "Runs tree, lineage_set, analyze, qa\n",
      "\n",
      "positional arguments:\n",
      "  bin_dir               directory containing bins (fasta format)\n",
      "  output_dir            directory to write output files\n",
      "\n",
      "optional arguments:\n",
      "  -h, --help            show this help message and exit\n",
      "  -r, --reduced_tree    use reduced tree (requires <16GB of memory) for determining lineage of each bin\n",
      "  --ali                 generate HMMER alignment file for each bin\n",
      "  --nt                  generate nucleotide gene sequences for each bin\n",
      "  -g, --genes           bins contain genes as amino acids instead of nucleotide contigs\n",
      "  -u, --unique UNIQUE   minimum number of unique phylogenetic markers required to use lineage-specific marker set (default: 10)\n",
      "  -m, --multi MULTI     maximum number of multi-copy phylogenetic markers before defaulting to domain-level marker set (default: 10)\n",
      "  --force_domain        use domain-level sets for all bins\n",
      "  --no_refinement       do not perform lineage-specific marker set refinement\n",
      "  --individual_markers  treat marker as independent (i.e., ignore co-located set structure)\n",
      "  --skip_adj_correction\n",
      "                        do not exclude adjacent marker genes when estimating contamination\n",
      "  --skip_pseudogene_correction\n",
      "                        skip identification and filtering of pseudogenes\n",
      "  --aai_strain AAI_STRAIN\n",
      "                        AAI threshold used to identify strain heterogeneity (default: 0.9)\n",
      "  -a, --alignment_file ALIGNMENT_FILE\n",
      "                        produce file showing alignment of multi-copy genes and their AAI identity\n",
      "  --ignore_thresholds   ignore model-specific score thresholds\n",
      "  -e, --e_value E_VALUE\n",
      "                        e-value cut off (default: 1e-10)\n",
      "  -l, --length LENGTH   percent overlap between target and query (default: 0.7)\n",
      "  -f, --file FILE       print results to file (default: stdout)\n",
      "  --tab_table           print tab-separated values table\n",
      "  -x, --extension EXTENSION\n",
      "                        extension of bins (other files in directory are ignored) (default: fna)\n",
      "  -t, --threads THREADS\n",
      "                        number of threads (default: 1)\n",
      "  --pplacer_threads PPLACER_THREADS\n",
      "                        number of threads used by pplacer (memory usage increases linearly with additional threads) (default: 1)\n",
      "  -q, --quiet           suppress console output\n",
      "  --tmpdir TMPDIR       specify an alternative directory for temporary files\n",
      "\n",
      "Example: checkm lineage_wf ./bins ./output\n"
     ]
    }
   ],
   "source": [
    "!checkm lineage_wf -h"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:metabat]",
   "language": "python",
   "name": "conda-env-metabat-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
