{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %config IPCompleter.greedy=False\n",
    "%config Completer.use_jedi = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.SeqFeature import SeqFeature, FeatureLocation\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "find: ‘bwa-mem2’: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "!find bwa-mem2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/gladkov2/anaconda3/bin/metabat\n",
      "/home/gladkov2/anaconda3/pkgs/metabat2-2.15-h137b6e9_0/bin/metabat\n",
      "/home/gladkov2/anaconda3/envs/metabat\n",
      "/home/gladkov2/anaconda3/envs/metabat/bin/metabat\n"
     ]
    }
   ],
   "source": [
    "!find  /home/gladkov2/anaconda3/ -name \"metabat\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/gladkov2/storage/temp/serg/checkm\n"
     ]
    }
   ],
   "source": [
    "!find  /home/gladkov2/storage/ -name \"checkm\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### vamb env"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/storage/metagenome/bact_4x\n",
      "/mnt/storage/metagenome/bact_4x/binning\n",
      "total 26G\n",
      "-rw-rw-r-- 1 gladkov2 3.8G Oct  8 13:01 26_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 3.7G Oct  8 13:09 46_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 3.0G Oct  8 13:00 26_ont.fasta\n",
      "-rw-rw-r-- 1 gladkov2 2.9G Oct  8 13:09 46_ont.fasta\n",
      "-rw-rw-r-- 1 gladkov2 2.6G Oct  8 12:57 14_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 2.2G Oct  8 13:06 29_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 2.1G Oct  8 12:56 14_ont.fasta\n",
      "-rw-rw-r-- 1 gladkov2 1.8G Oct  8 13:06 29_ont.fasta\n",
      "-rw-rw-r-- 1 gladkov2 1.3G Oct  8 13:06 26_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 1.3G Oct  8 13:14 46_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 842M Oct  8 13:00 14_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 712M Oct  8 13:09 29_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 109M Oct  8 12:55 46.fq\n",
      "-rw-rw-r-- 1 gladkov2  95M Oct  8 12:55 26.fq\n",
      "-rw-rw-r-- 1 gladkov2  89M Oct  8 12:55 14.fq\n",
      "-rw-rw-r-- 1 gladkov2  88M Oct  8 12:55 29.fq\n",
      "-rw-rw-r-- 1 gladkov2  73K Oct  8 14:02 14_depth.txt\n",
      "-rw-rw-r-- 1 gladkov2  64K Oct  8 14:03 46_depth.txt\n",
      "-rw-rw-r-- 1 gladkov2  60K Oct  8 14:02 26_depth.txt\n",
      "-rw-rw-r-- 1 gladkov2  59K Oct  8 14:02 29_depth.txt\n",
      "drwxrwxr-x 2 gladkov2  20K Oct  8 15:06 \u001b[0m\u001b[01;34mmags\u001b[0m/\n",
      "-rw-rw-r-- 1 gladkov2 8.5K Oct 13 01:03 log_ann_err.txt\n",
      "drwxrwxr-x 4 gladkov2 4.0K Oct 22 14:16 \u001b[01;34mcheckm\u001b[0m/\n",
      "drwxrwxr-x 2 gladkov2 4.0K Oct 12 13:44 \u001b[01;34mclassify_output\u001b[0m/\n",
      "drwxrwxr-x 5 gladkov2 4.0K Oct 12 13:36 \u001b[01;34mclassify_wf_out\u001b[0m/\n",
      "drwxrwxr-x 3 gladkov2 4.0K Oct 11 13:25 \u001b[01;34mdram\u001b[0m/\n",
      "drwxrwxr-x 3 gladkov2 4.0K Oct 11 20:14 \u001b[01;34mdram2\u001b[0m/\n",
      "drwxrwxr-x 3 gladkov2 4.0K Oct 13 01:13 \u001b[01;34mdram3\u001b[0m/\n",
      "drwxrwxr-x 2 gladkov2 4.0K Oct 13 13:10 \u001b[01;34mgenome_summaries\u001b[0m/\n",
      "drwxrwxr-x 2 gladkov2 4.0K Oct 11 13:18 \u001b[01;34mh_mags\u001b[0m/\n",
      "-rw-rw-r-- 1 gladkov2  420 Apr 21  2021 rrr.sh\n",
      "-rw-rw-r-- 1 gladkov2  351 Oct 11 12:05 high_q_bins_list.txt\n",
      "-rw-rw-r-- 1 gladkov2    0 Oct 11 12:24 16S\n"
     ]
    }
   ],
   "source": [
    "%cd ~/storage/metagenome/bact_4x/\n",
    "# %mkdir binning\n",
    "%cd binning/\n",
    "\n",
    "%ll -hS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'14': ['14.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode01/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/14_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/14_R_wa.fq'],\n",
       " '26': ['26.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode02/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/26_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/26_R_wa.fq'],\n",
       " '29': ['29.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode03/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/29_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/29_R_wa.fq'],\n",
       " '46': ['46.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode04/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/46_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/46_R_wa.fq']}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = {\"14\":[\"14.fq\",\n",
    "                    \"/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode01/all.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/14_F_wa.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/14_R_wa.fq\"],\n",
    "     \"26\":[\"26.fq\",\n",
    "                    \"/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode02/all.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/26_F_wa.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/26_R_wa.fq\"],\n",
    "     \"29\":[\"29.fq\",\n",
    "                    \"/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode03/all.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/29_F_wa.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/29_R_wa.fq\"],\n",
    "     \"46\":[\"46.fq\",\n",
    "                    \"/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode04/all.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/46_F_wa.fq\",\n",
    "                    \"/home/gladkov2/storage/metagenome/bact_4x/new_con/46_R_wa.fq\"],\n",
    "    }\n",
    "\n",
    "d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: /home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode01/all.fq: Permission denied\n"
     ]
    }
   ],
   "source": [
    "!$ont"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!ont = d['14'][1]\n",
    "# !head $ont\n",
    "!sed -n '1~4s/^@/>/p;2~4p' $ont | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "merge not a valid action\n",
      "\n",
      "    available actions:\n",
      "        `extract`: extract sequences from a fasta file\n",
      "        `info`: show info about the fasta file and exit.\n",
      "        `split`: split a large fasta file into separate files\n",
      "                 and/or into K-mers.\n",
      "        `flatten`: flatten a fasta file inplace so that later\n",
      "                   command-line (and programmattic) access via\n",
      "                   pyfasta will use the inplace flattened version\n",
      "                   rather than creating another .flat copy of the\n",
      "                   sequence.\n",
      "\n",
      "    to view the help for a particular action, use:\n",
      "        pyfasta [action] --help\n",
      "    e.g.:\n",
      "        pyfasta extract --help\n",
      "    \n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "%rm *\n",
    "!cp /home/gladkov2/storage/metagenome/bact_4x/new_con/diamond/con/14.faa 14.fq\n",
    "!cp /home/gladkov2/storage/metagenome/bact_4x/new_con/diamond/con/26.faa 26.fq\n",
    "!cp /home/gladkov2/storage/metagenome/bact_4x/new_con/diamond/con/29.faa 29.fq\n",
    "!cp /home/gladkov2/storage/metagenome/bact_4x/new_con/diamond/con/46.faa 46.fq\n",
    "# !sed -n '1~4s/^@/>/p;2~4p' $ont | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'd' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-3-5fe1ba162f8e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mont\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0md\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msystem\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'head $ont'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'd' is not defined"
     ]
    }
   ],
   "source": [
    "ont = d[f][1]\n",
    "!head $ont"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!bwa-mem2 mem -t 50 14.fq 14_ont.fasta > test.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 5 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "!samtools sort test.sam > test.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for f in d:\n",
    "    con = d[f][0]\n",
    "    mi_f = d[f][2]\n",
    "    mi_r = d[f][3]\n",
    "    ont = d[f][1]\n",
    "    !sed -n '1~4s/^@/>/p;2~4p' $ont > $f\\_ont.fasta\n",
    "    !minimap2 -ax map-ont $con  $f\\_ont.fasta $mi_f $mi_r -t 50  > $f\\_bwa.sam\n",
    "    !samtools sort $f\\_bwa.sam > $f\\_bwa_sorted.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1084633 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "29790 + 0 secondary\n",
      "27551 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "802588 + 0 mapped (74.00% : N/A)\n",
      "0 + 0 paired in sequencing\n",
      "0 + 0 read1\n",
      "0 + 0 read2\n",
      "0 + 0 properly paired (N/A : N/A)\n",
      "0 + 0 with itself and mate mapped\n",
      "0 + 0 singletons (N/A : N/A)\n",
      "0 + 0 with mate mapped to a different chr\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::2.160*1.32] collected minimizers\n",
      "[M::mm_idx_gen::2.632*2.70] sorted minimizers\n",
      "[M::main::2.633*2.70] loaded/built the index for 1903 target sequence(s)\n",
      "[M::mm_mapopt_update::2.837*2.58] mid_occ = 32\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1903\n",
      "[M::mm_idx_stat::2.940*2.53] distinct minimizers: 11752665 (78.56% are singletons); average occurrences: 1.458; average spacing: 5.359\n",
      "[M::worker_pipeline::12.510*32.57] mapped 165675 sequences\n",
      "[M::worker_pipeline::16.321*36.56] mapped 167791 sequences\n",
      "[M::worker_pipeline::20.448*39.13] mapped 168228 sequences\n",
      "[M::worker_pipeline::23.738*36.84] mapped 167010 sequences\n",
      "[M::worker_pipeline::24.031*36.40] mapped 18490 sequences\n",
      "[M::worker_pipeline::24.477*35.97] mapped 170049 sequences\n",
      "[M::worker_pipeline::24.939*35.55] mapped 170049 sequences\n",
      "[M::main] Version: 2.17-r941\n",
      "[M::main] CMD: minimap2 -ax map-ont -t 50 14.fq 14_ont.fasta /home/gladkov2/storage/metagenome/bact_4x/new_con/14_F_wa.fq /home/gladkov2/storage/metagenome/bact_4x/new_con/14_R_wa.fq\n",
      "[M::main] Real time: 25.041 sec; CPU: 886.746 sec; Peak RSS: 4.688 GB\n",
      "[bam_sort_core] merging from 4 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "minimap2 -ax map-ont 14.fq  14_ont.fasta /home/gladkov2/storage/metagenome/bact_4x/new_con/14_F_wa.fq  /home/gladkov2/storage/metagenome/bact_4x/new_con/14_R_wa.fq -t 50 > test5.sam \n",
    "samtools sort test5.sam > test5.bam\n",
    "samtools flagstat test5.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "rm test4.sam test4.bam\n",
    "bwa-mem2 mem -t 50 14.fq 14_temp_f.fq /home/gladkov2/storage/metagenome/bact_4x/new_con/14_R_wa.fq  > test4.sam\n",
    "samtools sort test4.sam > test4.bam\n",
    "samtools flagstat test4.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "14\n",
      "1084633 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "29790 + 0 secondary\n",
      "27551 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "802588 + 0 mapped (74.00% : N/A)\n",
      "0 + 0 paired in sequencing\n",
      "0 + 0 read1\n",
      "0 + 0 read2\n",
      "0 + 0 properly paired (N/A : N/A)\n",
      "0 + 0 with itself and mate mapped\n",
      "0 + 0 singletons (N/A : N/A)\n",
      "0 + 0 with mate mapped to a different chr\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "\n",
      "26\n",
      "1360549 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "27936 + 0 secondary\n",
      "31783 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "1209509 + 0 mapped (88.90% : N/A)\n",
      "0 + 0 paired in sequencing\n",
      "0 + 0 read1\n",
      "0 + 0 read2\n",
      "0 + 0 properly paired (N/A : N/A)\n",
      "0 + 0 with itself and mate mapped\n",
      "0 + 0 singletons (N/A : N/A)\n",
      "0 + 0 with mate mapped to a different chr\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "\n",
      "29\n",
      "955338 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "25870 + 0 secondary\n",
      "19664 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "740807 + 0 mapped (77.54% : N/A)\n",
      "0 + 0 paired in sequencing\n",
      "0 + 0 read1\n",
      "0 + 0 read2\n",
      "0 + 0 properly paired (N/A : N/A)\n",
      "0 + 0 with itself and mate mapped\n",
      "0 + 0 singletons (N/A : N/A)\n",
      "0 + 0 with mate mapped to a different chr\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "\n",
      "46\n",
      "1671656 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "41636 + 0 secondary\n",
      "27770 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "1400623 + 0 mapped (83.79% : N/A)\n",
      "0 + 0 paired in sequencing\n",
      "0 + 0 read1\n",
      "0 + 0 read2\n",
      "0 + 0 properly paired (N/A : N/A)\n",
      "0 + 0 with itself and mate mapped\n",
      "0 + 0 singletons (N/A : N/A)\n",
      "0 + 0 with mate mapped to a different chr\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\n"
     ]
    }
   ],
   "source": [
    "for f in d:\n",
    "    print(f'\\n{f}')\n",
    "    !samtools flagstat $f\\_bwa_sorted.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### change env to metabat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 26G\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2  89M Oct  8 12:55 14.fq\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 2.6G Oct  8 12:57 14_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 842M Oct  8 13:00 14_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 2.1G Oct  8 12:56 14_ont.fasta\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2  95M Oct  8 12:55 26.fq\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 3.8G Oct  8 13:01 26_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 1.3G Oct  8 13:06 26_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 3.0G Oct  8 13:00 26_ont.fasta\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2  88M Oct  8 12:55 29.fq\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 2.2G Oct  8 13:06 29_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 712M Oct  8 13:09 29_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 1.8G Oct  8 13:06 29_ont.fasta\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 109M Oct  8 12:55 46.fq\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 3.7G Oct  8 13:09 46_bwa.sam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 1.3G Oct  8 13:14 46_bwa_sorted.bam\n",
      "-rw-rw-r-- 1 gladkov2 gladkov2 2.9G Oct  8 13:09 46_ont.fasta\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'14': ['14.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode01/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/14_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/14_R_wa.fq'],\n",
       " '26': ['26.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode02/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/26_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/26_R_wa.fq'],\n",
       " '29': ['29.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode03/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/29_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/29_R_wa.fq'],\n",
       " '46': ['46.fq',\n",
       "  '/home/gladkov2/storage/Bact_comm_4x/20200225_1435_MN24173_FAK85519_967910e6_3_6_1_barcoded/barcode04/all.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/46_F_wa.fq',\n",
       "  '/home/gladkov2/storage/metagenome/bact_4x/new_con/46_R_wa.fq']}"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!ls -lh\n",
    "d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output depth matrix to 14_depth.txt\n",
      "Reference fasta file 14.fq\n",
      "Output matrix to 14_depth.txt\n",
      "Reading reference fasta file: 14.fq\n",
      "... 1903 sequences\n",
      "Opening 1 bams\n",
      "Consolidating headers\n",
      "Processing bam files\n",
      "Thread 0 finished: 14_bwa_sorted.bam with 1084633 reads and 169783 readsWellMapped\n",
      "Creating depth matrix file: 14_depth.txt\n",
      "Closing most bam files\n",
      "Closing last bam file\n",
      "Finished\n",
      "Output depth matrix to 26_depth.txt\n",
      "Reference fasta file 26.fq\n",
      "Output matrix to 26_depth.txt\n",
      "Reading reference fasta file: 26.fq\n",
      "... 1589 sequences\n",
      "Opening 1 bams\n",
      "Consolidating headers\n",
      "Processing bam files\n",
      "Thread 0 finished: 26_bwa_sorted.bam with 1360549 reads and 191790 readsWellMapped\n",
      "Creating depth matrix file: 26_depth.txt\n",
      "Closing most bam files\n",
      "Closing last bam file\n",
      "Finished\n",
      "Output depth matrix to 29_depth.txt\n",
      "Reference fasta file 29.fq\n",
      "Output matrix to 29_depth.txt\n",
      "Reading reference fasta file: 29.fq\n",
      "... 1519 sequences\n",
      "Opening 1 bams\n",
      "Consolidating headers\n",
      "Processing bam files\n",
      "Thread 0 finished: 29_bwa_sorted.bam with 955338 reads and 166009 readsWellMapped\n",
      "Creating depth matrix file: 29_depth.txt\n",
      "Closing most bam files\n",
      "Closing last bam file\n",
      "Finished\n",
      "Output depth matrix to 46_depth.txt\n",
      "Reference fasta file 46.fq\n",
      "Output matrix to 46_depth.txt\n",
      "Reading reference fasta file: 46.fq\n",
      "... 1682 sequences\n",
      "Opening 1 bams\n",
      "Consolidating headers\n",
      "Processing bam files\n",
      "Thread 0 finished: 46_bwa_sorted.bam with 1671656 reads and 419608 readsWellMapped\n",
      "Creating depth matrix file: 46_depth.txt\n",
      "Closing most bam files\n",
      "Closing last bam file\n",
      "Finished\n"
     ]
    }
   ],
   "source": [
    "\n",
    "for f in d:\n",
    "    !jgi_summarize_bam_contig_depths --outputDepth $f\\_depth.txt --referenceFasta $f\\.fq  $f\\_bwa_sorted.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "2 bins (4266870 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "22 bins (88398034 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "0 bins (0 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "19 bins (95396352 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "0 bins (0 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "22 bins (88789908 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "3 bins (12204608 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "32 bins (109310530 bases in total) formed.\n",
      "[2021-10-08 14:32:18] INFO: CheckM v1.1.3\n",
      "[2021-10-08 14:32:18] INFO: checkm lineage_wf -x fa -f checkm/N2_MAGs_checkm.tab -t 50 mags/ checkm/\n",
      "[2021-10-08 14:32:18] INFO: [CheckM - tree] Placing bins in reference genome tree.\n",
      "[2021-10-08 14:32:21] INFO: Identifying marker genes in 100 bins with 50 threads:\n",
      "    Finished processing 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:34:56] INFO: Saving HMM info to file.\n",
      "[2021-10-08 14:34:56] INFO: Calculating genome statistics for 100 bins with 50 threads:\n",
      "    Finished processing 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:34:57] INFO: Extracting marker genes to align.\n",
      "[2021-10-08 14:34:57] INFO: Parsing HMM hits to marker genes:\n",
      "    Finished parsing hits for 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:35:02] INFO: Extracting 43 HMMs with 50 threads:\n",
      "    Finished extracting 43 of 43 (100.00%) HMMs.\n",
      "[2021-10-08 14:35:02] INFO: Aligning 43 marker genes with 50 threads:\n",
      "    Finished aligning 43 of 43 (100.00%) marker genes.\n",
      "[2021-10-08 14:35:04] INFO: Reading marker alignment files.\n",
      "[2021-10-08 14:35:04] INFO: Concatenating alignments.\n",
      "[2021-10-08 14:35:04] INFO: Placing 100 bins into the genome tree with pplacer (be patient).\n",
      "[2021-10-08 14:45:23] INFO: { Current stage: 0:13:04.949 || Total: 0:13:04.949 }\n",
      "[2021-10-08 14:45:23] INFO: [CheckM - lineage_set] Inferring lineage-specific marker sets.\n",
      "[2021-10-08 14:45:23] INFO: Reading HMM info from file.\n",
      "[2021-10-08 14:45:23] INFO: Parsing HMM hits to marker genes:\n",
      "    Finished parsing hits for 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:45:27] INFO: Determining marker sets for each genome bin.\n",
      "    Finished processing 100 of 100 (100.00%) bins (current: 14_gen_cov.1). \n",
      "[2021-10-08 14:45:36] INFO: Marker set written to: checkm/lineage.ms\n",
      "[2021-10-08 14:45:36] INFO: { Current stage: 0:00:12.574 || Total: 0:13:17.524 }\n",
      "[2021-10-08 14:45:36] INFO: [CheckM - analyze] Identifying marker genes in bins.\n",
      "[2021-10-08 14:45:38] INFO: Identifying marker genes in 100 bins with 50 threads:\n",
      "    Finished processing 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:51:23] INFO: Saving HMM info to file.\n",
      "[2021-10-08 14:51:26] INFO: { Current stage: 0:05:50.556 || Total: 0:19:08.080 }\n",
      "[2021-10-08 14:51:26] INFO: Parsing HMM hits to marker genes:\n",
      "    Finished parsing hits for 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:51:58] INFO: Aligning marker genes with multiple hits in a single bin:\n",
      "    Finished processing 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:54:34] INFO: { Current stage: 0:03:07.884 || Total: 0:22:15.964 }\n",
      "[2021-10-08 14:54:34] INFO: Calculating genome statistics for 100 bins with 50 threads:\n",
      "    Finished processing 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:54:35] INFO: { Current stage: 0:00:01.030 || Total: 0:22:16.995 }\n",
      "[2021-10-08 14:54:35] INFO: [CheckM - qa] Tabulating genome statistics.\n",
      "[2021-10-08 14:54:35] INFO: Calculating AAI between multi-copy marker genes.\n",
      "[2021-10-08 14:54:36] INFO: Reading HMM info from file.\n",
      "[2021-10-08 14:54:38] INFO: Parsing HMM hits to marker genes:\n",
      "    Finished parsing hits for 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:55:10] INFO: QA information written to: checkm/N2_MAGs_checkm.tab\n",
      "[2021-10-08 14:55:10] INFO: { Current stage: 0:00:34.806 || Total: 0:22:51.802 }\n",
      "[2021-10-08 14:55:11] INFO: CheckM v1.1.3\n",
      "[2021-10-08 14:55:11] INFO: checkm qa ./checkm/lineage.ms ./checkm -o 2 -f res --tab_table -t 50\n",
      "[2021-10-08 14:55:11] INFO: [CheckM - qa] Tabulating genome statistics.\n",
      "[2021-10-08 14:55:11] INFO: Calculating AAI between multi-copy marker genes.\n",
      "[2021-10-08 14:55:11] INFO: Reading HMM info from file.\n",
      "[2021-10-08 14:55:13] INFO: Parsing HMM hits to marker genes:\n",
      "    Finished parsing hits for 100 of 100 (100.00%) bins.\n",
      "[2021-10-08 14:55:46] INFO: QA information written to: res\n",
      "[2021-10-08 14:55:46] INFO: { Current stage: 0:00:34.753 || Total: 0:00:34.753 }\n",
      "/bin/bash: mags/*.fa: ambiguous redirect\n"
     ]
    }
   ],
   "source": [
    "for f in d:\n",
    "    !metabat2 -i $f\\.fq -o mags/$f\\_gen_cov -a  $f\\_depth.txt\n",
    "    !metabat2 -i $f\\.fq -o mags/$f\\_gen_w_cov\n",
    "!checkm lineage_wf -x fa  -f checkm/N2_MAGs_checkm.tab -t 50 mags/ checkm/\n",
    "!checkm qa ./checkm/lineage.ms ./checkm -o 2 -f res --tab_table -t 50\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "FILES=\"mags/*\"\n",
    "for f in $FILES\n",
    "do\n",
    "    barrnap -o $f\\_rrna.fa < $f > $f\\_rrna.gff\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "!rm -rf mags"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "2 bins (4266870 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "0 bins (0 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "0 bins (0 bases in total) formed.\n",
      "MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. \n",
      "3 bins (12204608 bases in total) formed.\n"
     ]
    }
   ],
   "source": [
    "for f in d:\n",
    "    !metabat2 -i $f\\.fq -o mags/$f\\_gen_cov -a  $f\\_depth.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "checkm lineage_wf -x fa  -f checkm/N2_MAGs_checkm.tab -t 50 metabat2_bins_3/ out/\n",
    "checkm qa ./out/lineage.ms ./out -o 2 -f res --tab_table -t 20"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### env "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use theshhold - "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"res\", delimiter=\"\\t\")\n",
    "df.columns\n",
    "df = df[['Bin Id','Marker lineage', 'Completeness','Contamination', 'Strain heterogeneity',\n",
    "       'Genome size (bp)', 'N50 (contigs)', 'GC']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bin Id</th>\n",
       "      <th>Marker lineage</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>Strain heterogeneity</th>\n",
       "      <th>Genome size (bp)</th>\n",
       "      <th>N50 (contigs)</th>\n",
       "      <th>GC</th>\n",
       "      <th>Quality</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>46_gen_w_cov.31</td>\n",
       "      <td>o__Actinomycetales (UID1572)</td>\n",
       "      <td>99.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3403813</td>\n",
       "      <td>3403813</td>\n",
       "      <td>73.9</td>\n",
       "      <td>99.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>46_gen_cov.1</td>\n",
       "      <td>o__Actinomycetales (UID1572)</td>\n",
       "      <td>99.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3403813</td>\n",
       "      <td>3403813</td>\n",
       "      <td>73.9</td>\n",
       "      <td>99.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>46_gen_w_cov.30</td>\n",
       "      <td>o__Rhizobiales (UID3642)</td>\n",
       "      <td>98.86</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>5300980</td>\n",
       "      <td>1767233</td>\n",
       "      <td>61.5</td>\n",
       "      <td>98.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14_gen_w_cov.2</td>\n",
       "      <td>o__Actinomycetales (UID1572)</td>\n",
       "      <td>98.55</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>4068525</td>\n",
       "      <td>3263006</td>\n",
       "      <td>73.2</td>\n",
       "      <td>98.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>26_gen_w_cov.14</td>\n",
       "      <td>o__Actinomycetales (UID1572)</td>\n",
       "      <td>98.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3916782</td>\n",
       "      <td>3916782</td>\n",
       "      <td>73.3</td>\n",
       "      <td>98.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>46_gen_w_cov.24</td>\n",
       "      <td>k__Bacteria (UID203)</td>\n",
       "      <td>87.93</td>\n",
       "      <td>92.79</td>\n",
       "      <td>17.82</td>\n",
       "      <td>12610198</td>\n",
       "      <td>103590</td>\n",
       "      <td>65.4</td>\n",
       "      <td>-376.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>26_gen_w_cov.3</td>\n",
       "      <td>k__Bacteria (UID203)</td>\n",
       "      <td>100.00</td>\n",
       "      <td>108.82</td>\n",
       "      <td>14.57</td>\n",
       "      <td>15282497</td>\n",
       "      <td>142215</td>\n",
       "      <td>65.0</td>\n",
       "      <td>-444.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>26_gen_w_cov.10</td>\n",
       "      <td>k__Bacteria (UID203)</td>\n",
       "      <td>95.30</td>\n",
       "      <td>114.08</td>\n",
       "      <td>43.22</td>\n",
       "      <td>10650099</td>\n",
       "      <td>75717</td>\n",
       "      <td>63.9</td>\n",
       "      <td>-475.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>26_gen_w_cov.7</td>\n",
       "      <td>f__Xanthomonadaceae (UID4214)</td>\n",
       "      <td>98.38</td>\n",
       "      <td>139.19</td>\n",
       "      <td>17.09</td>\n",
       "      <td>11453579</td>\n",
       "      <td>91794</td>\n",
       "      <td>69.0</td>\n",
       "      <td>-597.57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>14_gen_w_cov.14</td>\n",
       "      <td>k__Bacteria (UID203)</td>\n",
       "      <td>100.00</td>\n",
       "      <td>288.39</td>\n",
       "      <td>7.99</td>\n",
       "      <td>22796128</td>\n",
       "      <td>109616</td>\n",
       "      <td>66.8</td>\n",
       "      <td>-1341.95</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Bin Id                 Marker lineage  Completeness  \\\n",
       "92  46_gen_w_cov.31   o__Actinomycetales (UID1572)         99.23   \n",
       "65     46_gen_cov.1   o__Actinomycetales (UID1572)         99.23   \n",
       "91  46_gen_w_cov.30       o__Rhizobiales (UID3642)         98.86   \n",
       "13   14_gen_w_cov.2   o__Actinomycetales (UID1572)         98.55   \n",
       "29  26_gen_w_cov.14   o__Actinomycetales (UID1572)         98.22   \n",
       "..              ...                            ...           ...   \n",
       "84  46_gen_w_cov.24           k__Bacteria (UID203)         87.93   \n",
       "36   26_gen_w_cov.3           k__Bacteria (UID203)        100.00   \n",
       "25  26_gen_w_cov.10           k__Bacteria (UID203)         95.30   \n",
       "40   26_gen_w_cov.7  f__Xanthomonadaceae (UID4214)         98.38   \n",
       "7   14_gen_w_cov.14           k__Bacteria (UID203)        100.00   \n",
       "\n",
       "    Contamination  Strain heterogeneity  Genome size (bp)  N50 (contigs)  \\\n",
       "92           0.00                  0.00           3403813        3403813   \n",
       "65           0.00                  0.00           3403813        3403813   \n",
       "91           0.00                  0.00           5300980        1767233   \n",
       "13           0.00                  0.00           4068525        3263006   \n",
       "29           0.00                  0.00           3916782        3916782   \n",
       "..            ...                   ...               ...            ...   \n",
       "84          92.79                 17.82          12610198         103590   \n",
       "36         108.82                 14.57          15282497         142215   \n",
       "25         114.08                 43.22          10650099          75717   \n",
       "40         139.19                 17.09          11453579          91794   \n",
       "7          288.39                  7.99          22796128         109616   \n",
       "\n",
       "      GC  Quality  \n",
       "92  73.9    99.23  \n",
       "65  73.9    99.23  \n",
       "91  61.5    98.86  \n",
       "13  73.2    98.55  \n",
       "29  73.3    98.22  \n",
       "..   ...      ...  \n",
       "84  65.4  -376.02  \n",
       "36  65.0  -444.10  \n",
       "25  63.9  -475.10  \n",
       "40  69.0  -597.57  \n",
       "7   66.8 -1341.95  \n",
       "\n",
       "[100 rows x 9 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Quality'] = df['Completeness'] - 5*df['Contamination']\n",
    "df.sort_values(by='Quality', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_rows', 500)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bin Id</th>\n",
       "      <th>Marker lineage</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>Strain heterogeneity</th>\n",
       "      <th>Genome size (bp)</th>\n",
       "      <th>N50 (contigs)</th>\n",
       "      <th>GC</th>\n",
       "      <th>Quality</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14_gen_cov.2</td>\n",
       "      <td>f__Xanthomonadaceae (UID4214)</td>\n",
       "      <td>98.29</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3537401</td>\n",
       "      <td>587144</td>\n",
       "      <td>70.9</td>\n",
       "      <td>96.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>14_gen_w_cov.17</td>\n",
       "      <td>f__Xanthomonadaceae (UID4214)</td>\n",
       "      <td>98.29</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3542827</td>\n",
       "      <td>587144</td>\n",
       "      <td>70.9</td>\n",
       "      <td>96.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>14_gen_w_cov.19</td>\n",
       "      <td>f__Xanthomonadaceae (UID4214)</td>\n",
       "      <td>91.87</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3219570</td>\n",
       "      <td>582876</td>\n",
       "      <td>66.0</td>\n",
       "      <td>91.52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14_gen_w_cov.2</td>\n",
       "      <td>o__Actinomycetales (UID1572)</td>\n",
       "      <td>98.55</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4068525</td>\n",
       "      <td>3263006</td>\n",
       "      <td>73.2</td>\n",
       "      <td>98.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>26_gen_w_cov.11</td>\n",
       "      <td>c__Gammaproteobacteria (UID4444)</td>\n",
       "      <td>95.31</td>\n",
       "      <td>0.79</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4783894</td>\n",
       "      <td>788776</td>\n",
       "      <td>54.7</td>\n",
       "      <td>91.36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>26_gen_w_cov.14</td>\n",
       "      <td>o__Actinomycetales (UID1572)</td>\n",
       "      <td>98.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3916782</td>\n",
       "      <td>3916782</td>\n",
       "      <td>73.3</td>\n",
       "      <td>98.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>26_gen_w_cov.15</td>\n",
       "      <td>k__Bacteria (UID2982)</td>\n",
       "      <td>95.21</td>\n",
       "      <td>2.91</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4042972</td>\n",
       "      <td>2689941</td>\n",
       "      <td>63.5</td>\n",
       "      <td>80.66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>26_gen_w_cov.16</td>\n",
       "      <td>p__Bacteroidetes (UID2591)</td>\n",
       "      <td>91.80</td>\n",
       "      <td>0.50</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3859528</td>\n",
       "      <td>230523</td>\n",
       "      <td>44.4</td>\n",
       "      <td>89.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>26_gen_w_cov.4</td>\n",
       "      <td>c__Alphaproteobacteria (UID3422)</td>\n",
       "      <td>94.15</td>\n",
       "      <td>1.14</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3427767</td>\n",
       "      <td>1780775</td>\n",
       "      <td>64.4</td>\n",
       "      <td>88.45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>29_gen_w_cov.10</td>\n",
       "      <td>c__Alphaproteobacteria (UID3422)</td>\n",
       "      <td>96.43</td>\n",
       "      <td>1.95</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4136404</td>\n",
       "      <td>2264918</td>\n",
       "      <td>60.0</td>\n",
       "      <td>86.68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>29_gen_w_cov.19</td>\n",
       "      <td>o__Actinomycetales (UID1572)</td>\n",
       "      <td>96.48</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2538242</td>\n",
       "      <td>2538242</td>\n",
       "      <td>68.1</td>\n",
       "      <td>96.48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>29_gen_w_cov.20</td>\n",
       "      <td>f__Xanthomonadaceae (UID4214)</td>\n",
       "      <td>94.50</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4370568</td>\n",
       "      <td>857265</td>\n",
       "      <td>70.3</td>\n",
       "      <td>89.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>29_gen_w_cov.3</td>\n",
       "      <td>f__Xanthomonadaceae (UID4214)</td>\n",
       "      <td>99.03</td>\n",
       "      <td>0.69</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4402214</td>\n",
       "      <td>2697263</td>\n",
       "      <td>66.8</td>\n",
       "      <td>95.58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>29_gen_w_cov.4</td>\n",
       "      <td>o__Cytophagales (UID2936)</td>\n",
       "      <td>95.83</td>\n",
       "      <td>1.74</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6760168</td>\n",
       "      <td>995369</td>\n",
       "      <td>36.5</td>\n",
       "      <td>87.13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>46_gen_cov.1</td>\n",
       "      <td>o__Actinomycetales (UID1572)</td>\n",
       "      <td>99.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3403813</td>\n",
       "      <td>3403813</td>\n",
       "      <td>73.9</td>\n",
       "      <td>99.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>46_gen_cov.2</td>\n",
       "      <td>f__Rhodocyclaceae (UID3972)</td>\n",
       "      <td>98.89</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4345908</td>\n",
       "      <td>1400303</td>\n",
       "      <td>66.4</td>\n",
       "      <td>96.84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>46_gen_cov.3</td>\n",
       "      <td>o__Rhizobiales (UID3447)</td>\n",
       "      <td>97.01</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4454887</td>\n",
       "      <td>762926</td>\n",
       "      <td>61.9</td>\n",
       "      <td>96.81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>46_gen_w_cov.1</td>\n",
       "      <td>o__Rhodospirillales (UID3754)</td>\n",
       "      <td>96.72</td>\n",
       "      <td>3.53</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5144195</td>\n",
       "      <td>1246017</td>\n",
       "      <td>67.3</td>\n",
       "      <td>79.07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>46_gen_w_cov.25</td>\n",
       "      <td>o__Rhizobiales (UID3447)</td>\n",
       "      <td>97.01</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4513373</td>\n",
       "      <td>762926</td>\n",
       "      <td>61.9</td>\n",
       "      <td>96.81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>46_gen_w_cov.30</td>\n",
       "      <td>o__Rhizobiales (UID3642)</td>\n",
       "      <td>98.86</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5300980</td>\n",
       "      <td>1767233</td>\n",
       "      <td>61.5</td>\n",
       "      <td>98.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>46_gen_w_cov.31</td>\n",
       "      <td>o__Actinomycetales (UID1572)</td>\n",
       "      <td>99.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3403813</td>\n",
       "      <td>3403813</td>\n",
       "      <td>73.9</td>\n",
       "      <td>99.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>46_gen_w_cov.5</td>\n",
       "      <td>o__Rhizobiales (UID3447)</td>\n",
       "      <td>96.39</td>\n",
       "      <td>0.82</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4370784</td>\n",
       "      <td>705289</td>\n",
       "      <td>63.7</td>\n",
       "      <td>92.29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Bin Id                    Marker lineage  Completeness  \\\n",
       "1      14_gen_cov.2     f__Xanthomonadaceae (UID4214)         98.29   \n",
       "10  14_gen_w_cov.17     f__Xanthomonadaceae (UID4214)         98.29   \n",
       "12  14_gen_w_cov.19     f__Xanthomonadaceae (UID4214)         91.87   \n",
       "13   14_gen_w_cov.2      o__Actinomycetales (UID1572)         98.55   \n",
       "26  26_gen_w_cov.11  c__Gammaproteobacteria (UID4444)         95.31   \n",
       "29  26_gen_w_cov.14      o__Actinomycetales (UID1572)         98.22   \n",
       "30  26_gen_w_cov.15             k__Bacteria (UID2982)         95.21   \n",
       "31  26_gen_w_cov.16        p__Bacteroidetes (UID2591)         91.80   \n",
       "37   26_gen_w_cov.4  c__Alphaproteobacteria (UID3422)         94.15   \n",
       "44  29_gen_w_cov.10  c__Alphaproteobacteria (UID3422)         96.43   \n",
       "53  29_gen_w_cov.19      o__Actinomycetales (UID1572)         96.48   \n",
       "55  29_gen_w_cov.20     f__Xanthomonadaceae (UID4214)         94.50   \n",
       "58   29_gen_w_cov.3     f__Xanthomonadaceae (UID4214)         99.03   \n",
       "59   29_gen_w_cov.4         o__Cytophagales (UID2936)         95.83   \n",
       "65     46_gen_cov.1      o__Actinomycetales (UID1572)         99.23   \n",
       "66     46_gen_cov.2       f__Rhodocyclaceae (UID3972)         98.89   \n",
       "67     46_gen_cov.3          o__Rhizobiales (UID3447)         97.01   \n",
       "68   46_gen_w_cov.1     o__Rhodospirillales (UID3754)         96.72   \n",
       "85  46_gen_w_cov.25          o__Rhizobiales (UID3447)         97.01   \n",
       "91  46_gen_w_cov.30          o__Rhizobiales (UID3642)         98.86   \n",
       "92  46_gen_w_cov.31      o__Actinomycetales (UID1572)         99.23   \n",
       "95   46_gen_w_cov.5          o__Rhizobiales (UID3447)         96.39   \n",
       "\n",
       "    Contamination  Strain heterogeneity  Genome size (bp)  N50 (contigs)  \\\n",
       "1            0.34                   0.0           3537401         587144   \n",
       "10           0.34                   0.0           3542827         587144   \n",
       "12           0.07                   0.0           3219570         582876   \n",
       "13           0.00                   0.0           4068525        3263006   \n",
       "26           0.79                   0.0           4783894         788776   \n",
       "29           0.00                   0.0           3916782        3916782   \n",
       "30           2.91                   0.0           4042972        2689941   \n",
       "31           0.50                   0.0           3859528         230523   \n",
       "37           1.14                   0.0           3427767        1780775   \n",
       "44           1.95                   0.0           4136404        2264918   \n",
       "53           0.00                   0.0           2538242        2538242   \n",
       "55           0.99                   0.0           4370568         857265   \n",
       "58           0.69                   0.0           4402214        2697263   \n",
       "59           1.74                   0.0           6760168         995369   \n",
       "65           0.00                   0.0           3403813        3403813   \n",
       "66           0.41                   0.0           4345908        1400303   \n",
       "67           0.04                   0.0           4454887         762926   \n",
       "68           3.53                   0.0           5144195        1246017   \n",
       "85           0.04                   0.0           4513373         762926   \n",
       "91           0.00                   0.0           5300980        1767233   \n",
       "92           0.00                   0.0           3403813        3403813   \n",
       "95           0.82                   0.0           4370784         705289   \n",
       "\n",
       "      GC  Quality  \n",
       "1   70.9    96.59  \n",
       "10  70.9    96.59  \n",
       "12  66.0    91.52  \n",
       "13  73.2    98.55  \n",
       "26  54.7    91.36  \n",
       "29  73.3    98.22  \n",
       "30  63.5    80.66  \n",
       "31  44.4    89.30  \n",
       "37  64.4    88.45  \n",
       "44  60.0    86.68  \n",
       "53  68.1    96.48  \n",
       "55  70.3    89.55  \n",
       "58  66.8    95.58  \n",
       "59  36.5    87.13  \n",
       "65  73.9    99.23  \n",
       "66  66.4    96.84  \n",
       "67  61.9    96.81  \n",
       "68  67.3    79.07  \n",
       "85  61.9    96.81  \n",
       "91  61.5    98.86  \n",
       "92  73.9    99.23  \n",
       "95  63.7    92.29  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.query(\"Completeness > 90 and Contamination < 5\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_h = df.query(\"Completeness > 90 and Contamination < 5\")\n",
    "df_h.to_csv('checkm_stat_hq.tsv', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_h = df.query(\"Completeness > 90 and Contamination < 5\")\n",
    "df_h['Bin Id']\n",
    "# df_h.shape[0]\n",
    "with open('high_q_bins_list.txt', 'a') as f:\n",
    "    dfAsString = df_h['Bin Id'].to_string(header=False, index=False)\n",
    "    f.write(dfAsString)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   14_gen_cov.2\n",
      "14_gen_w_cov.17\n",
      "14_gen_w_cov.19\n",
      " 14_gen_w_cov.2\n",
      "26_gen_w_cov.11\n",
      "26_gen_w_cov.14\n",
      "26_gen_w_cov.15\n",
      "26_gen_w_cov.16\n",
      " 26_gen_w_cov.4\n",
      "29_gen_w_cov.10\n",
      "29_gen_w_cov.19\n",
      "29_gen_w_cov.20\n",
      " 29_gen_w_cov.3\n",
      " 29_gen_w_cov.4\n",
      "   46_gen_cov.1\n",
      "   46_gen_cov.2\n",
      "   46_gen_cov.3\n",
      " 46_gen_w_cov.1\n",
      "46_gen_w_cov.25\n",
      "46_gen_w_cov.30\n",
      "46_gen_w_cov.31\n",
      " 46_gen_w_cov.5"
     ]
    }
   ],
   "source": [
    "!cat high_q_bins_list.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14_gen_cov.1.fa              29_gen_w_cov.16.fa\n",
      "14_gen_cov.1.fa_rrna.fa      29_gen_w_cov.16.fa_rrna.fa\n",
      "14_gen_cov.1.fa_rrna.gff     29_gen_w_cov.16.fa_rrna.gff\n",
      "14_gen_cov.2.fa              29_gen_w_cov.17.fa\n",
      "14_gen_cov.2.fa_rrna.fa      29_gen_w_cov.17.fa_rrna.fa\n",
      "14_gen_cov.2.fa_rrna.gff     29_gen_w_cov.17.fa_rrna.gff\n",
      "14_gen_w_cov.1.fa            29_gen_w_cov.18.fa\n",
      "14_gen_w_cov.1.fa_rrna.fa    29_gen_w_cov.18.fa_rrna.fa\n",
      "14_gen_w_cov.1.fa_rrna.gff   29_gen_w_cov.18.fa_rrna.gff\n",
      "14_gen_w_cov.10.fa           29_gen_w_cov.19.fa\n",
      "14_gen_w_cov.10.fa_rrna.fa   29_gen_w_cov.19.fa_rrna.fa\n",
      "14_gen_w_cov.10.fa_rrna.gff  29_gen_w_cov.19.fa_rrna.gff\n",
      "14_gen_w_cov.11.fa           29_gen_w_cov.2.fa\n",
      "14_gen_w_cov.11.fa_rrna.fa   29_gen_w_cov.2.fa_rrna.fa\n",
      "14_gen_w_cov.11.fa_rrna.gff  29_gen_w_cov.2.fa_rrna.gff\n",
      "14_gen_w_cov.12.fa           29_gen_w_cov.20.fa\n",
      "14_gen_w_cov.12.fa_rrna.fa   29_gen_w_cov.20.fa_rrna.fa\n",
      "14_gen_w_cov.12.fa_rrna.gff  29_gen_w_cov.20.fa_rrna.gff\n",
      "14_gen_w_cov.13.fa           29_gen_w_cov.21.fa\n",
      "14_gen_w_cov.13.fa_rrna.fa   29_gen_w_cov.21.fa_rrna.fa\n",
      "14_gen_w_cov.13.fa_rrna.gff  29_gen_w_cov.21.fa_rrna.gff\n",
      "14_gen_w_cov.14.fa           29_gen_w_cov.22.fa\n",
      "14_gen_w_cov.14.fa_rrna.fa   29_gen_w_cov.22.fa_rrna.fa\n",
      "14_gen_w_cov.14.fa_rrna.gff  29_gen_w_cov.22.fa_rrna.gff\n",
      "14_gen_w_cov.15.fa           29_gen_w_cov.3.fa\n",
      "14_gen_w_cov.15.fa_rrna.fa   29_gen_w_cov.3.fa_rrna.fa\n",
      "14_gen_w_cov.15.fa_rrna.gff  29_gen_w_cov.3.fa_rrna.gff\n",
      "14_gen_w_cov.16.fa           29_gen_w_cov.4.fa\n",
      "14_gen_w_cov.16.fa_rrna.fa   29_gen_w_cov.4.fa_rrna.fa\n",
      "14_gen_w_cov.16.fa_rrna.gff  29_gen_w_cov.4.fa_rrna.gff\n",
      "14_gen_w_cov.17.fa           29_gen_w_cov.5.fa\n",
      "14_gen_w_cov.17.fa_rrna.fa   29_gen_w_cov.5.fa_rrna.fa\n",
      "14_gen_w_cov.17.fa_rrna.gff  29_gen_w_cov.5.fa_rrna.gff\n",
      "14_gen_w_cov.18.fa           29_gen_w_cov.6.fa\n",
      "14_gen_w_cov.18.fa_rrna.fa   29_gen_w_cov.6.fa_rrna.fa\n",
      "14_gen_w_cov.18.fa_rrna.gff  29_gen_w_cov.6.fa_rrna.gff\n",
      "14_gen_w_cov.19.fa           29_gen_w_cov.7.fa\n",
      "14_gen_w_cov.19.fa_rrna.fa   29_gen_w_cov.7.fa_rrna.fa\n",
      "14_gen_w_cov.19.fa_rrna.gff  29_gen_w_cov.7.fa_rrna.gff\n",
      "14_gen_w_cov.2.fa            29_gen_w_cov.8.fa\n",
      "14_gen_w_cov.2.fa_rrna.fa    29_gen_w_cov.8.fa_rrna.fa\n",
      "14_gen_w_cov.2.fa_rrna.gff   29_gen_w_cov.8.fa_rrna.gff\n",
      "14_gen_w_cov.20.fa           29_gen_w_cov.9.fa\n",
      "14_gen_w_cov.20.fa_rrna.fa   29_gen_w_cov.9.fa_rrna.fa\n",
      "14_gen_w_cov.20.fa_rrna.gff  29_gen_w_cov.9.fa_rrna.gff\n",
      "14_gen_w_cov.21.fa           46_gen_cov.1.fa\n",
      "14_gen_w_cov.21.fa_rrna.fa   46_gen_cov.1.fa_rrna.fa\n",
      "14_gen_w_cov.21.fa_rrna.gff  46_gen_cov.1.fa_rrna.gff\n",
      "14_gen_w_cov.22.fa           46_gen_cov.2.fa\n",
      "14_gen_w_cov.22.fa_rrna.fa   46_gen_cov.2.fa_rrna.fa\n",
      "14_gen_w_cov.22.fa_rrna.gff  46_gen_cov.2.fa_rrna.gff\n",
      "14_gen_w_cov.3.fa            46_gen_cov.3.fa\n",
      "14_gen_w_cov.3.fa_rrna.fa    46_gen_cov.3.fa_rrna.fa\n",
      "14_gen_w_cov.3.fa_rrna.gff   46_gen_cov.3.fa_rrna.gff\n",
      "14_gen_w_cov.4.fa            46_gen_w_cov.1.fa\n",
      "14_gen_w_cov.4.fa_rrna.fa    46_gen_w_cov.1.fa_rrna.fa\n",
      "14_gen_w_cov.4.fa_rrna.gff   46_gen_w_cov.1.fa_rrna.gff\n",
      "14_gen_w_cov.5.fa            46_gen_w_cov.10.fa\n",
      "14_gen_w_cov.5.fa_rrna.fa    46_gen_w_cov.10.fa_rrna.fa\n",
      "14_gen_w_cov.5.fa_rrna.gff   46_gen_w_cov.10.fa_rrna.gff\n",
      "14_gen_w_cov.6.fa            46_gen_w_cov.11.fa\n",
      "14_gen_w_cov.6.fa_rrna.fa    46_gen_w_cov.11.fa_rrna.fa\n",
      "14_gen_w_cov.6.fa_rrna.gff   46_gen_w_cov.11.fa_rrna.gff\n",
      "14_gen_w_cov.7.fa            46_gen_w_cov.12.fa\n",
      "14_gen_w_cov.7.fa_rrna.fa    46_gen_w_cov.12.fa_rrna.fa\n",
      "14_gen_w_cov.7.fa_rrna.gff   46_gen_w_cov.12.fa_rrna.gff\n",
      "14_gen_w_cov.8.fa            46_gen_w_cov.13.fa\n",
      "14_gen_w_cov.8.fa_rrna.fa    46_gen_w_cov.13.fa_rrna.fa\n",
      "14_gen_w_cov.8.fa_rrna.gff   46_gen_w_cov.13.fa_rrna.gff\n",
      "14_gen_w_cov.9.fa            46_gen_w_cov.14.fa\n",
      "14_gen_w_cov.9.fa_rrna.fa    46_gen_w_cov.14.fa_rrna.fa\n",
      "14_gen_w_cov.9.fa_rrna.gff   46_gen_w_cov.14.fa_rrna.gff\n",
      "26_gen_w_cov.1.fa            46_gen_w_cov.15.fa\n",
      "26_gen_w_cov.1.fa_rrna.fa    46_gen_w_cov.15.fa_rrna.fa\n",
      "26_gen_w_cov.1.fa_rrna.gff   46_gen_w_cov.15.fa_rrna.gff\n",
      "26_gen_w_cov.10.fa           46_gen_w_cov.16.fa\n",
      "26_gen_w_cov.10.fa_rrna.fa   46_gen_w_cov.16.fa_rrna.fa\n",
      "26_gen_w_cov.10.fa_rrna.gff  46_gen_w_cov.16.fa_rrna.gff\n",
      "26_gen_w_cov.11.fa           46_gen_w_cov.17.fa\n",
      "26_gen_w_cov.11.fa_rrna.fa   46_gen_w_cov.17.fa_rrna.fa\n",
      "26_gen_w_cov.11.fa_rrna.gff  46_gen_w_cov.17.fa_rrna.gff\n",
      "26_gen_w_cov.12.fa           46_gen_w_cov.18.fa\n",
      "26_gen_w_cov.12.fa_rrna.fa   46_gen_w_cov.18.fa_rrna.fa\n",
      "26_gen_w_cov.12.fa_rrna.gff  46_gen_w_cov.18.fa_rrna.gff\n",
      "26_gen_w_cov.13.fa           46_gen_w_cov.19.fa\n",
      "26_gen_w_cov.13.fa_rrna.fa   46_gen_w_cov.19.fa_rrna.fa\n",
      "26_gen_w_cov.13.fa_rrna.gff  46_gen_w_cov.19.fa_rrna.gff\n",
      "26_gen_w_cov.14.fa           46_gen_w_cov.2.fa\n",
      "26_gen_w_cov.14.fa_rrna.fa   46_gen_w_cov.2.fa_rrna.fa\n",
      "26_gen_w_cov.14.fa_rrna.gff  46_gen_w_cov.2.fa_rrna.gff\n",
      "26_gen_w_cov.15.fa           46_gen_w_cov.20.fa\n",
      "26_gen_w_cov.15.fa_rrna.fa   46_gen_w_cov.20.fa_rrna.fa\n",
      "26_gen_w_cov.15.fa_rrna.gff  46_gen_w_cov.20.fa_rrna.gff\n",
      "26_gen_w_cov.16.fa           46_gen_w_cov.21.fa\n",
      "26_gen_w_cov.16.fa_rrna.fa   46_gen_w_cov.21.fa_rrna.fa\n",
      "26_gen_w_cov.16.fa_rrna.gff  46_gen_w_cov.21.fa_rrna.gff\n",
      "26_gen_w_cov.17.fa           46_gen_w_cov.22.fa\n",
      "26_gen_w_cov.17.fa_rrna.fa   46_gen_w_cov.22.fa_rrna.fa\n",
      "26_gen_w_cov.17.fa_rrna.gff  46_gen_w_cov.22.fa_rrna.gff\n",
      "26_gen_w_cov.18.fa           46_gen_w_cov.23.fa\n",
      "26_gen_w_cov.18.fa_rrna.fa   46_gen_w_cov.23.fa_rrna.fa\n",
      "26_gen_w_cov.18.fa_rrna.gff  46_gen_w_cov.23.fa_rrna.gff\n",
      "26_gen_w_cov.19.fa           46_gen_w_cov.24.fa\n",
      "26_gen_w_cov.19.fa_rrna.fa   46_gen_w_cov.24.fa_rrna.fa\n",
      "26_gen_w_cov.19.fa_rrna.gff  46_gen_w_cov.24.fa_rrna.gff\n",
      "26_gen_w_cov.2.fa            46_gen_w_cov.25.fa\n",
      "26_gen_w_cov.2.fa_rrna.fa    46_gen_w_cov.25.fa_rrna.fa\n",
      "26_gen_w_cov.2.fa_rrna.gff   46_gen_w_cov.25.fa_rrna.gff\n",
      "26_gen_w_cov.3.fa            46_gen_w_cov.26.fa\n",
      "26_gen_w_cov.3.fa_rrna.fa    46_gen_w_cov.26.fa_rrna.fa\n",
      "26_gen_w_cov.3.fa_rrna.gff   46_gen_w_cov.26.fa_rrna.gff\n",
      "26_gen_w_cov.4.fa            46_gen_w_cov.27.fa\n",
      "26_gen_w_cov.4.fa_rrna.fa    46_gen_w_cov.27.fa_rrna.fa\n",
      "26_gen_w_cov.4.fa_rrna.gff   46_gen_w_cov.27.fa_rrna.gff\n",
      "26_gen_w_cov.5.fa            46_gen_w_cov.28.fa\n",
      "26_gen_w_cov.5.fa_rrna.fa    46_gen_w_cov.28.fa_rrna.fa\n",
      "26_gen_w_cov.5.fa_rrna.gff   46_gen_w_cov.28.fa_rrna.gff\n",
      "26_gen_w_cov.6.fa            46_gen_w_cov.29.fa\n",
      "26_gen_w_cov.6.fa_rrna.fa    46_gen_w_cov.29.fa_rrna.fa\n",
      "26_gen_w_cov.6.fa_rrna.gff   46_gen_w_cov.29.fa_rrna.gff\n",
      "26_gen_w_cov.7.fa            46_gen_w_cov.3.fa\n",
      "26_gen_w_cov.7.fa_rrna.fa    46_gen_w_cov.3.fa_rrna.fa\n",
      "26_gen_w_cov.7.fa_rrna.gff   46_gen_w_cov.3.fa_rrna.gff\n",
      "26_gen_w_cov.8.fa            46_gen_w_cov.30.fa\n",
      "26_gen_w_cov.8.fa_rrna.fa    46_gen_w_cov.30.fa_rrna.fa\n",
      "26_gen_w_cov.8.fa_rrna.gff   46_gen_w_cov.30.fa_rrna.gff\n",
      "26_gen_w_cov.9.fa            46_gen_w_cov.31.fa\n",
      "26_gen_w_cov.9.fa_rrna.fa    46_gen_w_cov.31.fa_rrna.fa\n",
      "26_gen_w_cov.9.fa_rrna.gff   46_gen_w_cov.31.fa_rrna.gff\n",
      "29_gen_w_cov.1.fa            46_gen_w_cov.32.fa\n",
      "29_gen_w_cov.1.fa_rrna.fa    46_gen_w_cov.32.fa_rrna.fa\n",
      "29_gen_w_cov.1.fa_rrna.gff   46_gen_w_cov.32.fa_rrna.gff\n",
      "29_gen_w_cov.10.fa           46_gen_w_cov.4.fa\n",
      "29_gen_w_cov.10.fa_rrna.fa   46_gen_w_cov.4.fa_rrna.fa\n",
      "29_gen_w_cov.10.fa_rrna.gff  46_gen_w_cov.4.fa_rrna.gff\n",
      "29_gen_w_cov.11.fa           46_gen_w_cov.5.fa\n",
      "29_gen_w_cov.11.fa_rrna.fa   46_gen_w_cov.5.fa_rrna.fa\n",
      "29_gen_w_cov.11.fa_rrna.gff  46_gen_w_cov.5.fa_rrna.gff\n",
      "29_gen_w_cov.12.fa           46_gen_w_cov.6.fa\n",
      "29_gen_w_cov.12.fa_rrna.fa   46_gen_w_cov.6.fa_rrna.fa\n",
      "29_gen_w_cov.12.fa_rrna.gff  46_gen_w_cov.6.fa_rrna.gff\n",
      "29_gen_w_cov.13.fa           46_gen_w_cov.7.fa\n",
      "29_gen_w_cov.13.fa_rrna.fa   46_gen_w_cov.7.fa_rrna.fa\n",
      "29_gen_w_cov.13.fa_rrna.gff  46_gen_w_cov.7.fa_rrna.gff\n",
      "29_gen_w_cov.14.fa           46_gen_w_cov.8.fa\n",
      "29_gen_w_cov.14.fa_rrna.fa   46_gen_w_cov.8.fa_rrna.fa\n",
      "29_gen_w_cov.14.fa_rrna.gff  46_gen_w_cov.8.fa_rrna.gff\n",
      "29_gen_w_cov.15.fa           46_gen_w_cov.9.fa\n",
      "29_gen_w_cov.15.fa_rrna.fa   46_gen_w_cov.9.fa_rrna.fa\n",
      "29_gen_w_cov.15.fa_rrna.gff  46_gen_w_cov.9.fa_rrna.gff\n"
     ]
    }
   ],
   "source": [
    "ls mags/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir h_mags"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "!for file in $(<high_q_bins_list.txt); do cp mags/$file\\.fa h_mags; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14_gen_cov.2.fa     26_gen_w_cov.15.fa\t29_gen_w_cov.3.fa  46_gen_w_cov.25.fa\n",
      "14_gen_w_cov.17.fa  26_gen_w_cov.16.fa\t29_gen_w_cov.4.fa  46_gen_w_cov.30.fa\n",
      "14_gen_w_cov.19.fa  26_gen_w_cov.4.fa\t46_gen_cov.1.fa    46_gen_w_cov.31.fa\n",
      "14_gen_w_cov.2.fa   29_gen_w_cov.10.fa\t46_gen_cov.2.fa    46_gen_w_cov.5.fa\n",
      "26_gen_w_cov.11.fa  29_gen_w_cov.19.fa\t46_gen_cov.3.fa\n",
      "26_gen_w_cov.14.fa  29_gen_w_cov.20.fa\t46_gen_w_cov.1.fa\n"
     ]
    }
   ],
   "source": [
    "!ls h_mags/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "!for file in $(<high_q_bins_list.txt); do cp mags/$file\\.fa_rrna.fa  h_mags; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14_gen_cov.2.fa\t\t    29_gen_w_cov.20.fa\n",
      "14_gen_cov.2.fa_rrna.fa     29_gen_w_cov.20.fa_rrna.fa\n",
      "14_gen_w_cov.17.fa\t    29_gen_w_cov.3.fa\n",
      "14_gen_w_cov.17.fa_rrna.fa  29_gen_w_cov.3.fa_rrna.fa\n",
      "14_gen_w_cov.19.fa\t    29_gen_w_cov.4.fa\n",
      "14_gen_w_cov.19.fa_rrna.fa  29_gen_w_cov.4.fa_rrna.fa\n",
      "14_gen_w_cov.2.fa\t    46_gen_cov.1.fa\n",
      "14_gen_w_cov.2.fa_rrna.fa   46_gen_cov.1.fa_rrna.fa\n",
      "26_gen_w_cov.11.fa\t    46_gen_cov.2.fa\n",
      "26_gen_w_cov.11.fa_rrna.fa  46_gen_cov.2.fa_rrna.fa\n",
      "26_gen_w_cov.14.fa\t    46_gen_cov.3.fa\n",
      "26_gen_w_cov.14.fa_rrna.fa  46_gen_cov.3.fa_rrna.fa\n",
      "26_gen_w_cov.15.fa\t    46_gen_w_cov.1.fa\n",
      "26_gen_w_cov.15.fa_rrna.fa  46_gen_w_cov.1.fa_rrna.fa\n",
      "26_gen_w_cov.16.fa\t    46_gen_w_cov.25.fa\n",
      "26_gen_w_cov.16.fa_rrna.fa  46_gen_w_cov.25.fa_rrna.fa\n",
      "26_gen_w_cov.4.fa\t    46_gen_w_cov.30.fa\n",
      "26_gen_w_cov.4.fa_rrna.fa   46_gen_w_cov.30.fa_rrna.fa\n",
      "29_gen_w_cov.10.fa\t    46_gen_w_cov.31.fa\n",
      "29_gen_w_cov.10.fa_rrna.fa  46_gen_w_cov.31.fa_rrna.fa\n",
      "29_gen_w_cov.19.fa\t    46_gen_w_cov.5.fa\n",
      "29_gen_w_cov.19.fa_rrna.fa  46_gen_w_cov.5.fa_rrna.fa\n"
     ]
    }
   ],
   "source": [
    "!ls h_mags/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "35\n"
     ]
    }
   ],
   "source": [
    "!cat h_mags/*.fa_rrna.fa | grep '>16S' | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!sed -n '/^12$/,/^15$/p'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat h_mags/*.fa_rrna.fa | sed -n '/^\\>16S$/,/^\\>$/p'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">16S_rRNA::c109:2328-3869(-)\n",
      "TGAAGAGTTTGATCCTGGCTCAGAGTGAACGCTGGCGGTAGGCCTAACACATGCAAGTCGAACGGCAGCACAGGAGAGCTTGCTCTCTGGGTGGCGAGTGGCGGACGGGTGAGGAACACATCGGAATCTACTCTGTCGTGGGGGATAACGTAGGGAAACTTACGCTAATACCGCATACGACCTACGGGTGAAAGTGGGGGACCTTCGGGCCTCACGCGATTGAATGAGCCGATGTCGGATTAGCTAGTTGGCGGGGTAAAGGCCCACCAAGGCGACGATCCGTAGCTGGTCTGAGAGGATGATCAGCCACACTGGAACTGAGACACGGTCCAGACTCCTACGGGAGGCAGCAGTGGGGAATATTGGACAATGGGCGCAAGCCTGATCCAGCCATACCGCGTGGGTGAAGAAGGCCTTCGGGTTGTAAAGCCCTTTTGTTGGGAAAGAAATCCAGCCGGTTAATACCTGGTTGGGATGACGGTACCCAAAGAATAAGCACCGGCTAACTTCGTGCCAGCAGCCGCGGTAATACGAAGGGTGCAAGCGTTACTCGGAATTACTGGGCGTAAAGCGTGCGTAGGTGGTGATTTAAGTCCGTTGTGAAAGCCCTGGGCTCAACCTGGGAATTGCAGTGGATACTGGGTCACTAGAGTGTGGTAGAGGGTGGCGGAATTCCCGGTGTAGCAGTGAAATGCGTAGAGATCGGGAGGAACATCCGTGGCGAAGGCGGCCACCTGGGCCAACACTGACACTGAGGCACGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCCTAAACGATGCGAACTGGATGTTGGGTTCAACTTGGAACCCAGTATCGAAGCTAACGCGTTAAGTTCGCCGCCTGGGGAGTACGGTCGCAAGACTGAAACTCAAAGGAATTGACGGGGGCCCGCACAAGCGGTGGAGTATGTGGTTTAATTCGATGCAACGCGAAGAACCTTACCTGGTCTTGACATCCACGGAACTTTCCAGAGATGGATTGGTGCCTTCGGGAACCGTGAGACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTCGTGAGATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTGTCCTTAGTTGCCAGCACGTAATGGTGGGAACTCTAAGGAGACCGCCGGTGACAAACCGGAGGAAGGTGGGGATGACGTCAAGTCATCATGGCCCTTACGACCAGGGCTACACACGTACTACAATGGTGGGGACAGAGGGCTGCAATCCCGCGAGGGTGAGCCAATCCCAGAAACCCTATCTCAGTCCGGATTGGAGTCTGCAACTCGACTCCATGAAGTCGGAATCGCTAGTAATCGCAGATCAGCATTGCTGCGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCATGGGAGTTTGTTGCACCAGAAGCAGGTAGCTTAACCTTCGGGAGGGCGCTTGCCACGGTGTGGCCGATGACTGGGGTGAAGTCGTAACAAGGTAGCCGTATCGGAAGGTGCGGCTGGATCACCTCCTTT\n",
      ">16S_rRNA::c1812:25671-27212(-)\n",
      "TGAAGAGTTTGATCCTGGCTCAGAGTGAACGCTGGCGGTAGGCCTAACACATGCAAGTCGAACGGCAGCACAGGAGAGCTTGCTCTCTGGGTGGCGAGTGGCGGACGGGTGAGGAACACATCGGAATCTACTCTGTCGTGGGGGATAACGTAGGGAAACTTACGCTAATACCGCATACGACCTACGGGTGAAAGTGGGGGACCTTCGGGCCTCACGCGATTGAATGAGCCGATGTCGGATTAGCTAGTTGGCGGGGTAAAGGCCCACCAAGGCGACGATCCGTAGCTGGTCTGAGAGGATGATCAGCCACACTGGAACTGAGACACGGTCCAGACTCCTACGGGAGGCAGCAGTGGGGAATATTGGACAATGGGCGCAAGCCTGATCCAGCCATACCGCGTGGGTGAAGAAGGCCTTCGGGTTGTAAAGCCCTTTTGTTGGGAAAGAAATCCAGCCGGTTAATACCTGGTTGGGATGACGGTACCCAAAGAATAAGCACCGGCTAACTTCGTGCCAGCAGCCGCGGTAATACGAAGGGTGCAAGCGTTACTCGGAATTACTGGGCGTAAAGCGTGCGTAGGTGGTGATTTAAGTCCGTTGTGAAAGCCCTGGGCTCAACCTGGGAATTGCAGTGGATACTGGGTCACTAGAGTGTGGTAGAGGGTGGCGGAATTCCCGGTGTAGCAGTGAAATGCGTAGAGATCGGGAGGAACATCCGTGGCGAAGGCGGCCACCTGGGCCAACACTGACACTGAGGCACGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCCTAAACGATGCGAACTGGATGTTGGGTTCAACTTGGAACCCAGTATCGAAGCTAACGCGTTAAGTTCGCCGCCTGGGGAGTACGGTCGCAAGACTGAAACTCAAAGGAATTGACGGGGGCCCGCACAAGCGGTGGAGTATGTGGTTTAATTCGATGCAACGCGAAGAACCTTACCTGGTCTTGACATCCACGGAACTTTCCAGAGATGGATTGGTGCCTTCGGGAACCGTGAGACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTCGTGAGATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTGTCCTTAGTTGCCAGCACGTAATGGTGGGAACTCTAAGGAGACCGCCGGTGACAAACCGGAGGAAGGTGGGGATGACGTCAAGTCATCATGGCCCTTACGACCAGGGCTACACACGTACTACAATGGTGGGGACAGAGGGCTGCAATCCCGCGAGGGTGAGCCAATCCCAGAAACCCTATCTCAGTCCGGATTGGAGTCTGCAACTCGACTCCATGAAGTCGGAATCGCTAGTAATCGCAGATCAGCATTGCTGCGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCATGGGAGTTTGTTGCACCAGAAGCAGGTAGCTTAACCTTCGGGAGGGCGCTTGCCACGGTGTGGCCGATGACTGGGGTGAAGTCGTAACAAGGTAGCCGTATCGGAAGGTGCGGCTGGATCACCTCCTTT\n",
      ">16S_rRNA::c550:31-1485(+)\n",
      "CGAGAGTGGCGGACGGGTGAAGGACACATCGGAATCTACTCTGTCGTGGGGGATAACGTAGGGAAACTTACGCTAATACCGCATACGACCTACGGGTGAAAAAGTGGGGGACCTTCGGGCCTCACGCGATTGAATGAGCCGATGTCGGATTAGCTAGTTGGCGGGGTAAAGGCCCACCCAAGGCGACGATCCGTAGCTGGTCTGAGAGGATGATCAGCCACACTGGAACTGAGACACGGTCCAGACTCCTACGGGGAGGCAGCAGTGGGGAATATTGGACAATGGGCGCAAGCCTGATCCAGCCATACCGCGTGGGTGAAGAAGGCCTTCGGGTTGTAAAGCCCTTTTGTTGGGAAAGAAATCCAGCCGGTTAATACCTGGTTGGGATGACGGTACCCAAAGAATAAGCACCGGCTAACTTCGTGCCAGCAGCCGCGGTAATACGAAGGGTGCAAGCGTTACTCGGAATTACTGGGCGTAAAGCGTGCGTAGGTGGTGATTTAAGTCCGTTGTGAAAGCCCTGGGCTCAACCTGGGAATTGCAGTGGATACTGGGTCACTAGAGTGTGGTAGAGGGTGGCGGAATTCCCGGTGTAGCAGTGAAATGCGTAGAGATCGGGAGGAACATCCGTGGCGAAGGCGGCCACCTGGGCCAACACTGACACTGAGGCACGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTGGGTCCACGCCCTAAACGATGCGAACTGGATGTTGGGTTCAACTTGGAACCCAGTATCGAAGCTAACGCGTTAAGTTCGCCGCCTGGGGAGTACGGTCGCAAGACTGAAACTCAAAGGAATTGACGGGGGCCCGCACAAGCGGTGGAGTATGTGGTTTAATTCGATGCAACGCGAAGAACCTTACCTGGTCTTGACATCCACGGAACTTTCCAGAGATGGATTGGTGCCTTCGGGAACCGTGAGACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTCGTGAGATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTGTCCTTAGTTGCCAGCACGTAATGGTGGGAACTCTAAGGAGACCGCCGGTGACAAACCGGAGGAAGGTGGGGATGACGTCAAGTCATCATGGCCCTTACGACCAGGGCTACACACGTACTACAATGGTGGGGACAGAGGGCTGCAATCCCGCGAGGGTGAGCCAATCCCAGAAACCCTATCTCAGTCCGGATTGGAGTCTGCAACTCGACTCCATGAAGTCGGAATCGCTAGTAATCGCAGATCAGCATTGCTGCGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCATGGGAGTTTGTTGCACCAGAAGCAGGTAGCTTAACCTTCGGGAGGGCGCTTGCCACGGTGTGGCCGATGACTGGGGTGAAGTCGTAACAAGGTAGCCGTATCGGAAGGTGCGGCTGGATCACCTCCTTT\n",
      ">23S_rRNA::c1812:22249-25124(-)\n",
      "TCAAGCGAATAAGCGCACACGGTGGATGCCTTGGCGGTCAGAGGCGATGAAGGACGTGACAGCCTGCGAAAAGCGCGGGGGAGCTGGCAATAAGCTTTGATCCCGCGATGTCCGAATGGGGAAACCCACTGCTTCGGCAGTATCCTGCAGTGAATACATAGCTGCTGGAAGCGAACCCGGTGAACTGAAATATCTAAGTAACCGGAGGAAAAGAAATCAACCGAGATTCCCTAAGTAGTGACGAGCGAACGGGGACTAGCCCTTAAGCTGGAATGGTTCTAGGAAAACAACCTGGAAAGGTTGGCCATAGAAGGTGACAGCCCTGTATTCGAAAGGGCCACTCCAGTGAAGACGAGTAAGGCGGGGCACGTGAAACCCTGTCCGAACATGGGGGGACCATCCTCCAAGGCTAAATACTCCTGACCGACCGATAGTGAACCAGTACCGTGAGGGAAAGGCGAAAAGAACCCCGGAGAGGGGAGTGAAATAGACCCTGAAACCGTGTGCGTACAAGCAGTAGGAGCCCGCAAGGGTGACTGCGTACCTTTTGTATAATGGGTCAGCGACTTACAGTTCGTGGCAAGCTTAACCGTATAGGGGAGGCGAAGGGAAACCGAGTCTGAATAGGGCGCATAGTCGCGGGCTGTAGACCCGAAACCGGGTGATCTAGTCATGCCCAGGGTGAAGGTGCCGTAACAGGTACTGGAGGCCCGAACCCACTCCCGTTGCAAAGGTAGGGGATGAGGTGTGATTAGGAGTGAAAAGCTAATCGAACCCGGAGATAGCTGGTTCTCCTCGAAAGCTATTTAGGTAGCGCCTCATATGTATCCTCTCGGGGGTAGAGCACTGTTATGGCTAGGGGGTCATCGCGACTTACCAAACCATTGCAAACTCCGAATACCGAGACGGACTGTATGGGAGACACACGGCGGGTGCTAACGTCCGTCGTGAAAAGGGAAACAACCCAGACCCACAGCTAAGGTCCCAAATTACCGCTAAGTGGGAAACGATGTGGAAAGGCACAGACAGCCAGGAGGTTGGCTTAGAAGCAGCCACCCTTTAAAGAAAGCGTAATAGCTCACTGGTCGAGTCGGTCTGCGCGGAAGATTTAACGGGGCTAAGCGGTAAACCGAAGCTTGGGGTGCACAACTTGTTTGTGCGCGGTAGAGGAGCGTTCCGTAAGCCTGTGAAGGTGGATTGAGAAGTCTGCTGGAGGTATCGGAAGTGCGAATGCTGACATGAGTAACGATAATGCGGGTGAAAAGCCCGCACGCCGAAAGCCCAAGGTTTCCTTGCGCAACGTTAATCGACGCAGGGTTAGTCGGTCCCTAAGGCGAGGGCGAAAGCCGTAGTCGATGGGAAGCAGGTTAATATTCCTGCACCTCGCGTGAGTGCGATGGAGGGACGGAGAAGGTTAGGTGTACCGGGCGTTGGTTGTCCCGGGGAAAGGCGGTAGGTTTGGATCTTTGGCAAATCCGGGATCCTTTAAGACCGAGCACCGAGACGAGCCCTTTAGGGCGAAGTCACTGATACCACGCTTCCAGGAAAAGCTCCTAAGCTTCAGCTCACGCAGACCGTACCGTAAACCGACACAGGTGGGTAGGATGAGAATTCTCAGGCGCTTGAGAGAACTCGGGTGAAGGAACTAGGCAACATGGCACCGTAACTTCGGGAGAAGGTGCACCCCCTTTGGTGGCCCATGCGGGCTATAGCTGAGGGGGGTCGCAGAAACCAGGCCGCTGCGACTGTTTATCAAAAACACAGCACTCTGCAAACACGAAAGTGGACGTATAGGGTGTGACGCCTGCCCGGTGCTGGAAGGTTAATTGATGGGGTCAGCCGCAAGGCGAAGCTCTTGATCGAAGCCCCAGTAAACGGCGGCCGTAACTATAACGGTCCTAAGGTAGCGAAATTCCTTGTCGGGTAAGTTCCGACCTGCACGAATGGCGTAACGACAGCGGCGCTGTCTCCACCCGAGACTCAGTGAAATTGAAATCGCTGTGAAGATGCAGCGTTCCCGTGGCAAGACGGAAAGACCCCGTGAACCTTTACTATAGCTTTACATTGAACGTTGAGTTCGTCTGTGTAGGATAGGTGGGAGGCTTTGAAGTGTCGGCGCTAGCTGGCATGGAGCCATCCTTGAAATACCACCCTGATGTGCTTGACGTTCTAACCTGGGCCCGTAATCCGGGCCGGGGACCATGTATGGTGGGTAGTTTGACTGGGGCGGTCTCCTCCCAAAGAGTAACGGAGGAGCTCGAAGGTACGCTCAGCGCGGTCGGACATCGCGCACTGTGTGCAAAGGCATAAGCGTGCTTGACTGCAAGATCGACGGATCAAGCAGGTACGAAAGTAGGACTTAGTGATCCGGTGGTTCTGTATGGAAGGGCCATCGCTCAACGGATAAAAGGTACTCCGGGGATAACAGGCTGATACCGCCCAAGAGTTCATATCGACGGCGGTGTTTGGCACCTCGATGTCGGCTCATCACATCCTGGGGCTGTAGTCGGTCCCAAGGGTATGGCTGTTCGCCATTTAAAGTGGTACGCGAGCTGGGTTCAGAACGTCGTGAGACAGTTCGGTCCCTATCTGCCATGGGCGTTGGAGATTTGAGAGGGGCTGCTCCTAGTACGAGAGGACCGGAGTGGACGAACCTCTGGTGTTCCGGTTGTCACGCCAGTGGCATTGCCGGGTAGCTATGTTCGGAAGCGATAACCGCTGAAAGCATCTAAGCGGGAAGCGCGCCTCAAGATGAGATCTCCCGGGACACAAGTCCCCTAAAGGAACCATCAAGACTAGGTGGTTGATAGGTCAGGTGTGTAAGCGCAGCAATGCGTTGAGCTAACTGATACTAATGATCCGTGTGGCTTGACCA\n",
      ">23S_rRNA::c550:2033-4908(+)\n",
      "TCAAGCGAATAAGCGCACACGGTGGATGCCTTGGCGGTCAGAGGCGATGAAGGACGTGACAGCCTGCGAAAAGCGCGGGGGAGCTGGCAATAAGCTTTGATCCCGCGATGTCCGAATGGGGAAACCCACTGCTTCGGCAGTATCCTGCAGTGAATACATAGCTGCTGGAAGCGAACCCGGTGAACTGAAATATCTAAGTAACCGGAGGAAAAGAAATCAACCGAGATTCCCTAAGTAGTGACGAGCGAACGGGGACTAGCCCTTAAGCTGGAATGGTTCTAGGAAAACAACCTGGAAAGGTTGGCCATAGAAGGTGACAGCCCTGTATTCGAAAGGGCCACTCCAGTGAAGACGAGTAAGGCGGGGCACGTGAAACCCTGTCCGAACATGGGGGGGACCATCCTCCAAGGCTAAATACTCCTGACCGACCGATAGTGAACCAGTACCGTGAGGGAAAGGCGAAAAGAACCCCGGAGAGGGGAGTGAAATAGACCCTGAAACCGTGTGCGTACAAGCAGTAGGAGCCCGCAAGGGTGACTGCGTACCTTTTGTATAATGGGTCAGCGACTTACAGTTCGTGGCAAGCTTAACCGTATAGGGGAGGCGAAGGGAAACCGAGTCTGAATAGGGCGCATAGTCGCGGGCTGTAGACCCGAAACCGGGTGATCTAGTCATGCCCAGGGTGAAGGTGCCGTAACAGGTACTGGAGGCCCGAACCCACTCCCGTTGCAAAGGTAGGGGATGAGGTGTGATTAGGAGTGAAAAGCTAATCGAACCCGGAGATAGCTGGTTCTCCTCGAAAGCTATTTAGGTAGCGCCTCATATGTATCCTCTCGGGGGTAGAGCACTGTTATGGCTAGGGGGTCATCGCGACTTACCAAACCATTGCAAACTCCGAATACCGAGACGGACTGTATGGGAGACACACGGCGGGTGCTAACGTCCGTCGTGAAAAGGGAAACAACCCAGACCCACAGCTAAGGTCCCAAATTACCGCTAAGTGGGAAACGATGTGGAAAGGCACAGACAGCCAGGAGGTTGGCTTAGAAGCAGCCACCCTTTAAAGAAAGCGTAATAGCTCACTGGTCGAGTCGGTCTGCGCGGAAGATTTAACGGGGCTAAGCGGTAAACCGAAGCTTGGGGTGCACAACTTGTTTGTGCGCGGTAGAGGAGCGTTCCGTAAGCCTGTGAAGGTGGATTGAGAAGTCTGCTGGAGGTATCGGAAGTGCGAATGCTGACATGAGTAACGATAATGCGGGTGAAAAGCCCGCACGCCGAAAGCCCAAGGTTTCCTTGCGCAACGTTAATCGACGCAGGGTTAGTCGGTCCCTAAGGCGAGGGCGAAAGCCGTAGTCGATGGGAAGCAGGTTAATATTCCTGCACCTCGCGTGAGTGCGATGGAGGGACGGAGAAGGTTAGGTGTACCGGGCGTTGGTTGTCCCGGGGAAAGGCGGTAGGTTTGGATCTTTGGCAAATCCGGGATCCTTTAAGACCGAGCACCGAGACGAGCCCTTTAGGGCGAAGTCACTGATACCACGCTTCCAGGAAAAGCTCCTAAGCTTCAGCTCACGCAGACCGTACCGTAAACCGACACAGGTGGGTAGGATGAGAATTCTCAGGCGCTTGAGAGAACTCGGGTGAAGGAACTAGGCAACATGGCACCGTAACTTCGGGAGAAGGTGCACCCCCTTTGGTGGCCCATGCGGGCTATAGCTGAGGGGGTCGCAGAAACCAGGCCGCTGCGACTGTTTATCAAAAACACAGCACTCTGCAAACACGAAAGTGGACGTATAGGGTGTGACGCCTGCCCGGTGCTGGAAGGTTAATTGATGGGGTCAGCCGCAAGGCGAAGCTCTTGATCGAAGCCCCAGTAAACGGCGGCCGTAACTATAACGGTCCTAAGGTAGCGAAATTCCTTGTCGGGTAAGTTCCGACCTGCACGAATGGCGTAACGACAGCGGCGCTGTCTCCACCCGAGACTCAGTGAAATTGAAATCGCTGTGAAGATGCAGCGTTCCCGTGGCAAGACGGAAAGACCCCGTGAACCTTTACTATAGCTTTACATTGAACGTTGAGTTCGTCTGTGTAGGATAGGTGGGAGGCTTTGAAGTGTCGGCGCTAGCTGGCATGGAGCCATCCTTGAAATACCACCCTGATGTGCTTGACGTTCTAACCTGGGCCCGTAATCCGGGCCGGGGACCATGTATGGTGGGTAGTTTGACTGGGGCGGTCTCCTCCCAAAGAGTAACGGAGGAGCTCGAAGGTACGCTCAGCGCGGTCGGACATCGCGCACTGTGTGCAAAGGCATAAGCGTGCTTGACTGCAAGATCGACGGATCAAGCAGGTACGAAAGTAGGACTTAGTGATCCGGTGGTTCTGTATGGAAGGGCCATCGCTCAACGGATAAAAGGTACTCCGGGGATAACAGGCTGATACCGCCCAAGAGTTCATATCGACGGCGGTGTTTGGCACCTCGATGTCGGCTCATCACATCCTGGGGCTGTAGTCGGTCCCAAGGGTATGGCTGTTCGCCATTTAAAGTGGTACGCGAGCTGGGTTCAGAACGTCGTGAGACAGTTCGGTCCCTATCTGCCATGGGCGTTGGAGATTTGAGAGGGGCTGCTCCTAGTACGAGAGGACCGGAGTGGACGAACCTCTGGTGTTCCGGTTGTCACGCCAGTGGCATTGCCGGGTAGCTATGTTCGGAAGCGATAACCGCTGAAAGCATCTAAGCGGGAAGCGCGCCTCAAGATGAGATCTCCCGGGACACAAGTCCCCTGAAGGAACCATCAAGACTAGGTGGTTGATAGGTCAGGTGTGTAAGCGCAGCAATGCGTTGAGCTAACTGATACTAATGATCCGTGTGGCTTGACCA\n",
      ">23S_rRNA::c109:0-1781(-)\n",
      "TCAAGCGAATAAGCGCACACGGTGGATGCCTTGGCGGTCAGAGGCGATGAAGGACGTGACAGCCTGCGAAAAGCGCGGGGGAGCTGGCAATAAGCTTTGATCCCGCGATGTCCGAATGGGGAAACCCACTGCTTCGGCAGTATCCTGCAGTGAATACATAGCTGCTGGAAGCGAACCCGGTGAACTGAAATATCTAAGTAACCGGAGGAAAAGAAATCAACCGAGATTCCCTAAGTAGTGACGAGCGAACGGGGACTAGCCCTTAAGCTGGAATGGTTCTAGGAAAACAACCTGGAAAGGTTGGCCATAGAAGGTGACAGCCCTGTATTCGAAAGGGCCACTCCAGTGAAGACGAGTAAGGCGGGGCACGTGAAACCCTGTCCGAACATGGGGGGACCATCCTCCAAGGCTAAATACTCCTGACCGACCGATAGTGAACCAGTACCGTGAGGGAAAGGCGAAAAGAACCCCGGAGAGGGGAGTGAAATAGACCCTGAAACCGTGTGCGTACAAGCAGTAGGAGCCCGCAAGGGTGACTGCGTACCTTTTGTATAATGGGTCAGCGACTTACAGTTCGTGGCAAGCTTAACCGTATAGGGGAGGCGAAGGGAAACCGAGTCTGAATAGGGCGCATAGTCGCGGGCTGTAGACCCGAAACCGGGTGATCTAGTCATGCCCAGGGTGAAGGTGCCGTAACAGGTACTGGAGGCCCGAACCCACTCCCGTTGCAAAGGTAGGGGATGAGGTGTGATTAGGAGTGAAAAGCTAATCGAACCCGGAGATAGCTGGTTCTCCTCGAAAGCTATTTAGGTAGCGCCTCATATGTATCCTCTCGGGGGTAGAGCACTGTTATGGCTAGGGGGTCATCGCGACTTACCAAACCATTGCAAACTCCGAATACCGAGACGGACTGTATGGGAGACACACGGCGGGTGCTAACGTCCGTCGTGAAAAGGGAAACAACCCAGACCCACAGCTAAGGTCCCAAATTACCGCTAAGTGGGAAACGATGTGGAAAGGCACAGACAGCCAGGAGGTTGGCTTAGAAGCAGCCACCCTTTAAAGAAAGCGTAATAGCTCACTGGTCGAGTCGGTCTGCGCGGAAGATTTAACGGGGCTAAGCGGTAAACCGAAGCTTGGGGTGCACAACTTGTTTGTGCGCGGTAGAGGAGCGTTCCGTAAGCCTGTGAAGGTGGATTGAGAAGTCTGCTGGAGGTATCGGAAGTGCGAATGCTGACATGAGTAACGATAATGCGGGTGAAAAGCCCGCACGCCGAAAGCCCAAGGTTTCCTTGCGCAACGTTAATCGACGCAGGGTTAGTCGGTCCCTAAGGCGAGGGCGAAAGCCGTAGTCGATGGGAAGCAGGTTAATATTCCTGCACCTCGCGTGAGTGCGATGGAGGGACGGAGAAGGTTAGGTGTACCGGGCGTTGGTTGTCCCGGGGAAAGGCGGTAGGTTTGGATCTTTGGCAAATCCGGGATCCTTTAAGACCGAGCACCGAGACGAGCCCTTTAGGGCGAAGTCACTGATACCACGCTTCCAGGAAAAGCTCCTAAGCTTCAGCTCACGCAGACCGTACCGTAAACCGACACAGGTGGGTAGGATGAGAATTCTCAGGCGCTTGAGAGAACTCGGGTGAAGGAACTAGGCAACATGGCACCGTAACTTCGGGAGAAGGTGCACCCCTTTGGTGGCCCATGCGGGCTATAGCTGAGGGGGTCGCAGAAACCAGGCCGCTGCGACTGTTTATCAAAAACACAGCACTCTGCAAACACGAAAG\n",
      ">5S_rRNA::c1812:21996-22099(-)\n",
      "AAATTAGCGTTGTGGAACCACCCGATCCCATCCCGAACTCGGAAGTGAAACGCAACAGCGCCGATGGTAGTGTGGGCTTCCCATGCGAGAGTAGGTCATCGCC\n",
      ">5S_rRNA::c550:5058-5161(+)\n",
      "AAATTAGCGTTGTGGAACCACCCGATCCCATCCCGAACTCGGAAGTGAAACGCAACAGCGCCGATGGTAGTGTGGGCTTCCCATGCGAGAGTAGGTCATCGCC\n"
     ]
    }
   ],
   "source": [
    "!cat h_mags/14_gen_cov.2.fa_rrna.fa"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:vamb]",
   "language": "python",
   "name": "conda-env-vamb-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
