{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.SeqFeature import SeqFeature, FeatureLocation\n",
    "from Bio import SeqIO\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import os\n",
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/storage/metagenome/bact_4x/new_con/diamond\n",
      "14.0.faa\t26.0.faa_dbcan\t29.1.faa\t46.1.faa_dbcan\tleft\n",
      "14.0.faa_dbcan\t26.1.faa\t29.1.faa_dbcan\t46.faa.flat\tright\n",
      "14.1.faa\t26.1.faa_dbcan\t29.faa.flat\t46.faa.gdx\ttemp.57557\n",
      "14.1.faa_dbcan\t26.faa.flat\t29.faa.gdx\t46_nice.tsv\ttmp1\n",
      "14.faa.flat\t26.faa.gdx\t29_nice.tsv\t46_prokka\ttmp2\n",
      "14.faa.gdx\t26_nice.tsv\t29_prokka\tHotpep\n",
      "14_nice.tsv\t26_prokka\t46.0.faa\tcon\n",
      "14_prokka\t29.0.faa\t46.0.faa_dbcan\tdb\n",
      "26.0.faa\t29.0.faa_dbcan\t46.1.faa\tfake_otu\n"
     ]
    }
   ],
   "source": [
    "%cd ~/storage/metagenome/bact_4x/new_con/diamond/\n",
    "!ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Prokka import\n",
    "\n",
    "Import common info about contig and gene id "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### NOT RUN\n",
    "\n",
    "gbk SeqIO based parser \\\n",
    "This parser didnt work correctly. Too low number of seqs. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-3-5db6f345bab2>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      8\u001b[0m     \u001b[0mprokka_gene_annotation\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m     \u001b[0mprokka_gene_id\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m     \u001b[0;32mfor\u001b[0m \u001b[0mrecord\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mSeqIO\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparse\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mf'{f}'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"genbank\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrecord\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfeatures\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtype\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"CDS\"\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;34m\"gene\"\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mqualifiers\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/Bio/SeqIO/Interfaces.py\u001b[0m in \u001b[0;36m__next__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     71\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__next__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     72\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 73\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mnext\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecords\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     74\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mException\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     75\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshould_close_stream\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/Bio/GenBank/Scanner.py\u001b[0m in \u001b[0;36mparse_records\u001b[0;34m(self, handle, do_features)\u001b[0m\n\u001b[1;32m    514\u001b[0m         \u001b[0;32mwith\u001b[0m \u001b[0mas_handle\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mhandle\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    515\u001b[0m             \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 516\u001b[0;31m                 \u001b[0mrecord\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparse\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdo_features\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    517\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mrecord\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    518\u001b[0m                     \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/Bio/GenBank/Scanner.py\u001b[0m in \u001b[0;36mparse\u001b[0;34m(self, handle, do_features)\u001b[0m\n\u001b[1;32m    497\u001b[0m         )\n\u001b[1;32m    498\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 499\u001b[0;31m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfeed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mconsumer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdo_features\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    500\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mconsumer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    501\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/Bio/GenBank/Scanner.py\u001b[0m in \u001b[0;36mfeed\u001b[0;34m(self, handle, consumer, do_features)\u001b[0m\n\u001b[1;32m    468\u001b[0m         \u001b[0;31m# Features (common to both EMBL and GenBank):\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    469\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mdo_features\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 470\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_feed_feature_table\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mconsumer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparse_features\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mskip\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    471\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    472\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparse_features\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mskip\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# ignore the data\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/Bio/GenBank/Scanner.py\u001b[0m in \u001b[0;36m_feed_feature_table\u001b[0;34m(consumer, feature_tuples)\u001b[0m\n\u001b[1;32m    423\u001b[0m                     \u001b[0mconsumer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfeature_qualifier\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mq_key\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mq_value\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    424\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 425\u001b[0;31m                     \u001b[0mconsumer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfeature_qualifier\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mq_key\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mq_value\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreplace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"\\n\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\" \"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    426\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    427\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_feed_misc_lines\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mconsumer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlines\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/Bio/GenBank/__init__.py\u001b[0m in \u001b[0;36mfeature_qualifier\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   1281\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1282\u001b[0m         \u001b[0;31m# Remove enclosing quotation marks\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1283\u001b[0;31m         \u001b[0mvalue\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mre\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msub\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'^\"|\"$'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1284\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1285\u001b[0m         \u001b[0;31m# Handle NCBI escaping\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/re.py\u001b[0m in \u001b[0;36msub\u001b[0;34m(pattern, repl, string, count, flags)\u001b[0m\n\u001b[1;32m    208\u001b[0m     \u001b[0ma\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mit\u001b[0m\u001b[0;31m'\u001b[0m\u001b[0ms\u001b[0m \u001b[0mpassed\u001b[0m \u001b[0mthe\u001b[0m \u001b[0mMatch\u001b[0m \u001b[0mobject\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mmust\u001b[0m \u001b[0;32mreturn\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    209\u001b[0m     a replacement string to be used.\"\"\"\n\u001b[0;32m--> 210\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0m_compile\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpattern\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mflags\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msub\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrepl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstring\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcount\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    211\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    212\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0msubn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpattern\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrepl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstring\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcount\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mflags\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/re.py\u001b[0m in \u001b[0;36m_compile\u001b[0;34m(pattern, flags)\u001b[0m\n\u001b[1;32m    291\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mflags\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mRegexFlag\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    292\u001b[0m         \u001b[0mflags\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mflags\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 293\u001b[0;31m     \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    294\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0m_cache\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpattern\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpattern\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mflags\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    295\u001b[0m     \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# for f in glob.glob('./*faa_dbcan/overview.txt'):\n",
    "#     df = pd.read_csv(f'{f}', sep='\\t')\n",
    "#     files_l.append(df)\n",
    "files_in_dir=['./14_prokka/PROKKA_06282021.gbk', './26_prokka/PROKKA_06282021.gbk', './29_prokka/PROKKA_06282021.gbk', './46_prokka/PROKKA_06282021.gbk']\n",
    "files_l = []\n",
    "for f in files_in_dir:\n",
    "    prokka_read_id = []\n",
    "    prokka_gene_annotation = []\n",
    "    prokka_gene_id = []\n",
    "    for record in SeqIO.parse(f'{f}', \"genbank\"):\n",
    "        for f in record.features:\n",
    "            if f.type == \"CDS\" and \"gene\" in f.qualifiers:\n",
    "                gene = f.qualifiers[\"gene\"][0]\n",
    "                prokka_read_id.append(record.id)\n",
    "                prokka_gene_annotation.append(f.qualifiers[\"gene\"][0])\n",
    "                prokka_gene_id.append(f.qualifiers[\"locus_tag\"][0])\n",
    "    df = pd.DataFrame(list(zip(prokka_read_id, prokka_gene_id, prokka_gene_annotation)),\n",
    "                columns=['contig', 'Gene_ID', 'Annotation'])\n",
    "    files_l.append(df)\n",
    "df = pd.concat(files_l, keys=['14', '26', '29', '46'])\n",
    "df.reset_index(inplace=True)\n",
    "community = df['level_0'].str.split('.',expand=True)[0]\n",
    "df.insert(0, 'community', community)\n",
    "df_prokka = df.drop(columns=['level_0', 'level_1'])\n",
    "df_prokka"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "gff grep based parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community</th>\n",
       "      <th>contig</th>\n",
       "      <th>Gene_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14</td>\n",
       "      <td>427</td>\n",
       "      <td>MEBDDFFA_00001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14</td>\n",
       "      <td>427</td>\n",
       "      <td>MEBDDFFA_00002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14</td>\n",
       "      <td>427</td>\n",
       "      <td>MEBDDFFA_00003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14</td>\n",
       "      <td>427</td>\n",
       "      <td>MEBDDFFA_00004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14</td>\n",
       "      <td>427</td>\n",
       "      <td>MEBDDFFA_00005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121953</th>\n",
       "      <td>46</td>\n",
       "      <td>68</td>\n",
       "      <td>AAKEFBLG_121954</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121954</th>\n",
       "      <td>46</td>\n",
       "      <td>68</td>\n",
       "      <td>AAKEFBLG_121955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121955</th>\n",
       "      <td>46</td>\n",
       "      <td>68</td>\n",
       "      <td>AAKEFBLG_121956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121956</th>\n",
       "      <td>46</td>\n",
       "      <td>1467</td>\n",
       "      <td>AAKEFBLG_121957</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121957</th>\n",
       "      <td>46</td>\n",
       "      <td>1467</td>\n",
       "      <td>AAKEFBLG_121958</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>442759 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       community contig          Gene_ID\n",
       "0             14    427   MEBDDFFA_00001\n",
       "1             14    427   MEBDDFFA_00002\n",
       "2             14    427   MEBDDFFA_00003\n",
       "3             14    427   MEBDDFFA_00004\n",
       "4             14    427   MEBDDFFA_00005\n",
       "...          ...    ...              ...\n",
       "121953        46     68  AAKEFBLG_121954\n",
       "121954        46     68  AAKEFBLG_121955\n",
       "121955        46     68  AAKEFBLG_121956\n",
       "121956        46   1467  AAKEFBLG_121957\n",
       "121957        46   1467  AAKEFBLG_121958\n",
       "\n",
       "[442759 rows x 3 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files_in_dir=['./14_prokka/PROKKA_06282021.gff', './26_prokka/PROKKA_06282021.gff', './29_prokka/PROKKA_06282021.gff', './46_prokka/PROKKA_06282021.gff']\n",
    "files_l = []\n",
    "for f in files_in_dir:   \n",
    "    !grep -oP \"(s|c)[0-9]+\\s.*ID=[A-Z]+_[0-9]+\" $f | sed 's/[\\t].*[\\t]ID=/\\t/g' > contig_ids_temp.tsv\n",
    "    df = pd.read_csv('contig_ids_temp.tsv', sep='\\t',  names=['contig', 'Gene_ID'])\n",
    "    community = f'{f}'.split(\"/\")[1][:2]\n",
    "    df.insert(0, 'community', community)\n",
    "    files_l.append(df)\n",
    "    !rm contig_ids_temp.tsv\n",
    "df_ids = pd.concat(files_l)\n",
    "df_ids[\"contig\"] = df_ids[\"contig\"].str[1:]\n",
    "df_ids"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### OPTIONAL\n",
    "\n",
    "In work\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "files_in_dir=['./14_prokka/PROKKA_06282021.gff', './26_prokka/PROKKA_06282021.gff', './29_prokka/PROKKA_06282021.gff', './46_prokka/PROKKA_06282021.gff']\n",
    "files_l = []\n",
    "for f in files_in_dir:   \n",
    "    !grep -oP \"(s|c)[0-9]+\\s.*ID=[A-Z]+_[0-9]+\" $f | sed 's/[\\t].*[\\t]ID=/\\t/g' > contig_ids_temp.tsv\n",
    "    !f = grep -oP \"product=.+\\n\" \n",
    "    !sed -i \"s/$/\\t$f/\"  contig_ids_temp\n",
    "    df = pd.read_csv('contig_ids_temp.tsv', sep='\\t',  names=['contig', 'Gene_ID'])\n",
    "    community = f'{f}'.split(\"/\")[1][:2]\n",
    "    df.insert(0, 'community', community)\n",
    "    files_l.append(df)\n",
    "    \n",
    "    pd.read_csv\n",
    "    !rm contig_ids_temp.tsv\n",
    "df_ids = pd.concat(files_l)\n",
    "df_ids[\"contig\"] = df_ids[\"contig\"].str[1:]\n",
    "df_ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "f = './14_prokka/PROKKA_06282021.gff'\n",
    "# !grep -oP \"(s|c)[0-9]+\\s.*ID=[A-Z]+_[0-9]+\" $f | sed 's/[\\t].*[\\t]ID=/\\t/g' > contig_ids_temp.tsv\n",
    "!grep -oP \"(product=.*\\n\" $f | head\n",
    "# !paste contig_ids_temp.tsv right > contig_ids_temp.tsv\n",
    "# !head contig_ids_temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!head -n 50000 ./14_prokka/PROKKA_06282021.gff | tail -n 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 735,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community</th>\n",
       "      <th>contig</th>\n",
       "      <th>Gene_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [community, contig, Gene_ID]\n",
       "Index: []"
      ]
     },
     "execution_count": 735,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_prokka_wc.loc[df_prokka_wc['Gene_ID'] == 'AAKEFBLG_00264']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'df_prokka_wc' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-54-7271e815ee4d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mdf_prokka_wc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'community'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalue_counts\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'df_prokka_wc' is not defined"
     ]
    }
   ],
   "source": [
    "df_prokka_wc['community'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = {\"14.0\":\"14.0.faa\",\n",
    "     \"14.1\":\"14.1.faa\",\n",
    "     \"26.0\":\"26.0.faa\",\n",
    "     \"26.1\":\"26.1.faa\",\n",
    "     \"29.0\":\"29.0.faa\",\n",
    "     \"29.1\":\"29.1.faa\",\n",
    "     \"46.0\":\"46.0.faa\",\n",
    "     \"46.1\":\"46.1.faa\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for file in con/kraken2_res/*tsv; do\n",
    "    comm=$(basename $file | cut -c1-2)\n",
    "    cat $file | cut -f2,6 > con/kraken2_res/$comm\\_nice.tsv\n",
    "#     sed -i 'g/\\;[a-zA-Z\\s]+group\\;//i' con/kraken2_res/$comm\\_nice.tsv\n",
    "done;\n",
    "# grep \"Pseudomonas oleovorans\" con/kraken2_res/*_res_kraken.tsv\n",
    "# grep \"Pseudomonas oleovorans\" con/kraken2_res/*_nice.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t427\tcellular organisms\tBacteria\tBacteroidetes\tCytophagia\tCytophagales\tCytophagaceae\tSporocytophaga\tSporocytophaga myxococcoides\n",
      "\t1457\tcellular organisms\tBacteria\tBacteroidetes\tCytophagia\tCytophagales\tCytophagaceae\tSporocytophaga\tSporocytophaga myxococcoides\n",
      "\t1813\tcellular organisms\tBacteria\tVerrucomicrobia\tOpitutae\tOpitutales\tOpitutaceae\tOpitutus\tOpitutus terrae\n",
      "\t768\tcellular organisms\tBacteria\tProteobacteria\tAlphaproteobacteria\tHyphomicrobiales\tPhyllobacteriaceae\tMesorhizobium\n",
      "\t1440\tcellular organisms\tBacteria\tProteobacteria\tAlphaproteobacteria\tHyphomicrobiales\tBradyrhizobiaceae\tBradyrhizobium\n",
      "\t897\tcellular organisms\tBacteria\tVerrucomicrobia\tOpitutae\tOpitutales\tOpitutaceae\tOpitutus\tOpitutus terrae\n",
      "\t1041\tcellular organisms\tBacteria\n",
      "\t1454\tcellular organisms\tBacteria\tVerrucomicrobia\tOpitutae\tOpitutales\tOpitutaceae\tunclassified Opitutaceae\tOpitutaceae bacterium EW11\n",
      "\t608\tcellular organisms\tBacteria\tActinobacteria\tActinomycetia\tMicrococcales\tBeutenbergiaceae\tMiniimonas\tunclassified Miniimonas\n",
      "\t694\tcellular organisms\tBacteria\tBacteroidetes\tCytophagia\tCytophagales\tCytophagaceae\tSporocytophaga\tSporocytophaga myxococcoides\n",
      "rm: cannot remove 'con/kraken2_res/_n_nice.tsv': No such file or directory\n"
     ]
    }
   ],
   "source": [
    "for f in glob.glob('con/kraken2_res/*_nice.tsv'):\n",
    "    l_1 = []\n",
    "    counter = 0\n",
    "    counter2 = 0\n",
    "    with open(f, 'r') as file:      \n",
    "        for l in file:\n",
    "\n",
    "            if counter > 0:\n",
    "                l_1[-1] = l_1[-1] + '\\n'\n",
    "            counter = 0\n",
    "            for i in l.split(\";\"):\n",
    "                if (not i.endswith(\"group\")) and (counter < 8):              \n",
    "                    l_1.append(i.rstrip())\n",
    "                    if counter2 == 0:\n",
    "                        l_1[0] = '\\t' + l_1[0]\n",
    "                    counter += 1\n",
    "                    counter2 += 1\n",
    "    with open(f, 'w') as file:\n",
    "        file.write('\\t'.join(l_1) + '\\n')\n",
    "\n",
    "!head con/kraken2_res/14_nice.tsv\n",
    "%rm con/kraken2_res/_n_nice.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>HMMER</th>\n",
       "      <th>Hotpep</th>\n",
       "      <th>DIAMOND</th>\n",
       "      <th>NumberofTools</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00098</td>\n",
       "      <td>GT1(11-421)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00146</td>\n",
       "      <td>GH2(21-413)</td>\n",
       "      <td>-</td>\n",
       "      <td>GH2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00264</td>\n",
       "      <td>GH154(15-332)</td>\n",
       "      <td>GH154(12)</td>\n",
       "      <td>GH154</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00268</td>\n",
       "      <td>GH42(14-380)</td>\n",
       "      <td>GH42(3)</td>\n",
       "      <td>GH42</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00282</td>\n",
       "      <td>CBM67(117-293)+GH78(320-842)</td>\n",
       "      <td>GH78(1)+CBM67(23)</td>\n",
       "      <td>GH78</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13356</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_117026</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PL8</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13357</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_35246</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>GH2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13358</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_78144</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>GH36</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13359</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_106008</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>CBM48+GH13_10</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13360</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_04286</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PL22</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13361 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      community          Gene_ID                         HMMER  \\\n",
       "0            46   AAKEFBLG_00098                   GT1(11-421)   \n",
       "1            46   AAKEFBLG_00146                   GH2(21-413)   \n",
       "2            46   AAKEFBLG_00264                 GH154(15-332)   \n",
       "3            46   AAKEFBLG_00268                  GH42(14-380)   \n",
       "4            46   AAKEFBLG_00282  CBM67(117-293)+GH78(320-842)   \n",
       "...         ...              ...                           ...   \n",
       "13356        46  AAKEFBLG_117026                             -   \n",
       "13357        46   AAKEFBLG_35246                             -   \n",
       "13358        46   AAKEFBLG_78144                             -   \n",
       "13359        46  AAKEFBLG_106008                             -   \n",
       "13360        46   AAKEFBLG_04286                             -   \n",
       "\n",
       "                  Hotpep        DIAMOND  NumberofTools  \n",
       "0                      -              -              1  \n",
       "1                      -            GH2              2  \n",
       "2              GH154(12)          GH154              3  \n",
       "3                GH42(3)           GH42              3  \n",
       "4      GH78(1)+CBM67(23)           GH78              3  \n",
       "...                  ...            ...            ...  \n",
       "13356                  -            PL8              1  \n",
       "13357                  -            GH2              1  \n",
       "13358                  -           GH36              1  \n",
       "13359                  -  CBM48+GH13_10              1  \n",
       "13360                  -           PL22              1  \n",
       "\n",
       "[13361 rows x 6 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files_l = []\n",
    "names = []\n",
    "for f in glob.glob('./*faa_dbcan/overview.txt'):\n",
    "    df = pd.read_csv(f'{f}', sep='\\t')\n",
    "    filename = f'{f}'.split(\"/\")[1][:4]\n",
    "    df.insert(0, 'filename', filename)\n",
    "    files_l.append(df)\n",
    "df = pd.concat(files_l)\n",
    "df.reset_index(inplace=True)\n",
    "community = df['filename'].str.split('.',expand=True)[0]\n",
    "df.insert(0, 'community', community)\n",
    "df_cazy = df.drop(columns=['filename', 'index'])\n",
    "df_cazy.columns = ['community', 'Gene_ID', 'HMMER', 'Hotpep', 'DIAMOND', 'NumberofTools']\n",
    "df_cazy\n",
    "\n",
    "\n",
    "# files_l = []\n",
    "# for f in glob.glob('./con/kraken2_res/*nice.tsv'):\n",
    "#     df = pd.read_csv(f'{f}', sep='\\t',  names=['zero', 'contig', \"d\" ,\"Kingdom\", \"Phylum\",  \"Class\" ,  \"Order\",  \"Family\",  \"Genus\",   \"Species\"])\n",
    "#     community = f'{f}'.split(\"/\")[-1][:2]\n",
    "#     df.insert(0, 'community', community)\n",
    "#     files_l.append(df)\n",
    "# df = pd.concat(files_l)\n",
    "# df = df.drop(['zero', 'd'], axis=1)\n",
    "# df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>HMMER</th>\n",
       "      <th>Hotpep</th>\n",
       "      <th>DIAMOND</th>\n",
       "      <th>NumberofTools</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00098</td>\n",
       "      <td>GT1(11-421)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00146</td>\n",
       "      <td>GH2(21-413)</td>\n",
       "      <td>-</td>\n",
       "      <td>GH2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00264</td>\n",
       "      <td>GH154(15-332)</td>\n",
       "      <td>GH154(12)</td>\n",
       "      <td>GH154</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00268</td>\n",
       "      <td>GH42(14-380)</td>\n",
       "      <td>GH42(3)</td>\n",
       "      <td>GH42</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00282</td>\n",
       "      <td>CBM67(117-293)+GH78(320-842)</td>\n",
       "      <td>GH78(1)+CBM67(23)</td>\n",
       "      <td>GH78</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13356</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_117026</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PL8</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13357</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_35246</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>GH2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13358</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_78144</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>GH36</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13359</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_106008</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>CBM48+GH13_10</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13360</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_04286</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PL22</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13361 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      community          Gene_ID                         HMMER  \\\n",
       "0            46   AAKEFBLG_00098                   GT1(11-421)   \n",
       "1            46   AAKEFBLG_00146                   GH2(21-413)   \n",
       "2            46   AAKEFBLG_00264                 GH154(15-332)   \n",
       "3            46   AAKEFBLG_00268                  GH42(14-380)   \n",
       "4            46   AAKEFBLG_00282  CBM67(117-293)+GH78(320-842)   \n",
       "...         ...              ...                           ...   \n",
       "13356        46  AAKEFBLG_117026                             -   \n",
       "13357        46   AAKEFBLG_35246                             -   \n",
       "13358        46   AAKEFBLG_78144                             -   \n",
       "13359        46  AAKEFBLG_106008                             -   \n",
       "13360        46   AAKEFBLG_04286                             -   \n",
       "\n",
       "                  Hotpep        DIAMOND  NumberofTools  \n",
       "0                      -              -              1  \n",
       "1                      -            GH2              2  \n",
       "2              GH154(12)          GH154              3  \n",
       "3                GH42(3)           GH42              3  \n",
       "4      GH78(1)+CBM67(23)           GH78              3  \n",
       "...                  ...            ...            ...  \n",
       "13356                  -            PL8              1  \n",
       "13357                  -            GH2              1  \n",
       "13358                  -           GH36              1  \n",
       "13359                  -  CBM48+GH13_10              1  \n",
       "13360                  -           PL22              1  \n",
       "\n",
       "[13361 rows x 6 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cazy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>HMMER</th>\n",
       "      <th>Hotpep</th>\n",
       "      <th>DIAMOND</th>\n",
       "      <th>NumberofTools</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00146</td>\n",
       "      <td>GH2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GH2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00264</td>\n",
       "      <td>GH154</td>\n",
       "      <td>GH154</td>\n",
       "      <td>GH154</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00268</td>\n",
       "      <td>GH42</td>\n",
       "      <td>GH42</td>\n",
       "      <td>GH42</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00282</td>\n",
       "      <td>CBM67</td>\n",
       "      <td>GH78</td>\n",
       "      <td>GH78</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00283</td>\n",
       "      <td>GH2</td>\n",
       "      <td>GH2</td>\n",
       "      <td>GH2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12995</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_60620</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12998</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_102001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12999</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_67159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13003</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_113036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CBM6</td>\n",
       "      <td>GH2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13006</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_46317</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CBM9</td>\n",
       "      <td>CBM9</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7385 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      community          Gene_ID  HMMER Hotpep DIAMOND  NumberofTools\n",
       "1            46   AAKEFBLG_00146    GH2    NaN     GH2              2\n",
       "2            46   AAKEFBLG_00264  GH154  GH154   GH154              3\n",
       "3            46   AAKEFBLG_00268   GH42   GH42    GH42              3\n",
       "4            46   AAKEFBLG_00282  CBM67   GH78    GH78              3\n",
       "5            46   AAKEFBLG_00283    GH2    GH2     GH2              3\n",
       "...         ...              ...    ...    ...     ...            ...\n",
       "12995        46   AAKEFBLG_60620    NaN  CBM50   CBM50              2\n",
       "12998        46  AAKEFBLG_102001    NaN  CBM50   CBM50              2\n",
       "12999        46   AAKEFBLG_67159    NaN  CBM50   CBM50              2\n",
       "13003        46  AAKEFBLG_113036    NaN   CBM6     GH2              2\n",
       "13006        46   AAKEFBLG_46317    NaN   CBM9    CBM9              2\n",
       "\n",
       "[7385 rows x 6 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cazy_f = df_cazy[df_cazy['NumberofTools'] > 1]\n",
    "df_cazy_f = df_cazy_f.replace('-', value=np.nan)\n",
    "# df.replace(regex={r'^ba.$': 'new'})\n",
    "df_cazy_f = df_cazy_f.replace('\\(.*\\)', '', regex=True)\n",
    "df_cazy_f"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    GT2_Glycos_transf_2\n",
       "dtype: object"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# df_cazy_f['is_equal'] = (df_cazy_f['HMMER'] == df_cazy_f['Hotpep'])\n",
    "df_cazy_f.HMMER.mode()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GT2_Glycos_transf_2',\n",
       " 'GT4',\n",
       " 'GH3',\n",
       " 'GT51',\n",
       " 'GH23',\n",
       " 'GH2',\n",
       " 'GH103',\n",
       " 'GH51',\n",
       " 'GH1',\n",
       " 'GH10']"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cazy_f['HMMER'].value_counts()[:10].index.tolist()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### mode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>HMMER</th>\n",
       "      <th>Hotpep</th>\n",
       "      <th>DIAMOND</th>\n",
       "      <th>NumberofTools</th>\n",
       "      <th>mode</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00146</td>\n",
       "      <td>GH2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GH2</td>\n",
       "      <td>2</td>\n",
       "      <td>GH2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00264</td>\n",
       "      <td>GH154</td>\n",
       "      <td>GH154</td>\n",
       "      <td>GH154</td>\n",
       "      <td>3</td>\n",
       "      <td>GH154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00268</td>\n",
       "      <td>GH42</td>\n",
       "      <td>GH42</td>\n",
       "      <td>GH42</td>\n",
       "      <td>3</td>\n",
       "      <td>GH42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00282</td>\n",
       "      <td>CBM67</td>\n",
       "      <td>GH78</td>\n",
       "      <td>GH78</td>\n",
       "      <td>3</td>\n",
       "      <td>GH78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00283</td>\n",
       "      <td>GH2</td>\n",
       "      <td>GH2</td>\n",
       "      <td>GH2</td>\n",
       "      <td>3</td>\n",
       "      <td>GH2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12995</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_60620</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>2</td>\n",
       "      <td>CBM50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12998</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_102001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>2</td>\n",
       "      <td>CBM50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12999</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_67159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>2</td>\n",
       "      <td>CBM50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13003</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_113036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CBM6</td>\n",
       "      <td>GH2</td>\n",
       "      <td>2</td>\n",
       "      <td>CBM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13006</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_46317</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CBM9</td>\n",
       "      <td>CBM9</td>\n",
       "      <td>2</td>\n",
       "      <td>CBM9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7385 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      community          Gene_ID  HMMER Hotpep DIAMOND  NumberofTools   mode\n",
       "1            46   AAKEFBLG_00146    GH2    NaN     GH2              2    GH2\n",
       "2            46   AAKEFBLG_00264  GH154  GH154   GH154              3  GH154\n",
       "3            46   AAKEFBLG_00268   GH42   GH42    GH42              3   GH42\n",
       "4            46   AAKEFBLG_00282  CBM67   GH78    GH78              3   GH78\n",
       "5            46   AAKEFBLG_00283    GH2    GH2     GH2              3    GH2\n",
       "...         ...              ...    ...    ...     ...            ...    ...\n",
       "12995        46   AAKEFBLG_60620    NaN  CBM50   CBM50              2  CBM50\n",
       "12998        46  AAKEFBLG_102001    NaN  CBM50   CBM50              2  CBM50\n",
       "12999        46   AAKEFBLG_67159    NaN  CBM50   CBM50              2  CBM50\n",
       "13003        46  AAKEFBLG_113036    NaN   CBM6     GH2              2   CBM6\n",
       "13006        46   AAKEFBLG_46317    NaN   CBM9    CBM9              2   CBM9\n",
       "\n",
       "[7385 rows x 7 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cazy_f[\"mode\"] = df_cazy_f[['HMMER', 'Hotpep', 'DIAMOND']].T.mode().values.tolist()[0]\n",
    "df_cazy_f"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_cazy_simple = df_cazy_f[]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'community' is not a valid function for 'Series' object",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-25-5c788d2cccb7>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mdf_max\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf_cazy_f\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'community'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'mode'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0magg\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'community'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgroupby\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'mode'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcount\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msort_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'community'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mascending\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_option\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'display.max_rows'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mdf_max\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36maggregate\u001b[0;34m(self, func, axis, *args, **kwargs)\u001b[0m\n\u001b[1;32m   7576\u001b[0m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   7577\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 7578\u001b[0;31m             \u001b[0mresult\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_aggregate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfunc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   7579\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mTypeError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   7580\u001b[0m             exc = TypeError(\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_aggregate\u001b[0;34m(self, arg, axis, *args, **kwargs)\u001b[0m\n\u001b[1;32m   7607\u001b[0m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mT\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mresult\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   7608\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 7609\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0maggregate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0marg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   7610\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   7611\u001b[0m     \u001b[0magg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0maggregate\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/pandas/core/aggregation.py\u001b[0m in \u001b[0;36maggregate\u001b[0;34m(obj, arg, *args, **kwargs)\u001b[0m\n\u001b[1;32m    584\u001b[0m         \u001b[0;31m# we require a list, but not an 'str'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    585\u001b[0m         \u001b[0marg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcast\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mList\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mAggFuncTypeBase\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 586\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0magg_list_like\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0marg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_axis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0m_axis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    587\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    588\u001b[0m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/pandas/core/aggregation.py\u001b[0m in \u001b[0;36magg_list_like\u001b[0;34m(obj, arg, _axis)\u001b[0m\n\u001b[1;32m    651\u001b[0m             \u001b[0mcolg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_gotitem\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcol\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mndim\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msubset\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mselected_obj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    652\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 653\u001b[0;31m                 \u001b[0mnew_res\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcolg\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maggregate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    654\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mTypeError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mDataError\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    655\u001b[0m                 \u001b[0;32mpass\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36maggregate\u001b[0;34m(self, func, axis, *args, **kwargs)\u001b[0m\n\u001b[1;32m   3972\u001b[0m             \u001b[0mfunc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3973\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3974\u001b[0;31m         \u001b[0mresult\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0maggregate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3975\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mresult\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3976\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/pandas/core/aggregation.py\u001b[0m in \u001b[0;36maggregate\u001b[0;34m(obj, arg, *args, **kwargs)\u001b[0m\n\u001b[1;32m    584\u001b[0m         \u001b[0;31m# we require a list, but not an 'str'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    585\u001b[0m         \u001b[0marg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcast\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mList\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mAggFuncTypeBase\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 586\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0magg_list_like\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0marg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_axis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0m_axis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    587\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    588\u001b[0m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/pandas/core/aggregation.py\u001b[0m in \u001b[0;36magg_list_like\u001b[0;34m(obj, arg, _axis)\u001b[0m\n\u001b[1;32m    635\u001b[0m             \u001b[0mcolg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_gotitem\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mselected_obj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mndim\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msubset\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mselected_obj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    636\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 637\u001b[0;31m                 \u001b[0mnew_res\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcolg\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maggregate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    638\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    639\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36maggregate\u001b[0;34m(self, func, axis, *args, **kwargs)\u001b[0m\n\u001b[1;32m   3972\u001b[0m             \u001b[0mfunc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3973\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3974\u001b[0;31m         \u001b[0mresult\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0maggregate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3975\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mresult\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3976\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/pandas/core/aggregation.py\u001b[0m in \u001b[0;36maggregate\u001b[0;34m(obj, arg, *args, **kwargs)\u001b[0m\n\u001b[1;32m    577\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    578\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 579\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_try_aggregate_string_function\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    580\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0mis_dict_like\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    581\u001b[0m         \u001b[0marg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcast\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mAggFuncTypeDict\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/vamb/lib/python3.8/site-packages/pandas/core/base.py\u001b[0m in \u001b[0;36m_try_aggregate_string_function\u001b[0;34m(self, arg, *args, **kwargs)\u001b[0m\n\u001b[1;32m    327\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    328\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 329\u001b[0;31m         raise AttributeError(\n\u001b[0m\u001b[1;32m    330\u001b[0m             \u001b[0;34mf\"'{arg}' is not a valid function for '{type(self).__name__}' object\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    331\u001b[0m         )\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'community' is not a valid function for 'Series' object"
     ]
    }
   ],
   "source": [
    "df_max = df_cazy_f[['community','mode']].groupby(['mode']).count().sort_values('community', ascending=False)\n",
    "pd.set_option('display.max_rows', None)\n",
    "df_max"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.reset_option('all')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 737,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'sns' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-737-fc53153720d4>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;31m# fake_otu\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;31m# !pwd\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 7\u001b[0;31m \u001b[0msns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mboxplot\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'mode'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'community'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfake_otu\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'sns' is not defined"
     ]
    }
   ],
   "source": [
    "# df_cazy_f[['community','mode']].agg(\"mode\").count()\n",
    "fake_otu = df_cazy_f.groupby(['community','mode']).size().reset_index(name='counts')\n",
    "fake_otu = fake_otu.pivot(index='mode', columns='community', values='counts').fillna('0')\n",
    "fake_otu.to_csv('fake_otu',index=True, sep = '\\t')\n",
    "# fake_otu\n",
    "# !pwd\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 813,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mode</th>\n",
       "      <th>14</th>\n",
       "      <th>26</th>\n",
       "      <th>29</th>\n",
       "      <th>46</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AA1</td>\n",
       "      <td>13.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>18.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AA10</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AA12</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AA1_3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AA2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>480</th>\n",
       "      <td>PL9</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481</th>\n",
       "      <td>PL9_1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>482</th>\n",
       "      <td>PL9_2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>483</th>\n",
       "      <td>PL9_4</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>484</th>\n",
       "      <td>SLH</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>485 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      mode    14    26    29    46\n",
       "0      AA1  13.0  19.0  11.0  18.0\n",
       "1     AA10   4.0   3.0   3.0   1.0\n",
       "2     AA12   6.0   5.0   4.0   2.0\n",
       "3    AA1_3   0.0   0.0   0.0   1.0\n",
       "4      AA2   2.0   0.0   0.0   1.0\n",
       "..     ...   ...   ...   ...   ...\n",
       "480    PL9   2.0   2.0   4.0   5.0\n",
       "481  PL9_1   0.0   0.0   0.0   3.0\n",
       "482  PL9_2   0.0   0.0   0.0   1.0\n",
       "483  PL9_4   1.0   1.0   1.0   1.0\n",
       "484    SLH   0.0   0.0   3.0  17.0\n",
       "\n",
       "[485 rows x 5 columns]"
      ]
     },
     "execution_count": 813,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "td = pd.read_csv(\"fake_otu\", sep='\\t')\n",
    "td = td.drop_duplicates()\n",
    "td"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 845,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CBM50',\n",
       " 'CE10',\n",
       " 'GH23',\n",
       " 'GH3',\n",
       " 'GT2',\n",
       " 'GT4',\n",
       " 'GT51',\n",
       " 'GH2',\n",
       " 'GT2_Glycos_transf_2',\n",
       " 'CE1',\n",
       " 'CE4',\n",
       " 'GH0',\n",
       " 'GT0']"
      ]
     },
     "execution_count": 845,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sortedl = l[['count','mode']].groupby(\"mode\").count().sort_values('count', ascending=False)\n",
    "sortedl.index.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 861,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CBM50',\n",
       " 'CE10',\n",
       " 'GH23',\n",
       " 'GH3',\n",
       " 'GT2',\n",
       " 'GT4',\n",
       " 'GT51',\n",
       " 'GH2',\n",
       " 'GT2_Glycos_transf_2',\n",
       " 'CE1',\n",
       " 'CE4',\n",
       " 'GH0',\n",
       " 'GT0']"
      ]
     },
     "execution_count": 861,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "liss = sortedl.index.tolist()\n",
    "liss"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 862,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mode</th>\n",
       "      <th>community</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>CBM50</td>\n",
       "      <td>14</td>\n",
       "      <td>81.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>CE1</td>\n",
       "      <td>14</td>\n",
       "      <td>55.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>CE10</td>\n",
       "      <td>14</td>\n",
       "      <td>75.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>CE4</td>\n",
       "      <td>14</td>\n",
       "      <td>52.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>GH0</td>\n",
       "      <td>14</td>\n",
       "      <td>41.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>GH2</td>\n",
       "      <td>14</td>\n",
       "      <td>60.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263</th>\n",
       "      <td>GH23</td>\n",
       "      <td>14</td>\n",
       "      <td>73.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271</th>\n",
       "      <td>GH3</td>\n",
       "      <td>14</td>\n",
       "      <td>62.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379</th>\n",
       "      <td>GT0</td>\n",
       "      <td>14</td>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>GT2</td>\n",
       "      <td>14</td>\n",
       "      <td>203.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>GT2_Glycos_transf_2</td>\n",
       "      <td>14</td>\n",
       "      <td>59.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417</th>\n",
       "      <td>GT4</td>\n",
       "      <td>14</td>\n",
       "      <td>226.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423</th>\n",
       "      <td>GT51</td>\n",
       "      <td>14</td>\n",
       "      <td>81.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>600</th>\n",
       "      <td>CBM50</td>\n",
       "      <td>26</td>\n",
       "      <td>107.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>644</th>\n",
       "      <td>CE1</td>\n",
       "      <td>26</td>\n",
       "      <td>69.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>646</th>\n",
       "      <td>CE10</td>\n",
       "      <td>26</td>\n",
       "      <td>105.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>656</th>\n",
       "      <td>CE4</td>\n",
       "      <td>26</td>\n",
       "      <td>65.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>664</th>\n",
       "      <td>GH0</td>\n",
       "      <td>26</td>\n",
       "      <td>54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>745</th>\n",
       "      <td>GH2</td>\n",
       "      <td>26</td>\n",
       "      <td>57.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>748</th>\n",
       "      <td>GH23</td>\n",
       "      <td>26</td>\n",
       "      <td>105.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>756</th>\n",
       "      <td>GH3</td>\n",
       "      <td>26</td>\n",
       "      <td>96.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864</th>\n",
       "      <td>GT0</td>\n",
       "      <td>26</td>\n",
       "      <td>59.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>881</th>\n",
       "      <td>GT2</td>\n",
       "      <td>26</td>\n",
       "      <td>263.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>894</th>\n",
       "      <td>GT2_Glycos_transf_2</td>\n",
       "      <td>26</td>\n",
       "      <td>42.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>902</th>\n",
       "      <td>GT4</td>\n",
       "      <td>26</td>\n",
       "      <td>263.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>908</th>\n",
       "      <td>GT51</td>\n",
       "      <td>26</td>\n",
       "      <td>118.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1085</th>\n",
       "      <td>CBM50</td>\n",
       "      <td>29</td>\n",
       "      <td>75.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1129</th>\n",
       "      <td>CE1</td>\n",
       "      <td>29</td>\n",
       "      <td>50.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1131</th>\n",
       "      <td>CE10</td>\n",
       "      <td>29</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1141</th>\n",
       "      <td>CE4</td>\n",
       "      <td>29</td>\n",
       "      <td>50.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149</th>\n",
       "      <td>GH0</td>\n",
       "      <td>29</td>\n",
       "      <td>50.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1230</th>\n",
       "      <td>GH2</td>\n",
       "      <td>29</td>\n",
       "      <td>67.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1233</th>\n",
       "      <td>GH23</td>\n",
       "      <td>29</td>\n",
       "      <td>73.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1241</th>\n",
       "      <td>GH3</td>\n",
       "      <td>29</td>\n",
       "      <td>91.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1349</th>\n",
       "      <td>GT0</td>\n",
       "      <td>29</td>\n",
       "      <td>41.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1366</th>\n",
       "      <td>GT2</td>\n",
       "      <td>29</td>\n",
       "      <td>201.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1379</th>\n",
       "      <td>GT2_Glycos_transf_2</td>\n",
       "      <td>29</td>\n",
       "      <td>61.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1387</th>\n",
       "      <td>GT4</td>\n",
       "      <td>29</td>\n",
       "      <td>213.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1393</th>\n",
       "      <td>GT51</td>\n",
       "      <td>29</td>\n",
       "      <td>86.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1570</th>\n",
       "      <td>CBM50</td>\n",
       "      <td>46</td>\n",
       "      <td>112.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1614</th>\n",
       "      <td>CE1</td>\n",
       "      <td>46</td>\n",
       "      <td>61.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1616</th>\n",
       "      <td>CE10</td>\n",
       "      <td>46</td>\n",
       "      <td>81.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1626</th>\n",
       "      <td>CE4</td>\n",
       "      <td>46</td>\n",
       "      <td>102.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1634</th>\n",
       "      <td>GH0</td>\n",
       "      <td>46</td>\n",
       "      <td>90.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1715</th>\n",
       "      <td>GH2</td>\n",
       "      <td>46</td>\n",
       "      <td>101.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1718</th>\n",
       "      <td>GH23</td>\n",
       "      <td>46</td>\n",
       "      <td>120.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1726</th>\n",
       "      <td>GH3</td>\n",
       "      <td>46</td>\n",
       "      <td>122.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1834</th>\n",
       "      <td>GT0</td>\n",
       "      <td>46</td>\n",
       "      <td>54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1851</th>\n",
       "      <td>GT2</td>\n",
       "      <td>46</td>\n",
       "      <td>279.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1864</th>\n",
       "      <td>GT2_Glycos_transf_2</td>\n",
       "      <td>46</td>\n",
       "      <td>70.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1872</th>\n",
       "      <td>GT4</td>\n",
       "      <td>46</td>\n",
       "      <td>323.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1878</th>\n",
       "      <td>GT51</td>\n",
       "      <td>46</td>\n",
       "      <td>98.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     mode community  count\n",
       "115                 CBM50        14   81.0\n",
       "159                   CE1        14   55.0\n",
       "161                  CE10        14   75.0\n",
       "171                   CE4        14   52.0\n",
       "179                   GH0        14   41.0\n",
       "260                   GH2        14   60.0\n",
       "263                  GH23        14   73.0\n",
       "271                   GH3        14   62.0\n",
       "379                   GT0        14   37.0\n",
       "396                   GT2        14  203.0\n",
       "409   GT2_Glycos_transf_2        14   59.0\n",
       "417                   GT4        14  226.0\n",
       "423                  GT51        14   81.0\n",
       "600                 CBM50        26  107.0\n",
       "644                   CE1        26   69.0\n",
       "646                  CE10        26  105.0\n",
       "656                   CE4        26   65.0\n",
       "664                   GH0        26   54.0\n",
       "745                   GH2        26   57.0\n",
       "748                  GH23        26  105.0\n",
       "756                   GH3        26   96.0\n",
       "864                   GT0        26   59.0\n",
       "881                   GT2        26  263.0\n",
       "894   GT2_Glycos_transf_2        26   42.0\n",
       "902                   GT4        26  263.0\n",
       "908                  GT51        26  118.0\n",
       "1085                CBM50        29   75.0\n",
       "1129                  CE1        29   50.0\n",
       "1131                 CE10        29   76.0\n",
       "1141                  CE4        29   50.0\n",
       "1149                  GH0        29   50.0\n",
       "1230                  GH2        29   67.0\n",
       "1233                 GH23        29   73.0\n",
       "1241                  GH3        29   91.0\n",
       "1349                  GT0        29   41.0\n",
       "1366                  GT2        29  201.0\n",
       "1379  GT2_Glycos_transf_2        29   61.0\n",
       "1387                  GT4        29  213.0\n",
       "1393                 GT51        29   86.0\n",
       "1570                CBM50        46  112.0\n",
       "1614                  CE1        46   61.0\n",
       "1616                 CE10        46   81.0\n",
       "1626                  CE4        46  102.0\n",
       "1634                  GH0        46   90.0\n",
       "1715                  GH2        46  101.0\n",
       "1718                 GH23        46  120.0\n",
       "1726                  GH3        46  122.0\n",
       "1834                  GT0        46   54.0\n",
       "1851                  GT2        46  279.0\n",
       "1864  GT2_Glycos_transf_2        46   70.0\n",
       "1872                  GT4        46  323.0\n",
       "1878                 GT51        46   98.0"
      ]
     },
     "execution_count": 862,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "l = pd.melt(td, id_vars='mode', value_vars=['14', '26', '29', '46'], var_name='community', value_name='count')\n",
    "# l = l.sort_values(by=['count'], ascending=False)[0:40]\n",
    "todraw = l[l['mode'].isin(liss)]\n",
    "todraw"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 851,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mode</th>\n",
       "      <th>community</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AA1</td>\n",
       "      <td>14</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AA10</td>\n",
       "      <td>14</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AA12</td>\n",
       "      <td>14</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AA1_3</td>\n",
       "      <td>14</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AA2</td>\n",
       "      <td>14</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1935</th>\n",
       "      <td>PL9</td>\n",
       "      <td>46</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1936</th>\n",
       "      <td>PL9_1</td>\n",
       "      <td>46</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1937</th>\n",
       "      <td>PL9_2</td>\n",
       "      <td>46</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1938</th>\n",
       "      <td>PL9_4</td>\n",
       "      <td>46</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1939</th>\n",
       "      <td>SLH</td>\n",
       "      <td>46</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1940 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       mode community  count\n",
       "0       AA1        14   13.0\n",
       "1      AA10        14    4.0\n",
       "2      AA12        14    6.0\n",
       "3     AA1_3        14    0.0\n",
       "4       AA2        14    2.0\n",
       "...     ...       ...    ...\n",
       "1935    PL9        46    5.0\n",
       "1936  PL9_1        46    3.0\n",
       "1937  PL9_2        46    1.0\n",
       "1938  PL9_4        46    1.0\n",
       "1939    SLH        46   17.0\n",
       "\n",
       "[1940 rows x 3 columns]"
      ]
     },
     "execution_count": 851,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'sns' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-2-c212a4d01d8a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m d = sns.catplot(data=fake_otu, hue=\"community\",\n\u001b[0m\u001b[1;32m      2\u001b[0m                 \u001b[0my\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"mode\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'count'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkind\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"bar\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maspect\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m                 height=10,legend= False, orient=\"h\")\n\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mplt\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mplot\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlabel\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'Community'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'sns' is not defined"
     ]
    }
   ],
   "source": [
    "d = sns.catplot(data=todraw, hue=\"community\",\n",
    "                y = \"mode\", x='count', kind=\"bar\", aspect=2,\n",
    "                height=10,legend= False, orient=\"h\")\n",
    "\n",
    "plt.plot([1,2,3], label = 'Community')\n",
    "plt.ylabel('glycosylases', fontsize = 16)\n",
    "plt.xlabel('count', fontsize = 20)\n",
    "plt.legend(fontsize = 18)\n",
    "plt.xticks(fontsize = 20)\n",
    "plt.yticks(fontsize = 17)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 736,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>community</th>\n",
       "      <th>14</th>\n",
       "      <th>26</th>\n",
       "      <th>29</th>\n",
       "      <th>46</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mode</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AA1</th>\n",
       "      <td>13</td>\n",
       "      <td>19</td>\n",
       "      <td>11</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AA10</th>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AA12</th>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AA1_3</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AA2</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PL9</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PL9_1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PL9_2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PL9_4</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SLH</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>485 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "community  14  26  29  46\n",
       "mode                     \n",
       "AA1        13  19  11  18\n",
       "AA10        4   3   3   1\n",
       "AA12        6   5   4   2\n",
       "AA1_3       0   0   0   1\n",
       "AA2         2   0   0   1\n",
       "...        ..  ..  ..  ..\n",
       "PL9         2   2   4   5\n",
       "PL9_1       0   0   0   3\n",
       "PL9_2       0   0   0   1\n",
       "PL9_4       1   1   1   1\n",
       "SLH         0   0   3  17\n",
       "\n",
       "[485 rows x 4 columns]"
      ]
     },
     "execution_count": 736,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fake_otu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 871,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>zero</th>\n",
       "      <th>d</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Actinomycetia</td>\n",
       "      <td>Streptomycetales</td>\n",
       "      <td>Streptomycetaceae</td>\n",
       "      <td>Streptomyces</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Hyphomicrobiales</td>\n",
       "      <td>Boseaceae</td>\n",
       "      <td>Bosea</td>\n",
       "      <td>unclassified Bosea</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Flavobacteriia</td>\n",
       "      <td>Flavobacteriales</td>\n",
       "      <td>Weeksellaceae</td>\n",
       "      <td>Kaistella</td>\n",
       "      <td>Kaistella jeonii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Actinomycetia</td>\n",
       "      <td>Corynebacteriales</td>\n",
       "      <td>Mycobacteriaceae</td>\n",
       "      <td>Mycobacterium</td>\n",
       "      <td>Mycobacterium shimoidei</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Hyphomicrobiales</td>\n",
       "      <td>Phyllobacteriaceae</td>\n",
       "      <td>Mesorhizobium</td>\n",
       "      <td>unclassified Mesorhizobium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1534</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Hyphomicrobiales</td>\n",
       "      <td>Phyllobacteriaceae</td>\n",
       "      <td>Mesorhizobium</td>\n",
       "      <td>Mesorhizobium plurifarium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1535</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Hyphomicrobiales</td>\n",
       "      <td>Xanthobacteraceae</td>\n",
       "      <td>Xanthobacter</td>\n",
       "      <td>unclassified Xanthobacter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1547</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Sphingomonadales</td>\n",
       "      <td>Erythrobacteraceae</td>\n",
       "      <td>Croceicoccus</td>\n",
       "      <td>Croceicoccus bisphenolivorans</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1557</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Xanthomonadales</td>\n",
       "      <td>Xanthomonadaceae</td>\n",
       "      <td>Luteimonas</td>\n",
       "      <td>Luteimonas mephitis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1581</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Xanthomonadales</td>\n",
       "      <td>Xanthomonadaceae</td>\n",
       "      <td>Lysobacter</td>\n",
       "      <td>Lysobacter defluvii</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>875 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      zero                   d   Kingdom          Phylum                Class  \\\n",
       "0      NaN  cellular organisms  Bacteria  Actinobacteria        Actinomycetia   \n",
       "1      NaN  cellular organisms  Bacteria  Proteobacteria  Alphaproteobacteria   \n",
       "2      NaN  cellular organisms  Bacteria   Bacteroidetes       Flavobacteriia   \n",
       "3      NaN  cellular organisms  Bacteria  Actinobacteria        Actinomycetia   \n",
       "4      NaN  cellular organisms  Bacteria  Proteobacteria  Alphaproteobacteria   \n",
       "...    ...                 ...       ...             ...                  ...   \n",
       "1534   NaN  cellular organisms  Bacteria  Proteobacteria  Alphaproteobacteria   \n",
       "1535   NaN  cellular organisms  Bacteria  Proteobacteria  Alphaproteobacteria   \n",
       "1547   NaN  cellular organisms  Bacteria  Proteobacteria  Alphaproteobacteria   \n",
       "1557   NaN  cellular organisms  Bacteria  Proteobacteria  Gammaproteobacteria   \n",
       "1581   NaN  cellular organisms  Bacteria  Proteobacteria  Gammaproteobacteria   \n",
       "\n",
       "                  Order              Family          Genus  \\\n",
       "0      Streptomycetales   Streptomycetaceae   Streptomyces   \n",
       "1      Hyphomicrobiales           Boseaceae          Bosea   \n",
       "2      Flavobacteriales       Weeksellaceae      Kaistella   \n",
       "3     Corynebacteriales    Mycobacteriaceae  Mycobacterium   \n",
       "4      Hyphomicrobiales  Phyllobacteriaceae  Mesorhizobium   \n",
       "...                 ...                 ...            ...   \n",
       "1534   Hyphomicrobiales  Phyllobacteriaceae  Mesorhizobium   \n",
       "1535   Hyphomicrobiales   Xanthobacteraceae   Xanthobacter   \n",
       "1547   Sphingomonadales  Erythrobacteraceae   Croceicoccus   \n",
       "1557    Xanthomonadales    Xanthomonadaceae     Luteimonas   \n",
       "1581    Xanthomonadales    Xanthomonadaceae     Lysobacter   \n",
       "\n",
       "                            Species  \n",
       "0                               NaN  \n",
       "1                unclassified Bosea  \n",
       "2                  Kaistella jeonii  \n",
       "3           Mycobacterium shimoidei  \n",
       "4        unclassified Mesorhizobium  \n",
       "...                             ...  \n",
       "1534      Mesorhizobium plurifarium  \n",
       "1535      unclassified Xanthobacter  \n",
       "1547  Croceicoccus bisphenolivorans  \n",
       "1557            Luteimonas mephitis  \n",
       "1581            Lysobacter defluvii  \n",
       "\n",
       "[875 rows x 9 columns]"
      ]
     },
     "execution_count": 871,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_taxa = df.drop(['community', 'contig'], axis=1)\n",
    "taxa_duplicates_dropped = df_taxa.drop_duplicates()\n",
    "taxa_duplicates_dropped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 872,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community</th>\n",
       "      <th>contig</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>29</td>\n",
       "      <td>540</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Actinomycetia</td>\n",
       "      <td>Streptomycetales</td>\n",
       "      <td>Streptomycetaceae</td>\n",
       "      <td>Streptomyces</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>29</td>\n",
       "      <td>471</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Hyphomicrobiales</td>\n",
       "      <td>Boseaceae</td>\n",
       "      <td>Bosea</td>\n",
       "      <td>unclassified Bosea</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>29</td>\n",
       "      <td>109</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Flavobacteriia</td>\n",
       "      <td>Flavobacteriales</td>\n",
       "      <td>Weeksellaceae</td>\n",
       "      <td>Kaistella</td>\n",
       "      <td>Kaistella jeonii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>29</td>\n",
       "      <td>954</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Actinomycetia</td>\n",
       "      <td>Corynebacteriales</td>\n",
       "      <td>Mycobacteriaceae</td>\n",
       "      <td>Mycobacterium</td>\n",
       "      <td>Mycobacterium shimoidei</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>29</td>\n",
       "      <td>297</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Hyphomicrobiales</td>\n",
       "      <td>Phyllobacteriaceae</td>\n",
       "      <td>Mesorhizobium</td>\n",
       "      <td>unclassified Mesorhizobium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1584</th>\n",
       "      <td>26</td>\n",
       "      <td>1387</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Xanthomonadales</td>\n",
       "      <td>Xanthomonadaceae</td>\n",
       "      <td>Luteimonas</td>\n",
       "      <td>unclassified Luteimonas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1585</th>\n",
       "      <td>26</td>\n",
       "      <td>964</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Hyphomicrobiales</td>\n",
       "      <td>Devosiaceae</td>\n",
       "      <td>Devosia</td>\n",
       "      <td>unclassified Devosia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1586</th>\n",
       "      <td>26</td>\n",
       "      <td>1016</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Hyphomicrobiales</td>\n",
       "      <td>Phyllobacteriaceae</td>\n",
       "      <td>Mesorhizobium</td>\n",
       "      <td>Mesorhizobium ephedrae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1587</th>\n",
       "      <td>26</td>\n",
       "      <td>35</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Hyphomicrobiales</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1588</th>\n",
       "      <td>26</td>\n",
       "      <td>1124</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Xanthomonadales</td>\n",
       "      <td>Xanthomonadaceae</td>\n",
       "      <td>Lysobacter</td>\n",
       "      <td>Lysobacter tabacisoli</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6693 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     community  contig   Kingdom          Phylum                Class  \\\n",
       "0           29     540  Bacteria  Actinobacteria        Actinomycetia   \n",
       "1           29     471  Bacteria  Proteobacteria  Alphaproteobacteria   \n",
       "2           29     109  Bacteria   Bacteroidetes       Flavobacteriia   \n",
       "3           29     954  Bacteria  Actinobacteria        Actinomycetia   \n",
       "4           29     297  Bacteria  Proteobacteria  Alphaproteobacteria   \n",
       "...        ...     ...       ...             ...                  ...   \n",
       "1584        26    1387  Bacteria  Proteobacteria  Gammaproteobacteria   \n",
       "1585        26     964  Bacteria  Proteobacteria  Alphaproteobacteria   \n",
       "1586        26    1016  Bacteria  Proteobacteria  Alphaproteobacteria   \n",
       "1587        26      35  Bacteria  Proteobacteria  Alphaproteobacteria   \n",
       "1588        26    1124  Bacteria  Proteobacteria  Gammaproteobacteria   \n",
       "\n",
       "                  Order              Family          Genus  \\\n",
       "0      Streptomycetales   Streptomycetaceae   Streptomyces   \n",
       "1      Hyphomicrobiales           Boseaceae          Bosea   \n",
       "2      Flavobacteriales       Weeksellaceae      Kaistella   \n",
       "3     Corynebacteriales    Mycobacteriaceae  Mycobacterium   \n",
       "4      Hyphomicrobiales  Phyllobacteriaceae  Mesorhizobium   \n",
       "...                 ...                 ...            ...   \n",
       "1584    Xanthomonadales    Xanthomonadaceae     Luteimonas   \n",
       "1585   Hyphomicrobiales         Devosiaceae        Devosia   \n",
       "1586   Hyphomicrobiales  Phyllobacteriaceae  Mesorhizobium   \n",
       "1587   Hyphomicrobiales                 NaN            NaN   \n",
       "1588    Xanthomonadales    Xanthomonadaceae     Lysobacter   \n",
       "\n",
       "                         Species  \n",
       "0                            NaN  \n",
       "1             unclassified Bosea  \n",
       "2               Kaistella jeonii  \n",
       "3        Mycobacterium shimoidei  \n",
       "4     unclassified Mesorhizobium  \n",
       "...                          ...  \n",
       "1584     unclassified Luteimonas  \n",
       "1585        unclassified Devosia  \n",
       "1586      Mesorhizobium ephedrae  \n",
       "1587                         NaN  \n",
       "1588       Lysobacter tabacisoli  \n",
       "\n",
       "[6693 rows x 9 columns]"
      ]
     },
     "execution_count": 872,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files_l = []\n",
    "for f in glob.glob('./con/kraken2_res/*nice.tsv'):\n",
    "    df = pd.read_csv(f'{f}', sep='\\t',  names=['zero', 'contig', \"d\" ,\"Kingdom\", \"Phylum\",  \"Class\" ,  \"Order\",  \"Family\",  \"Genus\",   \"Species\"])\n",
    "    community = f'{f}'.split(\"/\")[-1][:2]\n",
    "    df.insert(0, 'community', community)\n",
    "    files_l.append(df)\n",
    "df = pd.concat(files_l)\n",
    "df_taxa = df.drop(['zero', 'd'], axis=1)\n",
    "df_taxa\n",
    "# taxa = df['taxa'].str.split(';',expand=True).loc[:, 1:7]\n",
    "# taxa.columns = [\"Kingdom\", \"Phylum\",  \"Class\" ,  \"Order\",  \"Family\",  \"Genus\",   \"Species\"]\n",
    "# taxa.query(\"Kingdom != 'Bacteria'\")\n",
    "# df.query(\"Kingdom != 'Bacteria'\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 919,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 919,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cazy_f.duplicated(subset=['Gene_ID']).sum()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 920,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_taxa[\"ID\"] = df_taxa[\"community\"].astype(str) + \"_\" + df_taxa[\"contig\"].astype(str)\n",
    "df_ids[\"ID\"] = df_ids[\"community\"].astype(str) + \"_\" + df_ids[\"contig\"].astype(str)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 697,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community</th>\n",
       "      <th>level_0</th>\n",
       "      <th>level_1</th>\n",
       "      <th>contig</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Annotation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>c427</td>\n",
       "      <td>MEBDDFFA_00005</td>\n",
       "      <td>amaA_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>c427</td>\n",
       "      <td>MEBDDFFA_00007</td>\n",
       "      <td>hipO_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>c427</td>\n",
       "      <td>MEBDDFFA_00009</td>\n",
       "      <td>secA_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>3</td>\n",
       "      <td>c427</td>\n",
       "      <td>MEBDDFFA_00010</td>\n",
       "      <td>secA_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>c427</td>\n",
       "      <td>MEBDDFFA_00011</td>\n",
       "      <td>secA_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165413</th>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>49190</td>\n",
       "      <td>c68</td>\n",
       "      <td>AAKEFBLG_121937</td>\n",
       "      <td>arnT_10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165414</th>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>49191</td>\n",
       "      <td>c68</td>\n",
       "      <td>AAKEFBLG_121939</td>\n",
       "      <td>rgtE_11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165415</th>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>49192</td>\n",
       "      <td>c68</td>\n",
       "      <td>AAKEFBLG_121943</td>\n",
       "      <td>chaB_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165416</th>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>49193</td>\n",
       "      <td>c68</td>\n",
       "      <td>AAKEFBLG_121951</td>\n",
       "      <td>bphC_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165417</th>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>49194</td>\n",
       "      <td>c68</td>\n",
       "      <td>AAKEFBLG_121955</td>\n",
       "      <td>btuD_394</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>165418 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       community level_0  level_1 contig          Gene_ID Annotation\n",
       "0             14      14        0   c427   MEBDDFFA_00005     amaA_1\n",
       "1             14      14        1   c427   MEBDDFFA_00007     hipO_1\n",
       "2             14      14        2   c427   MEBDDFFA_00009     secA_1\n",
       "3             14      14        3   c427   MEBDDFFA_00010     secA_2\n",
       "4             14      14        4   c427   MEBDDFFA_00011     secA_3\n",
       "...          ...     ...      ...    ...              ...        ...\n",
       "165413        46      46    49190    c68  AAKEFBLG_121937    arnT_10\n",
       "165414        46      46    49191    c68  AAKEFBLG_121939    rgtE_11\n",
       "165415        46      46    49192    c68  AAKEFBLG_121943     chaB_2\n",
       "165416        46      46    49193    c68  AAKEFBLG_121951     bphC_5\n",
       "165417        46      46    49194    c68  AAKEFBLG_121955   btuD_394\n",
       "\n",
       "[165418 rows x 6 columns]"
      ]
     },
     "execution_count": 697,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cazy_f df_prokka_wc df_taxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 703,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>HMMER</th>\n",
       "      <th>Hotpep</th>\n",
       "      <th>DIAMOND</th>\n",
       "      <th>NumberofTools</th>\n",
       "      <th>mode</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00098</td>\n",
       "      <td>GT1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>GT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00146</td>\n",
       "      <td>GH2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GH2</td>\n",
       "      <td>2</td>\n",
       "      <td>GH2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00264</td>\n",
       "      <td>GH154</td>\n",
       "      <td>GH154</td>\n",
       "      <td>GH154</td>\n",
       "      <td>3</td>\n",
       "      <td>GH154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00268</td>\n",
       "      <td>GH42</td>\n",
       "      <td>GH42</td>\n",
       "      <td>GH42</td>\n",
       "      <td>3</td>\n",
       "      <td>GH42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00282</td>\n",
       "      <td>CBM67</td>\n",
       "      <td>GH78</td>\n",
       "      <td>GH78</td>\n",
       "      <td>3</td>\n",
       "      <td>GH78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13356</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_117026</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PL8</td>\n",
       "      <td>1</td>\n",
       "      <td>PL8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13357</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_35246</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GH2</td>\n",
       "      <td>1</td>\n",
       "      <td>GH2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13358</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_78144</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GH36</td>\n",
       "      <td>1</td>\n",
       "      <td>GH36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13359</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_106008</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CBM48+GH13_10</td>\n",
       "      <td>1</td>\n",
       "      <td>CBM48+GH13_10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13360</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_04286</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PL22</td>\n",
       "      <td>1</td>\n",
       "      <td>PL22</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13361 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      community          Gene_ID  HMMER Hotpep        DIAMOND  NumberofTools  \\\n",
       "0            46   AAKEFBLG_00098    GT1    NaN            NaN              1   \n",
       "1            46   AAKEFBLG_00146    GH2    NaN            GH2              2   \n",
       "2            46   AAKEFBLG_00264  GH154  GH154          GH154              3   \n",
       "3            46   AAKEFBLG_00268   GH42   GH42           GH42              3   \n",
       "4            46   AAKEFBLG_00282  CBM67   GH78           GH78              3   \n",
       "...         ...              ...    ...    ...            ...            ...   \n",
       "13356        46  AAKEFBLG_117026    NaN    NaN            PL8              1   \n",
       "13357        46   AAKEFBLG_35246    NaN    NaN            GH2              1   \n",
       "13358        46   AAKEFBLG_78144    NaN    NaN           GH36              1   \n",
       "13359        46  AAKEFBLG_106008    NaN    NaN  CBM48+GH13_10              1   \n",
       "13360        46   AAKEFBLG_04286    NaN    NaN           PL22              1   \n",
       "\n",
       "                mode  \n",
       "0                GT1  \n",
       "1                GH2  \n",
       "2              GH154  \n",
       "3               GH42  \n",
       "4               GH78  \n",
       "...              ...  \n",
       "13356            PL8  \n",
       "13357            GH2  \n",
       "13358           GH36  \n",
       "13359  CBM48+GH13_10  \n",
       "13360           PL22  \n",
       "\n",
       "[13361 rows x 7 columns]"
      ]
     },
     "execution_count": 703,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cazy_f"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 922,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "community_x         0\n",
       "Gene_ID             0\n",
       "HMMER            4283\n",
       "Hotpep           5863\n",
       "DIAMOND          4769\n",
       "NumberofTools       0\n",
       "mode                0\n",
       "community_y         0\n",
       "contig_x            0\n",
       "ID                  0\n",
       "community           0\n",
       "contig_y            0\n",
       "Kingdom             9\n",
       "Phylum             52\n",
       "Class              91\n",
       "Order             117\n",
       "Family            132\n",
       "Genus             181\n",
       "Species           509\n",
       "dtype: int64"
      ]
     },
     "execution_count": 922,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge3.isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 921,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community_x</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>HMMER</th>\n",
       "      <th>Hotpep</th>\n",
       "      <th>DIAMOND</th>\n",
       "      <th>NumberofTools</th>\n",
       "      <th>mode</th>\n",
       "      <th>community_y</th>\n",
       "      <th>contig_x</th>\n",
       "      <th>ID</th>\n",
       "      <th>community</th>\n",
       "      <th>contig_y</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00098</td>\n",
       "      <td>GT1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>GT1</td>\n",
       "      <td>46</td>\n",
       "      <td>435</td>\n",
       "      <td>46_435</td>\n",
       "      <td>46</td>\n",
       "      <td>435</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>unclassified Paenibacillus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00146</td>\n",
       "      <td>GH2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GH2</td>\n",
       "      <td>2</td>\n",
       "      <td>GH2</td>\n",
       "      <td>46</td>\n",
       "      <td>435</td>\n",
       "      <td>46_435</td>\n",
       "      <td>46</td>\n",
       "      <td>435</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>unclassified Paenibacillus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00264</td>\n",
       "      <td>GH154</td>\n",
       "      <td>GH154</td>\n",
       "      <td>GH154</td>\n",
       "      <td>3</td>\n",
       "      <td>GH154</td>\n",
       "      <td>46</td>\n",
       "      <td>186</td>\n",
       "      <td>46_186</td>\n",
       "      <td>46</td>\n",
       "      <td>186</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>unclassified Paenibacillus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00268</td>\n",
       "      <td>GH42</td>\n",
       "      <td>GH42</td>\n",
       "      <td>GH42</td>\n",
       "      <td>3</td>\n",
       "      <td>GH42</td>\n",
       "      <td>46</td>\n",
       "      <td>186</td>\n",
       "      <td>46_186</td>\n",
       "      <td>46</td>\n",
       "      <td>186</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>unclassified Paenibacillus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00282</td>\n",
       "      <td>CBM67</td>\n",
       "      <td>GH78</td>\n",
       "      <td>GH78</td>\n",
       "      <td>3</td>\n",
       "      <td>GH78</td>\n",
       "      <td>46</td>\n",
       "      <td>186</td>\n",
       "      <td>46_186</td>\n",
       "      <td>46</td>\n",
       "      <td>186</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>unclassified Paenibacillus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13356</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_117026</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PL8</td>\n",
       "      <td>1</td>\n",
       "      <td>PL8</td>\n",
       "      <td>46</td>\n",
       "      <td>1142</td>\n",
       "      <td>46_1142</td>\n",
       "      <td>46</td>\n",
       "      <td>1142</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>Paenibacillus paeoniae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13357</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_35246</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GH2</td>\n",
       "      <td>1</td>\n",
       "      <td>GH2</td>\n",
       "      <td>46</td>\n",
       "      <td>118</td>\n",
       "      <td>46_118</td>\n",
       "      <td>46</td>\n",
       "      <td>118</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhodospirillales</td>\n",
       "      <td>Azospirillaceae</td>\n",
       "      <td>Azospirillum</td>\n",
       "      <td>unclassified Azospirillum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13358</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_78144</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GH36</td>\n",
       "      <td>1</td>\n",
       "      <td>GH36</td>\n",
       "      <td>46</td>\n",
       "      <td>1301</td>\n",
       "      <td>46_1301</td>\n",
       "      <td>46</td>\n",
       "      <td>1301</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Eubacteriales</td>\n",
       "      <td>Oscillospiraceae</td>\n",
       "      <td>Ruminiclostridium</td>\n",
       "      <td>Ruminiclostridium cellobioparum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13359</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_106008</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CBM48+GH13_10</td>\n",
       "      <td>1</td>\n",
       "      <td>CBM48+GH13_10</td>\n",
       "      <td>46</td>\n",
       "      <td>140</td>\n",
       "      <td>46_140</td>\n",
       "      <td>46</td>\n",
       "      <td>140</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhodospirillales</td>\n",
       "      <td>Azospirillaceae</td>\n",
       "      <td>Azospirillum</td>\n",
       "      <td>unclassified Azospirillum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13360</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_04286</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PL22</td>\n",
       "      <td>1</td>\n",
       "      <td>PL22</td>\n",
       "      <td>46</td>\n",
       "      <td>1573</td>\n",
       "      <td>46_1573</td>\n",
       "      <td>46</td>\n",
       "      <td>1573</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>Opitutae</td>\n",
       "      <td>Opitutales</td>\n",
       "      <td>Opitutaceae</td>\n",
       "      <td>Opitutus</td>\n",
       "      <td>unclassified Opitutus</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13361 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      community_x          Gene_ID  HMMER Hotpep        DIAMOND  \\\n",
       "0              46   AAKEFBLG_00098    GT1    NaN            NaN   \n",
       "1              46   AAKEFBLG_00146    GH2    NaN            GH2   \n",
       "2              46   AAKEFBLG_00264  GH154  GH154          GH154   \n",
       "3              46   AAKEFBLG_00268   GH42   GH42           GH42   \n",
       "4              46   AAKEFBLG_00282  CBM67   GH78           GH78   \n",
       "...           ...              ...    ...    ...            ...   \n",
       "13356          46  AAKEFBLG_117026    NaN    NaN            PL8   \n",
       "13357          46   AAKEFBLG_35246    NaN    NaN            GH2   \n",
       "13358          46   AAKEFBLG_78144    NaN    NaN           GH36   \n",
       "13359          46  AAKEFBLG_106008    NaN    NaN  CBM48+GH13_10   \n",
       "13360          46   AAKEFBLG_04286    NaN    NaN           PL22   \n",
       "\n",
       "       NumberofTools           mode community_y contig_x       ID community  \\\n",
       "0                  1            GT1          46      435   46_435        46   \n",
       "1                  2            GH2          46      435   46_435        46   \n",
       "2                  3          GH154          46      186   46_186        46   \n",
       "3                  3           GH42          46      186   46_186        46   \n",
       "4                  3           GH78          46      186   46_186        46   \n",
       "...              ...            ...         ...      ...      ...       ...   \n",
       "13356              1            PL8          46     1142  46_1142        46   \n",
       "13357              1            GH2          46      118   46_118        46   \n",
       "13358              1           GH36          46     1301  46_1301        46   \n",
       "13359              1  CBM48+GH13_10          46      140   46_140        46   \n",
       "13360              1           PL22          46     1573  46_1573        46   \n",
       "\n",
       "       contig_y   Kingdom           Phylum                Class  \\\n",
       "0           435  Bacteria       Firmicutes              Bacilli   \n",
       "1           435  Bacteria       Firmicutes              Bacilli   \n",
       "2           186  Bacteria       Firmicutes              Bacilli   \n",
       "3           186  Bacteria       Firmicutes              Bacilli   \n",
       "4           186  Bacteria       Firmicutes              Bacilli   \n",
       "...         ...       ...              ...                  ...   \n",
       "13356      1142  Bacteria       Firmicutes              Bacilli   \n",
       "13357       118  Bacteria   Proteobacteria  Alphaproteobacteria   \n",
       "13358      1301  Bacteria       Firmicutes           Clostridia   \n",
       "13359       140  Bacteria   Proteobacteria  Alphaproteobacteria   \n",
       "13360      1573  Bacteria  Verrucomicrobia             Opitutae   \n",
       "\n",
       "                  Order            Family              Genus  \\\n",
       "0            Bacillales  Paenibacillaceae      Paenibacillus   \n",
       "1            Bacillales  Paenibacillaceae      Paenibacillus   \n",
       "2            Bacillales  Paenibacillaceae      Paenibacillus   \n",
       "3            Bacillales  Paenibacillaceae      Paenibacillus   \n",
       "4            Bacillales  Paenibacillaceae      Paenibacillus   \n",
       "...                 ...               ...                ...   \n",
       "13356        Bacillales  Paenibacillaceae      Paenibacillus   \n",
       "13357  Rhodospirillales   Azospirillaceae       Azospirillum   \n",
       "13358     Eubacteriales  Oscillospiraceae  Ruminiclostridium   \n",
       "13359  Rhodospirillales   Azospirillaceae       Azospirillum   \n",
       "13360        Opitutales       Opitutaceae           Opitutus   \n",
       "\n",
       "                               Species  \n",
       "0           unclassified Paenibacillus  \n",
       "1           unclassified Paenibacillus  \n",
       "2           unclassified Paenibacillus  \n",
       "3           unclassified Paenibacillus  \n",
       "4           unclassified Paenibacillus  \n",
       "...                                ...  \n",
       "13356           Paenibacillus paeoniae  \n",
       "13357        unclassified Azospirillum  \n",
       "13358  Ruminiclostridium cellobioparum  \n",
       "13359        unclassified Azospirillum  \n",
       "13360            unclassified Opitutus  \n",
       "\n",
       "[13361 rows x 19 columns]"
      ]
     },
     "execution_count": 921,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge1 = df_cazy_f.merge(df_ids, on='Gene_ID', how='left')\n",
    "merge3 = merge1.merge(df_taxa, on='ID', how='left')\n",
    "merge3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 979,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community_x</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>HMMER</th>\n",
       "      <th>Hotpep</th>\n",
       "      <th>DIAMOND</th>\n",
       "      <th>NumberofTools</th>\n",
       "      <th>mode</th>\n",
       "      <th>community_y</th>\n",
       "      <th>contig_x</th>\n",
       "      <th>ID</th>\n",
       "      <th>community</th>\n",
       "      <th>contig_y</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00098</td>\n",
       "      <td>GT1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>GT1</td>\n",
       "      <td>46</td>\n",
       "      <td>435</td>\n",
       "      <td>46_435</td>\n",
       "      <td>46</td>\n",
       "      <td>435</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>unclassified Paenibacillus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00146</td>\n",
       "      <td>GH2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GH2</td>\n",
       "      <td>2</td>\n",
       "      <td>GH2</td>\n",
       "      <td>46</td>\n",
       "      <td>435</td>\n",
       "      <td>46_435</td>\n",
       "      <td>46</td>\n",
       "      <td>435</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>unclassified Paenibacillus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00264</td>\n",
       "      <td>GH154</td>\n",
       "      <td>GH154</td>\n",
       "      <td>GH154</td>\n",
       "      <td>3</td>\n",
       "      <td>GH154</td>\n",
       "      <td>46</td>\n",
       "      <td>186</td>\n",
       "      <td>46_186</td>\n",
       "      <td>46</td>\n",
       "      <td>186</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>unclassified Paenibacillus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00268</td>\n",
       "      <td>GH42</td>\n",
       "      <td>GH42</td>\n",
       "      <td>GH42</td>\n",
       "      <td>3</td>\n",
       "      <td>GH42</td>\n",
       "      <td>46</td>\n",
       "      <td>186</td>\n",
       "      <td>46_186</td>\n",
       "      <td>46</td>\n",
       "      <td>186</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>unclassified Paenibacillus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_00282</td>\n",
       "      <td>CBM67</td>\n",
       "      <td>GH78</td>\n",
       "      <td>GH78</td>\n",
       "      <td>3</td>\n",
       "      <td>GH78</td>\n",
       "      <td>46</td>\n",
       "      <td>186</td>\n",
       "      <td>46_186</td>\n",
       "      <td>46</td>\n",
       "      <td>186</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>unclassified Paenibacillus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13356</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_117026</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PL8</td>\n",
       "      <td>1</td>\n",
       "      <td>PL8</td>\n",
       "      <td>46</td>\n",
       "      <td>1142</td>\n",
       "      <td>46_1142</td>\n",
       "      <td>46</td>\n",
       "      <td>1142</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>Paenibacillus paeoniae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13357</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_35246</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GH2</td>\n",
       "      <td>1</td>\n",
       "      <td>GH2</td>\n",
       "      <td>46</td>\n",
       "      <td>118</td>\n",
       "      <td>46_118</td>\n",
       "      <td>46</td>\n",
       "      <td>118</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhodospirillales</td>\n",
       "      <td>Azospirillaceae</td>\n",
       "      <td>Azospirillum</td>\n",
       "      <td>unclassified Azospirillum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13358</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_78144</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GH36</td>\n",
       "      <td>1</td>\n",
       "      <td>GH36</td>\n",
       "      <td>46</td>\n",
       "      <td>1301</td>\n",
       "      <td>46_1301</td>\n",
       "      <td>46</td>\n",
       "      <td>1301</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Eubacteriales</td>\n",
       "      <td>Oscillospiraceae</td>\n",
       "      <td>Ruminiclostridium</td>\n",
       "      <td>Ruminiclostridium cellobioparum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13359</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_106008</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CBM48+GH13_10</td>\n",
       "      <td>1</td>\n",
       "      <td>CBM48+GH13_10</td>\n",
       "      <td>46</td>\n",
       "      <td>140</td>\n",
       "      <td>46_140</td>\n",
       "      <td>46</td>\n",
       "      <td>140</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhodospirillales</td>\n",
       "      <td>Azospirillaceae</td>\n",
       "      <td>Azospirillum</td>\n",
       "      <td>unclassified Azospirillum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13360</th>\n",
       "      <td>46</td>\n",
       "      <td>AAKEFBLG_04286</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PL22</td>\n",
       "      <td>1</td>\n",
       "      <td>PL22</td>\n",
       "      <td>46</td>\n",
       "      <td>1573</td>\n",
       "      <td>46_1573</td>\n",
       "      <td>46</td>\n",
       "      <td>1573</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>Opitutae</td>\n",
       "      <td>Opitutales</td>\n",
       "      <td>Opitutaceae</td>\n",
       "      <td>Opitutus</td>\n",
       "      <td>unclassified Opitutus</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13361 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      community_x          Gene_ID  HMMER Hotpep        DIAMOND  \\\n",
       "0              46   AAKEFBLG_00098    GT1    NaN            NaN   \n",
       "1              46   AAKEFBLG_00146    GH2    NaN            GH2   \n",
       "2              46   AAKEFBLG_00264  GH154  GH154          GH154   \n",
       "3              46   AAKEFBLG_00268   GH42   GH42           GH42   \n",
       "4              46   AAKEFBLG_00282  CBM67   GH78           GH78   \n",
       "...           ...              ...    ...    ...            ...   \n",
       "13356          46  AAKEFBLG_117026    NaN    NaN            PL8   \n",
       "13357          46   AAKEFBLG_35246    NaN    NaN            GH2   \n",
       "13358          46   AAKEFBLG_78144    NaN    NaN           GH36   \n",
       "13359          46  AAKEFBLG_106008    NaN    NaN  CBM48+GH13_10   \n",
       "13360          46   AAKEFBLG_04286    NaN    NaN           PL22   \n",
       "\n",
       "       NumberofTools           mode community_y contig_x       ID community  \\\n",
       "0                  1            GT1          46      435   46_435        46   \n",
       "1                  2            GH2          46      435   46_435        46   \n",
       "2                  3          GH154          46      186   46_186        46   \n",
       "3                  3           GH42          46      186   46_186        46   \n",
       "4                  3           GH78          46      186   46_186        46   \n",
       "...              ...            ...         ...      ...      ...       ...   \n",
       "13356              1            PL8          46     1142  46_1142        46   \n",
       "13357              1            GH2          46      118   46_118        46   \n",
       "13358              1           GH36          46     1301  46_1301        46   \n",
       "13359              1  CBM48+GH13_10          46      140   46_140        46   \n",
       "13360              1           PL22          46     1573  46_1573        46   \n",
       "\n",
       "       contig_y   Kingdom           Phylum                Class  \\\n",
       "0           435  Bacteria       Firmicutes              Bacilli   \n",
       "1           435  Bacteria       Firmicutes              Bacilli   \n",
       "2           186  Bacteria       Firmicutes              Bacilli   \n",
       "3           186  Bacteria       Firmicutes              Bacilli   \n",
       "4           186  Bacteria       Firmicutes              Bacilli   \n",
       "...         ...       ...              ...                  ...   \n",
       "13356      1142  Bacteria       Firmicutes              Bacilli   \n",
       "13357       118  Bacteria   Proteobacteria  Alphaproteobacteria   \n",
       "13358      1301  Bacteria       Firmicutes           Clostridia   \n",
       "13359       140  Bacteria   Proteobacteria  Alphaproteobacteria   \n",
       "13360      1573  Bacteria  Verrucomicrobia             Opitutae   \n",
       "\n",
       "                  Order            Family              Genus  \\\n",
       "0            Bacillales  Paenibacillaceae      Paenibacillus   \n",
       "1            Bacillales  Paenibacillaceae      Paenibacillus   \n",
       "2            Bacillales  Paenibacillaceae      Paenibacillus   \n",
       "3            Bacillales  Paenibacillaceae      Paenibacillus   \n",
       "4            Bacillales  Paenibacillaceae      Paenibacillus   \n",
       "...                 ...               ...                ...   \n",
       "13356        Bacillales  Paenibacillaceae      Paenibacillus   \n",
       "13357  Rhodospirillales   Azospirillaceae       Azospirillum   \n",
       "13358     Eubacteriales  Oscillospiraceae  Ruminiclostridium   \n",
       "13359  Rhodospirillales   Azospirillaceae       Azospirillum   \n",
       "13360        Opitutales       Opitutaceae           Opitutus   \n",
       "\n",
       "                               Species  \n",
       "0           unclassified Paenibacillus  \n",
       "1           unclassified Paenibacillus  \n",
       "2           unclassified Paenibacillus  \n",
       "3           unclassified Paenibacillus  \n",
       "4           unclassified Paenibacillus  \n",
       "...                                ...  \n",
       "13356           Paenibacillus paeoniae  \n",
       "13357        unclassified Azospirillum  \n",
       "13358  Ruminiclostridium cellobioparum  \n",
       "13359        unclassified Azospirillum  \n",
       "13360            unclassified Opitutus  \n",
       "\n",
       "[13361 rows x 19 columns]"
      ]
     },
     "execution_count": 979,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 933,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mode</th>\n",
       "      <th>ID</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>community</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>2873</td>\n",
       "      <td>2873</td>\n",
       "      <td>2867</td>\n",
       "      <td>2852</td>\n",
       "      <td>2840</td>\n",
       "      <td>2831</td>\n",
       "      <td>2825</td>\n",
       "      <td>2816</td>\n",
       "      <td>2704</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>3316</td>\n",
       "      <td>3316</td>\n",
       "      <td>3316</td>\n",
       "      <td>3316</td>\n",
       "      <td>3316</td>\n",
       "      <td>3314</td>\n",
       "      <td>3314</td>\n",
       "      <td>3293</td>\n",
       "      <td>3213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>2879</td>\n",
       "      <td>2879</td>\n",
       "      <td>2879</td>\n",
       "      <td>2854</td>\n",
       "      <td>2841</td>\n",
       "      <td>2826</td>\n",
       "      <td>2821</td>\n",
       "      <td>2814</td>\n",
       "      <td>2746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>4293</td>\n",
       "      <td>4293</td>\n",
       "      <td>4290</td>\n",
       "      <td>4287</td>\n",
       "      <td>4273</td>\n",
       "      <td>4273</td>\n",
       "      <td>4269</td>\n",
       "      <td>4257</td>\n",
       "      <td>4189</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           mode    ID  Kingdom  Phylum  Class  Order  Family  Genus  Species\n",
       "community                                                                   \n",
       "14         2873  2873     2867    2852   2840   2831    2825   2816     2704\n",
       "26         3316  3316     3316    3316   3316   3314    3314   3293     3213\n",
       "29         2879  2879     2879    2854   2841   2826    2821   2814     2746\n",
       "46         4293  4293     4290    4287   4273   4273    4269   4257     4189"
      ]
     },
     "execution_count": 933,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "taxa = merge3.iloc[: , [6, 9, 10, 12, 13, 14, 15, 16, 17 ,18]]\n",
    "merge3.iloc[: , [6, 9, 10, 12, 13, 14, 15, 16, 17 ,18]].groupby(\"community\").count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 911,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "community_x         0\n",
       "Gene_ID             0\n",
       "HMMER            4283\n",
       "Hotpep           5863\n",
       "DIAMOND          4769\n",
       "NumberofTools       0\n",
       "mode                0\n",
       "community_y         0\n",
       "contig              0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 911,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge1['community_y'].value_counts()\n",
    "merge1.isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1013,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community</th>\n",
       "      <th>mode</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14</td>\n",
       "      <td>AA1</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14</td>\n",
       "      <td>AA1</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14</td>\n",
       "      <td>AA1</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14</td>\n",
       "      <td>AA1</td>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14</td>\n",
       "      <td>AA10</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2420</th>\n",
       "      <td>46</td>\n",
       "      <td>PL9_1</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2421</th>\n",
       "      <td>46</td>\n",
       "      <td>PL9_2</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2422</th>\n",
       "      <td>46</td>\n",
       "      <td>PL9_4</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2423</th>\n",
       "      <td>46</td>\n",
       "      <td>SLH</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2424</th>\n",
       "      <td>46</td>\n",
       "      <td>SLH</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2425 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     community   mode           Phylum  counts\n",
       "0           14    AA1   Actinobacteria       1\n",
       "1           14    AA1    Bacteroidetes       1\n",
       "2           14    AA1   Proteobacteria       9\n",
       "3           14    AA1  Verrucomicrobia       2\n",
       "4           14   AA10   Actinobacteria       2\n",
       "...        ...    ...              ...     ...\n",
       "2420        46  PL9_1       Firmicutes       3\n",
       "2421        46  PL9_2       Firmicutes       1\n",
       "2422        46  PL9_4   Proteobacteria       1\n",
       "2423        46    SLH       Firmicutes      14\n",
       "2424        46    SLH   Proteobacteria       3\n",
       "\n",
       "[2425 rows x 4 columns]"
      ]
     },
     "execution_count": 1013,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fake_otu = df_cazy_f.groupby(['community','mode']).size().reset_index(name='counts')\n",
    "# fake_otu = fake_otu.pivot(index='mode', columns='community', values='counts').fillna('0')\n",
    "# fake_otu.to_csv('fake_otu',index=True, sep = '\\t')\n",
    "\n",
    "fake_taxa = taxa.groupby(['community','mode', 'Phylum']).size().reset_index(name='counts')\n",
    "fake_taxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 996,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community</th>\n",
       "      <th>mode</th>\n",
       "      <th>counts</th>\n",
       "      <th>Abditibacteriota</th>\n",
       "      <th>Actinobacteria</th>\n",
       "      <th>Armatimonadetes</th>\n",
       "      <th>Bacteroidetes</th>\n",
       "      <th>Chlamydiae</th>\n",
       "      <th>Cyanobacteria</th>\n",
       "      <th>Firmicutes</th>\n",
       "      <th>Planctomycetes</th>\n",
       "      <th>Proteobacteria</th>\n",
       "      <th>Verrucomicrobia</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Phylum</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Proteobacteria</th>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Actinobacteria</th>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroidetes</th>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Verrucomicrobia</th>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Firmicutes</th>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Planctomycetes</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cyanobacteria</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abditibacteriota</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Armatimonadetes</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chlamydiae</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  community  mode  counts  Abditibacteriota  Actinobacteria  \\\n",
       "Phylum                                                                        \n",
       "Proteobacteria           52    52      52                52              52   \n",
       "Actinobacteria           51    51      51                51              51   \n",
       "Bacteroidetes            47    47      47                47              47   \n",
       "Verrucomicrobia          38    38      38                38              38   \n",
       "Firmicutes               24    24      24                24              24   \n",
       "Planctomycetes            8     8       8                 8               8   \n",
       "Cyanobacteria             2     2       2                 2               2   \n",
       "Abditibacteriota          1     1       1                 1               1   \n",
       "Armatimonadetes           1     1       1                 1               1   \n",
       "Chlamydiae                1     1       1                 1               1   \n",
       "\n",
       "                  Armatimonadetes  Bacteroidetes  Chlamydiae  Cyanobacteria  \\\n",
       "Phylum                                                                        \n",
       "Proteobacteria                 52             52          52             52   \n",
       "Actinobacteria                 51             51          51             51   \n",
       "Bacteroidetes                  47             47          47             47   \n",
       "Verrucomicrobia                38             38          38             38   \n",
       "Firmicutes                     24             24          24             24   \n",
       "Planctomycetes                  8              8           8              8   \n",
       "Cyanobacteria                   2              2           2              2   \n",
       "Abditibacteriota                1              1           1              1   \n",
       "Armatimonadetes                 1              1           1              1   \n",
       "Chlamydiae                      1              1           1              1   \n",
       "\n",
       "                  Firmicutes  Planctomycetes  Proteobacteria  Verrucomicrobia  \n",
       "Phylum                                                                         \n",
       "Proteobacteria            52              52              52               52  \n",
       "Actinobacteria            51              51              51               51  \n",
       "Bacteroidetes             47              47              47               47  \n",
       "Verrucomicrobia           38              38              38               38  \n",
       "Firmicutes                24              24              24               24  \n",
       "Planctomycetes             8               8               8                8  \n",
       "Cyanobacteria              2               2               2                2  \n",
       "Abditibacteriota           1               1               1                1  \n",
       "Armatimonadetes            1               1               1                1  \n",
       "Chlamydiae                 1               1               1                1  "
      ]
     },
     "execution_count": 996,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "draws3.groupby(['Phylum']).count().sort_values('community', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 959,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community</th>\n",
       "      <th>mode</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>14</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>14</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>14</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>14</td>\n",
       "      <td>CBM50</td>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>14</td>\n",
       "      <td>CE1</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2344</th>\n",
       "      <td>46</td>\n",
       "      <td>GT4</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2345</th>\n",
       "      <td>46</td>\n",
       "      <td>GT4</td>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2354</th>\n",
       "      <td>46</td>\n",
       "      <td>GT51</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2355</th>\n",
       "      <td>46</td>\n",
       "      <td>GT51</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2356</th>\n",
       "      <td>46</td>\n",
       "      <td>GT51</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>225 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     community   mode           Phylum  counts\n",
       "90          14  CBM50   Actinobacteria       3\n",
       "91          14  CBM50    Bacteroidetes       2\n",
       "92          14  CBM50   Proteobacteria      74\n",
       "93          14  CBM50  Verrucomicrobia       2\n",
       "129         14    CE1   Actinobacteria       5\n",
       "...        ...    ...              ...     ...\n",
       "2344        46    GT4   Proteobacteria     208\n",
       "2345        46    GT4  Verrucomicrobia      15\n",
       "2354        46   GT51   Actinobacteria       5\n",
       "2355        46   GT51       Firmicutes      12\n",
       "2356        46   GT51   Proteobacteria      81\n",
       "\n",
       "[225 rows x 4 columns]"
      ]
     },
     "execution_count": 959,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "todraw2 = fake_taxa[fake_taxa['mode'].isin(liss)]\n",
    "todraw2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1014,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community</th>\n",
       "      <th>mode</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14</td>\n",
       "      <td>AA1</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14</td>\n",
       "      <td>AA1</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14</td>\n",
       "      <td>AA1</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14</td>\n",
       "      <td>AA1</td>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14</td>\n",
       "      <td>AA10</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2420</th>\n",
       "      <td>46</td>\n",
       "      <td>PL9_1</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2421</th>\n",
       "      <td>46</td>\n",
       "      <td>PL9_2</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2422</th>\n",
       "      <td>46</td>\n",
       "      <td>PL9_4</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2423</th>\n",
       "      <td>46</td>\n",
       "      <td>SLH</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2424</th>\n",
       "      <td>46</td>\n",
       "      <td>SLH</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2425 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     community   mode           Phylum  counts\n",
       "0           14    AA1   Actinobacteria       1\n",
       "1           14    AA1    Bacteroidetes       1\n",
       "2           14    AA1   Proteobacteria       9\n",
       "3           14    AA1  Verrucomicrobia       2\n",
       "4           14   AA10   Actinobacteria       2\n",
       "...        ...    ...              ...     ...\n",
       "2420        46  PL9_1       Firmicutes       3\n",
       "2421        46  PL9_2       Firmicutes       1\n",
       "2422        46  PL9_4   Proteobacteria       1\n",
       "2423        46    SLH       Firmicutes      14\n",
       "2424        46    SLH   Proteobacteria       3\n",
       "\n",
       "[2425 rows x 4 columns]"
      ]
     },
     "execution_count": 1014,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fake_taxa[\"community\"] = fake_taxa.to_numeric(fake_taxa[\"community\"])\n",
    "# fake_taxa.groupby(['Phylum']).count().sort_values('community', ascending=False)\n",
    "fake_taxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 997,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mode</th>\n",
       "      <th>ID</th>\n",
       "      <th>community</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GT1</td>\n",
       "      <td>46_435</td>\n",
       "      <td>46</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>unclassified Paenibacillus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GH2</td>\n",
       "      <td>46_435</td>\n",
       "      <td>46</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>unclassified Paenibacillus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GH154</td>\n",
       "      <td>46_186</td>\n",
       "      <td>46</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>unclassified Paenibacillus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GH42</td>\n",
       "      <td>46_186</td>\n",
       "      <td>46</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>unclassified Paenibacillus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GH78</td>\n",
       "      <td>46_186</td>\n",
       "      <td>46</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>unclassified Paenibacillus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13356</th>\n",
       "      <td>PL8</td>\n",
       "      <td>46_1142</td>\n",
       "      <td>46</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>Paenibacillus paeoniae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13357</th>\n",
       "      <td>GH2</td>\n",
       "      <td>46_118</td>\n",
       "      <td>46</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhodospirillales</td>\n",
       "      <td>Azospirillaceae</td>\n",
       "      <td>Azospirillum</td>\n",
       "      <td>unclassified Azospirillum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13358</th>\n",
       "      <td>GH36</td>\n",
       "      <td>46_1301</td>\n",
       "      <td>46</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Eubacteriales</td>\n",
       "      <td>Oscillospiraceae</td>\n",
       "      <td>Ruminiclostridium</td>\n",
       "      <td>Ruminiclostridium cellobioparum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13359</th>\n",
       "      <td>CBM48+GH13_10</td>\n",
       "      <td>46_140</td>\n",
       "      <td>46</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhodospirillales</td>\n",
       "      <td>Azospirillaceae</td>\n",
       "      <td>Azospirillum</td>\n",
       "      <td>unclassified Azospirillum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13360</th>\n",
       "      <td>PL22</td>\n",
       "      <td>46_1573</td>\n",
       "      <td>46</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>Opitutae</td>\n",
       "      <td>Opitutales</td>\n",
       "      <td>Opitutaceae</td>\n",
       "      <td>Opitutus</td>\n",
       "      <td>unclassified Opitutus</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13361 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                mode       ID community   Kingdom           Phylum  \\\n",
       "0                GT1   46_435        46  Bacteria       Firmicutes   \n",
       "1                GH2   46_435        46  Bacteria       Firmicutes   \n",
       "2              GH154   46_186        46  Bacteria       Firmicutes   \n",
       "3               GH42   46_186        46  Bacteria       Firmicutes   \n",
       "4               GH78   46_186        46  Bacteria       Firmicutes   \n",
       "...              ...      ...       ...       ...              ...   \n",
       "13356            PL8  46_1142        46  Bacteria       Firmicutes   \n",
       "13357            GH2   46_118        46  Bacteria   Proteobacteria   \n",
       "13358           GH36  46_1301        46  Bacteria       Firmicutes   \n",
       "13359  CBM48+GH13_10   46_140        46  Bacteria   Proteobacteria   \n",
       "13360           PL22  46_1573        46  Bacteria  Verrucomicrobia   \n",
       "\n",
       "                     Class             Order            Family  \\\n",
       "0                  Bacilli        Bacillales  Paenibacillaceae   \n",
       "1                  Bacilli        Bacillales  Paenibacillaceae   \n",
       "2                  Bacilli        Bacillales  Paenibacillaceae   \n",
       "3                  Bacilli        Bacillales  Paenibacillaceae   \n",
       "4                  Bacilli        Bacillales  Paenibacillaceae   \n",
       "...                    ...               ...               ...   \n",
       "13356              Bacilli        Bacillales  Paenibacillaceae   \n",
       "13357  Alphaproteobacteria  Rhodospirillales   Azospirillaceae   \n",
       "13358           Clostridia     Eubacteriales  Oscillospiraceae   \n",
       "13359  Alphaproteobacteria  Rhodospirillales   Azospirillaceae   \n",
       "13360             Opitutae        Opitutales       Opitutaceae   \n",
       "\n",
       "                   Genus                          Species  \n",
       "0          Paenibacillus       unclassified Paenibacillus  \n",
       "1          Paenibacillus       unclassified Paenibacillus  \n",
       "2          Paenibacillus       unclassified Paenibacillus  \n",
       "3          Paenibacillus       unclassified Paenibacillus  \n",
       "4          Paenibacillus       unclassified Paenibacillus  \n",
       "...                  ...                              ...  \n",
       "13356      Paenibacillus           Paenibacillus paeoniae  \n",
       "13357       Azospirillum        unclassified Azospirillum  \n",
       "13358  Ruminiclostridium  Ruminiclostridium cellobioparum  \n",
       "13359       Azospirillum        unclassified Azospirillum  \n",
       "13360           Opitutus            unclassified Opitutus  \n",
       "\n",
       "[13361 rows x 10 columns]"
      ]
     },
     "execution_count": 997,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "taxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 946,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "keyword argument repeated (<ipython-input-946-484dc70924b3>, line 1)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-946-484dc70924b3>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0m\n\u001b[0;31m    sns.catplot(data=todraw2, hue=\"community\", x = \"mode\", y='counts', kind=\"bar\", hue=\"Phylum\", aspect=1.5, height=10)\u001b[0m\n\u001b[0m                                                                                   ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m keyword argument repeated\n"
     ]
    }
   ],
   "source": [
    "sns.catplot(data=todraw2, hue=\"community\", x = \"mode\", y='counts', kind=\"bar\", aspect=1.5, height=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 999,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>community</th>\n",
       "      <th>mode</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>counts</th>\n",
       "      <th>Abditibacteriota</th>\n",
       "      <th>Acidobacteria</th>\n",
       "      <th>Actinobacteria</th>\n",
       "      <th>Armatimonadetes</th>\n",
       "      <th>Bacteroidetes</th>\n",
       "      <th>Chlamydiae</th>\n",
       "      <th>Cyanobacteria</th>\n",
       "      <th>Firmicutes</th>\n",
       "      <th>Opisthokonta</th>\n",
       "      <th>Planctomycetes</th>\n",
       "      <th>Proteobacteria</th>\n",
       "      <th>Verrucomicrobia</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14</td>\n",
       "      <td>AA1</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14</td>\n",
       "      <td>AA1</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14</td>\n",
       "      <td>AA1</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14</td>\n",
       "      <td>AA1</td>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14</td>\n",
       "      <td>AA10</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2420</th>\n",
       "      <td>46</td>\n",
       "      <td>PL9_1</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2421</th>\n",
       "      <td>46</td>\n",
       "      <td>PL9_2</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2422</th>\n",
       "      <td>46</td>\n",
       "      <td>PL9_4</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2423</th>\n",
       "      <td>46</td>\n",
       "      <td>SLH</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>14</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2424</th>\n",
       "      <td>46</td>\n",
       "      <td>SLH</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2425 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     community   mode           Phylum  counts  Abditibacteriota  \\\n",
       "0           14    AA1   Actinobacteria       1               0.0   \n",
       "1           14    AA1    Bacteroidetes       1               0.0   \n",
       "2           14    AA1   Proteobacteria       9               0.0   \n",
       "3           14    AA1  Verrucomicrobia       2               0.0   \n",
       "4           14   AA10   Actinobacteria       2               0.0   \n",
       "...        ...    ...              ...     ...               ...   \n",
       "2420        46  PL9_1       Firmicutes       3               0.0   \n",
       "2421        46  PL9_2       Firmicutes       1               0.0   \n",
       "2422        46  PL9_4   Proteobacteria       1               0.0   \n",
       "2423        46    SLH       Firmicutes      14               0.0   \n",
       "2424        46    SLH   Proteobacteria       3               0.0   \n",
       "\n",
       "      Acidobacteria  Actinobacteria  Armatimonadetes  Bacteroidetes  \\\n",
       "0               0.0             1.0              0.0            0.0   \n",
       "1               0.0             0.0              0.0            1.0   \n",
       "2               0.0             0.0              0.0            0.0   \n",
       "3               0.0             0.0              0.0            0.0   \n",
       "4               0.0             2.0              0.0            0.0   \n",
       "...             ...             ...              ...            ...   \n",
       "2420            0.0             0.0              0.0            0.0   \n",
       "2421            0.0             0.0              0.0            0.0   \n",
       "2422            0.0             0.0              0.0            0.0   \n",
       "2423            0.0             0.0              0.0            0.0   \n",
       "2424            0.0             0.0              0.0            0.0   \n",
       "\n",
       "      Chlamydiae  Cyanobacteria  Firmicutes  Opisthokonta  Planctomycetes  \\\n",
       "0            0.0            0.0         0.0           0.0             0.0   \n",
       "1            0.0            0.0         0.0           0.0             0.0   \n",
       "2            0.0            0.0         0.0           0.0             0.0   \n",
       "3            0.0            0.0         0.0           0.0             0.0   \n",
       "4            0.0            0.0         0.0           0.0             0.0   \n",
       "...          ...            ...         ...           ...             ...   \n",
       "2420         0.0            0.0         3.0           0.0             0.0   \n",
       "2421         0.0            0.0         1.0           0.0             0.0   \n",
       "2422         0.0            0.0         0.0           0.0             0.0   \n",
       "2423         0.0            0.0        14.0           0.0             0.0   \n",
       "2424         0.0            0.0         0.0           0.0             0.0   \n",
       "\n",
       "      Proteobacteria  Verrucomicrobia  \n",
       "0                0.0              0.0  \n",
       "1                0.0              0.0  \n",
       "2                9.0              0.0  \n",
       "3                0.0              2.0  \n",
       "4                0.0              0.0  \n",
       "...              ...              ...  \n",
       "2420             0.0              0.0  \n",
       "2421             0.0              0.0  \n",
       "2422             1.0              0.0  \n",
       "2423             0.0              0.0  \n",
       "2424             3.0              0.0  \n",
       "\n",
       "[2425 rows x 16 columns]"
      ]
     },
     "execution_count": 999,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phylums = fake_taxa.pivot(columns='Phylum', values='counts').fillna(\"0\").apply(pd.to_numeric)\n",
    "draws3 = pd.concat([fake_taxa, phylums], axis=1)\n",
    "draws3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1018,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import matplotlib.patches as mpatches\n",
    "\n",
    "# plt.figure(figsize=(8, 13))\n",
    "# # sns.barplot(data=todraw2, x=\"mode\", y = \"counts\", hue=\"community\")\n",
    "\n",
    "\n",
    "# s1 = sns.barplot(x = 'community', y = 'Proteobacteria', data = draws3, color = 'violet')\n",
    "# s2 = sns.barplot(x = 'community', y = 'Actinobacteria', data = draws3, color = 'cyan')\n",
    "# s3 = sns.barplot(x = 'community', y = 'Bacteroidetes', data = draws3, color = 'yellow')\n",
    "# s4 = sns.barplot(x = 'community', y = 'Verrucomicrobia', data = draws3, color = 'lightpink')\n",
    "# s5 = sns.barplot(x = 'community', y = 'Firmicutes', data = draws3, color = 'darkgreen')\n",
    "\n",
    "# bar_1 = mpatches.Patch(color='violet', label='Proteobacteria')\n",
    "# bar_2 = mpatches.Patch(color='cyan', label='Actinobacteria')\n",
    "# bar_3 = mpatches.Patch(color='yellow', label='Bacteroidetes')\n",
    "# bar_4 = mpatches.Patch(color='lightpink', label='Verrucomicrobia')\n",
    "# bar_5 = mpatches.Patch(color='darkgreen', label='Firmicutes')\n",
    "\n",
    "# plt.legend(handles=[bar_1, bar_2,bar_3,bar_4,bar_5])\n",
    "\n",
    "draw_max = fake_taxa[fake_taxa['Phylum'].isin(['Proteobacteria',\n",
    "                                                       'Actinobacteria', \n",
    "                                                       'Bacteroidetes',\n",
    "                                                       'Verrucomicrobia',\n",
    "                                                       'Firmicutes'\n",
    "                                                      ])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1034,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>mode</th>\n",
       "      <th>counts</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Phylum</th>\n",
       "      <th>community</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Proteobacteria</th>\n",
       "      <th>26</th>\n",
       "      <td>254</td>\n",
       "      <td>254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Firmicutes</th>\n",
       "      <th>46</th>\n",
       "      <td>246</td>\n",
       "      <td>246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">Proteobacteria</th>\n",
       "      <th>46</th>\n",
       "      <td>236</td>\n",
       "      <td>236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>236</td>\n",
       "      <td>236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Verrucomicrobia</th>\n",
       "      <th>14</th>\n",
       "      <td>155</td>\n",
       "      <td>155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Actinobacteria</th>\n",
       "      <th>46</th>\n",
       "      <td>133</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroidetes</th>\n",
       "      <th>29</th>\n",
       "      <td>133</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Actinobacteria</th>\n",
       "      <th>29</th>\n",
       "      <td>125</td>\n",
       "      <td>125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>113</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroidetes</th>\n",
       "      <th>26</th>\n",
       "      <td>106</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Actinobacteria</th>\n",
       "      <th>26</th>\n",
       "      <td>99</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Verrucomicrobia</th>\n",
       "      <th>29</th>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroidetes</th>\n",
       "      <th>14</th>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Firmicutes</th>\n",
       "      <th>29</th>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Verrucomicrobia</th>\n",
       "      <th>26</th>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroidetes</th>\n",
       "      <th>46</th>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Firmicutes</th>\n",
       "      <th>14</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           mode  counts\n",
       "Phylum          community              \n",
       "Proteobacteria  26          254     254\n",
       "Firmicutes      46          246     246\n",
       "Proteobacteria  46          236     236\n",
       "                14          236     236\n",
       "                29          203     203\n",
       "Verrucomicrobia 14          155     155\n",
       "Actinobacteria  46          133     133\n",
       "Bacteroidetes   29          133     133\n",
       "Actinobacteria  29          125     125\n",
       "                14          113     113\n",
       "Bacteroidetes   26          106     106\n",
       "Actinobacteria  26           99      99\n",
       "Verrucomicrobia 29           96      96\n",
       "Bacteroidetes   14           79      79\n",
       "Firmicutes      29           57      57\n",
       "Verrucomicrobia 26           43      43\n",
       "                46           43      43\n",
       "Bacteroidetes   46           25      25\n",
       "Firmicutes      14            4       4"
      ]
     },
     "execution_count": 1034,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "draw_amazind"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1038,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Phylum</th>\n",
       "      <th>community</th>\n",
       "      <th>mode</th>\n",
       "      <th>counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>26</td>\n",
       "      <td>254</td>\n",
       "      <td>254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>46</td>\n",
       "      <td>246</td>\n",
       "      <td>246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>46</td>\n",
       "      <td>236</td>\n",
       "      <td>236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>14</td>\n",
       "      <td>236</td>\n",
       "      <td>236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>29</td>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>14</td>\n",
       "      <td>155</td>\n",
       "      <td>155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>46</td>\n",
       "      <td>133</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>29</td>\n",
       "      <td>133</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>29</td>\n",
       "      <td>125</td>\n",
       "      <td>125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>14</td>\n",
       "      <td>113</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>26</td>\n",
       "      <td>106</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>26</td>\n",
       "      <td>99</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>29</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>14</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>29</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>26</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>46</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>46</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Phylum community  mode  counts\n",
       "0    Proteobacteria        26   254     254\n",
       "1        Firmicutes        46   246     246\n",
       "2    Proteobacteria        46   236     236\n",
       "3    Proteobacteria        14   236     236\n",
       "4    Proteobacteria        29   203     203\n",
       "5   Verrucomicrobia        14   155     155\n",
       "6    Actinobacteria        46   133     133\n",
       "7     Bacteroidetes        29   133     133\n",
       "8    Actinobacteria        29   125     125\n",
       "9    Actinobacteria        14   113     113\n",
       "10    Bacteroidetes        26   106     106\n",
       "11   Actinobacteria        26    99      99\n",
       "12  Verrucomicrobia        29    96      96\n",
       "13    Bacteroidetes        14    79      79\n",
       "14       Firmicutes        29    57      57\n",
       "15  Verrucomicrobia        26    43      43\n",
       "16  Verrucomicrobia        46    43      43\n",
       "17    Bacteroidetes        46    25      25\n",
       "18       Firmicutes        14     4       4"
      ]
     },
     "execution_count": 1038,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "draw_amazind = draw_max.groupby(['Phylum','community']).count().sort_values('counts', ascending=False)\n",
    "draw_amazind.reset_index(inplace=True)\n",
    "draw_amazind"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1044,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<AxesSubplot:xlabel='community', ylabel='counts'>"
      ]
     },
     "execution_count": 1044,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAiwAAAHgCAYAAACGtTa7AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjMuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/d3fzzAAAACXBIWXMAAAsTAAALEwEAmpwYAAAxjElEQVR4nO3deZhV1Znv8e8LGHFABSFeIyql7YBQRQEFMimgEYzzRARRMai0UWISh0TtK1fpcDvpRKXFm/BwW8XYijgENbYxBpkCaLBAVCYVIlGCNwFRBAGVYt0/6nBSQCEF1qE28P08D0+ds/fa67xVW4qfa6+9V6SUkCRJyrJ6dV2AJEnSthhYJElS5hlYJElS5hlYJElS5hlYJElS5hlYJElS5jWo6wK+iqZNm6YWLVrUdRmSJKmWzJw5c3lKqdnm23fpwNKiRQvKy8vrugxJklRLIuIv1W33kpAkSco8A4skSco8A4skScq8XXoOiyRpz/DFF1+wZMkS1q1bV9elqJY0bNiQ5s2bs9dee9WovYFFkpR5S5YsoVGjRrRo0YKIqOty9BWllPjwww9ZsmQJRUVFNTrGS0KSpMxbt24dBx98sGFlNxERHHzwwds1YmZgkSTtEgwru5ftPZ8GFkmSakH9+vUpLS2ldevW9OnThzVr1rB48WJat269Xf306NHDZ4xVw8AiSVIt2GeffZg9ezZz5szha1/7GiNHjqzrknYrBhZJkmrZSSedxMKFCwGoqKjg6quvplWrVvTq1Yu1a9eyaNEi2rVrl2//zjvv0L59+y362X///fOvn3zySa644goArrjiCr773e/Ss2dPjjrqKCZPnszAgQNp2bJlvs3uxsAiSVItWr9+Pb/73e8oLi4GKsPIddddx9y5cznooIN46qmnOProoznwwAOZPXs2AA8++OB2B42PPvqICRMmcM8993D22Wfzwx/+kLlz5/Lmm2/m+92dGFgkSaoFa9eupbS0lLKyMo444giuvPJKAIqKiigtLQWgffv2LF68GICrrrqKBx98kIqKCsaOHcsll1yyXZ939tlnExEUFxdzyCGHUFxcTL169WjVqlX+M3YnPodFkqRasHEOy+b23nvv/Ov69euzdu1aAC688ELuvPNOTjnlFNq3b8/BBx+8xbFV76TZ/Bbgjf3Wq1dvk8+oV68e69ev/0rfSxY5wiJJUh1o2LAhvXv35rvf/S7f+c53qm1zyCGHMH/+fDZs2MC4ceN2coXZYmCRJKmO9O/fn4igV69e1e7/6U9/yllnncUpp5zCoYceupOry5ZIKdV1DTusrKwsea+6JO3+5s+fT8uWLeu6jFr3i1/8gpUrV/Kv//qvdV1KnajuvEbEzJRS2eZtncMiSVIdOP/881m0aBETJkyo61J2CQYWSZLqwJ4+J2V7OYdFkiRl3h43wtL+5l/Xan8zf355rfYnSZK25AiLJEnKvIIFlog4PCImRsT8iJgbEd/Pbb8jIv4aEbNzf86ocsytEbEwIt6KiN6Fqk2SJO1aCjnCsh64MaXUEugEXBcRJ+T23ZNSKs39eR4gt68v0Ao4HfhlRNQvYH2SJNVY/fr1KS0tpXXr1vTp04c1a9bU+NjZs2fz/PPP7/Bnjx49msGDB+/w8RtNmjSJ6dOnb/dx5eXlXH/99V/587+Kgs1hSSl9AHyQe70qIuYDh33JIecCj6WUPgPejYiFQEfg5ULVKEnaNdXFfMSqj97v378/I0eO5IYbbsjvr6iooH796v8/e/bs2ZSXl3PGGWdUu39nmTRpEvvvvz9dunSp8THr16+nrKyMsrItHo2yU+2UOSwR0QJoC/wpt2lwRLwREQ9EROPctsOA96sctoQvDziSJNWJk046iYULFzJp0iR69uzJJZdcQnFxMevWreM73/kOxcXFtG3blokTJ/L5558zZMgQxo4dS2lpKWPHjuXTTz9l4MCBdOjQgbZt2/LMM88AVHv8Ru+//z6nn346xx13HHfeeWd++3nnnUf79u1p1aoVo0aNym9/4YUXaNeuHW3atOHUU09l8eLFjBw5knvuuYfS0lL++Mc/smzZMi688EI6dOhAhw4dmDZtGgB33HEHgwYNolevXlx++eVMmjSJs846C4AZM2bQpUsX2rZtS5cuXXjrrbd2xo+88HcJRcT+wFPAD1JKn0TEr4B/BVLu613AQCCqOXyLx/BGxCBgEMARRxxRqLIlSarW+vXr+d3vfsfpp58OVP4DPmfOHIqKirjrrrsAePPNN1mwYAG9evXi7bffZujQoZSXl3PfffcBcNttt3HKKafwwAMP8PHHH9OxY0e++c1vMnLkyGqPr/o5++67Lx06dODMM8+krKyMBx54gCZNmrB27Vo6dOjAhRdeyIYNG7j66quZMmUKRUVFrFixgiZNmnDNNdew//77c9NNNwFwySWX8MMf/pBu3brx3nvv0bt3b+bPnw/AzJkzmTp1Kvvssw+TJk3Kf//HH388U6ZMoUGDBowfP57bbruNp556quA/94IGlojYi8qw8khK6TcAKaW/Vdn/f4Hncm+XAIdXObw5sHTzPlNKo4BRUPlo/sJULknSptauXUtpaSlQOcJy5ZVXMn36dDp27EhRUREAU6dO5Xvf+x5Q+Q/7kUcemQ8cVb344os8++yz/OIXvwAqR1bee++9Lz3+tNNOy6/ofMEFFzB16lTKysq499578w+he//993nnnXdYtmwZJ598cr6uJk2aVPs9jR8/nnnz5uXff/LJJ6xatQqAc845h3322WeLY1auXMmAAQN45513iAi++OKL7fgp7riCBZaoXBP7fmB+SunuKtsPzc1vATgfmJN7/SzwaETcDXwDOAaYUaj6JEnaHlXnsFS133775V/XdH2+lBJPPfUUxx133Bbbt6byn9VN30+aNInx48fz8ssvs++++9KjRw/WrVtHSmmL9tXZsGEDL7/8crXBpOr3VdXtt99Oz549GTduHIsXL6ZHjx7b/JzaUMg5LF2By4BTNruF+d8j4s2IeAPoCfwQIKU0F3gcmAe8AFyXUqooYH2SJNWqk08+mUceeQSAt99+m/fee4/jjjuORo0a5UcuAHr37s2IESPyAeW111770uMB/vCHP7BixQrWrl3L008/TdeuXVm5ciWNGzdm3333ZcGCBbzyyisAdO7cmcmTJ/Puu+8CsGLFCoAt6ujVq1f+MhVQbSDb3MqVKznssMoppqNHj97un9GOKlhgSSlNTSlFSqmk6i3MKaXLUkrFue3nVBltIaU0LKV0dErpuJTS7wpVmyRJhXDttddSUVFBcXExF198MaNHj2bvvfemZ8+ezJs3Lz/p9vbbb+eLL76gpKSE1q1bc/vtt3/p8QDdunXjsssuo7S0lAsvvJCysjJOP/101q9fT0lJCbfffjudOnUCoFmzZowaNYoLLriANm3acPHFFwNw9tlnM27cuPyk23vvvZfy8nJKSko44YQT8nNovsyPfvQjbr31Vrp27UpFxc4bV4iaDl9lUVlZWSovL9+uY3w0vyTteubPn0/Lli3rugzVsurOa0TMTCltcQ+1j+aXJEmZt8ctfijVhveGFtdqf0cMebNW+5Ok3Y0jLJIkKfMMLJIkKfMMLJIkKfMMLJIkKfMMLJIk1UD9+vUpLS3N/1m8ePF2rXpcnZEjR/LrX+/Y4zaGDx/OmjVrvtLn70q8S0iStMupizv1qns0//Tp07doV1FRQf369Wv0uddcc02N2lVn+PDhXHrppey777473MeuxBEWSZJ20P777w/ApEmT6NmzJ5dccgnFxcVMmjSJ7t278+1vf5tjjz2WW265hUceeYSOHTtSXFzMokWLALjjjjvyCyAuXLiQb37zm7Rp04Z27dqxaNEiJk2axFlnnZX/vMGDBzN69Gjuvfdeli5dSs+ePenZsydQuaBi586dadeuHX369GH16tUA3HLLLZxwwgmUlJTkV2neFTnCIklSDVRdrbmoqCi/QvJGM2bMYM6cORQVFTFp0iRef/115s+fT5MmTTjqqKO46qqrmDFjBv/xH//BiBEjGD58+CbH9+/fn1tuuYXzzz+fdevWsWHDBt5///1qa7n++uu5++67mThxIk2bNmX58uX85Cc/Yfz48ey333787Gc/4+6772bw4MGMGzeOBQsWEBF8/PHHBfjJ7BwGFkmSamBrqzVv1LFjR4qKivLvO3TowKGHHgrA0UcfTa9evQAoLi5m4sSJmxy7atUq/vrXv3L++ecD0LBhw+2q7ZVXXmHevHl07doVgM8//5zOnTtzwAEH0LBhQ6666irOPPPMTUZrdjUGFkmSasF+++23yfuNixYC1KtXL/++Xr16rF+/fpO2W1vXr0GDBmzYsCH/ft26ddW2Sylx2mmnMWbMmC32zZgxg5deeonHHnuM++67jwkTJtTsG8oY57BIklTHDjjgAJo3b87TTz8NwGeffcaaNWs48sgjmTdvHp999hkrV67kpZdeyh/TqFEjVq1aBUCnTp2YNm0aCxcuBGDNmjW8/fbbrF69mpUrV3LGGWcwfPjwLx0hyjpHWCRJyoCHH36Yf/7nf2bIkCHstddePPHEExx11FF8+9vfpqSkhGOOOYa2bdvm2w8aNIhvfetbHHrooUycOJHRo0fTr18/PvvsMwB+8pOf0KhRI84991zWrVtHSol77rmnrr69ryy2Ngy1KygrK0vl5eXbdUz7m3fsfvetmfnzy2u1P+0aXPxQ2rnmz59Py5Yt67oM1bLqzmtEzEwplW3e1ktCkiQp8wwskiQp85zDot1ebV8GBBjXqNa7VC3z8q+0e3GERZIkZZ6BRZIkZZ6BRZIkZZ6BRZKkbejRowe///3vN9k2fPhwrr322jqqqOaWLl3KRRddVCt9bVzscXNDhgxh/PjxtfIZW+OkW0nSLqfriK612t+070370v39+vXjscceo3fv3vltjz32GD//+c+32XdFRQX169ff6vtC+8Y3vsGTTz5Z4/Y7Ut/QoUO3t6zt5giLJEnbcNFFF/Hcc8/lnyK7ePFili5dypo1a+jcuTPt2rWjT58+rF69GoAWLVowdOhQunXrxhNPPLHF+x49erDxwafLly+nRYsWQGVYuOmmmyguLqakpIQRI0YA8NJLL9G2bVuKi4sZOHBgvo4WLVpw22230blzZ8rKypg1axa9e/fm6KOPZuTIkflaW7du/aX9b17fmDFjKC4upnXr1vz4xz/e5Gdx44030q5dO0499VSWLVsGwBVXXJEPRUOHDqVDhw60bt2aQYMGbXWdpO1lYJEkaRsOPvhgOnbsyAsvvABUjq6ceuqpDBs2jPHjxzNr1izKysq4++6788c0bNiQqVOn0rdv32rfV2fUqFG8++67vPbaa7zxxhv079+fdevWccUVVzB27FjefPNN1q9fz69+9av8MYcffjgvv/wyJ510Uj44vPLKKwwZMqRG/W9e78knn8yPf/xjJkyYwOzZs3n11Vfzaxx9+umntGvXjlmzZtG9e3fuvPPOLT5j8ODBvPrqq8yZM4e1a9fy3HPPbd8PeysMLJIk1cDGy0JQGViKioqYN28eXbt2pbS0lIceeoi//OUv+fYXX3zxJsdv/r4648eP55prrqFBg8oZG02aNOGtt96iqKiIY489FoABAwYwZcqU/DHnnHMOAMXFxZx44ok0atSIZs2a0bBhQz7++ONt9r95fa+++io9evSgWbNmNGjQgP79++c/r169evl2l156KVOnTt3ie5g4cSInnngixcXFTJgwgblz527z+64J57BIklQD5513HjfccAOzZs1i7dq1tG3bltNOO40xY8ZU236//fbb6vsGDRqwYcMGANatW5ffnlIiIjY5bluXVPbee2+gMkxsfL3x/fr167foa/P+N69vey7hbN7XunXruPbaaykvL+fwww/njjvu2OT7+yocYZEkqQb2339/evTowcCBA+nXrx+dOnVi2rRpLFy4EIA1a9bw9ttv16ivFi1aMHPmTIBNJsT26tWLkSNH5oPGihUrOP7441m8eHH+cx5++GG6d+++Q99Ddf1v7sQTT2Ty5MksX76ciooKxowZk/+8DRs25Ot99NFH6dat2ybHbgwnTZs2ZfXq1ds12XdbDCySJNVQv379eP311+nbty/NmjVj9OjR9OvXj5KSEjp16sSCBQtq1M9NN93Er371K7p06cLy5cvz26+66iqOOOIISkpKaNOmDY8++igNGzbkwQcfpE+fPhQXF1OvXj2uueaaHaq/uv43d+ihh/Jv//Zv9OzZkzZt2tCuXTvOPfdcoHIUZu7cubRv354JEyZsMU/moIMO4uqrr6a4uJjzzjuPDh067FCd1Ynamr1bF8rKytLGWdY15foie57CrCW07VsZt8cRQ96s1f7k3/Xdzfz582nZsmVdl6FaVt15jYiZKaWyzds6wiJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJUg2NGzeOiNjm81aGDx/OmjVr8u/POOOMLR6TX1NVF0r8KjavqaaGDBnC+PHjv/Lnf1U+ml+StMuZfPKOPel1a7pPmVyjdmPGjKFbt2489thj3HHHHVttN3z4cC699FL23XdfAJ5//vnaKPMr2bymmqioqGDo0KEFrKrmHGGRJKkGVq9ezbRp07j//vvziyBWVFRw0003UVxcTElJCSNGjODee+9l6dKl9OzZk549ewKVj+Jfvnw5ixcvpmXLllx99dW0atWKXr16sXbtWgBmz55Np06dKCkp4fzzz+ejjz7Kf/Z//dd/0aVLF1q3bs2MGTMAmDFjBl26dKFt27Z06dKFt956a7tqevHFF+ncuTPt2rWjT58+rF69Ol/r0KFD6datG0888UR+BWiAoUOH0qFDB1q3bs2gQYO2a92hr8rAIklSDTz99NOcfvrpHHvssTRp0oRZs2YxatQo3n33XV577TXeeOMN+vfvz/XXX883vvENJk6cyMSJE7fo55133uG6665j7ty5HHTQQTz11FMAXH755fzsZz/jjTfeoLi4mDvvvDN/zKeffsr06dP55S9/ycCBAwE4/vjjmTJlCq+99hpDhw7ltttuA6hRTcuXL+cnP/kJ48ePZ9asWZSVlXH33XfnP69hw4ZMnTqVvn37blL74MGDefXVV5kzZw5r167lueeeq/Wf89Z4SUiSpBoYM2YMP/jBDwDo27cvY8aM4c9//jPXXHMNDRpU/nPapEmTbfZTVFREaWkpAO3bt2fx4sWsXLmSjz/+OL/I4IABA+jTp0/+mH79+gFw8skn88knn/Dxxx+zatUqBgwYwDvvvENE8MUXXwAwfvz4bdb0yiuvMG/ePLp27QrA559/TufOnfP7L7744mprnzhxIv/+7//OmjVrWLFiBa1ateLss8/e5vdcGwwskiRtw4cffsiECROYM2cOEUFFRQURQfv27YmI7epr7733zr+uX79+/pLQl9n8MyKC22+/nZ49ezJu3DgWL15Mjx49AEgpbbOmlBKnnXYaY8aMqXb/fvvtt8W2devWce2111JeXs7hhx/OHXfckV+deWfwkpAkSdvw5JNPcvnll/OXv/yFxYsX8/7771NUVES7du0YOXIk69evB2DFihUANGrUiFWrVtW4/wMPPJDGjRvzxz/+EYCHH344P9oCMHbsWACmTp3KgQceyIEHHsjKlSs57LDDABg9enS+ba9evbZZU6dOnZg2bRoLFy4EYM2aNbz99ttfWuPGcNK0aVNWr16dn9eysxhYJEnahjFjxnD++edvsu3CCy9k6dKlHHHEEZSUlNCmTRseffRRAAYNGsS3vvWt/ATXmnjooYe4+eabKSkpYfbs2QwZMiS/r3HjxnTp0oVrrrmG+++/H4Af/ehH3HrrrXTt2pWKiop826uuumqbNTVr1ozRo0fTr18/SkpK6NSp0zZv1T7ooIO4+uqrKS4u5rzzzqNDhw41/t5qQ+zMGb61raysLG3vvekuOb/nqe1zDjCu0c9rtb8jhrxZq/3Jv+u7m/nz59OyZcu6LkO1rLrzGhEzU0plm7d1hEWSJGWegUWSJGWegUWSJGWegUWSJGWegUWSJGWegUWSJGWegUWSpBqoX78+paWltGnThnbt2jF9+vQd6ufpp59m3rx5tVbXkCFDGD9+/BbbJ02axFlnnfWlx86ePTsTK0nXhI/mlyTtcu678be12t/gu7a9Hs4+++zD7NmzAfj973/PrbfeyuTJk7f7s55++mnOOussTjjhhBofs379+vzaQJsbOnTodtew0ezZsykvL+eMM87Y4T52FkdYJEnaTp988gmNGzcGYPXq1Zx66qm0a9eO4uJinnnmmXy7X//61/knzl522WVMnz6dZ599lptvvpnS0lIWLVrEokWLOP3002nfvj0nnXRS/omzV1xxBTfccAM9e/bkxz/+MbNnz6ZTp06UlJRw/vnn89FHH+XbbXxM/gsvvMDxxx9Pt27d+M1vfpOv49NPP2XgwIF06NCBtm3b8swzz/D5558zZMgQxo4dS2lpKWPHjq22HcDcuXPp2LEjpaWllJSU8M477+yUn3NVjrBIklQDa9eupbS0lHXr1vHBBx8wYcIEABo2bMi4ceM44IADWL58OZ06deKcc85h3rx5DBs2jGnTptG0aVNWrFhBkyZNOOecczjrrLO46KKLADj11FMZOXIkxxxzDH/605+49tpr832//fbbjB8/nvr161NSUsKIESPo3r07Q4YM4c4772T48OH5+tatW8fVV1/NhAkT+Kd/+qdNVlweNmwYp5xyCg888AAff/wxHTt25Jvf/CZDhw6lvLyc++67D4Dbbrut2nYjR47k+9//Pv379+fzzz/fZCmAncXAIklSDVS9JPTyyy9z+eWXM2fOHFJK3HbbbUyZMoV69erx17/+lb/97W9MmDCBiy66iKZNmwLQpEmTLfpcvXo106dPp0+fPvltn332Wf51nz59qF+/PitXruTjjz/OL4g4YMCATY4BWLBgAUVFRRxzzDEAXHrppYwaNQqAF198kWeffZZf/OIXQGW4ee+997aoZ2vtOnfuzLBhw1iyZAkXXHBB/jN2JgOLJEnbqXPnzixfvpxly5bx/PPPs2zZMmbOnMlee+1FixYtWLduHSklIuJL+9mwYQMHHXRQPghtbr/99tuuurb2eSklnnrqKY477rhNtv/pT3+qUbuWLVty4okn8t///d/07t2b//zP/+SUU07Zrtq+KuewSJK0nRYsWEBFRQUHH3wwK1eu5Otf/zp77bUXEydO5C9/+QtQeann8ccf58MPPwRgxYoVADRq1IhVq1YBcMABB1BUVMQTTzwBVAaG119/fYvPO/DAA2ncuDF//OMfAXj44Yfzoy0bHX/88bz77rssWrQIqFxheqPevXszYsQINi54/Nprr21Ry5e1+/Of/8xRRx3F9ddfzznnnMMbb7yxwz+7HWVgkSSpBjbOYSktLeXiiy/moYceon79+vTv35/y8nLKysp45JFHOP744wFo1aoV//Iv/0L37t1p06YNN9xwAwB9+/bl5z//OW3btmXRokU88sgj3H///bRp04ZWrVptMmm3qoceeoibb76ZkpISZs+ezZAhQzbZ37BhQ0aNGsWZZ55Jt27dOPLII/P7br/9dr744gtKSkpo3bo1t99+OwA9e/Zk3rx5+Um3W2s3duxYWrduTWlpKQsWLODyy3f+6uWxMUXtisrKylJ5efl2HeOS83ue2j7nAOMa/bxW+ztiyJu12p/8u767mT9/Pi1btqzrMlTLqjuvETEzpVS2eVtHWCRJUuYZWCRJUuYZWCRJUuYZWCRJu4Rdec6ltrS959PAIknKvIYNG/Lhhx8aWnYTKSU+/PBDGjZsWONjfHCcJCnzmjdvzpIlS1i2bFldl6Ja0rBhQ5o3b17j9gYWSVLm7bXXXhQVFdV1GapDXhKSJEmZZ2CRJEmZZ2CRJEmZZ2CRJEmZ56RbSdJuwzWkdl8FG2GJiMMjYmJEzI+IuRHx/dz2JhHxh4h4J/e1cZVjbo2IhRHxVkT0LlRtkiRp11LIS0LrgRtTSi2BTsB1EXECcAvwUkrpGOCl3Hty+/oCrYDTgV9GRP0C1idJknYRBQssKaUPUkqzcq9XAfOBw4BzgYdyzR4Czsu9Phd4LKX0WUrpXWAh0LFQ9UmSpF3HTpl0GxEtgLbAn4BDUkofQGWoAb6ea3YY8H6Vw5bktkmSpD1cwQNLROwPPAX8IKX0yZc1rWbbFotGRMSgiCiPiHIf0SxJ0p6hoIElIvaiMqw8klL6TW7z3yLi0Nz+Q4G/57YvAQ6vcnhzYOnmfaaURqWUylJKZc2aNStc8ZIkKTMKeZdQAPcD81NKd1fZ9SwwIPd6APBMle19I2LviCgCjgFmFKo+SZK06yjkc1i6ApcBb0bE7Ny224CfAo9HxJXAe0AfgJTS3Ih4HJhH5R1G16WUKgpYnyRJ2kUULLCklKZS/bwUgFO3cswwYFihapIkSbsmH80vSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyz8AiSZIyr0FdFyBJe6KuI7rWan/TvjetVvuTssYRFkmSlHkGFkmSlHkGFkmSlHkFCywR8UBE/D0i5lTZdkdE/DUiZuf+nFFl360RsTAi3oqI3oWqS5Ik7XoKOcIyGji9mu33pJRKc3+eB4iIE4C+QKvcMb+MiPoFrE2SJO1CChZYUkpTgBU1bH4u8FhK6bOU0rvAQqBjoWqTJEm7lrqYwzI4It7IXTJqnNt2GPB+lTZLctskSZJ2emD5FXA0UAp8ANyV2x7VtE3VdRARgyKiPCLKly1bVpAiJUlStuzUwJJS+ltKqSKltAH4v/zjss8S4PAqTZsDS7fSx6iUUllKqaxZs2aFLViSJGXCTg0sEXFolbfnAxvvIHoW6BsRe0dEEXAMMGNn1iZJkrKrYI/mj4gxQA+gaUQsAf4X0CMiSqm83LMY+GeAlNLciHgcmAesB65LKVUUqjZJkrRrKVhgSSn1q2bz/V/SfhgwrFD1SJKkXZdPupUkSZlnYJEkSZlnYJEkSZlnYJEkSZlnYJEkSZlnYJEkSZlnYJEkSZlnYJEkSZlXo8ASEd+PiAOi0v0RMSsiehW6OEmSJKj5CMvAlNInQC+gGfAd4KcFq0qSJKmKmgaWyH09A3gwpfR6lW2SJEkFVdPAMjMiXqQysPw+IhoBGwpXliRJ0j/UdPHDK4FS4M8ppTURcTCVl4UkSZIKrqYjLH9IKc1KKX0MkFL6ELinYFVJkiRV8aUjLBHRENgXaBoRjfnHvJUDgG8UuDZJkiRg25eE/hn4AZXhZCb/CCyfAP+ncGVJkiT9w5cGlpTSfwD/ERHfSymN2Ek1SZIkbaJGk25TSiMiogvQouoxKaVfF6guSZKkvBoFloh4GDgamA1U5DYnwMAiSZIKrqa3NZcBJ6SUUiGLkSRJqk5Nb2ueA/yPQhYiSZK0NTUdYWkKzIuIGcBnGzemlM4pSFWSJElV1DSw3FHIIiRJkr5MTe8SmlzoQiTVnsknd6/V/rpP8VeAtCe478bf1mp/g+86u9b6quldQquovCsI4GvAXsCnKaUDaq0SSZKkrajpCEujqu8j4jygYyEKkiRJ2lxN7xLaRErpaeCU2i1FkiSpejW9JHRBlbf1qHwui89kkSRJO0VN7xKqOmtmPbAYOLfWq5EkSapGTeewfKfQhUiSJG1NjeawRETziBgXEX+PiL9FxFMR0bzQxUmSJEHNJ90+CDwLfAM4DPhtbpskSVLB1TSwNEspPZhSWp/7MxpoVsC6JEmS8moaWJZHxKURUT/351Lgw0IWJkmStFFNA8tA4NvA/wM+AC4CnIgrSZJ2ipre1vyvwICU0kcAEdEE+AWVQUaSJKmgahpYSjaGFYCU0oqIaFugmiRJUg3U9kKndLipdvurRTW9JFQvIhpvfJMbYalp2JEkSfpKaho67gKmR8STVD6S/9vAsIJVJUmSVEVNn3T764gop3LBwwAuSCnNK2hlkiRJOTW+rJMLKIYUSZK009V0DoskSVKdMbBIkqTMM7BIkqTMM7BIkqTMM7BIkqTMM7BIkqTMM7BIkqTMM7BIkqTMM7BIkqTMM7BIkqTMM7BIkqTMM7BIkqTMM7BIkqTMM7BIkqTMM7BIkqTMa1DXBUiSvrrJJ3ev1f66T5lcq/1JX5UjLJIkKfMMLJIkKfMMLJIkKfMMLJIkKfMMLJIkKfMMLJIkKfMMLJIkKfMMLJIkKfMMLJIkKfMMLJIkKfN8NL+2y303/rZW+xt819m12p8kaffkCIskSco8A4skSco8A4skSco8A4skSco8A4skScq8ggWWiHggIv4eEXOqbGsSEX+IiHdyXxtX2XdrRCyMiLcioneh6pIkSbueQo6wjAZO32zbLcBLKaVjgJdy74mIE4C+QKvcMb+MiPoFrE2SJO1CChZYUkpTgBWbbT4XeCj3+iHgvCrbH0spfZZSehdYCHQsVG2SJGnXsrPnsBySUvoAIPf167nthwHvV2m3JLdNkiQpM5Nuo5ptqdqGEYMiojwiypctW1bgsiRJUhbs7Efz/y0iDk0pfRARhwJ/z21fAhxepV1zYGl1HaSURgGjAMrKyqoNNfqHySd3r90OO9xUu/1JklQDO3uE5VlgQO71AOCZKtv7RsTeEVEEHAPM2Mm1SZKkjCrYCEtEjAF6AE0jYgnwv4CfAo9HxJXAe0AfgJTS3Ih4HJgHrAeuSylVFKo2SZK0aylYYEkp9dvKrlO30n4YMKxQ9UiSpF1XVibdSpIkbdXOnnQrSdIeq+uIrrXa3//eg/4Zd4RFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlXoO6LkCSlD333fjbWu1v8F1n12p/2vM4wiJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjLPwCJJkjKvQV0XIElSVr03tLh2O2x8QO32twdxhEWSJGWegUWSJGWegUWSJGWegUWSJGWegUWSJGWegUWSJGWegUWSJGVenTyHJSIWA6uACmB9SqksIpoAY4EWwGLg2ymlj+qiPkmSlC11OcLSM6VUmlIqy72/BXgppXQM8FLuvSRJUqYuCZ0LPJR7/RBwXt2VIkmSsqSuAksCXoyImRExKLftkJTSBwC5r1+v7sCIGBQR5RFRvmzZsp1UriRJqkt1tZZQ15TS0oj4OvCHiFhQ0wNTSqOAUQBlZWWpUAVKkqTsqJMRlpTS0tzXvwPjgI7A3yLiUIDc17/XRW2SJCl7dnpgiYj9IqLRxtdAL2AO8CwwINdsAPDMzq5NkiRlU11cEjoEGBcRGz//0ZTSCxHxKvB4RFwJvAf0qYPaJElSBu30wJJS+jPQpprtHwKn7ux6JElS9mXptmZJkqRqGVgkSVLmGVgkSVLmGVgkSVLmGVgkSVLmGVgkSVLm1dWj+bUVXUd0rdX+/reneJfgeZekL+cIiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyrwGdV2ApOy778bf1mp/g+86u1b7k7T7c4RFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlnoFFkiRlXoO6LkCSdgXvDS2u3Q4bH1C7/Um7OUdYJElS5hlYJElS5mUusETE6RHxVkQsjIhb6roeSZJU9zIVWCKiPvB/gG8BJwD9IuKEuq1KkiTVtUwFFqAjsDCl9OeU0ufAY8C5dVyTJEmqY1kLLIcB71d5vyS3TZIk7cEipVTXNeRFRB+gd0rpqtz7y4COKaXvVWkzCBiUe3sc8NZOLzQbmgLL67oI7XSe9z2P53zPtCef9yNTSs0235i157AsAQ6v8r45sLRqg5TSKGDUziwqiyKiPKVUVtd1aOfyvO95POd7Js/7lrJ2SehV4JiIKIqIrwF9gWfruCZJklTHMjXCklJaHxGDgd8D9YEHUkpz67gsSZJUxzIVWABSSs8Dz9d1HbuAPf6y2B7K877n8ZzvmTzvm8nUpFtJkqTqZG0OiyRJ0hYMLBkXEYdHxMSImB8RcyPi+1X2fS+3jMHciPj3uqxThRER9SPitYh4rso2z/tuKiIeiIi/R8ScavbdFBEpIprWRW0qjK39jo+INhHxckS8GRG/jYg9fnnvzM1h0RbWAzemlGZFRCNgZkT8ATiEyqcAl6SUPouIr9dplSqU7wPzgQMAIqInnvfd2WjgPuDXVTdGxOHAacB7dVCTCmtrv+P/E7gppTQ5IgYCNwO312Whdc0RloxLKX2QUpqVe72Kyn+8DgO+C/w0pfRZbt/f665KFUJENAfOpPIX10ae991YSmkKsKKaXfcAPwKcdLib+ZLf8ccBU3LN/gBcWDcVZoeBZRcSES2AtsCfgGOBkyLiTxExOSI61GlxKoThVP4jtaHKNs/7HiYizgH+mlJ6va5rUWFt9jt+DnBOblcfNn2o6h7JwLKLiIj9gaeAH6SUPqHycl5joBOVQ4WPR0TUYYmqRRFxFvD3lNLMzXZ53vcgEbEv8C/AkLquRYVVze/4gcB1ETETaAR8Xpf1ZYFzWHYBEbEXlf8hP5JS+k1u8xLgN6nyvvQZEbGByrUnltVRmapdXYFzIuIMoCFwQET8F573Pc3RQBHwei6XNgdmRUTHlNL/q9PKVGuq+x2fUloA9MrtP5bKy8N7NEdYMi73f8/3A/NTSndX2fU0cEquzbHA19hzF8ra7aSUbk0pNU8ptaByiYoJKaVL8bzvUVJKb6aUvp5SapH7b2EJ0M6wsvvY2u/4jRPqI6Ie8D+BkXVTYXYYWLKvK3AZcEpEzM79OQN4ADgqd/vjY8CA5FMA9wSe991YRIwBXgaOi4glEXFlXdekgtva7/h+EfE2sIDKRYAfrMsis8An3UqSpMxzhEWSJGWegUWSJGWegUWSJGWegUWSJGWegUWSJGWegUXSbi8iromIy3Ovr4iIb9R1TZK2j7c1S9qjRMQkKlfBLa/rWiTVnCMsknZYRFweEW9ExOsR8XBEHBkRL+W2vRQRR+TajY6IX0XExIj4c0R0j4gHImJ+RIyu0t/qiPhZRMyMiPER0TEiJuWOOSfX5oqIuK/KMc9FRI8qxw/L1fNKRByS235HRNwUERcBZcAjuQd0nRkR46r0dVpEbFz+QlKGGFgk7ZCIaEXlwnynpJTaAN8H7gN+nVIqAR4B7q1ySGMqlxX4IfBb4B6gFVAcEaW5NvsBk1JK7YFVwE+A04DzgaE1KGs/4JVcPVOAq6vuTCk9CZQD/VNKpcDzQMuIaJZr8h18oqiUSQYWSTvqFODJlNJygJTSCqAz8Ghu/8NAtyrtf5tbRuBN4G+5dXI2AHOBFrk2nwMv5F6/CUxOKX2Re92CbfsceC73eua2jsnV8zBwaUQclKv/dzX4HEk7mas1S9pRAWxrElzV/Z/lvm6o8nrj+42/i76osjZSvl1KaUNEbGyznk3/Z6thlddVj6+gZr/jHqRyxGcd8ERKaX0NjpG0kznCImlHvQR8OyIOBoiIJsB0KleXBugPTC3A5y4GSiOiXkQcDnTczuNXAY02vkkpLaVycbn/CYyupRol1TJHWCTtkJTS3IgYBkyOiArgNeB64IGIuBlYRuWckNo2DXiXystEc4BZ23n8aGBkRKwFOqeU1lI536ZZSmlebRYqqfZ4W7OkPV7urqPXUkr313UtkqpnYJG0R4uImcCnwGkppc+21V5S3TCwSJKkzHPSrSRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyjwDiyRJyrz/D19jwWbPCfd/AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 648x576 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(9, 8))\n",
    "sns.barplot(data=draw_amazind, x=\"community\", y = \"counts\", hue=\"Phylum\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'draws3' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-17-c43d78099c07>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 7\u001b[0;31m \u001b[0msns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbarplot\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'community'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'Proteobacteria'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdraws3\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolor\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'violet'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'draws3' is not defined"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 576x936 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.patches as mpatches\n",
    "\n",
    "plt.figure(figsize=(8, 13))\n",
    "# sns.barplot(data=todraw2, x=\"mode\", y = \"counts\", hue=\"community\")\n",
    "\n",
    "\n",
    "sns.barplot(x = 'community', y = 'Proteobacteria', data = draws3, color = 'violet')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.catplot(data=todraw2, hue=\"Phylum\", x = \"mode\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:vamb]",
   "language": "python",
   "name": "conda-env-vamb-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
