---
title: "endo"
author: "GrGladkov"
date: "03 10 2021"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/home/gladkov/storage/Endo/")
path_trein_set = "~/storage/somebases/silva_nr_v138_train_set.fa"
path_trein_set_species = "~/storage/somebases/silva_species_assignment_v138.fa"
name_Brief = "nameBrief.txt"
```



```{r}

setwd("~/storage/Endo/")

library(Matrix)
library(ShortRead)
library(dada2)
library(Biostrings)
library(seqinr)
library(tidyverse)

```



```{r}

setwd("/home/gladkov/storage/Endo/")
path <- "in/"
list.files()

```

```{r}

library(naturalsort)
path <- "in/"
FWD <- "GGATTAGATACCCTGGTA" 
REV <- "CGACRRCCATGCANCACCT"

length(FWD)

cutFs <- sort(list.files(path.cut, pattern = "_R1_001.fastq.gz", full.names = TRUE))
cutRs <- sort(list.files(path.cut, pattern = "_R2_001.fastq.gz", full.names = TRUE))
fnFs <- sort(list.files(path, pattern = "R1_001.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern = "R2_001.fastq.gz", full.names = TRUE))
fnFs <- sort(list.files(path, pattern = "_F_filt.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern = "_R_filt.fastq.gz", full.names = TRUE))
get.sample.name <- function(fname) strsplit(basename(fname), "_")[[1]][1]
sample.names <- unname(sapply(cutFs, get.sample.name))
basename(fnFs)
sort(fnFs)
filenames_d <- data_frame(forward_reads = basename(fnFs), reverse_reads = basename(fnRs) )
filenames_d
fnFs <- naturalsort(fnFs)
fnRs <- naturalsort(fnRs)
plotQualityProfile(fnRs[8])


fnFs <- sort(list.files(path, pattern="_R1_001.fastq", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_001.fastq", full.names = TRUE))
  
plotQualityProfile(fnFs, aggregate = TRUE)
```


```{r}

plotQualityProfile(fnRs, aggregate = TRUE)

```


```{r}
length(fnRs)
```

```{r}
plotQualityProfile(fnRs)

mult = TRUE
mlt = NULL

packageVersion("Matrix")
BiocManager::install("ShortRead")
BiocManager::install("dada2")

install.packages("~/storage/temp/dada2-master//",
                 repos = NULL,
                 type = "source",
                 dependencies = c("Depends", "Suggests","Imports"))


remove.packages("Matrix")
devtools::install_version("Matrix", version = "1.3.2", repos = "http://cran.us.r-project.org")

```


```{r} 

sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
on.exit()
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(250,200), trimLeft=c(18,19), maxN=0, maxEE=c(2,6), rm.phix=TRUE, compress=TRUE, multithread=20)
errF <- learnErrors(filtFs, multithread=mult)
errR <- learnErrors(filtRs, multithread=mult)
derepFs <- derepFastq(filtFs, verbose=TRUE)
derepRs <- derepFastq(filtRs, verbose=TRUE)
# Name the derep-class objects by the sample names
names(derepFs) <- sample.names
names(derepRs) <- sample.names
dadaFs <- dada(derepFs, err=errF, multithread=mult, pool=FALSE)
dadaRs <- dada(derepRs, err=errR, multithread=mult, pool=FALSE)
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose=TRUE)
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
table(nchar(getSequences(seqtab)))
getN <- function(x) sum(getUniques(x))
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=mult, verbose=TRUE)


briefToSeq <- colnames(seqtab.nochim)
names(briefToSeq) <- paste0("Seq", seq(ncol(seqtab.nochim))) 
st.brief <- seqtab.nochim
colnames(st.brief) <- names(briefToSeq) 

write.table(briefToSeq, file = "endo_taxa", sep = "\t")

#dna <- DNAStringSet(briefToSeq)
#alignment <- AlignSeqs(DNAStringSet(dna), anchor=NA,verbose=FALSE, processors = mlt)
#writeXStringSet(alignment, file="align.fasta")

taxa.dada2 <- assignTaxonomy(briefToSeq,path_trein_set , multithread=mult)
taxa.dada2.species <- assignSpecies(briefToSeq, path_trein_set_species) # maybe use not seqs but brief 
rownames(taxa.dada2.species) <- rownames(briefToSeq)
briefToSeq.df <- data.frame(briefToSeq)
rownames(taxa.dada2.species) <- rownames(briefToSeq.df)
rownames(taxa.dada2) <- rownames(taxa.dada2.species)
taxa <- cbind2(taxa.dada2, taxa.dada2.species[,2])
colnames(taxa)[7] <- "Species"

getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
write.table(track, file = "track.tsv", sep= "\t", col.names = NA, quote=FALSE)

st.brief.t.df <- data.frame(t(st.brief))
write.table(st.brief.t.df, file = "otu_table.txt", sep= "\t", col.names = NA, quote=FALSE)

briefToSeq.ls <- as.list(briefToSeq.df[,c("briefToSeq")])
briefToSeq.names <- as.list(rownames(briefToSeq.df))
write.fasta( briefToSeq.ls, briefToSeq.names , "rep_seq.fasta", as.string = FALSE)

write.table(taxa, file = "taxa.txt", sep= "\t", col.names = NA, quote=FALSE)
as.data.frame(st.brief.t.df)
```


#### map, reexport

```{r}


library(phyloseq)
#some SEPP bash tree magic

map <- read_tsv("endo_map.tsv")
map
map <- column_to_rownames(map, "ID")

map

taxa <- read.csv("taxa.txt" , header=TRUE, sep="\t")
taxa <- column_to_rownames(taxa, 'X')
taxa <- as.matrix(taxa)


filt.otu <- read.csv("otu_table.txt" , header=TRUE, sep="\t")
filt.otu
colnames(filt.otu)[1] <- "ID"
filt.otu <- column_to_rownames(filt.otu, "ID")
filt.otu
library(naturalsort)
filt.otu <- filt.otu[c(naturalsort(colnames(filt.otu)))]
colnames(filt.otu)
colnames(filt.otu) <- rownames(map)
filt.otu
# class(filt.otu) <- "numeric"

filt.otu <- as.matrix(filt.otu)
filt.otu <- t(filt.otu)
rownames(filt.otu)
rownames(map)

tree <- read_tree(treefile="tree.nwk")

ps <- phyloseq(otu_table(filt.otu, taxa_are_rows=FALSE), 
               sample_data(map),
               tax_table(taxa)
               )

phy_tree(tree)
length(tree$tip.label)
ps@phy_tree$tip.label
ps

filt.otu[0,] = rownames(map)
rownames(filt.otu) == rownames(map)

rownames(taxa)

intersect(rownames(as.data.frame(filt.otu.matrix)), rownames(map)) %>% length()

colnames(filt.otu)
ps <- merge_phyloseq(ps, refseq(readDNAStringSet("rep_seq.fasta")))
ps

```


```{r}
delete_mit_chl_eu <- function(ps)  {
  
  pop_taxa <- function(physeq, badTaxa)
    {
  allTaxa = taxa_names(physeq)
  myTaxa <- allTaxa[!(allTaxa %in% badTaxa)]
  return(prune_taxa(myTaxa, physeq))
  }
  
  del_chl <- function(ps){
    if ("Chloroplast" %in% as.data.frame(ps@tax_table@.Data)$Order){
      badTaxa <- taxa_names(subset_taxa(ps, Order=="Chloroplast"))
      ps <- pop_taxa(ps, badTaxa)
    }
    else{
    return(ps)  
    }
  }
  
  del_mit <- function(ps)
    {
    if ("Mitochondria" %in% as.data.frame(ps@tax_table@.Data)$Family) {
        badTaxa <- taxa_names(subset_taxa(ps, Family=="Mitochondria"))
    ps <- pop_taxa(ps, badTaxa)
    }
    else{
    return(ps)  
    }
  }
  
    del_eu <- function(ps){
    if ("Eukaryota" %in% as.data.frame(ps@tax_table@.Data)$Kingdom){
      badTaxa <- taxa_names(subset_taxa(ps, Kingdom == "Eukaryota"))
      ps <- pop_taxa(ps, badTaxa)
    }
    else{
    return(ps)  
    }
  }
    
    del_phy <- function(ps){
    if (any(is.na(as.data.frame(ps@tax_table@.Data)$Phylum))){
      badTaxa <- taxa_names(subset_taxa(ps, is.na(Phylum)))
      ps <- pop_taxa(ps, badTaxa)
    }
    else{
    return(ps)  
    }
  }
      
  ps <- del_chl(ps)
  ps <- del_mit(ps)
  ps <- del_eu(ps)
  ps <- del_phy(ps)

  return(ps)
}


ps.f <- delete_mit_chl_eu(ps)
writeXStringSet(ps.f@refseq, file="ref_filt.fasta")
```

```{r}
ps.f
```

```{r}
ps
```

```{r}
map <- map %>%  mutate_if(sapply(map, is.character), as.factor)
sample_data(ps.f) <- map
```


```{r, message=FALSE,echo=TRUE, fig.height=8, fig.width=12}
# Plot alpha-diversity by selected metric
plot_alpha <- function(ps, metric, group) {
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  plot_richness(ps_a, x=group, measures=metric) + 
    geom_boxplot(, fill="#4DBBD5B2", alpha=0.4) +
    geom_point(size=1.2, alpha=0.3) + 
    theme_light() + scale_color_brewer(palette="Dark2") +
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
          axis.title.x=element_blank(),
          text = element_text(size=20)) +
    labs(y=paste("alpha-diversity index")) 
}

plot_alpha(ps.f, c("Observed","Shannon","InvSimpson"), "Repeat") 

```


```{r,fig.height=10, fig.width=8, results=FALSE}

phyloseq_to_amp <- function(ps){
    require(ampvis2)
    require(tibble)
    require(phyloseq)
    colnames(ps@tax_table@.Data) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
    OTU1 = as(otu_table(ps), "matrix")
    OTU1 <- t(OTU1)
    OTUdf = as.data.frame(OTU1)
    taxa.ps <- as(tax_table(ps), "matrix")
    taxa.df = as.data.frame(taxa.ps)
    my_otu_table <- merge(OTUdf, taxa.df, by=0)
    my_otu_table <- column_to_rownames(my_otu_table, var="Row.names")

    my_metadata <- as_tibble(sample_data(ps), rownames=NA)
    my_metadata <- rownames_to_column(my_metadata,var = "SampleID")
    # my_tree <- phy_tree(ps)
    amp.ps <- amp_load(otutable = my_otu_table, metadata = my_metadata)
    return(amp.ps)
}


amp <- phyloseq_to_amp(ps.f2)
amp_heatmap(amp,tax_show = 30, group_by = "Repeat", tax_aggregate = "Genus", tax_class = "Proteobacteria", tax_add = "Phylum") + theme_bw() + theme(text = element_text(size=15), legend.position = "none") + theme(axis.text.x=element_text(angle=45,hjust=1))
amp
```
```{r}
as.data.frame(ps.f@tax_table@.Data)
ps.f@sam_data
```


```{r}

ps.f2 <- readRDS("ps.f2")
tree <-  read_tree("tree.nwk")
ps.f2 <- merge_phyloseq(ps.f2, phy_tree(tree))
write_rds(ps.f2, "ps.f2")
ps.f2
```


```{r}
ps_vst <- function(ps){
  diagdds = phyloseq_to_deseq2(ps, ~ Repeat)                  
  diagdds = estimateSizeFactors(diagdds, type="poscounts")
  diagdds = estimateDispersions(diagdds, fitType = "local") 
  pst <- varianceStabilizingTransformation(diagdds)
  pst.dimmed <- t(as.matrix(assay(pst))) 
  pst.dimmed[pst.dimmed < 0.0] <- 0.0
  ps.varstab <- ps
  otu_table(ps.varstab) <- otu_table(pst.dimmed, taxa_are_rows = FALSE) 
  return(ps.varstab)
}

library(DESeq2)

ps.var <- ps_vst(ps.f2)
ps.f2@sam_data

```

```{r, message=FALSE,echo=TRUE,fig.height=8, fig.width=9, results=FALSE, fig.keep='all'}

beta_custom_norm_NMDS_elli <- function(ps, seed = 7888, normtype="vst", color="Description"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  require(DESeq2)
  library(ggforce)
  
  
  # beta_NMDS <- function(){
  #normalisation. unifrac - rarefaction; wunifrac,bray - varstab
  
  diagdds = phyloseq_to_deseq2(ps, ~ Repeat)                  
  diagdds = estimateSizeFactors(diagdds, type="poscounts")
  diagdds = estimateDispersions(diagdds, fitType = "local") 
  if (normtype =="vst")
    pst <- varianceStabilizingTransformation(diagdds)
  if (normtype =="log") 
    pst <- rlogTransformation(diagdds)
  
  pst.dimmed <- t(as.matrix(assay(pst))) 
  pst.dimmed[pst.dimmed < 0.0] <- 0.0
  ps.varstab <- ps
  otu_table(ps.varstab) <- otu_table(pst.dimmed, taxa_are_rows = FALSE) 

  #beta and ordination
  
  ordination.b <- ordinate(ps.varstab, "NMDS", "bray")

  #plotting
  p  <-  plot_ordination(ps, ordination.b, type="sample", color="Repeat", title="NMDS - Bray-Curtis", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes(group = Repeat, label = Repeat), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
  
  return(p)
}

p.beta <- beta_custom_norm_NMDS_elli(ps.f2)
p.beta

```

```{r, message=FALSE,echo=TRUE, fig.height=5, fig.width=8}
# Plot alpha-diversity by selected metric
plot_alpha <- function(ps, metric, group) {
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  plot_richness(ps_a, x=group, measures=metric) + 
    geom_boxplot(, fill="#4DBBD5B2", alpha=0.4) +
    geom_point(size=1.2, alpha=0.3) + 
    theme_light() + scale_color_brewer(palette="Dark2") +
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
          axis.title.x=element_blank(),
          text = element_text(size=20)) +
    labs(y=paste("alpha-diversity index")) 
}

plot_alpha(ps.f2, c("Observed","Shannon","InvSimpson"), "Soil") 

```



```{r}
ps.f2@sam_data$Genotype
```

```{r, message=FALSE,echo=TRUE,fig.height=8, fig.width=12}

library(ggpubr)

alpha.custom <- function(ps.f, arrga = "PD"){
  
  require(picante)
  
  pd <- pd(ps.f@otu_table@.Data, ps.f@phy_tree)
  
  pd1 <- estimate_richness(ps.f, measures=c("Observed", "InvSimpson", "Shannon"))
  pd <- cbind(pd, ps.f@sam_data, pd1 )
  
  bmi <- levels(as.factor(pd$Soil))
  bmi.pairs <- combn(seq_along(bmi), 2, simplify = FALSE, FUN = function(i)bmi[i])
  p1 <- ggviolin(pd, x = "Soil", y = arrga,
                 add = "boxplot", fill = "Soil") + stat_compare_means(comparisons = bmi.pairs, method = "wilcox.test") +
    scale_x_discrete(limits = c("ARRIAM_field", "chernevaya_taiga")) +
    theme(axis.title.x = element_blank(),
          legend.title = element_blank(),
          legend.position = "none",
          axis.text.x = element_text(angle = 45, hjust = 1),
          text = element_text(size=14))
  return(p1)
}

alpha.custom2 <- function(ps.f, arrga = "PD"){
  
  require(picante)
  
  pd <- pd(ps.f@otu_table@.Data, ps.f@phy_tree)
  
  pd1 <- estimate_richness(ps.f, measures=c("Observed", "InvSimpson", "Shannon"))
  pd <- cbind(pd, ps.f@sam_data, pd1 )
  
  bmi <- levels(as.factor(pd$Genotype))
  bmi.pairs <- combn(seq_along(bmi), 2, simplify = FALSE, FUN = function(i)bmi[i])
  p1 <- ggviolin(pd, x = "Genotype", y = arrga,
                 add = "boxplot", fill = "Genotype") + stat_compare_means(comparisons = bmi.pairs, method = "wilcox.test") +
    scale_x_discrete(limits = c("Vendevil", "Classic", "Triumph")) +
    theme(axis.title.x = element_blank(),
          legend.title = element_blank(),
          legend.position = "none",
          axis.text.x = element_text(angle = 45, hjust = 1),
          text = element_text(size=14))
  return(p1)
}

p.alpha.Cdt.oo <- alpha.custom(ps.f2, arrga = "Observed")
p.alpha.Cdt.sh <- alpha.custom(ps.f2, arrga = "PD")
p.alpha.Cdt.is <- alpha.custom(ps.f2, arrga = "Shannon")
p.alpha.Cdt.pd <- alpha.custom(ps.f2, arrga = "InvSimpson")

p.alpha.Cdt.oo2 <- alpha.custom2(ps.f2, arrga = "Observed")
p.alpha.Cdt.sh2 <- alpha.custom2(ps.f2, arrga = "PD")
p.alpha.Cdt.is2 <- alpha.custom2(ps.f2, arrga = "Shannon")
p.alpha.Cdt.pd2 <- alpha.custom2(ps.f2, arrga = "InvSimpson")

p1 <- ggarrange(p.alpha.Cdt.oo, p.alpha.Cdt.sh,p.alpha.Cdt.is, p.alpha.Cdt.pd ,
                p.alpha.Cdt.oo2, p.alpha.Cdt.sh2, p.alpha.Cdt.is2, p.alpha.Cdt.pd2,
                ncol = 4 ,label.x = 0.105, nrow = 2)
p1


ps.f2
``` 



```{r,fig.height=8, fig.width=12, results=FALSE}

ps.rand <- rarefy_even_depth(ps.f2)

ordination.b <- ordinate(ps.var, "PCoA", "bray")
ordination.u <- ordinate(ps.rand, "PCoA", "unifrac")
ordination.w <- ordinate(ps.var, "PCoA", "wunifrac")
  
  #plotting
  p1 <-  plot_ordination(ps.var, ordination.b, type="sample", color="Repeat", title="Bray-Curtis", 
                       axes = c(1,2) ) + theme_bw() + theme(text = element_text(size = 10)) + geom_point(size = 3) 
  

  p2 <-  plot_ordination(ps.rand, ordination.u, type="sample", color="Repeat", title="UniFrac", 
                       axes = c(1,2) ) + theme_bw() + theme(text = element_text(size = 10)) + geom_point(size = 3)
  
  p3 <-  plot_ordination(ps.var, ordination.w, type="sample", color="Repeat", title="Weighted UniFrac", 
                       axes = c(1,2) ) + theme_bw() + theme(text = element_text(size = 10)) + geom_point(size = 3) 
  
   p1 <-  plot_ordination(ps.var, ordination.b, type="Soil", color="Soil", title="PCoA - Bray-Curtis", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    theme(legend.position="bottom")
  
   p2 <-  plot_ordination(ps.rand, ordination.u, type="Soil", color="Soil", title="PCoA - UniFrac", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    theme(legend.position="bottom")
  
   p3 <-  plot_ordination(ps.var, ordination.w, type="Soil", color="Soil", title="PCoA - Weighted UniFrac", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    theme(legend.position="bottom")
   
    p12 <-  plot_ordination(ps.var, ordination.b, type="Genotype", color="Genotype", title="PCoA - Bray-Curtis", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    theme(legend.position="bottom")
  
   p22 <-  plot_ordination(ps.rand, ordination.u, type="Genotype", color="Genotype", title="PCoA - UniFrac", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    theme(legend.position="bottom")
  
   p32 <-  plot_ordination(ps.var, ordination.w, type="Genotype", color="Genotype", title="PCoA - Weighted UniFrac", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    theme(legend.position="bottom")

  
   ggarrange(p1, p2, p3, p12, p22, p32, 
             ncol = 3 , 
             nrow = 2, 
             common.legend = TRUE, 
             legend = "none", 
             font.label = list(size = 12, 
                               face = "bold", 
                               color ="black")
             )
  



```

```{r, message=FALSE, echo=FALSE}
des_w_simper <- function(ps, factor){
  require(DESeq2)
  require(vegan)
  require(tibble)
  require(phyloseq)
  
  diagdds = phyloseq_to_deseq2(ps, as.formula(paste( "~", factor)))                  
  diagdds = estimateSizeFactors(diagdds, type="poscounts")
  diagdds = estimateDispersions(diagdds, fitType = "local") 
  diagdds = DESeq(diagdds)
  samp <-sample_data(ps)
  aggdata <- t(aggregate.data.frame(as.data.frame(as.data.frame(t(diagdds@assays@data$mu))), by=list(samp[[factor]]), median))
  colnames(aggdata) <- aggdata[1,]
  aggdata <- aggdata[-1,]
  res = results(diagdds)
  res.df <- as.data.frame(res)
  nice <- cbind(res.df, as.data.frame(ps@tax_table@.Data[rownames(res.df),]), as.data.frame(aggdata)[rownames(res.df),])
} 

```

### Amasing DESeq


```{r, eval=FALSE}


des.soil <- des_w_simper(ps.f2, "Soil")
des.gen <- des_w_simper(ps.f2, "Genotype")

des.soil
des.soil %>% filter(padj < 0.1) %>% arrange(log2FoldChange)
des.gen %>% filter(baseMean > 0) %>% filter(padj < 0.1) %>% arrange(log2FoldChange)

```


```{r, echo=TRUE, rows.print = 112}

permanova.forloop.noNestedPlant <- function(factor){
  require(vegan)
  require(phyloseq)
  dist <- phyloseq::distance(physeq, "bray")
  metadata <- as(sample_data(physeq@sam_data), "data.frame")
  ad <- adonis2(as.formula(paste( "dist ~ ", factor)), data = metadata)
  ad <- cbind2(ad[1,][3], ad[1,][5])
  return(ad)
}

physeq <- ps.var
permanova.forloop.pos.noPlant <- possibly(permanova.forloop.noNestedPlant, otherwise = NA)
out <- map(colnames(physeq@sam_data[, 3:length(colnames(physeq@sam_data))]), permanova.forloop.pos.noPlant)
out.d.noPlant <- do.call("rbind", out)
out.d.noPlant

```

```{r, message=FALSE,echo=TRUE, fig.height=6, fig.width=6}

library(ggVennDiagram)

ps_f_family <- tax_glom(ps_f, taxrank="Family")

get_list_from_ps <- function(physeq, amaz_fac) {
  my_keys <- levels(sample_data(physeq)[[amaz_fac]])
  mylist <- vector(mode="list", length=length(my_keys))
  names(mylist) <- my_keys
  for (i in levels(sample_data(physeq)[[amaz_fac]])) { 
    x <- prune_samples(sample_data(physeq)[[amaz_fac]] %in% i, physeq)
    x <- prune_taxa(taxa_sums(x) > 0, x)
    mylist[[i]] <- taxa_names(x)
  }
  return(mylist)
}

library(ggVennDiagram)


?ggVennDiagram
ggVennDiagram(get_list_from_ps(ps_f_family, "Soil")) + 
  ggplot2::scale_fill_viridis_c(option = "D", begin=0.2)

ps.f@sam_data

ggVennDiagram(get_list_from_ps(ps.f2, "Genotype")) + 
  ggplot2::scale_fill_viridis_c(option = "D", begin=0.2)

```
```{r, fig.height=6, fig.width=6}

ggVennDiagram(get_list_from_ps(ps.f2, "Soil")) + 
  ggplot2::scale_fill_viridis_c(option = "D", begin=0.2)

```

```{r}

library(vegan)
physeq <- ps.f2
dist <- phyloseq::distance(physeq, "bray")
metadata <- as(sample_data(physeq@sam_data), "data.frame")
ad <- adonis2(dist ~ Soil, data = metadata, permutations = 9999)
ad
```
```{r}

library(vegan)
physeq <- ps.f2
dist <- phyloseq::distance(physeq, "bray")
metadata <- as(sample_data(physeq@sam_data), "data.frame")
ad <- adonis2(dist ~ Genotype, data = metadata, permutations = 9999)
ad
```
