---
title: "Untitled"
author: "GrGladkov"
date: "2023-06-30"
output: html_document
---


```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/storage/june_2023/")
rmarkdown::find_pandoc(dir = "/home/gladkov/.conda/envs/bi/bin")

```

## import 

```{r}

setwd("~/storage/june_2023/")

library(readxl)
library(tidyverse)
library(phyloseq)

list.files()
read_excel("bor_syk2023.xlsx")
path <- "bor_syk2023.xlsx"

mad <- path %>%
        excel_sheets() %>%
        set_names() %>%
       purrr::map_df(read_excel,
           path = path,
           .id = "type")

mad <- mad %>% 
  select(-c("type", "No"))

num_id <- unlist(purrr::map(stringr::str_split(mad$ID, "\\-", 2), function(x) x[[2]])) 
mad_id_mod <- paste0("Ab.", num_id)
mad['ID'] <- mad_id_mod
map <- mad %>% column_to_rownames("ID")

otu_table <- read_tsv("otu_table.txt") %>% 
  column_to_rownames("...1")
cols_otu <-  otu_table %>% 
  colnames()
num <- unlist(purrr::map(stringr::str_split(cols_otu, "\\.", 2), function(x) x[[2]])) 
cols_otu_mod <- paste0("Ab.", num)
colnames(otu_table) <- cols_otu_mod
otus <- otu_table %>% 
  as.matrix() %>% 
  t()

taxa <- read_tsv("taxa.txt") %>% 
  column_to_rownames("...1") %>% 
  as.matrix()

ps <- phyloseq(otu_table(otus, taxa_are_rows=FALSE), 
               sample_data(map),
               tax_table(taxa))

ps

```


### functions

```{r}

plot_rich_reads_samlenames_lm <- function(physeq, group = "Site", label = "Repeat"){
  rish <- estimate_richness(physeq, measures = "Observed")
  reads.sum <- as.data.frame(sample_sums(physeq))
  reads.summary <- cbind(rish, reads.sum)
  colnames(reads.summary) <- c("otus","reads")
  reads.summary["Repeat"] <-unlist(purrr::map(stringr::str_split(rownames(physeq@sam_data), "\\.", 2), function(x) x[[2]]))
  reads.summary["Site"] <- physeq@sam_data[[group]]
  library(ggrepel)
  require(ggforce)
  p1 <- ggplot(data=reads.summary) + 
    geom_point(aes(y=otus, x=log2(reads), color=Site),size=3) + 
    geom_text_repel(aes(y=otus, x=log2(reads), label=paste0(Repeat))) + 
    theme_bw() +
    geom_smooth(aes(y=otus, x=log2(reads), fill=Site, color=Site),method=lm, se=FALSE, ymin = 1) + 
    scale_x_continuous(sec.axis = sec_axis(sec.axis ~ 2**.)) 

  return(p1)
}


beta_custom_norm_NMDS_elli_w <- function(ps, seed = 7888, normtype="vst", Color="What", Group="Repeat"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  library(ggforce)
  
  ps@otu_table[ps@otu_table < 0] <- 0
  ordination.b <- ordinate(ps, "NMDS", "bray")t
  mds <- as.data.frame(ordination.b$points)
  p  <-  plot_ordination(ps,
                         ordination.b,
                         type="sample",
                         color = Color,
                         title="NMDS - Bray-Curtis",
                         # title=NULL,
                         axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    annotate("text",
    x=min(mds$MDS1) + abs(min(mds$MDS1))/4,
    y=max(mds$MDS2),
    label=paste0("Stress -- ", round(ordination.b$stress, 3))) +
    geom_mark_ellipse(aes_string(group = Group, label = Group),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                      con.colour='gray') +
    theme(legend.position = "none") +
    scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8)
  
  return(p)
}

plot_alpha_w_toc <- function(ps, group, metric) {
  
  require(phyloseq)
  require(ggplot2)
  
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  
  er <- estimate_richness(ps_a)
  df_er <- cbind(ps_a@sam_data, er)
  df_er <- df_er %>% select(c(group, metric))
  stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
    rstatix::tukey_hsd()
  y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))

  plot_richness(ps_a, x=group, measures=metric) + 
    geom_boxplot() +
    geom_point(size=1.2, alpha=0.3) +
    ggpubr::stat_pvalue_manual(
      stat.test, 
      label = "p.adj.signif", 
      y.position = y) +
    theme_light() + 
    scale_color_brewer(palette="Dark2") +
    theme(axis.text.x = element_text(angle = 90),
          axis.title.x=element_blank()) +
    labs(y=paste(metric, "index")) 
}

phyloseq_to_ampvis2 <- function(physeq) {
  #check object for class
  if(!any(class(physeq) %in% "phyloseq"))
    stop("physeq object must be of class \"phyloseq\"", call. = FALSE)
  
  #ampvis2 requires taxonomy and abundance table, phyloseq checks for the latter
  if(is.null(physeq@tax_table))
    stop("No taxonomy found in the phyloseq object and is required for ampvis2", call. = FALSE)
  
  #OTUs must be in rows, not columns
  if(phyloseq::taxa_are_rows(physeq))
    abund <- as.data.frame(phyloseq::otu_table(physeq)@.Data)
  else
    abund <- as.data.frame(t(phyloseq::otu_table(physeq)@.Data))
  
  #tax_table is assumed to have OTUs in rows too
  tax <- phyloseq::tax_table(physeq)@.Data
  
  #merge by rownames (OTUs)
  otutable <- merge(
    abund,
    tax,
    by = 0,
    all.x = TRUE,
    all.y = FALSE,
    sort = FALSE
  )
  colnames(otutable)[1] <- "OTU"
  
  #extract sample_data (metadata)
  if(!is.null(physeq@sam_data)) {
    metadata <- data.frame(
      phyloseq::sample_data(physeq),
      row.names = phyloseq::sample_names(physeq), 
      stringsAsFactors = FALSE, 
      check.names = FALSE
    )
    
    #check if any columns match exactly with rownames
    #if none matched assume row names are sample identifiers
    samplesCol <- unlist(lapply(metadata, function(x) {
      identical(x, rownames(metadata))}))
    
    if(any(samplesCol)) {
      #error if a column matched and it's not the first
      if(!samplesCol[[1]])
        stop("Sample ID's must be in the first column in the sample metadata, please reorder", call. = FALSE)
    } else {
      #assume rownames are sample identifiers, merge at the end with name "SampleID"
      if(any(colnames(metadata) %in% "SampleID"))
        stop("A column in the sample metadata is already named \"SampleID\" but does not seem to contain sample ID's", call. = FALSE)
      metadata$SampleID <- rownames(metadata)
      
      #reorder columns so SampleID is the first
      metadata <- metadata[, c(which(colnames(metadata) %in% "SampleID"), 1:(ncol(metadata)-1L)), drop = FALSE]
    }
  } else
    metadata <- NULL
  
  #extract phylogenetic tree, assumed to be of class "phylo"
  if(!is.null(physeq@phy_tree)) {
    tree <- phyloseq::phy_tree(physeq)
  } else
    tree <- NULL
  
  #extract OTU DNA sequences, assumed to be of class "XStringSet"
  if(!is.null(physeq@refseq)) {
    #convert XStringSet to DNAbin using a temporary file (easiest)
    fastaTempFile <- tempfile(pattern = "ampvis2_", fileext = ".fa")
    Biostrings::writeXStringSet(physeq@refseq, filepath = fastaTempFile)
  } else
    fastaTempFile <- NULL
  
  #load as normally with amp_load
  ampvis2::amp_load(
    otutable = otutable,
    metadata = metadata,
    tree = tree,
    fasta = fastaTempFile
  )
}

detachAllPackages <- function() {

  basic.packages <- c("package:stats","package:graphics","package:grDevices","package:utils","package:datasets","package:methods","package:base")

  package.list <- search()[ifelse(unlist(gregexpr("package:",search()))==1,TRUE,FALSE)]

  package.list <- setdiff(package.list,basic.packages)

  if (length(package.list)>0)  for (package in package.list) detach(package, character.only=TRUE, force = TRUE)

}

ps_vst <- function(ps, factor){
  diagdds = phyloseq::phyloseq_to_deseq2(ps, as.formula(paste( "~", factor)))              
  diagdds = DESeq2::estimateSizeFactors(diagdds, type="poscounts")
  diagdds = DESeq2::estimateDispersions(diagdds, fitType = "local") 
  pst <- DESeq2::varianceStabilizingTransformation(diagdds)
  pst.dimmed <- t(as.matrix(SummarizedExperiment::assay(pst))) 
  # pst.dimmed[pst.dimmed < 0.0] <- 0.0
  ps.varstab <- ps
  otu_table(ps.varstab) <- otu_table(pst.dimmed, taxa_are_rows = FALSE) 
  return(ps.varstab)
} 

filter_chrmitnas <- function(physeq){
  ps_object <- physeq 
  ps_object <- subset_taxa(ps_object, Phylum != "NA")
  
  ps_object@tax_table[is.na(ps_object@tax_table)] <- ps@tax_table@.Data
  ps_object <- subset_taxa(ps_object,
                           !(Family  == "Mitochondria" |
                               Class   == "Chloroplast" |
                               Order   == "Chloroplast"))
  ps_object@tax_table <- dplyr::na_if(ps_object@tax_table, TRUE)
  ps.f <- ps_object
  
  out_old <- capture.output(physeq)[4] %>% 
    str_extract(regex("([1-9]).*(?= by)"))
  out_new <- capture.output(ps.f)[4] %>% 
    str_extract(regex("([1-9]).*(?= by)"))
  message(paste0("old = ", out_old))
  message(paste0("filtered = ", out_new))
  
  return(ps.f)
  
}

filter_chrmitnas <- function(physeq){
  ps_object <- physeq 
  ps_object <- subset_taxa(ps_object, Phylum != "NA")
  
  ps_object@tax_table <- ps_object@tax_table %>% 
    as.data.frame() %>% 
    mutate_all(~ replace_na(., "unknown")) %>% 
    as.matrix() %>% 
    tax_table()
  
  ps_object <- subset_taxa(ps_object,
                           !(Family  == "Mitochondria" | 
                               Class   == "Chloroplast" | 
                               Order   == "Chloroplast"))

  ps_object@tax_table <- ps_object@tax_table %>% 
    as.data.frame() %>% 
    mutate_all(~ na_if(., "unknown")) %>% 
    as.matrix() %>% 
    tax_table()
  
  ps.f <- ps_object
  
  out_old <- capture.output(physeq)[4] %>%
    str_extract(regex("([1-9]).*(?= by)"))
  out_new <- capture.output(ps.f)[4] %>%
    str_extract(regex("([1-9]).*(?= by)"))
  message(paste0("old = ", out_old))
  message(paste0("filtered = ", out_new))

  return(ps.f)
  
}

get_list_from_ps <- function(physeq, amaz_fac) {
  my_keys <- levels(sample_data(physeq)[[amaz_fac]])
  mylist <- vector(mode="list", length=length(my_keys))
  names(mylist) <- my_keys
  for (i in levels(sample_data(physeq)[[amaz_fac]])) { 
    x <- prune_samples(sample_data(physeq)[[amaz_fac]] %in% i, physeq)
    x <- prune_taxa(taxa_sums(x) > 0, x)
    mylist[[i]] <- taxa_names(x)
  }
  return(mylist)
}

lsf.str()

```


```{r, fig.height=6, fig.width=6}

ps.f <- filter_chrmitnas(ps)


ps.f@tax_table <- ps.f@tax_table@.Data %>% 
  data.frame() %>% 
  mutate(Phylum = str_replace_all(Phylum, c("Firmicutes" = "Bacillota",
                                            "Chloroflexi" = "Chloroflexota",
                                            "Proteobacteria" = "Pseudomonadota",
                                            "Desulfobacterota" = "Thermodesulfobacteriota",
                                            "Crenarchaeota" = "Thermoproteota",
                                            "Cyanobacteria" = "Cyanobacteriota"))) %>% 
  as.matrix() %>% 
  tax_table()

ps.syk <- prune_samples(sample_data(ps.f)$Region == "syktyvkar", ps.f)
ps.syk  <- prune_taxa(taxa_sums(ps.syk) > 0, ps.syk)

ps.bor <- prune_samples(sample_data(ps.f)$Region == "borovichi", ps.f)
ps.bor  <- prune_taxa(taxa_sums(ps.bor) > 0, ps.bor)

ps.syk@sam_data <- ps.syk@sam_data %>% 
  data.frame() %>% 
  rename('Vbasal' = 'BAS') %>% 
  sample_data()


```



```{r}

map_mod <- ps.syk@sam_data %>% 
  data.frame() %>% 
  mutate(Group = as.factor(c(
    rep("Oa pyr", 4),
   rep("Tirr", 4),
   rep("Qpyr", 4),
   rep("Oe", 4)
 ))
 )

map_mod %>% 
  select(c("Description", "Group"))

ps.syk@sam_data <- sample_data(map_mod)
ps.syk@sam_data


```


```{r, fig.height=6, fig.width=6}

ps.syk.clr <- ps.syk
syk.f2.otu <- as.data.frame(t(ps.syk.clr@otu_table)) %>% 
  compositions::clr() %>% 
  t()

ps.syk.clr@otu_table <-  otu_table(syk.f2.otu, taxa_are_rows = FALSE)

beta_custom_norm_PCA_elli_w <- function(ps, seed = 7888, normtype="vst", Color="What", Group="Repeat"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  library(ggforce)
  
  ps@otu_table[ps@otu_table < 0] <- 0
  ordination.b <- ordinate(ps, "PCoA", "euclidean")
  mds <- as.data.frame(ordination.b$points)
  p  <-  plot_ordination(ps,
                         ordination.b,
                         type="sample",
                         color = Color,
                         title="PCA - clr-transformed -euclidian",
                         # title=NULL,
                         axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes_string(group = Group, label = Group),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                      con.colour='gray') +
    theme(legend.position = "none") +
    scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8)
  
  return(p)
}

beta_clr <- beta_custom_norm_PCA_elli_w(ps.syk.clr, Group = "Group", Color = "Group") +
  scale_color_manual(values=viridisLite::magma(n = 7, begin = 0, end = 0.8)[-c(7, 3)])


```


```{r, fig.height=6, fig.width=6}
beta_custom_norm_NMDS_elli_w <- function(ps, seed = 7888, normtype="vst", Color="What", Group="Repeat"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  library(ggforce)
  
  ps@otu_table[ps@otu_table < 0] <- 0
  ordination.b <- ordinate(ps,
                           "NMDS",
                           "bray",
                           k = 2,
                           maxit = 10000,
                           trymax = 5000,
                           parallel = 10)
  
  mds <- as.data.frame(ordination.b$points)
  p  <-  plot_ordination(ps,
                         ordination.b,
                         type="sample",
                         color = Color,
                         title="NMDS - Bray-Curtis",
                         # title=NULL,
                         axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    annotate("text",
    x=min(mds$MDS1) + abs(min(mds$MDS1))/4,
    y=max(mds$MDS2),
    label=paste0("Stress -- ", round(ordination.b$stress, 3))) +
    geom_mark_ellipse(aes_string(group = Group, label = Group),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                      con.colour='gray') +
    theme(legend.position = "none") +
    scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8)
  
  return(p)
}

beta_nmds <- beta_custom_norm_NMDS_elli_w(ps.syk, Group = "Group", Color = "Group")

beta_custom_norm_NMDS_elli_w(ps.syk, Group = "Group", Color = "Group")

```

```{r, fig.height=4, fig.width=6}

ggarrange(beta_nmds, beta_clr, ncol=2)
plot_rich_reads_samlenames_lm(ps.syk, group = "Group", label = "Group")

```

```{r, fig.height=6, fig.width=6}

ps.syk.r <- rarefy_even_depth(ps.syk)

beta_custom_norm_PCoA_elli_w <- function(ps, seed = 7888, normtype="vst", Color="What", Group="Repeat"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  library(ggforce)
  
  ps@otu_table[ps@otu_table < 0] <- 0
  ordination.b <- ordinate(ps, "PCoA", "bray")
  # p <- phyloseq::plot_scree(ordination.b) + theme(axis.text.x = element_text(size = 6))
  mds <- as.data.frame(ordination.b$points)
  p  <-  plot_ordination(ps,
                         ordination.b,
                         type="sample",
                         color = Color,
                         title="PCoA - bray - rarefied",
                         # title=NULL,
                         axes = c(1,2) ) +
    theme_bw() +
    theme(text = element_text(size = 10)) +
    geom_point(size = 3) +
    geom_mark_ellipse(aes_string(group = Group, label = Group),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                      con.colour='gray') +
    theme(legend.position = "none") +
    scale_colour_viridis_d(option = "magma",
                       aesthetics = "color",
                       begin = 0,
                       end = 0.8)
  
  return(p)
}

beta_rar <- beta_custom_norm_PCoA_elli_w(ps.syk.r, Color="Group", Group="Group")

beta_rar <- beta_custom_norm_PCoA_elli_w(ps.syk.clr, Group = "Group", Color = "Group") +
  scale_color_manual(values=viridisLite::magma(n = 7, begin = 0, end = 0.8)[-c(7, 3)])

ggsave(beta_rar, file="~/storage/june_2023/beta_rar.png", width=6, height=4, dpi = 600)
beta_rar


  ordination.b <- ordinate(ps.syk.r, "PCoA", "bray")
  phyloseq::plot_scree(ordination.b) + theme(axis.text.x = element_text(size = 6))

```



```{r, fig.height=6, fig.width=7}

library(ampvis2)

amp_f2 <- phyloseq_to_ampvis2(ps.syk)
amp_f2
ps.syk

amp_phyla <- amp_heatmap(amp_f2,
            tax_show = 12,
            group_by = "Group",
            tax_aggregate = "Phylum",
            tax_add = "Kingdom",
            normalise=TRUE,
            plot_values_size = 5.5,
            showRemainingTaxa = TRUE,
            color_vector = colorspace::lighten(viridis::magma(n = 3, begin = 0.35, direction = -1), 0.2)) +
            theme(axis.text.x = element_text(size=14, angle=45, hjust=1.2, vjust=1.2),
              axis.text.y = element_text(size=14)) 

ggsave(amp_phyla, file="~/storage/june_2023/heat_ph.png", width=7, height=6, dpi = 600)


```


```{r, fig.height=10, fig.width=12}

amp_genus <-amp_heatmap(amp_f2,
            tax_show = 40,
            group_by = "Group",
            tax_aggregate = "Genus",
            # tax_add = "Phylum", 
            tax_class = "Pseudomonadota",
            normalise=TRUE,
            plot_values_size = 5.5,
            showRemainingTaxa = TRUE,
            color_vector = colorspace::lighten(viridis::magma(n = 3, begin = 0.35, direction = -1), 0.2)) +
            theme(axis.text.x = element_text(size=14, angle=45, hjust=1.2, vjust=1.2),
              axis.text.y = element_text(size=14)) 

ggsave(amp_genus, file="~/storage/june_2023/heat_genus.png", width=12, height=10, dpi = 600)

```


```{r}

rows_phyl <- amp_heatmap(amp_f2,
            tax_show = 12,
            group_by = "Group",
            tax_aggregate = "Phylum",
            tax_add = "Kingdom",
            normalise=TRUE,
            plot_values_size = 5.5,
            showRemainingTaxa = TRUE,textmap = TRUE) %>% 
  rownames() 

rows_phyl <- rows_phyl[-length(rows_phyl)]
rows_phyl <- unlist(purrr::map(stringr::str_split(rows_phyl, "\\; ", 2), function(x) x[[2]]))

```

нету


```{r, fig.width= 7, fig.height=5}

scale_colour_viridis_d(option = "magma", 
                   aesthetics = "color", 
                   begin = 0, 
                   end = 0.8)

viridis::magma(n = 3, begin = 0.2)

ps.re@sam_data %>% 
  data.frame()

ps.f2@sam_data %>% 
  data.frame() %>%
  # rownames_to_column("shit") %>% 
  # column_to_rownames("Name") %>% 
  select(which(sapply(., is.numeric)), Group) %>% 
  group_by(Group) %>% 
  summarise(across(everything(), mean),
            .groups = 'drop') %>% 
  column_to_rownames("Group") %>% 
  scale() %>% 
  heatmaply::ggheatmap(color = viridis::magma(n = 3, begin = 0.1, direction = -1)) 


```

```{r}

unregister <- function() {
  env <- foreach:::.foreachGlobals
  rm(list=ls(name=env), pos=env)
}

unregister()

```


### cca

```{r, fig.height=5, fig.width=6}

library(ggrepel)

# physeq <- ps.clr
physeq <- ps.syk

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}

otus.ps.vegan <- veganifyOTU(physeq)
rownames(otus.ps.vegan) <- physeq@sam_data$Group
metadata <- as(sample_data(physeq), "data.frame") 

otus.ps.vegan %>% 
  data.frame()
metadata

cca <- vegan::cca(otus.ps.vegan ~  Vbasal + C + pH , data=metadata)

biplot <- as.data.frame(cca$CCA$biplot)
wa <- as.data.frame(cca$CCA$wa)

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot)))) %>% 
  mutate(
  Type = rep("biplot", length(rownames(biplot))),
)

wa <- rownames_to_column(wa, "Label") %>% 
  add_column(Score = rep("sites", length(rownames(wa)))) %>% mutate(
  Type = metadata$Group,
)

fdat_amazing <- rbind(biplot, wa) 

fdat_amazing <- fdat_amazing %>%
  mutate(label_size = ifelse(Score == "biplot", 4.5, 3)) %>% 
  mutate_if(is.character, as.factor)

list.pe <- summary.cca$cont$importance %>% 
  as.data.frame() %>% 
  select(c("CCA1", "CCA2")) %>% 
  dplyr::slice(2L) %>% 
  t() %>% 
  as.vector()



# kate.ggcca.sites <- function(vare.cca){
# 
#   require(tidyverse)
#   biplot <- as.data.frame(vare.cca$CCA$biplot)
#   wa <- as.data.frame(vare.cca$CCA$wa)
# 
#   biplot <- rownames_to_column(biplot, "Label") %>% 
#     add_column(Score = rep("biplot", length(rownames(biplot))))
#   wa <- rownames_to_column(wa, "Label") %>% 
#     add_column(Score = rep("sites", length(rownames(wa))))
#   fdat_amazing <- rbind(biplot, wa) 
#   fdat_amazing <- fdat_amazing %>%
#     mutate(label_size = ifelse(Score == "biplot", 4.5, 3))
#  
#   p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
#     # geom_abline(intercept=seq(-100, 100, 25), slope=1, colour="grey", alpha=0.5) +
#     # geom_abline(intercept=seq(-100, 100, 25), slope=-1, colour="grey", alpha=0.5) +
#     geom_vline(xintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
#     geom_hline(yintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
#     geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), 
#                mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
#     geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), 
#                  aes(x = 0, xend = CCA1, y = 0, yend = CCA2),
#                   size = 0.6, 
#                  alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
#   geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=fdat_amazing$label_size) + 
#   xlab(paste0(round(cca$CCA$eig[1], 3)*100, "% CCA1")) +
#   ylab(paste0(round(cca$CCA$eig[2], 3)*100, "% CCA2")) +
#   grids(linetype = "dashed") +
#   theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
#   return(p)
# }
# 
# kate.ggcca.sites(cca)

cca1 <- ggplot(fdat_amazing) + 
  geom_vline(xintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  geom_hline(yintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  geom_point(data = fdat_amazing %>% 
               dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score)))   + 
  ggforce::geom_mark_ellipse(data = fdat_amazing %>% 
                                      dplyr::filter(Score == "sites"),
                             aes(x=CCA1, y=CCA2, group = Type, label = Type, col = 'grey'),
                  label.fontsize = 13,
                  label.buffer = unit(0, "mm"),
                  label.hjust = 0,
                  label.minwidth = unit(0, "mm"),
                  con.cap = unit(0, "mm"),
                  label.margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "pt"),
                  con.colour='salmon',
                  label.colour='salmon'
                  ) + 
  geom_segment(data = fdat_amazing %>% 
                 dplyr::filter(Score == "biplot"), 
               aes(x = 0, xend = CCA1, y = 0, yend = CCA2),
               size = 0.6, 
               alpha=0.8, 
               color = "red",
               arrow = arrow(angle = 3)) +
  ggrepel::geom_text_repel(data = fdat_amazing %>% 
                 dplyr::filter(Score == "biplot"), aes(x=CCA1, y=CCA2, label= Label), size=6, nudge_x=0.07) +
  xlab(paste0(round(list.pe[1], 3)*100, "% CCA1")) +
  ylab(paste0(round(list.pe[2], 3)*100, "% CCA2")) +
  grids(linetype = "dashed") +
  theme(legend.position = "none", 
        panel.background = element_rect(fill = "white", colour = "grey50"))

ggsave(cca1, file="~/storage/june_2023/cca1.png", width=6, height=6, dpi = 600)

```


## alpha

```{r, fig.height=4, fig.width=8}


plot_alpha_w_toc_mod <- function(ps, group, metric) {
  
  require(phyloseq)
  require(ggplot2)
  
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  
  er <- estimate_richness(ps_a)
  df_er <- cbind(ps_a@sam_data, er)
  df_er <- df_er %>% select(c(group, metric))
  stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
    rstatix::tukey_hsd() %>%
    filter(p.adj.signif != "ns")
  y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))

  plot_richness(ps_a, x=group, measures=metric, color="Group") + 
    geom_boxplot() +
    geom_point(size=1.2, alpha=0.3) +
    ggpubr::stat_pvalue_manual(
      stat.test, 
      label = "p.adj.signif", 
      y.position = y,
      tip.length = 0.005, 
      bracket.nudge.y = 1.5,
      vjust = 0.6, 
      size = 3) +
    theme_light() + 
        scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8) +
    theme(axis.text.x = element_text(angle = 45, hjust=1),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),
          legend.position = "None") +
    labs(y=paste(metric, "index")) 
}

p_a1 <-  plot_alpha_w_toc_mod(ps.syk, group="Group", metric="Observed") 
p_a2 <- plot_alpha_w_toc_mod(ps.syk, group="Group", metric="Shannon") 
p_a3 <- plot_alpha_w_toc_mod(ps.syk, group="Group", metric="InvSimpson") 

p_alpha <- ggpubr::ggarrange(p_a1, p_a2, p_a3, nrow = 1)
p_alpha
ggsave(p_alpha, file="~/storage/june_2023/alpha.png", width=8, height=4, dpi = 600)

# 
# ggpubr::annotate_figure(p_alpha, 
#                         top = text_grob("NMDS - Bray-Curtis", 
#                                         color = "black", 
#                                         face = "bold", 
#                                         size = 12,
#                                         hjust = 2.45),
#                fig.lab.pos = "top.left")

```




```{r}

ps.syk.f2 <- phyloseq::filter_taxa(ps.syk, function(x) sum(x > 10) > (0.1*length(x)), TRUE)


out.f <-  ANCOMBC::ancombc2(data=ps.syk.f2, 
  fix_formula = "Group",
  prv_cut = 0,
  p_adj_method = "BH",
  tax_level = NULL,
  rand_formula= NULL,
  pseudo = 0, 
  pseudo_sens = FALSE,
  s0_perc = 0.05,
  group = "Group",
  struc_zero = FALSE,
  neg_lb = TRUE,
  alpha = 0.05,
  n_cl = 20, 
  verbose = TRUE,
  global = FALSE, 
  pairwise = FALSE, 
  dunnet = FALSE, 
  trend = FALSE
  )
  
out.f

unregister <- function() {
  env <- foreach:::.foreachGlobals
  rm(list=ls(name=env), pos=env)
}

saveRDS(out.f, "anc")
list.files()
# anc <- readRDS("")
out.f

unregister()

```


```{r}

ps.anc.f <- ps.syk.f2

samp_frac <-  out.f$samp_frac
samp_frac[is.na(samp_frac)] <-  0 
# Add pesudo-count (1) to avoid taking the log of 0
log_obs_abn = log(out.f$feature_table + 1)
# Adjust the log observed abundances
log_corr_abn = t(t(log_obs_abn) - samp_frac)
otu_table(ps.anc.f) <- otu_table(t(log_corr_abn), taxa_are_rows = FALSE)

beta_custom_norm_PCA_elli_w(ps.anc.f, Color = "Group", Group = "Group")

```


```{r}

library(WGCNA)

ps.vst.mod <- ps.anc.f

ps.vst.mod@sam_data
ps.vst.mod@sam_data <- ps.vst.mod@sam_data %>% 
  data.frame() %>% 
  mutate(Repeat = paste0(Group, "_", sapply(str_split(row.names(.), '\\.', 2), function(x) x[[2]]))
                         ) %>% 
  sample_data()

data3 <- ps.vst.mod@otu_table@.Data %>% 
  as.data.frame()


rownames(data3) <- as.character(ps.vst.mod@sam_data$Repeat)
powers <-  c(seq(from = 1, to=10, by=0.5), seq(from = 11, to=20, by=1))
sft3 <-  pickSoftThreshold(data3, powerVector = powers, verbose = 5, networkType = "signed hybrid")

sft3

plot(sft3$fitIndices[,1], -sign(sft3$fitIndices[,3])*sft3$fitIndices[,2], xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n", main = paste("Scale independence"))
text(sft3$fitIndices[,1], -sign(sft3$fitIndices[,3])*sft3$fitIndices[,2], labels=powers,cex=0.9,col="red")
abline(h=0.9,col="salmon")


```


```{r}

detachAllPackages()
library(WGCNA)


net3 <- WGCNA::blockwiseModules(data3,
                          power=10,
                          TOMType="signed",
                          networkType="signed hybrid",
                          nThreads=15)

library(phyloseq)
library(tidyverse)
library(ggpubr)
library(ampvis2)
library(heatmaply)
library(WGCNA)
library(phyloseq)
library(ggtree)
library(tidyverse)
library(KneeArrower)

net3$colors %>% 
  unique()

# mergedColors2 <-  WGCNA::labels2colors(net3$colors, colorSeq = c(net3$colors %>% 
  # unique()))

mergedColors2 <- net3$colors

plotDendroAndColors(
  net3$dendrograms[[1]],
  mergedColors2[net3$blockGenes[[1]]],
  "Module colors",
  dendroLabels = FALSE,
  hang = 0.03,
  addGuide = TRUE,
  guideHang = 0.05)

```
```{r}

modules_of_interest = mergedColors2 %>% 
  unique()

module_df <- data.frame(
  asv = names(net3$colors),
  colors = mergedColors2
)

# module_df[module_df == "yellow"] <- "salmon"

submod <-  module_df %>%
  subset(colors %in% modules_of_interest)

row.names(module_df) = module_df$asv

subexpr = as.data.frame(t(data3))[submod$asv,]

submod_df <- data.frame(subexpr) %>%
  mutate(
    asv = row.names(.)
  ) %>%
  pivot_longer(-asv) %>%
  mutate(
    module = module_df[asv,]$colors
  )
submod_df
submod_df <- submod_df %>% 
  mutate(name = gsub("\\_.*","",submod_df$name)) %>% 
  group_by(name, asv) %>% 
  summarise(value = mean(value), asv = asv, module = module) %>% 
  relocate(c(asv, name, value, module)) %>% 
  ungroup() %>% 
  mutate(module = as.factor(module))

p <- submod_df %>% 
  ggplot(., aes(x=name, y=value, group=asv)) +
  geom_line(aes(color = module),
            alpha = 0.2) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90),
    legend.position = "none") +
  facet_grid(rows = vars(module)) +
  labs(x = "day",
       y = "normalized abundance")

p
p + scale_color_manual(values = levels(submod_df$module))

+
#   scale_x_discrete(labels=(c("D01"="3",
#                                 "D03"="14",
#                                 "D05"="28",
#                                 "D07"="49" ,
#                                 "D08"="63",
#                                 "D10"="91",
#                                 "D12"="119",
#                                 "D13"="140",
#                                 "D14"="161",
#                                 "D15"="182")
#          ))


```

```{r, fig.height=8, fig.width=10}

clust.pyr <- net3$colors[net3$colors %in% c("blue")]
clust.nonpyr <- net3$colors[! net3$colors %in% c("blue")]

clust.pyr %>% length()
clust.nonpyr %>% length()


ps.rel  <-  phyloseq::transform_sample_counts(ps.syk, function(x) x / sum(x) * 100)
tx <- ps.anc.f %>% 
  tax_table() %>% 
  as.data.frame() %>% 
  mutate(Group = net3$colors) %>% 
  add_column(abnd = taxa_sums(ps.rel)[taxa_names(ps.anc.f)]) %>% 
  mutate(Group = ifelse(Group == "blue", "pyr", "nonpyr"))

treemap <- tx %>% 
  mutate(Phylum = case_when(Phylum %in% "Pseudomonadota" ~ Class,
                            !Phylum %in% "Pseudomonadota" ~ Phylum) %>% as.factor()) %>% 
  select(c("Genus", "Phylum", "Group", "abnd")) %>% 
  group_by(Genus, Phylum, Group ) %>%
  mutate(Genus = str_replace_na(Genus, replacement = ' ') %>% as.factor()) %>% 
  summarise(area = sum(abnd)) %>% 
  mutate_if(is.character, as.factor) %>% 
  drop_na() %>% 
  ggplot(aes(area = area, 
             subgroup = Phylum, 
             label = Genus,
             fill = Group)) +
  treemapify::geom_treemap() +
  treemapify::geom_treemap_subgroup_border(color = "white") +
  treemapify::geom_treemap_subgroup_text(place = "centre", 
                                         grow = T, 
                                         alpha = 0.7, 
                                         colour = "black", 
                                         fontface = "italic", 
                                         min.size = 2) +
  treemapify::geom_treemap_text(colour = "white", 
                                place = "topleft", 
                                grow = T, 
                                reflow = T, 
                                layout = 'squarified',
                                min.size = 4) +
  scale_colour_viridis_d(option = "magma", 
                         aesthetics = "fill", 
                         begin = 0.4, 
                         end = 0.8) +
  theme(legend.position="bottom") 

 ggsave(treemap, file="~/storage/june_2023/treemap.png", width=10, height=8, dpi = 600)


```


```{r}

ancom_phylum$res_pair %>% 
  filter(taxon %in% rows_phyl) %>% 
  select("taxon"|starts_with("p_"))


```


```{r, fig.height=8, fig.width=10}

ancom_phylum <-  ANCOMBC::ancombc2(data = ps.syk, 
  tax_level = "Phylum",
  fix_formula = "Group",
  rand_formula = NULL,
  p_adj_method = "fdr",
  pseudo = 0, 
  pseudo_sens = FALSE,
  prv_cut = 0.1,
  s0_perc = 0.05,
  group = "Group",
  struc_zero = FALSE,
  neg_lb = TRUE,
  alpha = 0.05,
  n_cl = 20, 
  verbose = TRUE,
  global = FALSE, 
  pairwise = TRUE, 
  dunnet = FALSE, 
  trend = FALSE,
  iter_control = list(tol = 1e-2, max_iter = 20, verbose = TRUE),
  em_control = list(tol = 1e-5, max_iter = 100),
  lme_control = lme4::lmerControl(),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
    nrow = 2,
    byrow = TRUE),
    matrix(c(-1, 0, 1, -1),
    nrow = 2,
    byrow = TRUE)),
   node = list(2, 2),
   solver = "ECOS",
  B = 100)
  )

phyl_anc <- ancom_phylum$res_pair %>% 
  filter(taxon %in% rows_phyl) %>% 
  select("taxon"|starts_with("p_")) %>% 
  rename_all(~ stringr::str_replace(., regex("p_Group", ignore_case = TRUE), "")) %>% 
  rename_all(~ stringr::str_replace(., regex("Group", ignore_case = TRUE), "")) %>% 
  rename_if( !stringr::str_detect(names(.), "_") & !stringr::str_detect(names(.), "taxon"),  ~ paste0(., "_Oa pyr")) %>%   reshape2::melt() %>% 
  rstatix::p_format(value, new.col = TRUE, digits = 2, accuracy = 1e-02) %>% 
  mutate(value.format = ifelse(value.format == "1", "", value.format)) %>%
  separate(variable,into=c('Var1', 'Var2'), sep = "_") %>% 
  mutate_if(is.character, as.factor) %>% 
  ggplot(aes(Var1, Var2, fill = value)) + 
  geom_tile() +
  scale_fill_gradientn(colours = c("#A5317EFF", "#FCFDBFFF",  "#FCFDBFFF" ),
                       values = scales::rescale(c(0, 0.05, 1))) +
  geom_text(aes(label = value.format)) +
  facet_wrap(~taxon) +
  theme_bw() +
  theme(legend.position = "none",
          axis.title.x=element_blank(),
          axis.title.y=element_blank())

phyl_anc
ggsave(phyl_anc, file="~/storage/june_2023/phyl_anc.png", width=10, height=8, dpi = 600)

```




```{r}

plot_rich_reads_samlenames_lm(ps.syk, group = "Group", label = "Group")  

```


```{r}

tx %>% 
  filter(Group == "nonpyr") %>% 
  arrange(desc(abnd))

```

```{r, fig.height=5, fig.width=10}

ggpubr::ggarrange(beta_nmds, beta_rar)

```


# Borovichi

```{r}

ps.bor@sam_data

```


```{r}

# ps.bor@sam_data <- ps.bor@sam_data %>% 
#   data.frame() %>% 
#   mutate(Group = case_when(
#     Site %in%  "backgound" ~ "background",
#     Site %in% "haying 30 years" ~ "hayfield_30",
#     Site %in% "pasture 30 year deposit" ~ "pasture_30",
#     Site %in% "plaggen" ~ "plaggen",
#     Site %in% "garden" ~ "garden",
#     Site %in% "forest 30 year deposit" ~ "forest_30")) %>% 
#   dplyr::rename(TOC = C) %>% 
#   # mutate(Region = case_when(type %in% "igan" ~ "polar ural",
#   #                           type %in% "kavkaz" ~ "caucasus",
#   #                           type %in% "baranovka" ~ "severnaya zemlya",
#   #                           type %in% "livingston"  ~ "antarctica")) %>%
#   # mutate(Cryoconite = case_when((sample_type  == "криоконит") ~ "cryoconite",
#   #                               (Region  != "caucasus") ~ "cryoconite",
#   #                               .default = "mud/soil"))  %>% 
#   mutate_if(is.character, as.factor) %>%
#   sample_data()

ps.bor@sam_data <- ps.bor@sam_data %>% 
  data.frame() %>% 
  mutate(Group = case_when(
    Site %in%  "backgound" ~ "Mature soil",
    Site %in% "haying 30 years" ~ "Hayfield",
    Site %in% "pasture 30 year deposit" ~ "Pasture",
    Site %in% "plaggen" ~ "Plaggen",
    Site %in% "garden" ~ "Garden",
    Site %in% "forest 30 year deposit" ~ "Forest")) %>% 
  dplyr::rename(TOC = C) %>% 
  mutate(Group=forcats::fct_relevel(Group, c("Mature soil","Garden", "Forest" ,"Pasture", "Hayfield","Plaggen"))) %>% 
  # mutate(Region = case_when(type %in% "igan" ~ "polar ural",
  #                           type %in% "kavkaz" ~ "caucasus",
  #                           type %in% "baranovka" ~ "severnaya zemlya",
  #                           type %in% "livingston"  ~ "antarctica")) %>%
  # mutate(Cryoconite = case_when((sample_type  == "криоконит") ~ "cryoconite",
  #                               (Region  != "caucasus") ~ "cryoconite",
  #                               .default = "mud/soil"))  %>% 
  mutate_if(is.character, as.factor) %>%
  sample_data()

ps.bor@sam_data$Group

```

```{r, fig.height=6, fig.width=6}

ps.bor.clr <- ps.bor
syk.bor.otu <- as.data.frame(t(ps.bor.clr@otu_table)) %>% 
  compositions::clr() %>% 
  t()

ps.bor.clr@otu_table <-  otu_table(syk.bor.otu, taxa_are_rows = FALSE)

beta_custom_norm_NMDS_elli_w(ps.bor.clr, Group = "Group", Color = "Group")

```

```{r}

plot_rich_reads_samlenames_lm(ps.bor, group = "Group", label = "Group") 

```

### nulls nulls everywhere

```{r, fig.height=8, fig.width=16}

physeq <- ps.bor
l <- get_list_from_ps(physeq, "Group")
core <- Reduce(intersect, l)
length(core)
ps.core <- prune_taxa(core, ps.bor)

amp.core <- phyloseq_to_ampvis2(ps.core)
# ps.core.ancm <- prune_taxa(zeros.all, ps.core)
# amp.core.ancm <- phyloseq_to_ampvis2(ps.core.ancm)

amp_heatmap(amp.core,
            tax_show = 155,
            tax_aggregate = "OTU",
            tax_add = "Genus",
            normalise=FALSE, plot_values_size = 2, facet_by = "Group",
            showRemainingTaxa = TRUE)

```

amcom-bc

```{r}

ancom_core <-  ANCOMBC::ancombc2(data = ps.core, 
  fix_formula = "Group",
  rand_formula = NULL,
  p_adj_method = "fdr",
  pseudo = 0, 
  pseudo_sens = TRUE,
  prv_cut = 0.10,
  s0_perc = 0.05,
  group = "Group",
  struc_zero = TRUE,
  neg_lb = TRUE,
  alpha = 0.05,
  n_cl = 15, 
  tax_level = NULL,
  verbose = TRUE,
  global = FALSE, 
  pairwise = FALSE, 
  dunnet = FALSE, 
  trend = FALSE,
  iter_control = list(tol = 1e-2, max_iter = 20, verbose = TRUE),
  em_control = list(tol = 1e-5, max_iter = 100),
  lme_control = lme4::lmerControl(),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
    nrow = 2,
    byrow = TRUE),
    matrix(c(-1, 0, 1, -1),
    nrow = 2,
    byrow = TRUE)),
   node = list(2, 2),
   solver = "ECOS",
  B = 1)
  )

non_str_otu <- ancom_core$res$taxon

ancom_core$zero_ind 

ancom_core$zero_ind %>%
  column_to_rownames("taxon") %>% 
  mutate_all(.funs = function(x) as.integer(as.logical(x))) %>% 
  as.matrix() %>% 
  heatmap()
  
zeros <- ancom_core$zero_ind %>% 
  column_to_rownames("taxon") %>% 
  rowSums() > 0L

zeros

zeros.all <- zeros[zeros == FALSE]
zeros.all <- names(zeros.all)

zeros.all %>% 
  length()

zeros.all

ps.cr@tax_table %>% 
  as.data.frame() %>%
  rownames_to_column("ID") %>% 
  filter(ID %in% zeros.all)


```
```{r}

ancom_core <-  ANCOMBC::ancombc2(data = ps.bor, 
  fix_formula = "Group",
  rand_formula = NULL,
  p_adj_method = "fdr",
  pseudo = 0, 
  pseudo_sens = TRUE,
  prv_cut = 0.10,
  s0_perc = 0.05,
  group = "Group",
  struc_zero = TRUE,
  neg_lb = TRUE,
  alpha = 0.05,
  n_cl = 15, 
  tax_level = NULL,
  verbose = TRUE,
  global = FALSE, 
  pairwise = FALSE, 
  dunnet = FALSE, 
  trend = FALSE,
  iter_control = list(tol = 1e-2, max_iter = 20, verbose = TRUE),
  em_control = list(tol = 1e-5, max_iter = 100),
  lme_control = lme4::lmerControl(),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
    nrow = 2,
    byrow = TRUE),
    matrix(c(-1, 0, 1, -1),
    nrow = 2,
    byrow = TRUE)),
   node = list(2, 2),
   solver = "ECOS",
  B = 1)
  )

non_str_otu <- ancom_core$res$taxon

ancom_core$zero_ind 

ancom_core$zero_ind %>%
  column_to_rownames("taxon") %>% 
  mutate_all(.funs = function(x) as.integer(as.logical(x))) %>% 
  as.matrix() %>% 
  heatmap()
  
zeros <- ancom_core$zero_ind %>% 
  column_to_rownames("taxon") %>% 
  rowSums() > 0L

zeros

zeros.all <- zeros[zeros == FALSE]
zeros.all <- names(zeros.all)

zeros.all %>% 
  length()

zeros.all

ps.cr@tax_table %>% 
  as.data.frame() %>%
  rownames_to_column("ID") %>% 
  filter(ID %in% zeros.all)


```

```{r}

metadata

```


```{r}

unregister <- function() {
  env <- foreach:::.foreachGlobals
  rm(list=ls(name=env), pos=env)
}

unregister()

```

```{r, fig.height=6, fig.width=7}

library(ggrepel)

# physeq <- ps.clr
physeq <- ps.bor

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}

otus.ps.vegan <- veganifyOTU(physeq)
rownames(otus.ps.vegan) <- physeq@sam_data$Group
metadata <- as(sample_data(physeq), "data.frame") 

otus.ps.vegan %>% 
  data.frame()
metadata

cca <- vegan::cca(otus.ps.vegan ~  pH + TOC + NH4 + NO3 + P + K + BAS + SIR, data=metadata)

biplot <- as.data.frame(cca$CCA$biplot)
wa <- as.data.frame(cca$CCA$wa)

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot)))) %>% 
  mutate(
  Type = rep("biplot", length(rownames(biplot))),
)

wa <- rownames_to_column(wa, "Label") %>% 
  add_column(Score = rep("sites", length(rownames(wa)))) %>% mutate(
  Type = metadata$Group,
)

fdat_amazing <- rbind(biplot, wa) 

fdat_amazing <- fdat_amazing %>%
  mutate(label_size = ifelse(Score == "biplot", 4.5, 3)) %>% 
  mutate_if(is.character, as.factor)

list.pe <- summary.cca$cont$importance %>% 
  as.data.frame() %>% 
  select(c("CCA1", "CCA2")) %>% 
  dplyr::slice(2L) %>% 
  t() %>% 
  as.vector()



# kate.ggcca.sites <- function(vare.cca){
# 
#   require(tidyverse)
#   biplot <- as.data.frame(vare.cca$CCA$biplot)
#   wa <- as.data.frame(vare.cca$CCA$wa)
# 
#   biplot <- rownames_to_column(biplot, "Label") %>% 
#     add_column(Score = rep("biplot", length(rownames(biplot))))
#   wa <- rownames_to_column(wa, "Label") %>% 
#     add_column(Score = rep("sites", length(rownames(wa))))
#   fdat_amazing <- rbind(biplot, wa) 
#   fdat_amazing <- fdat_amazing %>%
#     mutate(label_size = ifelse(Score == "biplot", 4.5, 3))
#  
#   p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
#     # geom_abline(intercept=seq(-100, 100, 25), slope=1, colour="grey", alpha=0.5) +
#     # geom_abline(intercept=seq(-100, 100, 25), slope=-1, colour="grey", alpha=0.5) +
#     geom_vline(xintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
#     geom_hline(yintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
#     geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), 
#                mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
#     geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), 
#                  aes(x = 0, xend = CCA1, y = 0, yend = CCA2),
#                   size = 0.6, 
#                  alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
#   geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=fdat_amazing$label_size) + 
#   xlab(paste0(round(cca$CCA$eig[1], 3)*100, "% CCA1")) +
#   ylab(paste0(round(cca$CCA$eig[2], 3)*100, "% CCA2")) +
#   grids(linetype = "dashed") +
#   theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
#   return(p)
# }
# 
# kate.ggcca.sites(cca)

ggplot(fdat_amazing) + 
  geom_vline(xintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  geom_hline(yintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  geom_point(data = fdat_amazing %>% 
               dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score)))   + 
  ggforce::geom_mark_ellipse(data = fdat_amazing %>% 
                                      dplyr::filter(Score == "sites"),
                             aes(x=CCA1, y=CCA2, group = Type, label = Type, col = 'grey'),
                  label.fontsize = 10,
                  label.buffer = unit(20, "mm"),
                  label.hjust = 0,
                  label.minwidth = unit(5, "mm"),
                  con.cap = unit(0.1, "mm"),
                  label.margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "pt"),
                  con.colour='salmon',
                  label.colour=''
                  ) + 
  geom_segment(data = fdat_amazing %>% 
                 dplyr::filter(Score == "biplot"), 
               aes(x = 0, xend = CCA1, y = 0, yend = CCA2),
               size = 0.6, 
               alpha=0.8, 
               color = "red",
               arrow = arrow(angle = 3)) +
  ggrepel::geom_text_repel(data = fdat_amazing %>% 
                 dplyr::filter(Score == "biplot"), aes(x=CCA1, y=CCA2, label= Label), size=5, nudge_x=0.15) +
  xlab(paste0(round(list.pe[1], 3)*100, "% CCA1")) +
  ylab(paste0(round(list.pe[2], 3)*100, "% CCA2")) +
  grids(linetype = "dashed") +
  theme(legend.position = "none", 
        panel.background = element_rect(fill = "white", colour = "grey50"))


```

```{r}

fdat_amazing

```


```{r}

summary.cca$cont

```

```{r}

list.pe <- summary.cca$cont$importance %>% 
  as.data.frame() %>% 
  select(c("CCA1", "CCA2")) %>% 
  dplyr::slice(2L) %>% 
  t() %>% 
  as.vector()

list.pe
```


```{r}

summary.cca <- summary(cca)
summary.cca$concont$importance[2][2]

```


```{r}

cca$CCA$biplot
library(vegan)
package_version("vegan")

```


```{r}

alias(cca, names=TRUE)

```

```{r}

p_a1 <-  plot_alpha_w_toc_mod(ps.bor, group="Group", metric="Observed") 
p_a2 <- plot_alpha_w_toc_mod(ps.bor, group="Group", metric="Shannon") 
p_a3 <- plot_alpha_w_toc_mod(ps.bor, group="Group", metric="InvSimpson") 

p_alpha <- ggpubr::ggarrange(p_a1, p_a2, p_a3, nrow = 1)
p_alpha

```



```{r, fig.width=9, fig.height=9}

library(ggVennDiagram)

plot_vienn <- function(ps, group){
  physeq <- prune_taxa(taxa_sums(ps) > 0, ps)
  groups <- levels(sample_data(physeq)[[group]] %>% as.factor())
  data <- merge_samples(physeq, group) %>% 
    psmelt() %>% 
    group_by(Sample, OTU) %>% 
    summarise(ASVs_abund = list(paste(OTU, 1:sum(Abundance))), Abund = sum(Abundance), .groups='keep') %>% 
    filter(Abund > 0)
  
  asvs <- data %>% select(Sample, OTU) %>% group_by(Sample) %>% summarise(ASVs = list(OTU)) %>% as.list()
  d1 <- asvs[[2]]
  names(d1) <- asvs[[1]]
  d1
  
  weighted.asvs <- data %>% select(Sample, ASVs_abund) %>% group_by(Sample) %>% summarise(ASVs = list(unlist(ASVs_abund)))
  d2 <- weighted.asvs[[2]]
  names(d2) <- weighted.asvs[[1]]
  d2
  
  
  list(ggVennDiagram(d1) + ggtitle("ASVs") + scale_fill_distiller(palette = "OrRd", trans = "reverse"),
       ggVennDiagram(d2) + ggtitle("Reads") + scale_fill_distiller(palette = "OrRd", trans = "reverse"))
}

plot_vienn(ps.bor, "Group")

```
```{r}

plot_alpha_w_toc_mod <- function(ps, group, metric) {
  
  require(phyloseq)
  require(ggplot2)
  
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  
  er <- estimate_richness(ps_a)
  df_er <- cbind(ps_a@sam_data, er)
  df_er <- df_er %>% select(c(group, metric))
  stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
    rstatix::tukey_hsd() %>%
    filter(p.adj.signif != "ns")
  y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))

```


```{r, fig.height=5.5, fig.width=9}

ps.bor
ps.f2.al <- ps.bor
# ps.f2.al@sam_data <- ps.f2.al@sam_data %>% 
#   data.frame() %>% 
#   mutate(Group=forcats::fct_relevel(Group, c("Pimpirev","IGAN", "Mush" ,"Garabashi", "Shkelda","Mud", "Soil"))) %>% 
#   sample_data()


plot_alpha_w_toc_mod <- function(ps, group, metric, test="HSD") {
  
  require(phyloseq)
  require(ggplot2)
  
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  
  er <- estimate_richness(ps_a)
  df_er <- cbind(ps_a@sam_data, er)
  df_er <- df_er %>% select(c(group, metric))

  if (test != "HSD"){
    stat.test <- df_er %>%
      rstatix::wilcox_test(as.formula(paste0(metric, "~", group)), p.adjust.method = "BH") %>%
      filter(p.adj.signif != "ns")
  }
    else
    {
    stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
      rstatix::tukey_hsd() %>%
      filter(p.adj.signif != "ns")
  }
  
  y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))

  plot_richness(ps_a, x=group, measures=metric, color="Group") + 
    geom_boxplot() +
    geom_point(size=1.2, alpha=0.3) +
    ggpubr::stat_pvalue_manual(
      stat.test, 
      label = "p.adj.signif", 
      y.position = y,
      tip.length = 0.005, 
      bracket.nudge.y = 1.5,
      vjust = 0.6, 
      size = 3) +
    theme_light() + 
        scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8) +
    theme(axis.text.x = element_text(angle = 45, hjust=1),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),
          legend.position = "None") +
    labs(y=paste(metric, "index")) 
}

p_a1 <-  plot_alpha_w_toc_mod(ps.f2.al, group="Group", metric="Observed") 
p_a2 <- plot_alpha_w_toc_mod(ps.f2.al, group="Group", metric="Shannon") 
p_a3 <- plot_alpha_w_toc_mod(ps.f2.al, group="Group", metric="InvSimpson") 

p_alpha <- ggpubr::ggarrange(p_a1, p_a2, p_a3, nrow = 1)
p_alpha

# 
# ggpubr::annotate_figure(p_alpha, 
#                         top = text_grob("NMDS - Bray-Curtis", 
#                                         color = "black", 
#                                         face = "bold", 
#                                         size = 12,
#                                         hjust = 2.45),
#                fig.lab.pos = "top.left")

```



```{r}
metric <- "Shannon"
group <- "Group"
ps_a <- ps.bor

ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)

er <- estimate_richness(ps_a)
df_er <- cbind(ps_a@sam_data, er)
df_er <- df_er %>% select(c(group, metric))
stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
  rstatix::tukey_hsd() %>%
  filter(p.adj.signif != "ns")
y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))
y

stat.test
y


df_er %>%
  rstatix::wilcox_test(as.formula(paste0(metric, "~", group)), p.adjust.method = "BH") %>%
  filter(p.adj.signif != "ns")

```


```{r, fig.height=5.5, fig.width=9}

plot_alpha_w_toc_mod_evil <- function(ps, group, metric, test="HSD") {
  
  require(phyloseq)
  require(ggplot2)
  
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  
  er <- estimate_richness(ps_a)
  df_er <- cbind(ps_a@sam_data, er)
  df_er <- df_er %>% select(c(group, metric))

  if (test != "HSD"){
    stat.test <- df_er %>%
      rstatix::wilcox_test(as.formula(paste0(metric, "~", group)), p.adjust.method = "none") %>%
      filter(p.adj.signif != "ns")
  }
    else
    {
    stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
      rstatix::tukey_hsd() %>%
      filter(p.adj.signif != "ns")
  }
  
  y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))

  plot_richness(ps_a, x=group, measures=metric, color="Group") + 
    geom_boxplot() +
    geom_point(size=1.2, alpha=0.3) +
    ggpubr::stat_pvalue_manual(
      stat.test, 
      label = "p.adj.signif", 
      y.position = y,
      tip.length = 0.005, 
      bracket.nudge.y = 1.5,
      vjust = 0.6, 
      size = 3) +
    theme_light() + 
        scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8) +
    theme(axis.text.x = element_text(angle = 45, hjust=1),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),
          legend.position = "None") +
    labs(y=paste(metric, "index")) 
}

ps_ar <- phyloseq::rarefy_even_depth(ps.bor, rngseed =  1488)

p_a1 <-  plot_alpha_w_toc_mod_evil(ps_ar, group="Group", metric="Observed", test="hui") 
p_a2 <- plot_alpha_w_toc_mod_evil(ps_ar, group="Group", metric="Shannon", test="hui") 
p_a3 <- plot_alpha_w_toc_mod_evil(ps_ar, group="Group", metric="InvSimpson", test="hui") 

p_alpha <- ggpubr::ggarrange(p_a1, p_a2, p_a3, nrow = 1)
p_alpha

```
```{r, fig.height=5.5, fig.width=9}

ps_ar <- phyloseq::rarefy_even_depth(ps.bor, rngseed =  1488)

p_a1 <-  plot_alpha_w_toc_mod(ps_ar, group="Group", metric="Observed", test="hui") 
p_a2 <- plot_alpha_w_toc_mod(ps_ar, group="Group", metric="Shannon", test="hui") 
p_a3 <- plot_alpha_w_toc_mod(ps_ar, group="Group", metric="InvSimpson", test="hui") 

p_alpha <- ggpubr::ggarrange(p_a1, p_a2, p_a3, nrow = 1)
p_alpha

```

```{r, fig.height=5.5, fig.width=9}

ps_ar <- phyloseq::rarefy_even_depth(ps.bor, rngseed =  1488)

p_a1 <-  plot_alpha_w_toc_mod(ps_ar, group="Group", metric="Observed", test="HSD") 
p_a2 <- plot_alpha_w_toc_mod(ps_ar, group="Group", metric="Shannon", test="HSD") 
p_a3 <- plot_alpha_w_toc_mod(ps_ar, group="Group", metric="InvSimpson", test="HSD") 

p_alpha <- ggpubr::ggarrange(p_a1, p_a2, p_a3, nrow = 1)
p_alpha

```

```{r, fig.height=4, fig.width=6}

library(breakaway)
library(magrittr)

ps.m <- merge_samples(ps.bor, "Group")

ft <- ps.bor@otu_table@.Data %>% 
  data.frame() %>% make_frequency_count_table

ft %>% 
  tail()


er <- ft %>% breakaway()

plot(er)

ps.bor %>% sample_data %>% head

res <- breakaway(ps.m)
summar <- summary(res)

summar
res

ggplot(summar, aes(x=sample_names, y=estimate)) + 
  geom_pointrange(aes(ymin=lower, ymax=upper), size =0.8, color="salmon", linetype=2, shape=1) +
      theme_light()



```


```{r}

plot_alpha_w_toc_mod_amy <- function(ps, group, metric) {
  
  require(phyloseq)
  require(ggplot2)
  
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  
  er <- estimate_richness(ps_a)
  df_er <- cbind(ps_a@sam_data, er)
  df_er <- df_er %>% select(c(group, metric))

  if (test != "HSD"){
    stat.test <- df_er %>%
      rstatix::wilcox_test(as.formula(paste0(metric, "~", group)), p.adjust.method = "none") %>%
      filter(p.adj.signif != "ns")
  }
    else
    {
    stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
      rstatix::tukey_hsd() %>%
      filter(p.adj.signif != "ns")
  }
  
  y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))

  plot_richness(ps_a, x=group, measures=metric, color="Group") + 
    geom_boxplot() +
    geom_point(size=1.2, alpha=0.3) +
    ggpubr::stat_pvalue_manual(
      stat.test, 
      label = "p.adj.signif", 
      y.position = y,
      tip.length = 0.005, 
      bracket.nudge.y = 1.5,
      vjust = 0.6, 
      size = 3) +
    theme_light() + 
        scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8) +
    theme(axis.text.x = element_text(angle = 45, hjust=1),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),
          legend.position = "None") +
    labs(y=paste(metric, "index")) 
}

ps_ar <- phyloseq::rarefy_even_depth(ps.bor, rngseed =  1488)

p_a1 <-  plot_alpha_w_toc_mod_evil(ps_ar, group="Group", metric="Observed", test="hui") 
p_a2 <- plot_alpha_w_toc_mod_evil(ps_ar, group="Group", metric="Shannon", test="hui") 
p_a3 <- plot_alpha_w_toc_mod_evil(ps_ar, group="Group", metric="InvSimpson", test="hui") 

p_alpha <- ggpubr::ggarrange(p_a1, p_a2, p_a3, nrow = 1)
p_alpha

stat.test

```


```{r, fig.height=12, fig.width=12}


amp_f2 <- phyloseq_to_ampvis2(ps.bor)

amp_heatmap(amp_f2,
            tax_show = 40,
            group_by = "Group",
            tax_aggregate = "Genus",
            tax_add = "Phylum",
            normalise=TRUE,
            plot_values_size = 4.5,
            showRemainingTaxa = TRUE,
            color_vector = colorspace::lighten(viridis::magma(n = 3, begin = 0.35, direction = -1), 0.2)) +
            theme(axis.text.x = element_text(size=14, angle=45, hjust=0.4),
              axis.text.y = element_text(size=13)) 

```

```{r, fig.height=7, fig.width=10}


amp_f2 <- phyloseq_to_ampvis2(ps.bor)

amp_heatmap(amp_f2,
            tax_show = 15,
            group_by = "Group",
            tax_aggregate = "Phylum",
            normalise=TRUE,
            plot_values_size = 4.5,
            showRemainingTaxa = TRUE, tax_class = "Pseudomonadota",
            color_vector = colorspace::lighten(viridis::magma(n = 3, begin = 0.35, direction = -1), 0.2)) +
            theme(axis.text.x = element_text(size=14, angle=45, hjust=0.4),
              axis.text.y = element_text(size=13)) 

```
```{r}

amp_f2

library("limma")
library("clusterProfiler")
library(edgeR)

```
### kegg docu

```{r}

dfkegg <- read_csv("~/storage/temp/kegg_df.csv", )
dfkegg <- dfkegg %>% 
  select(KEGG_ko, site_y, weight) %>%
  mutate(site_x = paste0(sapply(str_split(site_y, '\\.', 2), function(x) x[[1]])))

dfkegg %>%  ggplot() + 
    geom_boxplot(aes(x = site_y, y = weight, group = site_y))
 
dfkegg %>%  ggplot(aes(x=weight)) + 
  geom_density()

```



```{r}

dfkegg_wide <- dfkegg %>% 
  select(KEGG_ko, site_y, weight) %>% 
  pivot_wider(names_from = site_y, values_from = weight)

dfkegg_wide
dfkegg_wide <- dfkegg_wide %>% 
  column_to_rownames("KEGG_ko")

dfkegg_wide <- dfkegg_wide + 1
dfkegg_wide[is.na(dfkegg_wide)] <- 0
dfkegg_wide

d0 <- DGEList(dfkegg_wide)
group <- dfkegg$site_x %>% 
  unique() %>% 
  rep(each=3)

plotMDS(y$E, col = as.numeric(group))

```

```{r}

cutoff <- 1
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 

```



```{r, fig.height=5, fig.width=6}

plotMDS(y, col = as.numeric(group))

```
```{r, fig.height=5, fig.width=6}

plotMDS(d, col = as.numeric(group))

```


```{r}



mm <- model.matrix(~0 + group)
y <- voom(d, mm, lib.size = d$samples$lib.size,  plot = T)

```
```{r}
y$E
```
```{r}

dt <- d$counts %>% 
  as.data.frame() %>% 
  pivot_longer(everything(), names_to = "site", values_to = "values") %>% 
  mutate(site = as.factor(site))

ggplot(data = dt, aes(x=values, color = site)) + 
  geom_density()

```

```{r}

y$E %>% 
  as.data.frame() %>% 
  pivot_longer(everything(), names_to = "site", values_to = "values") %>% 
  ggplot(aes(x=values)) + 
  geom_density()

```

```{r}

fit <- lmFit(y, mm)
head(coef(fit))

```

```{r}

contr <- makeContrasts((group1 + group2 + group5)/3 - (group3 + group4)/2, levels = colnames(coef(fit)))
contr

```


```{r}

tmp <- contrasts.fit(fit, contr)
tmp <- eBayes(tmp)

```

```{r}

top.table <- topTable(tmp, sort.by = "P", n = Inf)

less <- top.table %>%
  filter(logFC < 0) %>% 
  filter(adj.P.Val < 0.05) %>% 
  rownames_to_column("KEGG_ko") %>% 
  select(KEGG_ko) %>% 
  mutate(id = row_number()) %>% 
  relocate(id)

write_tsv(less, "less_forest_kegg.csv")

more <- top.table %>%
  filter(logFC > 0) %>% 
  filter(adj.P.Val < 0.05) %>% 
  rownames_to_column("KEGG_ko") %>% 
  select(KEGG_ko) %>% 
  mutate(id = row_number()) %>% 
  relocate(id)

write_tsv(more, "more_forest_kegg.csv")

top.table %>%
  filter(logFC > 0) %>% 
  filter(adj.P.Val < 0.05)

top.table %>%
  filter(logFC > 0) %>% 
  filter(adj.P.Val < 0.05)

```


```{r}

getwd()

```

```{r}

contr1 <- makeContrasts((group1) - (group2 + group3 + group4 + group5)/4, levels = colnames(coef(fit)))
contr2 <- makeContrasts((group2) - (group3 + group4 + group5 + group1)/4, levels = colnames(coef(fit)))
contr3 <- makeContrasts((group3) - (group1 + group2 + group4 + group5)/4, levels = colnames(coef(fit)))
contr4 <- makeContrasts((group4) - (group1 + group2 + group3 + group5)/4, levels = colnames(coef(fit)))
contr5 <- makeContrasts((group5) - (group1 + group2 + group3 + group4)/4, levels = colnames(coef(fit)))

```


```{r}

tmp1 <- contrasts.fit(fit, contr1)
tmp1 <- eBayes(tmp1)

tmp2 <- contrasts.fit(fit, contr2)
tmp2 <- eBayes(tmp2)

tmp3 <- contrasts.fit(fit, contr3)
tmp3 <- eBayes(tmp3)

tmp4 <- contrasts.fit(fit, contr4)
tmp4 <- eBayes(tmp4)

tmp5 <- contrasts.fit(fit, contr5)
tmp5 <- eBayes(tmp5)

```



```{r}



for (i in seq(1, 5)){
  top.table <- topTable(get(paste0("tmp", i)), sort.by = "P", n = Inf)
  
  less <- top.table %>%
  filter(logFC < 0) %>% 
  filter(adj.P.Val < 0.05) %>% 
  rownames_to_column("KEGG_ko") %>% 
  select(KEGG_ko) %>% 
  mutate(id = row_number()) %>% 
  relocate(id)
  
  more <- top.table %>%
  filter(logFC > 0) %>% 
  filter(adj.P.Val < 0.05) %>% 
  rownames_to_column("KEGG_ko") %>% 
  select(KEGG_ko) %>% 
  mutate(id = row_number()) %>% 
  relocate(id)
  
  write_tsv(less, paste0(i, "_less_kegg.csv"))
  write_tsv(more, paste0(i, "_more_kegg.csv"))

}

top.table <- topTable(tmp2, sort.by = "P", n = Inf)

less <- top.table %>%
  filter(logFC < 0) %>% 
  filter(adj.P.Val < 0.05) %>% 
  rownames_to_column("KEGG_ko") %>% 
  select(KEGG_ko) %>% 
  mutate(id = row_number()) %>% 
  relocate(id)

write_tsv(less, "less_forest_kegg.csv")

more <- top.table %>%
  filter(logFC > 0) %>% 
  filter(adj.P.Val < 0.05) %>% 
  rownames_to_column("KEGG_ko") %>% 
  select(KEGG_ko) %>% 
  mutate(id = row_number()) %>% 
  relocate(id)

write_tsv(more, "more_forest_kegg.csv")


```

```{r}

dfkegg %>%  ggplot() + 
    geom_boxplot(aes(x = site_y, y = weight, group = site_y))

```


```{r}

y$E %>% 
  as.data.frame() %>% 
  pivot_longer(everything(), names_to = "site", values_to = "values") %>% 
  ggplot() + 
    geom_boxplot(aes(x = site, y = values))

```

```{r}

dt <- y$E %>% 
  as.data.frame() %>% 
  pivot_longer(everything(), names_to = "site", values_to = "values") %>% 
  mutate(site = as.factor(site))

ggplot(data = dt, aes(x=values, color = site)) + 
  geom_density()

```

