---
title: "Nal"
author: "GrGladkov"
date: "19 11 2020"
output: html_document
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = FALSE)
knitr::opts_knit$set(root.dir = "~/storage/nal/")

```



```{r}

list.files()

```

### add variables, set work directory, copy raw files 

```{r}

#first run - with controls(its), pool - true, 2.5ee, 220-180
# ls | grep Abacumov-Nal | xargs cp -t  ~/storage/nal/in

setwd("~/storage/nal/")
path = "in/"
path_trein_set = "~/storage/somebases/silva_nr_v138_train_set.fa"
path_trein_set_species = "~/storage/somebases/silva_species_assignment_v138.fa"
name_Brief = "nameBrief.txt"

truncLen = "220,180"
maxEE = "2,5"
mult = TRUE
mlt = NULL

```

### import libraries

```{r}

require(dada2)
require(Biostrings)
require(DECIPHER)
require(phyloseq)
require(seqinr)
require(data.table)
require(metagMisc)
require(tibble)
require(tidyverse)

```

### quallity plots

```{r}
# docker container run --rm -e AMPLICONLENGTH=[amplicon length] -e FORWARDPRIMERLENGTH=[forward primer length] \
    # -e REVERSEPRIMERLENGTH=[reverse primer length] -v /path/to/fastqs:/data/input \
    # -v / figaro

# docker container run --rm -e AMPLICONLENGTH=260 -e FORWARDPRIMERLENGTH=20 \
# -e REVERSEPRIMERLENGTH=19 -v /home/gladkov/storage/nal/in \
# -v /home/gladkov/storage/nal figaro

# python3 figaro.py -i /home/gladkov/storage/nal/in -o /home/gladkov/storage/nal -a 260 -f 20 -r 19


fnFs <- sort(list.files(path, pattern="_R1_001.fastq", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_001.fastq", full.names = TRUE))
  
plotQualityProfile(fnRs[1:3])

fnFs  
list.files()
ps.f@sam_data

saveRDS(ps.f, file = "ps.f.2")
```

#### dada2 pipeline

```{r} 

sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
on.exit()
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(185,185), trimLeft=20, maxN=0, maxEE=c(6,8), rm.phix=TRUE, compress=TRUE, multithread=mult)
errF <- learnErrors(filtFs, multithread=mult)
errR <- learnErrors(filtRs, multithread=mult)
derepFs <- derepFastq(filtFs, verbose=TRUE)
derepRs <- derepFastq(filtRs, verbose=TRUE)
# Name the derep-class objects by the sample names
names(derepFs) <- sample.names
names(derepRs) <- sample.names
dadaFs <- dada(derepFs, err=errF, multithread=mult, pool=FALSE)
dadaRs <- dada(derepRs, err=errR, multithread=mult, pool=FALSE)
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose=TRUE)
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
table(nchar(getSequences(seqtab)))
getN <- function(x) sum(getUniques(x))
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=mult, verbose=TRUE)


briefToSeq <- colnames(seqtab.nochim)
names(briefToSeq) <- paste0("Seq", seq(ncol(seqtab.nochim))) 
st.brief <- seqtab.nochim
colnames(st.brief) <- names(briefToSeq) 

write.table(briefToSeq, file = name_Brief, sep = "\t")

#dna <- DNAStringSet(briefToSeq)
#alignment <- AlignSeqs(DNAStringSet(dna), anchor=NA,verbose=FALSE, processors = mlt)
#writeXStringSet(alignment, file="align.fasta")

taxa.dada2 <- assignTaxonomy(briefToSeq,path_trein_set , multithread=mult)
taxa.dada2.species <- assignSpecies(briefToSeq, path_trein_set_species) # maybe use not seqs but brief 
rownames(taxa.dada2.species) <- rownames(briefToSeq)
briefToSeq.df <- data.frame(briefToSeq)
rownames(taxa.dada2.species) <- rownames(briefToSeq.df)
rownames(taxa.dada2) <- rownames(taxa.dada2.species)
taxa <- cbind2(taxa.dada2, taxa.dada2.species[,2])
colnames(taxa)[7] <- "Species"

getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
write.table(track, file = "track.tsv", sep= "\t", col.names = NA, quote=FALSE)

st.brief.t.df <- data.frame(t(st.brief))
write.table(st.brief.t.df, file = "otu_table.txt", sep= "\t", col.names = NA, quote=FALSE)

briefToSeq.ls <- as.list(briefToSeq.df[,c("briefToSeq")])
briefToSeq.names <- as.list(rownames(briefToSeq.df))
write.fasta( briefToSeq.ls, briefToSeq.names , "rep_seq.fasta", as.string = FALSE)

write.table(taxa, file = "taxa.txt", sep= "\t", col.names = NA, quote=FALSE)
  
```

#### tree from SEPP-QIIME2-plagin

sed -i '1s/^/#OTU_ID/' otu_table.txt 
biom convert -i otu_table.txt -o otu_table.biom --to-hdf5
qiime tools import \
--input-path rep_seq.fasta \
--output-path rep_seq.qza \
--type 'FeatureData[Sequence]'
qiime fragment-insertion sepp --i-representative-sequences rep_seq.qza --i-reference-database ~/storage/somebases/sepp-refs-silva-128.qza --o-tree insertion-tree.qza  --o-placements insertion-placements.qza --p-threads 20  ы
qiime tools import \
--input-path otu_table.biom \
--type 'FeatureTable[Frequency]' \
--input-format BIOMV210Format \
--output-path feature-table.qza
qiime fragment-insertion filter-features \
--i-table feature-table.qza \
--i-tree insertion-tree.qza \
--o-filtered-table filtered_table.qza \
--o-removed-table removed_table.qza
unzip -p filtered_table.qza > filtered_table.biom
biom convert -i  filtered_table.biom -o filtered_table.txt --to-tsv
sed -i '1d' filtered_table.txt
unzip -p insertion-tree.qza */data/* > tree.nwk


```{r}

as_tibble(track, rownames = "ID") %>% arrange(nonchim)
reps <- readFasta("rep_seq.fasta")

```

### build phyloseq object

```{r}

#some SEPP bash tree magic

map <- read_tsv("mapping_nalchic_2.tsv")
map <- column_to_rownames(map, "ID")

taxa <- read.csv("taxa.txt" , header=TRUE, sep="\t")
taxa <- column_to_rownames(taxa, 'X')
taxa <- as.matrix(taxa)

filt.otu <- read.csv("otu_table.txt" , header=TRUE, sep="\t")
filt.otu
colnames(filt.otu)[1] <- "ID"
filt.otu <- column_to_rownames(filt.otu, "ID")
filt.otu
library(naturalsort)
filt.otu <- filt.otu[c(naturalsort(colnames(filt.otu)))]
filt.otu
colnames(filt.otu) <- rownames(map)
filt.otu
# class(filt.otu) <- "numeric"

filt.otu <- as.matrix(filt.otu)
filt.otu <- t(filt.otu)


tree <- read_tree(treefile="tree.nwk")

ps <- phyloseq(otu_table(filt.otu, taxa_are_rows=FALSE), 
               sample_data(map),
               tax_table(taxa),
               phy_tree(tree))

filt.otu[0,] = rownames(map)
rownames(filt.otu) == rownames(map)

intersect(rownames(as.data.frame(filt.otu.matrix)), rownames(map)) %>% length()

ps
filt.otu
```

#### rep fasta lenght density plot

```{r, fig.height=4, fig.width=8}

library(tidyverse)
library(seqinr)

some_rep_fasta <- read.fasta("~/storage/mal/rep_seq.fasta")
d <- density(getLength(some_rep_fasta))
plot(d)
sort(getLength(some_rep_fasta), decreasing = TRUE)
quantile(getLength(some_rep_fasta))

```


```{r}

saveRDS(ps, file = "ps.rds")

```


### Read some previously generated data 

Delete mit and chl
ps, track,etc.
Add refseq data.

```{r}

track <- read_tsv("track.tsv")
ps <- read_rds("ps.rds")
ps <- merge_phyloseq(ps, refseq(readDNAStringSet("rep_seq.fasta")))

pop_taxa <- function(physeq, badTaxa){
  allTaxa = taxa_names(physeq)
  myTaxa <- allTaxa[!(allTaxa %in% badTaxa)]
  return(prune_taxa(myTaxa, physeq))
}

delete_mit_chl_eu <- function(ps){
  badTaxa <- taxa_names(subset_taxa(ps, Order=="Chloroplast"))
  ps <- pop_taxa(ps, badTaxa)
  badTaxa <- taxa_names(subset_taxa(ps, Family=="Mitochondria"))
  ps <- pop_taxa(ps, badTaxa)
  badTaxa <- taxa_names(subset_taxa(ps, Kingdom == "Eukaryota"))
  ps <- pop_taxa(ps, badTaxa)
  badTaxa <- taxa_names(subset_taxa(ps, is.na(Phylum)))
  ps <- pop_taxa(ps, badTaxa)
  return(ps)
}

saveRDS(ps.f, file = "ps_f.rds")
ps.f <- delete_mit_chl_eu(ps)
writeXStringSet(ps.f@refseq, file="ref_filt.fasta")

```


```{r,fig.height=6, fig.width=10}

pssd2veg <- function(physeq) {
  sd <- sample_data(physeq)
  return(as(sd,"data.frame"))
}

sample_data(ps.f2) <- pssd2veg(ps.f2) %>% 
  rownames_to_column("ID") %>% 
    mutate(
    Quarry_or_Benchmark = case_when(
      grepl('Quarry', What) ~ 'Quarry',
      grepl('Benchmark', What) ~ 'Benchmark'
    ) 
    ) %>%
  relocate(Quarry_or_Benchmark, .before = Insolation) %>% 
      column_to_rownames("ID")

```

### Richness/depth correlation.

```{r, message=FALSE,echo=TRUE, fig.height=4, fig.width=12}

library(phyloseq)
library(ggplot2)
plot_rich_reads_samlenames_lm <- function(physeq){
  rish <- estimate_richness(physeq, measures = "Observed")
  reads.sum <- as.data.frame(sample_sums(physeq))
  reads.summary <- cbind(rish, reads.sum)
  colnames(reads.summary) <- c("otus","reads")
  reads.summary["Repeat"] <-unlist(purrr::map(stringr::str_split(rownames(physeq@sam_data), "N_", 2), function(x) x[[2]]))
  reads.summary["Site"] <- physeq@sam_data$Site
  library(ggrepel)
  require(ggforce)
  p1 <- ggplot(data=reads.summary) + 
    geom_point(aes(y=otus, x=log2(reads), color=Site),size=3) + 
    geom_text_repel(aes(y=otus, x=log2(reads), label=paste0(Site, "_", Repeat))) + 
    theme_bw() +
    geom_smooth(aes(y=otus, x=log2(reads), fill=Site, color=Site),method=lm, se=FALSE, ymin = 1) + 
    scale_x_continuous(sec.axis = sec_axis(sec.axis ~ 2**.)) 
  # geom_mark_ellipse(aes(y = otus, x=reads, group = Repeats, label = Repeats, color = Repeats), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
  
  return(p1)
}

p.rich1 <- plot_rich_reads_samlenames_lm(ps.f)

ps.f2 <- subset_samples(ps.f, sample_names(ps.f) != 'N_95')
ps.f2 <- subset_samples(ps.f2, sample_names(ps.f2) != 'N_96')
ps.f2 <- subset_samples(ps.f2, sample_names(ps.f2) != 'N_89')
ps.f2 <- subset_samples(ps.f2, sample_names(ps.f2) != 'N_31')
# ps.f2 <- subset_samples(ps.f2, sample_names(ps.f2) != 'N_70')
ps.f2 <- subset_samples(ps.f2, Site != 'N8-2')
ps.f2 <- subset_samples(ps.f2, Site != 'N8-3')

p.rich2 <- plot_rich_reads_samlenames_lm(ps.f2)
sample_names(ps.f2)

ps.test <- subset_samples(ps.f2, Site == 'N3-1')
p.rich <- plot_rich_reads_samlenames_lm(ps.test)

ggarrange(p.rich1, p.rich2)

ps.f <- ps.f2

```


```{r}

ps.nc <- prune_samples(! sample_data(ps.f2)$Soil %in% c("Forest_gray"), ps.f2)
ps.nc  <- prune_taxa(taxa_sums(ps.nc) > 0, ps.nc)

ps.nc@sam_data

```

```{r}

amp <- phyloseq_to_amp(ps_f)
amp

```



```{r}

ps.f2@sam_data <- map2 

ps.f2@sam_data

p1 <- plot_ordination(ps.varstab, ordination.b, type="sample", color="Soil", title="Типы почвы", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    theme(legend.position="bottom")

p2 <- plot_ordination(ps.varstab, ordination.b, type="sample", color="Quarry_or_Benchmark", title="Повреждённая почва и контроли", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    theme(legend.position="bottom")

ggarrange(p1, p2, nrow = 1, ncol = 2)

```

#### reexport rs

```{r}

ps.f2@sam_data

ps.nc <- prune_samples(! sample_data(ps_f2)$Soil %in% c("Carbonate_chernozem"), ps_f2)
ps.nc  <- prune_taxa(taxa_sums(ps.nc) > 0, ps.nc)

ps.сc <- prune_samples(sample_data(ps_f2)$Soil %in% c("Carbonate_chernozem"), ps_f2)
ps.сc  <- prune_taxa(taxa_sums(ps.сc) > 0, ps.сc)

ps.nc@sam_data

```



## Import RDS data. Read new mapping.

add some new normalisation - temp
reformat dataset


```{r}

map <- read_tsv("mapping_nalchic_new.tsv", 
                col_types = cols(
                  pH = col_character()
                )
              )
map <- column_to_rownames(map, "Sample")
ps <- read_rds("ps_f.rds")
site_column <- ps@sam_data %>% 
  as.data.frame() %>% 
  pull('Site')
map <- map %>% 
  mutate(Type = str_replace(Type, " /", "/")) %>% 
  mutate(`Corg,_%` = str_replace(`Corg,_%`, ",", ".")) %>%
  mutate(`NO3_mg/kg` = str_replace(`NO3_mg/kg` , "," , ".")) %>% 
  mutate(pH = str_replace(pH, ",", ".")) %>% 
  mutate(final_concentration = str_replace(final_concentration, ",", ".")) %>% 
  mutate(pH = as.numeric(pH)) %>% 
  mutate(final_concentration = as.numeric(final_concentration)) %>% 
  mutate(`NO3_mg/kg` = as.numeric(`NO3_mg/kg`)) %>% 
  mutate(`Corg,_%` = as.numeric(`Corg,_%`)) %>%
  mutate(`Day_sampling` = as.factor(`Day_sampling`)) %>%
  mutate(
  Quarry_or_Benchmark = case_when(
    grepl('Quarry', What) ~ 'Quarry',
    grepl('Benchmark', What) ~ 'Benchmark'
  ) 
  ) %>%
      mutate(
    Region = case_when(
      grepl('meadow_humus', Soil) ~ 'Urvan',
      grepl('Carbonate_chernozem', Soil) ~ 'Progress',
      grepl('Forest_gray', Soil) ~ 'Gerpegesh',
    )
  ) %>% 
  relocate(Quarry_or_Benchmark, .before = Insolation) %>% 
  mutate(Repeat = site_column) %>% 
  relocate(Repeat, .after = Site)

map <- map %>%  mutate_if(sapply(map, is.character), as.factor)


map_m <- map
colnames(map_m)
colnames(map_m)[16] <- "C"
colnames(map_m)[17] <- "C_org"
colnames(map_m)[18] <- "P"
colnames(map_m)[19] <- "K"
colnames(map_m)[20] <- "NH4"
colnames(map_m)[21] <- "NO3"

colnames(map)

ps.f <- prune_samples(! sample_data(ps)$Soil %in% c("Forest_gray"), ps)
ps.f  <- prune_taxa(taxa_sums(ps.f) > 0, ps.f)
sample_data(ps.f) <- map_m


ps.nc <- prune_samples(sample_data(ps.f)$Day_sampling %in% c("1"), ps.f)
map3 <- data.frame(ps.nc@sam_data) %>%  mutate(temp = scale(Temperature))
ps.nc <- merge_phyloseq(ps.nc, sample_data(map3))

ps.nc1 <- prune_samples(sample_data(ps.f)$Day_sampling %in% c("2", "3"), ps.f)
map3 <- data.frame(ps.nc1@sam_data) %>%  mutate(temp = scale(Temperature))
ps.nc1 <- merge_phyloseq(ps.nc1, sample_data(map3))

ps.f <- merge_phyloseq(ps.nc, ps.nc1)

ps.merged <- phyloseq::merge_samples(ps.f, "Repeat")


# not run
# may be some bad import repair?

# sample_data(ps.nc)$Quarry_or_Benchmark = factor(sample_data(ps.nc)$Quarry_or_Benchmark)
# sample_data(ps.nc)$Feature = factor(sample_data(ps.nc)$Feature)
# ps.merged <- phyloseq::merge_samples(ps.nc, "Site")
# 
# sample_data(ps.merged)$Soil <- factor(sample_data(ps.merged)$Soil, labels=levels(sample_data(ps.nc)$Soil))
# sample_data(ps.merged)$Quarry_or_Benchmark <-  factor(sample_data(ps.merged)$Quarry_or_Benchmark, labels=levels(sample_data(ps.nc)$Quarry_or_Benchmark))
# sample_data(ps.merged)$Feature <- factor(sample_data(ps.merged)$Feature, labels=levels(sample_data(ps.nc)$Feature))
# 
# sample_data(ps.merged)
# factor(sample_data(ps.merged)$Soil, levels = levels(sample_data(ps.nc)$Soil))
# sample_data(c)$Soil
# 
# levels(as.factor(ps.f@sam_data$Repeat))
# 
# as.data.frame(ps.f@sam_data@.Data) 
# m <- as.data.frame(ps.f@sam_data@.Data) 
# class(m)
# map <- m %>%  mutate_if(sapply(m, is.character), as.character)
# ps.fc <- ps.f
# sample_data(ps.fc) <- map
# map
# 
ps.f@sam_data

```


#### vst dataset

```{r}

ps_vst <- function(ps, factor){
  
  require(DESeq2)
  require(phyloseq)
  
  diagdds = phyloseq_to_deseq2(ps, as.formula(paste( "~", factor)))              
  diagdds = estimateSizeFactors(diagdds, type="poscounts")
  diagdds = estimateDispersions(diagdds, fitType = "local") 
  pst <- varianceStabilizingTransformation(diagdds)
  pst.dimmed <- t(as.matrix(assay(pst))) 
  pst.dimmed[pst.dimmed < 0.0] <- 0.0
  ps.varstab <- ps
  otu_table(ps.varstab) <- otu_table(pst.dimmed, taxa_are_rows = FALSE) 
  return(ps.varstab)
}

ps.varstab <- ps_vst(ps.f, "Repeat")

```

### Beta - NMDS

```{r, message=FALSE,echo=TRUE,fig.height=7, fig.width=9, results=FALSE, fig.keep='all'}

beta_custom_norm_NMDS_elli <- function(ps, seed = 7888, normtype="vst", Color="What", Group="Repeat"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  require(DESeq2)
  library(ggforce)
  
  #normalisation. unifrac - rarefaction; wunifrac,bray - varstab
  #variant with only bray
  
  if (exists("ps.varstab") == FALSE){
    diagdds = phyloseq_to_deseq2(ps, ~ Site)                  
    diagdds = estimateSizeFactors(diagdds, type="poscounts")
    diagdds = estimateDispersions(diagdds, fitType = "local") 
    if (normtype =="vst")
      pst <- varianceStabilizingTransformation(diagdds)
    if (normtype =="log") 
      pst <- rlogTransformation(diagdds)
    
    pst.dimmed <- t(as.matrix(assay(pst))) 
    pst.dimmed[pst.dimmed < 0.0] <- 0.0
    ps.varstab <- ps
    otu_table(ps.varstab) <- otu_table(pst.dimmed, taxa_are_rows = FALSE) 
  }

  ordination.b <- ordinate(ps.varstab, "NMDS", "bray")
  p  <-  plot_ordination(ps.varstab,
                         ordination.b,
                         type="sample",
                         color = Color,
                         # title="NMDS - Bray-Curtis",
                         title=NULL,
                         axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes_string(group = Group, label = Group),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                      con.colour='gray') +
    theme(legend.position = "none") 

  return(p)
}

d = "descript"
d <- enquo(d)
xnew <- quo_name(d)
p.beta <- beta_custom_norm_NMDS_elli(ps.f, Group="Site", Color="Site")
p.beta

saveRDS(ps.varstab, file="ps.varstab.temp")
saveRDS(ps.f, file="ps.f.temp")


ggsave(file="plots/nmds.tiff", 
       plot=p.beta, 
       width=2200, 
       height=1850,
       units ="px", 
       dpi=300, 
       compression = "lzw")

?ggsave
?devSVG
?geom_mark_ellipse

```



```{r,fig.height=7, fig.width=8}
ordination.b <- ordinate(ps.varstab, "NMDS", "bray")
ps.f@sam_data

plot_ordination(ps.varstab, ordination.b, type="sample", color="What", title=NULL, axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes(group = What, label = What), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"),con.colour='gray') +
   theme(legend.position = "none") 

```

### Anova

```{r}

library(car)
colnames(map_m)
map_m
inter_model <- lm(Quarry_or_Benchmark*Soil ~ C + K + C_org + NH4 + pH + P + NO3, data = map_m)
inter <- Anova(mod = inter_model, error.estimate="pearson")
inter

```

```{r}

library(dplyr)
library(tidyr)
library(purrr)

df1 <- map_m %>% select()

map_m %>% 
  nest(-Site) %>% 
  mutate(model = map(data, ~anova(lm(Value ~ Sample, .))), 
         tidy = map(model, tidy)) %>% 
select(Class, td) %>% 
unnest()

```


```{r}

library(car)
library(multcomp)
colnames(map_m)
map_m
inter_model <- lm(C  ~ Quarry_or_Benchmark*Soil, data = map_m)
inter <- Anova(mod = inter_model)
inter
anova(inter_model)

```

```{r}

summary(glht(inter_model, mcp(Quarry_or_Benchmark="Tukey")))

```


### Heatmaps ampvis2
Heatmaps with relative abundance by ampvis2 package

```{r, message=FALSE,echo=TRUE}

map3 <- data.frame(ps.f2@sam_data)
sam_data(ps.f2) <- map3 %>% 
  rownames_to_column("ID") %>% 
    mutate(
    Soil = case_when(
      grepl('meadow_humus', Soil) ~ 'Луговая',
      grepl('Carbonate_chernozem', Soil) ~ 'Чернозём',
      grepl('Forest_gray', Soil) ~ 'Лесная',
    )
  ) %>% 
    mutate(
    Quarry_or_Benchmark = case_when(
      grepl('Quarry', What) ~ 'Quarry',
      grepl('Benchmark', What) ~ 'Benchmark'
    ) 
    )%>% 
      column_to_rownames("ID")


data.frame(map3))
library(tidyverse)
library(phyloseq)

phyloseq_to_amp <- function(ps){
    require(ampvis2)
    require(tibble)
    require(phyloseq)
    colnames(ps@tax_table@.Data) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
    OTU1 = as(otu_table(ps), "matrix")
    OTU1 <- t(OTU1)
    OTUdf = as.data.frame(OTU1)
    taxa.ps <- as(tax_table(ps), "matrix")
    taxa.df = as.data.frame(taxa.ps)
    my_otu_table <- merge(OTUdf, taxa.df, by=0)
    my_otu_table <- column_to_rownames(my_otu_table, var="Row.names")

    my_metadata <- as_tibble(sample_data(ps), rownames=NA)
    my_metadata <- rownames_to_column(my_metadata,var = "SampleID")
    my_tree <- phy_tree(ps)
    amp.ps <- amp_load(otutable = my_otu_table, metadata = my_metadata, tree = my_tree)
    return(amp.ps)
}


amp <- phyloseq_to_amp(ps.f)
ps.f@sam_data
```

### Split ps

```{r}

# ps.1 <- prune_samples(sample_data(ps.f2)$Soil %in% c("Луговая"),ps.f2)
# ps.1  <- prune_taxa(taxa_sums(ps.1) > 0, ps.1)
# 
# ps.2 <- prune_samples(sample_data(ps.f2)$Soil %in% c("Чернозём"),ps.f2)
# ps.2  <- prune_taxa(taxa_sums(ps.2) > 0, ps.2)
# 
# ps.12 <- prune_samples(! sample_data(ps.f2)$Soil %in% c("Лесная"),ps.f2)
# ps.12  <- prune_taxa(taxa_sums(ps.12) > 0, ps.12)

ps.nc <- prune_samples(sample_data(ps.f)$Region %in% c("Urvan"), ps.f)
ps.nc  <- prune_taxa(taxa_sums(ps.nc) > 0, ps.nc)

ps.cc <- prune_samples(sample_data(ps.f)$Region %in% c("Progress"), ps.f)
ps.cc  <- prune_taxa(taxa_sums(ps.cc) > 0, ps.cc)

ps.cc2 <- prune_samples(sample_data(ps.f)$Site %in% c("N1", "N3", "N5", "N6", "N7"), ps.f)
ps.cc2  <- prune_taxa(taxa_sums(ps.cc2) > 0, ps.cc2)

ps.n13 <- prune_samples(sample_data(ps.f)$Site %in% c("N1", "N3"), ps.f)
ps.n13  <- prune_taxa(taxa_sums(ps.n13) > 0, ps.n13)

ps.cc2 <- prune_samples(sample_data(ps.f)$Site %in% c("N2", "N4", ""), ps.f)
ps.cc2  <- prune_taxa(taxa_sums(ps.cc2) > 0, ps.cc2)

# ps.si <- prune_samples(sample_data(ps.f)$Soil %in% c("Carbonate_chernozem"), ps.f)
# ps.si  <- prune_taxa(taxa_sums(ps.si) > 0, ps.si)
# 
# ps.cc <- prune_samples(sample_data(ps.f)$Soil %in% c("Carbonate_chernozem"), ps.f)
# ps.cc  <- prune_taxa(taxa_sums(ps.cc) > 0, ps.cc)
# 
# ps.cc <- prune_samples(sample_data(ps.f)$Soil %in% c("Carbonate_chernozem"), ps.f)
# ps.cc  <- prune_taxa(taxa_sums(ps.cc) > 0, ps.cc)

ps.f@sam_data

```

### Heatmappa

```{r, fig.height=10, fig.width=12}

amp_heatmap(amp, tax_show = 40, group_by = "What", tax_aggregate = "Genus", tax_class = "Proteobacteria", tax_add = "Phylum") + theme_bw() + theme(text = element_text(size=15), legend.position = "none") + theme(axis.text.x=element_text(angle=45,hjust=1))

```

```{r, fig.height=7, fig.width=8}

p.heat.phyla <- amp_heatmap(amp,
            tax_show = 15,
            group_by = "Site",
            tax_aggregate = "Phylum",
            tax_class = "Proteobacteria") +
  theme_bw() +
  theme(text = element_text(size=15),
        legend.position = "none", 
        axis.text.x=element_text(angle=45,hjust=1))

ggsave(file="plots/heatta.tiff", 
       plot=p.heat.phyla, 
       width=1500, 
       height=1200,
       units ="px", 
       dpi=300, 
       compression = "lzw")

p.heat.phyla

```

```{r, fig.height=10, fig.width=12}

amp_heatmap(amp, tax_show = 30, group_by = "Site", tax_aggregate = "Class", tax_class = "Proteobacteria", tax_add = "Phylum") + theme_bw() + theme(text = element_text(size=15), legend.position = "none") + theme(axis.text.x=element_text(angle=45,hjust=1))

```



```{r}

library(vegan)
physeq <- ps.nc
dist <- phyloseq::distance(physeq, "bray")
metadata <- as(sample_data(physeq@sam_data), "data.frame")
ad <- adonis2(dist ~ Quarry_or_Benchmark, data = metadata)

ad

```

```{r}

library(vegan)
physeq <- ps.cc
dist <- phyloseq::distance(physeq, "bray")
metadata <- as(sample_data(physeq@sam_data), "data.frame")
ad <- adonis2(dist ~ Quarry_or_Benchmark, data = metadata)

ad

```

```{r,message=FALSE,echo=TRUE, fig.height=5, fig.width=5.8}

dist <- phyloseq::distance(ps.12, method = "bray")
groups <- ps.12@sam_data$Soil
mod <- vegan::betadisper(dist, groups)
mod.box <- boxplot(mod)
TukeyHSD(mod)
?TukeyHSD.betadisper
permutest(mod, permutations = 999)
anova(mod)
plot(TukeyHSD(mod))

```


```{r, message=FALSE,echo=TRUE, fig.height=6, fig.width=9}

amp_heatmap(amp, group_by = "Quarry_or_Benchmark",facet_by = "Soil",tax_show = 30, tax_aggregate = "Genus", tax_add = "Family", tax_class = "Proteobacteria") + theme_bw() + theme(text = element_text(size=15), legend.position = "none")
amp
```

### DESeq2

```{r, message=FALSE, echo=FALSE}

des_w_simper <- function(ps, factor){
  require(DESeq2)
  require(vegan)
  require(tibble)
  require(phyloseq)
  
  diagdds = phyloseq_to_deseq2(ps, as.formula(paste( "~", factor)))                  
  diagdds = estimateSizeFactors(diagdds, type="poscounts")
  diagdds = estimateDispersions(diagdds, fitType = "local") 
  diagdds = DESeq(diagdds)
  samp <-sample_data(ps)
  aggdata <- t(aggregate.data.frame(as.data.frame(as.data.frame(t(diagdds@assays@data$mu))), by=list(samp[[factor]]), median))
  colnames(aggdata) <- aggdata[1,]
  aggdata <- aggdata[-1,]
  res = results(diagdds)
  res.df <- as.data.frame(res)
  nice <- cbind(res.df, as.data.frame(ps@tax_table@.Data[rownames(res.df),]), as.data.frame(aggdata)[rownames(res.df),])
} 

```


```{r}
des.nc.filt

des.nc %>% 
  filter(baseMean > 10) %>% 
  filter(padj < 0.05) %>%
  arrange(log2FoldChange)

des.nc %>% 
  filter(baseMean > 10) %>% 
  filter(padj < 0.05) %>%
  filter(log2FoldChange < 0) %>% 
  arrange(log2FoldChange)

des.cc %>% 
  filter(baseMean > 10) %>% 
  filter(padj < 0.05) %>%
  filter(log2FoldChange < 0) %>% 
  arrange(log2FoldChange)

des.cc %>% 
  filter(baseMean > 10) %>% 
  filter(padj < 0.05) %>%
  filter(log2FoldChange > 0) %>% 
  arrange(log2FoldChange)


des.so %>% 
  filter(baseMean > 10) %>% 
  filter(padj < 0.05) %>%
  filter(log2FoldChange < 0) %>% 
  arrange(log2FoldChange)

des.so %>% 
  filter(baseMean > 10) %>% 
  filter(padj < 0.05) %>%
  filter(log2FoldChange < 0) %>% 
  arrange(log2FoldChange)

des.so %>% 
  filter(baseMean > 10) %>% 
  filter(padj < 0.05) %>%
  filter(log2FoldChange > 0) %>% 
  arrange(log2FoldChange)




```


```{r, fig.height=4, fig.width=10}

des.nc <- des_w_simper(ps.nc, "Quarry_or_Benchmark")
des.cc <- des_w_simper(ps.cc, "Quarry_or_Benchmark")
des.so <- des_w_simper(ps.f, "Region")

# des.nc %>% filter(baseMean > 10) %>% filter(padj < 0.05) %>% arrange(log2FoldChange)
# des.cc %>% filter(baseMean > 10) %>% filter(padj < 0.05) %>% arrange(log2FoldChange)
# des.so %>% filter(baseMean > 10) %>% filter(padj < 0.05) %>% arrange(log2FoldChange)

des.nc.filt <- des.nc %>% filter(baseMean > 10) %>% filter(padj < 0.05) %>% arrange(log2FoldChange)
des.cc.filt <- des.cc %>% filter(baseMean > 10) %>% filter(padj < 0.05) %>% arrange(log2FoldChange)
des.so.filt <- des.so %>% filter(baseMean > 10) %>% filter(padj < 0.05) %>% arrange(log2FoldChange)

f1 <- ggplot(des.nc.filt, aes(x=log2FoldChange, y=baseMean)) +
  geom_vline(xintercept = 0, size = 0.75, color = "red", alpha=0.3) +
  labs(title="Urvan, Quarry/Benchmark") +
  geom_point() +
  theme_bw() +
  theme(text = element_text(size=8))
  

f2 <- ggplot(des.cc.filt, aes(x=log2FoldChange, y=baseMean)) +
  geom_vline(xintercept = 0, size = 0.75, color = "red", alpha=0.3) +
  labs(title="Progress, Quarry/Benchmark") +
  geom_point() +
  theme_bw() +
  theme(text = element_text(size=8))

f3 <- ggplot(des.so.filt, aes(x=log2FoldChange, y=baseMean)) +
  geom_vline(xintercept = 0, size = 0.75, color = "red", alpha=0.3) +
  labs(title="Progress/Urvan") +
  geom_point() +
  theme_bw() +
  theme(text = element_text(size=8))

length(rownames(des.nc.filt))
length(rownames(des.cc.filt))
length(rownames(des.so.filt))

class(rownames(des.nc.filt))
row.des <- c(rownames(des.nc.filt), rownames(des.cc.filt))
row.des.so <- rownames(des.so.filt)
row.des <- unique(row.des)
length(row.des)

p.des <- ggarrange(f1,f2,f3,nrow = 1)
ggarrange(f1,f2,f3,nrow = 1)

ggsave(file="plots/deseq.tiff",
        device="tiff",
        plot=p.des,
        width=2200,
        height=700,
        units ="px",
        dpi=300,
        compression = "lzw"
        )


write.table(des.nc.filt, file = "des_nc_filt_new.tsv", sep= "\t", col.names = NA, quote=FALSE)
write.table(des.cc.filt, file = "des_cc_filt_new.tsv", sep= "\t", col.names = NA, quote=FALSE)
write.table(des.so.filt, file = "des_so_filt_new.tsv", sep= "\t", col.names = NA, quote=FALSE)

```

```{r}

length(rownames(des.nc.filt))
length(rownames(des.cc.filt))
length(rownames(des.so.filt))

```


```{r,fig.height=8, fig.width=20, results=FALSE}

ps.rand <- rarefy_even_depth(ps.f2)

ordination.b <- ordinate(ps.varstab, "PCoA", "bray")
ordination.u <- ordinate(ps.rand, "PCoA", "unifrac")
ordination.w <- ordinate(ps.varstab, "PCoA", "wunifrac")
  
  #plotting
  p1 <-  plot_ordination(ps, ordination.b, type="sample", color="Site", title="Bray-Curtis", 
                       axes = c(1,2) ) + theme_bw() + theme(text = element_text(size = 10)) + geom_point(size = 3) 
  

  p2 <-  plot_ordination(ps, ordination.u, type="sample", color="Site", title="UniFrac", 
                       axes = c(1,2) ) + theme_bw() + theme(text = element_text(size = 10)) + geom_point(size = 3)
  
  p3 <-  plot_ordination(ps, ordination.w, type="sample", color="Site", title="Weighted UniFrac", 
                       axes = c(1,2) ) + theme_bw() + theme(text = element_text(size = 10)) + geom_point(size = 3) 
  
  #merge by ggpubr
  
  
  
  p.all <- ggarrange(p1, p2, p3, ncol = 3 , nrow = 1, common.legend = TRUE, legend = "bottom", font.label = list(size = 12, face = "bold", color ="black"))
  
p.all
p.beta <- beta_custom_norm_NMDS_elli(ps.f)
ps.varstab@sam_data$Soil
```



```{r, message=FALSE,echo=TRUE,fig.height=8, fig.width=18}

library(ggpubr)

pd1 <- estimate_richness(ps.f2, measures=c("Observed", "InvSimpson", "Shannon"))

pdd  <- pd1 %>% rownames_to_column("Site") %>% mutate(Site = as.factor(Site)) %>% gather(Observed, Shannon, InvSimpson, value = "Index", key = "Type")
pdd <- cbind(pd1, ps.f2@sam_data)
pdd
p1 <- ggplot(data = pdd, aes(y = Observed, x = Quarry_or_Benchmark)) + geom_boxplot() + geom_point(alpha = 0.4) + labs(title = "Observed") + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_grid(~Soil) + theme(axis.title.x = element_blank())
p2 <- ggplot(data = pdd, aes(y = InvSimpson, x = Quarry_or_Benchmark)) + geom_boxplot() + geom_point(alpha = 0.4)  + labs(title = "InvSimpson") + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ facet_grid(~Soil)+ theme(axis.title.x = element_blank())
p3 <- ggplot(data = pdd, aes(y = Shannon, x = Quarry_or_Benchmark)) + geom_boxplot() + geom_point(alpha = 0.4)  + labs(title = "Shannon")+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ facet_grid(~Soil)+ theme(axis.title.x = element_blank())
p1
ggarrange(p1, p2,p3, ncol = 3 ,label.x = 0.105, nrow = 1, common.legend = TRUE) 


p_o
pdd

p.alpha.Cdt.oo <- alpha.custom(ps.f, arrga = "Observed")
p.alpha.Cdt.is <- alpha.custom(ps.f, arrga = "Shannon")
p.alpha.Cdt.pd <- alpha.custom(ps.f, arrga = "InvSimpson")

"Observed","Shannon","InvSimpson"


p1 <- ggarrange(p.alpha.Cdt.oo, p.alpha.Cdt.sh,p.alpha.Cdt.is, p.alpha.Cdt.pd , ncol = 4 ,label.x = 0.105, nrow = 1, common.legend = TRUE) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

p1
```


### Process before tree calculated

```{r}

list.files()



```


### Decontam

Run on all samples
non norm

```{r}

library(decontam)

conc <- read.csv("concentration_nalchic.csv") %>% select(c("ID", "Site", "final_concentration"))
sample_data(ps)$conc <- conc$final_concentration
sample_data(ps)$conc <- sample_data(ps)$conc + 0.000001
sample_data(ps)$conc

```

```{r}

ps.varstab
ps

```



```{r}
sample_sums(ps)

ps.f <- subset_samples(ps, sample_names(ps) != 'N_95')
ps.f <- subset_samples(ps.f, sample_names(ps.f) != 'N_89')
ps.f <- subset_samples(ps.f, sample_names(ps.f) != 'N_96')

ps.vst <- ps_vst(ps.f)

contamdf_ps <- isContaminant(ps.f, method="frequency", conc="conc")
contamdf_ps_vst <- isContaminant(ps.vst, method="frequency", conc="conc")

contamdf_ps["Seq703",]
contamdf_ps_vst["Seq703",]

contamdf_ps %>% arrange(p)
contamdf_ps_vst %>% arrange(p)

```

Seq666
Seq703


```{r}

table(contamdf_ps$contaminant)
cont <- contamdf.freq[contamdf_ps$contaminant == "TRUE",] 
as.data.frame(ps.f@tax_table[rownames(cont),])

```

Seq1123
Seq1461

```{r}

table(contamdf_ps_vst$contaminant)
cont_vst <- contamdf.freq[contamdf_ps_vst$contaminant == "TRUE",] 
as.data.frame(ps.f@tax_table[rownames(cont_vst),])

```



```{r}

plot_frequency(ps.f, "Seq703", conc="conc")

```

```{r}
plot_frequency(ps.vst, "Seq703", conc="conc")

```

```{r}
ps.f
```

```{r}

ps.f@sam_data
84

```


### CCA again

```{r, message=FALSE,echo=TRUE, fig.height=7, fig.width=7}

library(phyloseq)             
library(vegan)
library(ggrepel)

sample_data(ps.f)
physeq <- ps.merged

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}

physeq@sam_data

otus.ps.vegan <- veganifyOTU(physeq)

metadata <- as(sample_data(physeq), "data.frame") 

metadata

cca <- vegan::cca(otus.ps.vegan ~  pH + C_org + C +  K + NH4 + NO3 + temp + P, data=metadata)


kate.ggcca.sites <- function(vare.cca){

require(tidyverse)
biplot <- as.data.frame(vare.cca$CCA$biplot)
wa <- as.data.frame(vare.cca$CCA$wa)

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))
wa <- rownames_to_column(wa, "Label") %>% 
  add_column(Score = rep("sites", length(rownames(wa))))
fdat_amazing <- rbind(biplot, wa) 
fdat_amazing <- fdat_amazing %>%
  mutate(label_size = ifelse(Score == "biplot", 4.5, 3))
 
p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  # geom_abline(intercept=seq(-100, 100, 25), slope=1, colour="grey", alpha=0.5) +
  # geom_abline(intercept=seq(-100, 100, 25), slope=-1, colour="grey", alpha=0.5) +
  geom_vline(xintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  geom_hline(yintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), aes(x = 0, xend = CCA1, y = 0, yend = CCA2),
               size = 0.6, alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
  geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=fdat_amazing$label_size) + 
  xlab(paste0(round(cca$CCA$eig[1], 3)*100, "% CCA1")) +
  ylab(paste0(round(cca$CCA$eig[2], 3)*100, "% CCA2")) +
  grids(linetype = "dashed") +
  theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
  return(p)
}

cca.sites <- kate.ggcca.sites(cca)
cca.sites
ggsave(file="plots/cca-sites-2.tiff",
        device="tiff",
        plot=cca.sites, 
        width=2300, 
        height=2300,
        units ="px", 
        dpi=300,
        compression = "lzw"
        )

```


```{r}

anova(cca, parallel = 10)
anova(cca, by="terms", parallel = 10)
anova(cca, by="mar", parallel = 10)
vif.cca(cca)

```


#### Try another CCA models

```{r, message=FALSE,echo=TRUE, fig.height=7, fig.width=7}




cca_w <- vegan::cca(otus.ps.vegan ~  pH + C_org + C + NH4 + NO3 + temp, data=metadata)

kate.ggcca.sites(cca_w)

```

```{r}

anova(cca_w, parallel = 10)
anova(cca_w, by="terms", parallel = 10)
anova(cca_w, by="mar", parallel = 10)
vif.cca(cca_w)

```

#### Same with varstab normalisation

```{r, message=FALSE,echo=TRUE, fig.height=7, fig.width=7}

ps_vst <- function(ps){
  diagdds = phyloseq_to_deseq2(ps, ~ Repeat)                  
  diagdds = estimateSizeFactors(diagdds, type="poscounts")
  diagdds = estimateDispersions(diagdds, fitType = "local") 
  pst <- varianceStabilizingTransformation(diagdds)
  pst.dimmed <- t(as.matrix(assay(pst))) 
  pst.dimmed[pst.dimmed < 0.0] <- 0.0
  ps.varstab <- ps
  otu_table(ps.varstab) <- otu_table(pst.dimmed, taxa_are_rows = FALSE) 
  return(ps.varstab)
}

ps.varstab <- ps.f
ps.varstab <- ps_vst(ps.f)
ps.varstab.merged <- phyloseq::merge_samples(ps.varstab, "Repeat")
physeq <- ps.varstab.merged
ps



veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}

physeq@sam_data

otus.ps.vegan <- veganifyOTU(physeq)

metadata <- as(sample_data(physeq), "data.frame") 

cca_w_varstab <- vegan::cca(otus.ps.vegan ~  temp + pH + C_org + NH4,  data=metadata)
cca_f_varstab <- vegan::cca(otus.ps.vegan ~  Humidity + pH + C_org + C +  K + NH4 + NO3 + temp + P, data=metadata)


kate.ggcca.sites(cca_f_varstab)
cca_ew_varstab$CCA$u
cca_f_varstab$CCA


ps@sam_data

```


```{r}
# Humidity + pH + C_org + C +  K + NH4 + NO3 + temp

cca_w_varstab_asv <- vegan::cca(otus.ps.vegan ~  NO3 + pH + C_org + C +  K + NH4 + temp,  data=metadata)
anova(cca_w_varstab_asv, parallel = 10)
anova(cca_w_varstab_asv, by="terms", parallel = 10)
anova(cca_w_varstab_asv, by="mar", parallel = 10)
vif.cca(cca_w_varstab_asv)

```


#### CCA for taxons - family level

```{r echo=TRUE, fig.height=9, fig.width=9, message=FALSE}

ps.f
ps.ff <-  filter_taxa(ps.f, function(x) sum(x > 3) > (0.1*length(x)), TRUE)
ps.varstab <- ps_vst(ps.ff)
ps.varstab.merged <- phyloseq::merge_samples(ps.varstab, "Repeat")
ps.varstab.merged.family <- tax_glom(ps.varstab.merged, taxrank="Family")

physeq <- ps.varstab.merged.family

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}


otus.ps.vegan <- veganifyOTU(physeq)

metadata <- as(sample_data(physeq), "data.frame") 


cca_w_varstab_family <- vegan::cca(otus.ps.vegan ~  temp + pH + C_org + NH4,  data=metadata)

kate.ggcca.species <- function(vare.cca, physeq){


  
require(tidyverse)
biplot <- as.data.frame(vare.cca$CCA$biplot)

wa <- as.data.frame(vare.cca$CCA$v) %>% 
  rownames_to_column("ID")
seq <- wa
f <- as.data.frame(physeq@tax_table@.Data) %>% 
  select("Family") %>% 
  rownames_to_column("ID")
wa <- full_join(wa, f) %>% 
  select(-ID) %>% 
  rename("Family" = "Label") %>% 
  mutate(Score = "sites")

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))
fdat_amazing <- rbind(biplot, wa) %>% 
  mutate(Label = as.factor(Label))

 
p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), aes(x = 0, xend = CCA1, y = 0, yend = CCA2), alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
  geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=4) + 
  theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
  # geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
  return(p)
}

kate.ggcca.species(cca_w_varstab_family, physeq) 
cca_w_varstab_family$CCA
```


#### ASV level

```{r echo=TRUE, fig.height=10, fig.width=10, message=FALSE}

ps.ff <-  phyloseq::filter_taxa(ps.f, function(x) sum(x > 3) > (0.1*length(x)), TRUE)
ps.varstab <- ps_vst(ps.ff, "Repeat")
ps.varstab.merged <- phyloseq::merge_samples(ps.varstab, "Repeat")
# ps.varstab.merged.family <- tax_glom(ps.varstab.merged, taxrank="Family")

physeq <- ps.varstab.merged

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}


otus.ps.vegan <- veganifyOTU(physeq)

metadata <- as(sample_data(physeq), "data.frame") 


# cca_w_varstab_asv <- vegan::cca(otus.ps.vegan ~  Humidity + pH + C_org + C +  K + NH4 + NO3 + temp,  data=metadata)
# cca_w_varstab_asv <- vegan::cca(otus.ps.vegan ~  pH + C_org + NH4 + temp,  data=metadata)
cca_w_varstab_asv <- vegan::cca(otus.ps.vegan ~  pH + C_org + C +  K + NH4 + NO3 + temp + P,  data=metadata)

wa <- as.data.frame(cca_w_varstab_asv$CCA$v) %>% 
  rownames_to_column("ID")

taxa.pruned <- as.data.frame(ps.varstab@tax_table@.Data) %>% 
  rownames_to_column("ID")
taxa.pruned <- taxa.pruned %>%  mutate_all(as.character)
# old.tips <- tree$tip.label
# matches <- regmatches(tree$tip.label, gregexpr("[[:digit:]]+", tree$tip.label))
# taxa.pruned$number <- as.numeric(unlist(matches))

# Shitty taxa formating part
taxa.pruned$taxa <- ifelse(is.na(taxa.pruned$Genus),
                            ifelse(is.na(taxa.pruned$Family),
                            ifelse(is.na(taxa.pruned$Order), 
                            ifelse(is.na(taxa.pruned$Class), taxa.pruned$Phylum, taxa.pruned$Class) , taxa.pruned$Order), taxa.pruned$Family), taxa.pruned$Genus)
taxa.pruned[taxa.pruned == "Burkholderia-Caballeronia-Paraburkholderia"] <- "Burkholderia"
taxa.pruned[taxa.pruned == "Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium"] <- "Pararhizobium"

taxa.pruned$taxa2 <- ifelse(is.na(taxa.pruned$Species), with(taxa.pruned, paste0(taxa)), with(taxa.pruned, paste0(taxa, " ", Species )))

taxa.pruned$phylum <- ifelse(taxa.pruned$Phylum == "Proteobacteria", with(taxa.pruned, paste0(taxa.pruned$Class)), with(taxa.pruned, paste0(taxa.pruned$Phylum)))
taxa.pruned$Label <- paste0(taxa.pruned$ID, "_", taxa.pruned$taxa2)


wa <- full_join(wa, taxa.pruned, by="ID") %>% 
  mutate(Score = "sites")
wa

#For the top 50 most abundant taxa, skip if use des res
sw <- summarize_all(as.data.frame(ps.varstab@otu_table), sum)
head_asv <- as_tibble(t(sw), rownames = "ID") %>% 
  arrange(desc(V1)) %>% top_n(n = 100) %>% 
  pull(ID)

wa <- filter(wa, ID %in% head_asv)
# wa <- filter(wa, ID %in% row.des)
# wa <- filter(wa, ID %in% row.des.so)

wa

biplot <- as.data.frame(cca_w_varstab_asv$CCA$biplot)
biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))


fdat_amazing <- plyr::rbind.fill(biplot, wa) %>% 
  mutate(Label = as.factor(Label))
fdat_amazing <- fdat_amazing %>%
  mutate(label_size = ifelse(Score == "biplot", 4.5, 3))
fdat_amazing

p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2)) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), aes(x = 0, xend = CCA1, y = 0, yend = CCA2), alpha=0.8, color = "red", arrow = arrow(angle = 3)) +
  geom_text_repel(aes(x=CCA1, y=CCA2, label= Label, colour = phylum), size=fdat_amazing$label_size) + 
  xlab(paste0(round(cca$CCA$eig[1], 3)*100, "% CCA1")) +
  ylab(paste0(round(cca$CCA$eig[2], 3)*100, "% CCA2")) +
  grids(linetype = "dashed") +
  geom_vline(xintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  geom_hline(yintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  theme(legend.position = "top", panel.background = element_rect(fill = "white", colour = "grey50"))
  # geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
  
# p2 <- p + scale_color_manual(values=c("#5EA09E", "#6B440B", "#F499C2", "#7DA9D8", "#FEF898", "#0000FE", "#F27204", 
                                # "#FE0000", "#FEFF00", "#02F30E", "#F89679", "black","salmon", "red", "yellow", "orange"))

 p2 <- p + scale_color_manual(values=c("#fde725", "#d0e11c", "#F499C2", "#7DA9D8", "#FEF898", "#0000FE", "#F27204", 
                                 "#FE0000", "#FEFF00", "#02F30E", "#F89679", "black","salmon", "red", "yellow", "orange"))

p2
p
return(p)


# select(wa, -c("ID", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species", ))
wa

kate.ggcca.species <- function(vare.cca, physeq){

require(tidyverse)
  
biplot <- as.data.frame(vare.cca$CCA$biplot)

wa <- as.data.frame(vare.cca$CCA$v) %>% 
  rownames_to_column("ID")
seq <- wa
f <- as.data.frame(physeq@tax_table@.Data) %>% 
  select("phylum") %>% 
  rownames_to_column("ID")

wa <- full_join(wa, f, by="ID") %>% 
  select(-ID) %>% 
  rename("Family" = "Label") %>% 
  mutate(Score = "sites")

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))
biplot

fdat_amazing <- plyr::rbind.fill(biplot, wa) %>% 
  mutate(Label = as.factor(Label))
fdat_amazing
 
p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), aes(x = 0, xend = CCA1, y = 0, yend = CCA2), alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
  geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=4) + 
  theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
  # geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
  return(p)
}


kate.ggcca.species(cca_w_varstab_asv, physeq) 

cca_w_varstab_family$CCA
ps.varstab@tax_table@.Data

ggsave(file="plots/cca-genus.tiff",
        device="tiff",
        plot=p, 
        width=2300, 
        height=2400,
        units ="px", 
        dpi=300,
        compression = "lzw"
        )

```


```{r}

anova(cca_w_varstab_asv, parallel = 10)
anova(cca_w_varstab_asv, by="terms", parallel = 10)
anova(cca_w_varstab_asv, by="mar", parallel = 10)
vif.cca(cca_w_varstab_asv)

```

#### Family level with phylum color

```{r, fig.height=9, fig.width=9}

ps.ff <-  filter_taxa(ps.f, function(x) sum(x > 3) > (0.1*length(x)), TRUE)
ps.varstab <- ps_vst(ps.ff)
ps.varstab.merged <- phyloseq::merge_samples(ps.varstab, "Repeat")
ps.varstab.merged.family <- tax_glom(ps.varstab.merged, taxrank="Family")
ps.varstab.merged@sam_data
physeq <- ps.varstab.merged.family
physeq <- ps.varstab.merged

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}


otus.ps.vegan <- veganifyOTU(physeq)

metadata <- as(sample_data(physeq), "data.frame") 

metadata

# cca_w_varstab_asv <- vegan::cca(otus.ps.vegan ~  Humidity + pH + C_org + C +  K + NH4 + NO3 + temp,  data=metadata)
cca_w_varstab_asv <- vegan::cca(otus.ps.vegan ~  pH + C_org + NH4 + temp,  data=metadata)
cca_w_varstab_asv <- vegan::cca(otus.ps.vegan ~  pH + C_org + C +  K + NH4 + NO3 + temp + P,  data=metadata)



wa <- as.data.frame(cca_w_varstab_asv$CCA$v) %>% 
  rownames_to_column("ID")

taxa.pruned <- as.data.frame(physeq@tax_table@.Data) %>% 
  rownames_to_column("ID")
taxa.pruned
taxa.pruned <- taxa.pruned %>%  mutate_all(as.character)
# old.tips <- tree$tip.label
# matches <- regmatches(tree$tip.label, gregexpr("[[:digit:]]+", tree$tip.label))
# taxa.pruned$number <- as.numeric(unlist(matches))
taxa.pruned$phylum_pro <- ifelse(taxa.pruned$Phylum == "Proteobacteria", 
                                 with(taxa.pruned, paste0(taxa.pruned$Class)), 
                                 with(taxa.pruned, paste0(taxa.pruned$Phylum)))
taxa.pruned$Label <- paste0(taxa.pruned$Family)

wa <- full_join(wa, taxa.pruned, by="ID") %>% 
  mutate(Score = "sites")
wa

sw <- summarize_all(as.data.frame(ps.varstab@otu_table), sum)
head_asv <- as_tibble(t(sw), rownames = "ID") %>% 
  arrange(desc(V1)) %>% top_n(n = 50) %>% 
  pull(ID)

wa <- filter(wa, ID %in% head_asv)

wa

biplot <- as.data.frame(cca_w_varstab_asv$CCA$biplot)
biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))


fdat_amazing <- plyr::rbind.fill(biplot, wa) %>% 
  mutate(Label = as.factor(Label))
fdat_amazing

p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2)) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), 
               aes(x = 0, xend = CCA1, y = 0, yend = CCA2), 
               alpha=0.8, 
               color = "red",
               arrow = arrow(angle = 3)) +
  geom_text_repel(data = fdat_amazing %>% dplyr::filter(Score == "sites"), 
                  aes(x=CCA1, y=CCA2, label= Label, colour = phylum_pro),size=4) + 
  geom_text_repel(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), 
                  aes(x=CCA1, y=CCA2, label= Label),size=4) +
  theme(legend.position = "top", 
        panel.background = element_rect(fill = "white", colour = "grey50"))
  # geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
  
p2 <- p + scale_color_manual(values=c("#5EA09E", "#6B440B", "#F499C2", "#7DA9D8", "#FEF898", "#0000FE", "#F27204", 
                                "#FE0000", "#FEFF00", "#02F30E", "#F89679", "black","salmon"))
p2
p


# select(wa, -c("ID", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species", ))
wa

kate.ggcca.species <- function(vare.cca, physeq){

require(tidyverse)
  
biplot <- as.data.frame(vare.cca$CCA$biplot)

wa <- as.data.frame(vare.cca$CCA$v) %>% 
  rownames_to_column("ID")
seq <- wa
f <- as.data.frame(physeq@tax_table@.Data) %>% 
  select("phylum") %>% 
  rownames_to_column("ID")

wa <- full_join(wa, f, by="ID") %>% 
  select(-ID) %>% 
  rename("Family" = "Label") %>% 
  mutate(Score = "sites")

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))
biplot

fdat_amazing <- plyr::rbind.fill(biplot, wa) %>% 
  mutate(Label = as.factor(Label))
fdat_amazing
 
p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), aes(x = 0, xend = CCA1, y = 0, yend = CCA2), alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
  geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=4) + 
  theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
  # geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
  return(p)
}


kate.ggcca.species(cca_w_varstab_family, physeq) 
cca_w_varstab_family$CCA

pd.ff

```


#### CCA fuck my claster

```{r echo=TRUE, fig.height=9, fig.width=9, message=FALSE}


wa <- as.data.frame(cca_w_varstab_asv$CCA$v) %>% 
  rownames_to_column("ID")

taxa.pruned <- as.data.frame(ps.varstab@tax_table@.Data) %>% 
  rownames_to_column("ID")
taxa.pruned <- taxa.pruned %>%  mutate_all(as.character)
taxa.pruned$phylum_pro <- ifelse(taxa.pruned$Phylum == "Proteobacteria", with(taxa.pruned, paste0(taxa.pruned$Class)), with(taxa.pruned, paste0(taxa.pruned$Phylum)))

wa <- full_join(wa, taxa.pruned, by="ID") %>% 
  mutate(Score = "sites") %>% rename("ID" = "Label")
wa

biplot <- as.data.frame(cca_w_varstab_asv$CCA$biplot)
biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))


fdat_amazing <- plyr::rbind.fill(biplot, wa) %>% 
  mutate(Label = as.factor(Label))
fdat_amazing

p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, color = phylum_pro)) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), aes(x = 0, xend = CCA1, y = 0, yend = CCA2), alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
  geom_label(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), mapping = aes(x=CCA1, y=CCA2, label=Label)) +
  theme(legend.position = "top", panel.background = element_rect(fill = "white", colour = "grey50")) 
  # geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
  
p


```


### pls-da

run baby, lets rock this ordination with cross validation

#### pre processing

http://mixomics.org/mixmc/mixmc-pre-processing/



```{r, fig.height=9, fig.width=12}

library(mixOmics)

ps.ff

Tax <- tax_table(ps.ff)
Metadata <- ps.ff@sam_data
data <- otu_table(ps.ff)
data <- data + 1



pca.result <- pca(data, logratio = 'CLR')
plotIndiv(pca.result, group = Metadata$What, title = 'PCA plot with CLR by MixOmix', legend = TRUE)
Metadata


```

#### PLS-DA

```{r}

library(mixOmics)

X <- as.matrix(data)
Y <- as.factor(Metadata$Repeat)

plsda.res <- splsda(X, Y, ncomp = 15, keepX = 15)

perf.plsda <- perf(plsda.res, validation = "Mfold", folds = 4, 
                  progressBar = TRUE, auc = TRUE, nrepeat = 50, cpus = 15) 



```

```{r, fig.height=9, fig.width=9}

plotLoadings(plsda.res, comp = 2, method = 'mean', contrib = 'max', size.title = 1)

```

```{r, fig.height=7, fig.width=15}

name.var = paste(Tax[, 'Genus'], rownames(data), sep = '|')

plotLoadings(plsda.res, comp = 10, method = 'mean', contrib = 'max', ndisplay = 60, name.var = name.var, size.title = 1, size.name = 0.6, size.legend = 1)
?plotLoadings
```




```{r}

cim(plsda.res, row.sideColors = color.mixo(Y), threshold = 0.5)
?cim




```
```{r}

plot(plsda.res)

```


```{r, fig.height=9, fig.width=9}

plotIndiv(plsda.res, group = Metadata$Repeat, title = 'splsda', legend = TRUE, comp = c(1,10))

```



```{r}

grid.keepX = c(seq(5,1280, 5)) 
tune.splsda <- tune.splsda(X, Y, ncomp = 30, validation = 'Mfold', folds = 4, logratio = 'CLR',
                           progressBar = TRUE, dist = 'max.dist', nrepeat = 100, cpus = 10) 

choice.ncomp <- tune.splsda$choice.ncomp$ncomp
choice.keepX <- tune.splsda$choice.keepX[1:choice.ncomp]

plsda.res <- splsda(X, Y, ncomp = choice.ncomp, keepX = choice.keepX)

perf.plsda <- perf(plsda.res, validation = "Mfold", folds = 4, 
                  progressBar = TRUE, auc = TRUE, nrepeat = 100, cpus = 10) 


plot(perf.plsda, col = color.mixo(1:3), sd = TRUE, legend.position = "horizontal")
plot(perf.plsda)

perf.plsda
tune.splsda$choice.keepX
tune.splsda$choice.ncomp
```



```{r}


tune.splsda.what <- tune.splsda(X, Y, ncomp = 30, validation = 'Mfold', folds = 4, logratio = 'CLR',
                           progressBar = TRUE, dist = 'max.dist', nrepeat = 100, cpus = 10) 

choice.ncomp.what <- tune.splsda.what$choice.ncomp$ncomp
choice.keepX.what <- tune.splsda.what$choice.keepX[1:choice.ncomp]

plsda.res.what <- splsda(X, Y, ncomp = choice.ncomp.what, keepX = choice.keepX.what)

perf.plsda.what <- perf(plsda.res.what, validation = "Mfold", folds = 4, 
                  progressBar = TRUE, auc = TRUE, nrepeat = 100, cpus = 10) 



saveRDS(tune.splsda.what, file = "tune.splsda.what")
saveRDS(plsda.res.what, file = "plsda.res.what")
saveRDS(perf.plsda.what, file = "perf.plsda.what")


<!-- # plot(perf.plsda, col = color.mixo(1:3), sd = TRUE, legend.position = "horizontal") -->
<!-- # plot(perf.plsda) -->

<!-- # perf.plsda -->
<!-- # tune.splsda$choice.keepX -->
<!-- # tune.splsda$choice.ncomp -->

```


### PERMANOVA for all

```{r, echo=TRUE, rows.print = 112}

permanova.forloop.noNestedPlant <- function(factor){
  require(vegan)
  require(phyloseq)
  dist <- phyloseq::distance(physeq, "bray")
  metadata <- as(sample_data(physeq@sam_data), "data.frame")
  ad <- adonis2(as.formula(paste( "dist ~ ", factor)), data = metadata)
  ad <- cbind2(ad[1,][3], ad[1,][5])
  return(ad)
}

physeq <- ps.varstab
permanova.forloop.pos.noPlant <- possibly(permanova.forloop.noNestedPlant, otherwise = NA)
out <- map(colnames(physeq@sam_data[, 3:length(colnames(physeq@sam_data))]), permanova.forloop.pos.noPlant)
out.d.noPlant <- do.call("rbind", out)
out.d.noPlant

```

```{r}

cca_w_varstab_family$CCA$v["Seq10", ]
length(colnames(otus.ps.vegan))
length(rownames(cca_w_varstab_family$CCA$v))
anova(cca_w_varstab_family, parallel = 10)
anova(cca_w_varstab_family, by="terms", parallel = 10)
anova(cca_w_varstab_family, by="mar", parallel = 10)
vif.cca(cca_w_varstab_family)

```


### Alpha diversity violins

```{r, message=FALSE,echo=TRUE,fig.height=12, fig.width=20}

library(ggpubr)

ps.f <- subset_samples(ps.f, sample_names(ps.f) != 'N_31')
ps.f <- subset_samples(ps.f, sample_names(ps.f) != 'N_70')

alpha.custom <- function(ps.f, arrga = "PD"){
  
  require(picante)
  
  pd <- pd(ps.f@otu_table@.Data, ps.f@phy_tree)
  
  pd1 <- estimate_richness(ps.f, measures=c("Observed", "InvSimpson", "Shannon"))
  pd <- cbind(pd, ps.f@sam_data, pd1 )
  
  bmi <- levels(as.factor(pd$Site))
  bmi.pairs <- combn(seq_along(bmi), 2, simplify = FALSE, FUN = function(i)bmi[i])
  p1 <- ggviolin(pd, x = "Site", y = arrga,
                 add = "boxplot", fill = "Site") + stat_compare_means(comparisons = bmi.pairs, method = "wilcox.test") +
    scale_x_discrete(limits = c("N1", "N2", "N3", "N4", "N5", "N6", "N7")) + theme(axis.title.x = element_blank(), legend.title = element_blank())
  return(p1)
}

p.alpha.Cdt.oo <- alpha.custom(ps.f, arrga = "Observed")
p.alpha.Cdt.sh <- alpha.custom(ps.f, arrga = "PD")
p.alpha.Cdt.is <- alpha.custom(ps.f, arrga = "Shannon")
p.alpha.Cdt.pd <- alpha.custom(ps.f, arrga = "InvSimpson")

p1 <- ggarrange(p.alpha.Cdt.oo, p.alpha.Cdt.sh,p.alpha.Cdt.is, p.alpha.Cdt.pd , ncol = 4 ,label.x = 0.105, nrow = 1, common.legend = TRUE)
p1

``` 


```{r, message=FALSE,echo=TRUE, fig.height=8, fig.width=12}

library(phyloseq)
library(ggplot2)

plot_rich_reads_samlenames_lm <- function(physeq){
  rish <- estimate_richness(physeq, measures = "Observed")
  reads.sum <- as.data.frame(sample_sums(physeq))
  reads.summary <- cbind(rish, reads.sum)
  colnames(reads.summary) <- c("otus","reads")
  reads.summary["Repeat"] <-unlist(purrr::map(stringr::str_split(rownames(physeq@sam_data), "N_", 2), function(x) x[[2]]))
  reads.summary["Site"] <- physeq@sam_data$Site
  library(ggrepel)
  require(ggforce)
  p1 <- ggplot(data=reads.summary) + geom_point(aes(y=otus, x=log2(reads), color=Site),size=3) + geom_text_repel(aes(y=otus, x=log2(reads), label=paste0(Site, "_", Repeat))) + theme_bw()+
    geom_smooth(aes(y=otus, x=log2(reads), fill=Site, color=Site),method=lm, se=FALSE, ymin = 1) + scale_x_continuous(sec.axis = sec_axis(sec.axis ~ 2**.)) 
  # geom_mark_ellipse(aes(y = otus, x=reads, group = Repeats, label = Repeats, color = Repeats), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
  
  return(p1)
}

plot_rich_reads_samlenames_lm(ps_f)

ps_f2 <- ps_f
 #ps.f2 <- subset_samples(ps.f, sample_names(ps.f) != 'N_95')
 # ps.f2 <- subset_samples(ps.f2, sample_names(ps.f2) != 'N_96')
 # ps.f2 <- subset_samples(ps.f2, sample_names(ps.f2) != 'N_89')
ps_f2 <- subset_samples(ps_f2, sample_names(ps_f2) != 'N_31')
ps_f2 <- subset_samples(ps_f2, sample_names(ps_f2) != 'N_70')
 # ps.f2 <- subset_samples(ps.f2, Site != 'N8-2')
 # ps.f2 <- subset_samples(ps.f2, Site != 'N8-3')
ps_f2  <- prune_taxa(taxa_sums(ps_f2) > 0, ps_f2)
plot_rich_reads_samlenames_lm(ps_f2)

 # sample_names(ps_f2)
ps_f2
ps_f
 # ps.test <- subset_samples(ps.f2, Site == 'N3-1')
plot_rich_reads_samlenames_lm(ps_f2_2)

```



```{r}
map3 <- data.frame(ps_f2@sam_data)
sam_data(ps_f2) <- map3 %>% 
  rownames_to_column("ID") %>% 
    mutate(
    Soil = case_when(
      grepl('meadow_humus', Soil) ~ 'Umbric Gleyic',
      grepl('Carbonate_chernozem', Soil) ~ 'Calcareous Chernozem'
    )
  ) %>% 
    mutate(
    Quarry_or_Benchmark = case_when(
      grepl('Quarry', What) ~ 'Disturbed_soil',
      grepl('Benchmark', What) ~ 'Control'
    ) 
    )%>% 
    mutate(
    TH = Humidity/Temperature
    )  %>% 
      column_to_rownames("ID")


ps_f2@sam_data
ps.varstab <- ps_vst(ps_f2)
ordination.b <- ordinate(ps.varstab, "NMDS", "bray")

p1 <- plot_ordination(ps.varstab, ordination.b, type="sample", color="Soil", title="Soil types", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    theme(legend.position="bottom", legend.title = element_blank(), legend.text=element_text(size=9))

p2 <- plot_ordination(ps.varstab, ordination.b, type="sample", color="Quarry_or_Benchmark", title="Disturbed soil or control", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    theme(legend.position="bottom", legend.title = element_blank(), legend.text=element_text(size=9))

ggarrange(p1, p2, nrow = 1, ncol = 2)

pH + Temperature + Humidity + P_mg.kg + K_mg.kg + NH4_mg.kg + NO3_mg.kg

```

### Plot of geographic map

```{r}

library(rnaturalearth)
library(rnaturalearthdata)
library(tidyverse)
library(sf)
library(rgeos)
library(ggrepel)

world <- ne_countries(scale = "large", returnclass = "sf")
spdf_france_geounit <- ne_states(geounit = 'russia', returnclass = "sf")
probdata = tibble(site = c("Quarry on Chernozem", "Quarry on Gleyic"),
                  longitude = c(43.51137, 43.77252), 
                  latitude =  c(43.82965, 43.35022), 
                  xlabel = c(40, 47), 
                  ylabel = c(44.3, 43))

<!-- # rivers <- ne_download(scale = 10, type = 'rivers_lake_centerlines', category = 'physical', returnclass = "sf") -->
<!-- # lakes <- ne_download(scale = 50, type = 'lakes', category = 'physical', returnclass = "sf") -->

 ggplot(data = world) + 
  geom_sf() +
  geom_sf(data = spdf_france_geounit, fill="green",  alpha = 0.1 ) +
  # geom_sf(data = lakes ) +
  coord_sf(xlim = c(35, 50), ylim = c(40, 50), expand = FALSE) +
  geom_point(data = probdata, aes(x = longitude, y = latitude), size = 3, 
  shape = 20, colour = "red") +
  geom_text(data = probdata, aes(x = xlabel, y = ylabel, label = site), size = 5, colour="red") + 
  theme_bw() 

```


### Heterogenity 

```{r,message=FALSE,echo=TRUE, fig.height=5, fig.width=5.8}

dist <- phyloseq::distance(ps.f, method = "bray")
groups <- ps.f@sam_data$Region
mod <- vegan::betadisper(dist, groups)
mod.box <- boxplot(mod)
TukeyHSD(mod)
?TukeyHSD.betadisper
permutest(mod, permutations = 999)
anova(mod)
plot(TukeyHSD(mod))


```


```{r, message=FALSE,echo=TRUE, fig.height=7, fig.width=7}

library(phyloseq)             
library(vegan)

ps.f <- ps_f2_2
ps.merged <- merge_samples(ps.f, "Site")
physeq <- ps.merged

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}


otus.ps.vegan <- veganifyOTU(physeq)

metadata <- as(sample_data(physeq), "data.frame") 


vare.cca <- vegan::cca(otus.ps.vegan ~  pH + scaled_temp + P_mg.kg + K_mg.kg + NH4_mg.kg + NO3_mg.kg, data=metadata)


kate.ggcca.sites <- function(vare.cca){

require(tidyverse)
biplot <- as.data.frame(vare.cca$CCA$biplot)
wa <- as.data.frame(vare.cca$CCA$wa)

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))
wa <- rownames_to_column(wa, "Label") %>% 
  add_column(Score = rep("sites", length(rownames(wa))))
fdat_amazing <- rbind(biplot, wa)

 
p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), aes(x = 0, xend = CCA1, y = 0, yend = CCA2), alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
  geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=4) + 
  theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
  return(p)
}

kate.ggcca.sites(vare.cca)

```


```{r}

library(re)

M <- cor(met)
corrplot(met, method = "circle")

metadata %>% mutate(So)

met <- ps_f2_2@sam_data
dummy <- as.numeric(year == 1957)
dummy <- as.numeric(year == 1957)
map3 <- data.frame(ps_f2@sam_data)
met <- map3 %>% mutate(Soil_d = ifelse(Soil == "Umbric Gleyic", 1, 0)) %>%  mutate(Soil_d = ifelse(Soil == "Umbric Gleyic", 1, 0))
met
cor.test(met$Temperature, met$Soil_d)
cor.test(met$pH, met$Soil_d)
cor.test(met$scaled_temp, met$Humidity)
cor.test(met$Temperature, met$Humidity)

ps.nc <- prune_samples(sample_data(ps.varstab)$Soil %in% c("Umbric Gleyic"), ps.varstab)
map3 <- data.frame(ps.nc@sam_data) %>%  mutate(temp = scale(Temperature))
ps.nc <- merge_phyloseq(ps.nc, sample_data(map3))


ps.nc2 <- prune_samples(! sample_data(ps.varstab)$Soil %in% c("Umbric Gleyic"), ps.varstab)
map3 <- data.frame(ps.nc2@sam_data) %>%  mutate(temp = scale(Temperature))
ps.nc2 <- merge_phyloseq(ps.nc2, sample_data(map3))

ps_f2_2 <- merge_phyloseq(ps.nc, ps.nc2)

ps_f2_2@sam_data

ps.nc <- prune_samples(! sample_data(ps.f2)$Soil %in% c("Calcareous Chernozem"), ps_f2)
ps.nc  <- prune_taxa(taxa_sums(ps.nc) > 0, ps.nc)


filter()

```


```{r}

map_f3 <- data.frame(ps_f2_2@sam_data)
map_f3 <- map_f3 %>%  mutate_if(sapply(map_f3, is.character), as.factor)
ps.nc <- merge_phyloseq(ps_f2_2, sample_data(map_f3))
ps.merged <- phyloseq::merge_samples(ps.nc, "Site")


mutate_if(sapply(iris_char, is.character), as.factor)
sample_data(ps.nc)$Soil = factor(sample_data(ps.nc)$Soil)
sample_data(ps.nc)$Quarry_or_Benchmark = factor(sample_data(ps.nc)$Quarry_or_Benchmark)
sample_data(ps.nc)$Feature = factor(sample_data(ps.nc)$Feature)
ps.merged <- phyloseq::merge_samples(ps.nc, "Site")

sample_data(ps.merged)$Soil <- factor(sample_data(ps.merged)$Soil, labels=levels(sample_data(ps.nc)$Soil))
sample_data(ps.merged)$Quarry_or_Benchmark <-  factor(sample_data(ps.merged)$Quarry_or_Benchmark, labels=levels(sample_data(ps.nc)$Quarry_or_Benchmark))
sample_data(ps.merged)$Feature <- factor(sample_data(ps.merged)$Feature, labels=levels(sample_data(ps.nc)$Feature))

sample_data(ps.merged)
factor(sample_data(ps.merged)$Soil, levels = levels(sample_data(ps.nc)$Soil))
sample_data(c)$Soil

map_m <- data.frame(ps.merged@sam_data)


ggplot(data=map_m, aes(Soil, temp)) + geom_boxplot(alpha=0.5) + geom_point(alpha=0.5) + theme_bw()
ggplot(data=map_m, aes(Soil, Temperature)) + geom_boxplot(alpha=0.5) + geom_point(alpha=0.5) + theme_bw()
ggplot(data=map_m, aes(Soil, scale(Temperature))) + geom_bo
xplot(alpha=0.5) + geom_point(alpha=0.5) + theme_bw()

```


### CCA statistics

```{r}

anova(vare.cca)

```

#### axis

```{r}

anova(vare.cca, by="terms")

```

#### multicolleniarity 

```{r}

vif.cca(vare.cca)

```



#### margins
```{r}

anova(vare.cca, by="mar")

```

```{r, message=FALSE,echo=TRUE, fig.height=7, fig.width=7}

library(phyloseq)             
library(vegan)

ps.f <- ps_f2_2
ps.merged <- merge_samples(ps.f, "Site")
physeq <- ps.merged

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}


otus.ps.vegan <- veganifyOTU(physeq)

metadata <- as(sample_data(physeq), "data.frame") 


rda <- vegan::rda(otus.ps.vegan ~  pH + scaled_temp + P_mg.kg + K_mg.kg + NH4_mg.kg + NO3_mg.kg, data=metadata)


kate.ggcca.sites <- function(vare.cca){

require(tidyverse)
biplot <- as.data.frame(vare.cca$CCA$biplot)
wa <- as.data.frame(vare.cca$CCA$wa)

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))
wa <- rownames_to_column(wa, "Label") %>% 
  add_column(Score = rep("sites", length(rownames(wa))))
fdat_amazing <- rbind(biplot, wa)

 
p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), aes(x = 0, xend = CCA1, y = 0, yend = CCA2), alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
  geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=4) + 
  theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
  return(p)
}

kate.ggcca.sites(rda)

metadata

```


### NMDS of filtered phyloseq

```{r, fig.height=10, fig.width=12}

ordination.b <- ordinate(ps.f, "PCoA", "bray")

  #plotting
plot_ordination(ps.f, ordination.b, type="sample", color="Site", title="NMDS - Bray-Curtis", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))

```


```{r, fig.height=6, fig.width=7}
plot_ordination(ps_f2_2, ordination.b, type="sample", color="temp", title="NMDS - Bray-Curtis", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
```


```{r, fig.height=6, fig.width=7}
plot_ordination(ps_f2_2, ordination.b, type="sample", color="K_mg.kg", title="NMDS - Bray-Curtis", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
```

```{r, fig.height=6, fig.width=7}
plot_ordination(ps_f2_2, ordination.b, type="sample", color="pH", title="NMDS - Bray-Curtis", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
```

```{r, fig.height=6, fig.width=7}
plot_ordination(ps_f2, ordination.b, type="sample", color="What", title="NMDS - Bray-Curtis", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
```

```{r}

ps_f2_2@sam_data

```

```{r}
ps.merged@sam_data
```

```{r, fig.height=6, fig.width=8}
ps.merged@sam_data

phyloseq_to_amp <- function(ps){
    require(ampvis2)
    require(tibble)
    require(phyloseq)
    colnames(ps@tax_table@.Data) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
    OTU1 = as(otu_table(ps), "matrix")
    OTU1 <- t(OTU1)
    OTUdf = as.data.frame(OTU1)
    taxa.ps <- as(tax_table(ps), "matrix")
    taxa.df = as.data.frame(taxa.ps)
    my_otu_table <- merge(OTUdf, taxa.df, by=0)
    my_otu_table <- column_to_rownames(my_otu_table, var="Row.names")

    my_metadata <- as_tibble(sample_data(ps), rownames=NA)
    my_metadata <- rownames_to_column(my_metadata,var = "SampleID")
    my_tree <- phy_tree(ps)
    amp.ps <- amp_load(otutable = my_otu_table, metadata = my_metadata, tree = my_tree)
    return(amp.ps)
}

ps.n4 <- prune_samples(sample_data(ps_f2_2)$Site %in% c("N4-1", 	"N4-2"), ps_f2_2)
ps.n4  <- prune_taxa(taxa_sums(ps.n4) > 0, ps.n4)

amp.n4 <- phyloseq_to_amp(ps.n4)
ps.n4@otu_table[,1]
amp_heatmap(amp.n4, tax_show = 40, group_by = "Site", tax_aggregate = "Genus", tax_class = "Proteobacteria") + theme_bw() + theme(text = element_text(size=15), legend.position = "none") + theme(axis.text.x=element_text(angle=45,hjust=1))


amp.f <- phyloseq_to_amp(ps.f)

amp_heatmap(amp.f, tax_show = 15, group_by = "Site", tax_aggregate = "Phylum", tax_class = "Proteobacteria") + theme_bw() + theme(text = element_text(size=15), legend.position = "none") + theme(axis.text.x=element_text(angle=45,hjust=1))

```



```{r}

ps.n4 <- prune_samples(sample_data(ps_f2_2)$Site %in% c("N4-1", 	"N4-2"), ps.f2)
ps.n4  <- prune_taxa(taxa_sums(ps.n4) > 0, ps.n4)

ps.f2

```


```{r}

des_w_simper <- function(ps, factor){
  require(DESeq2)
  require(vegan)
  require(tibble)
  require(phyloseq)
  
  diagdds = phyloseq_to_deseq2(ps, as.formula(paste( "~", factor)))                  
  diagdds = estimateSizeFactors(diagdds, type="poscounts")
  diagdds = estimateDispersions(diagdds, fitType = "local") 
  diagdds = DESeq(diagdds)
  samp <-sample_data(ps)
  aggdata <- t(aggregate.data.frame(as.data.frame(as.data.frame(t(diagdds@assays@data$mu))), by=list(samp[[factor]]), median))
  colnames(aggdata) <- aggdata[1,]
  aggdata <- aggdata[-1,]
  res = results(diagdds)
  res.df <- as.data.frame(res)
  nice <- cbind(res.df, as.data.frame(ps@tax_table@.Data[rownames(res.df),]), as.data.frame(aggdata)[rownames(res.df),])
} 


ds.n4 <- des_w_simper(ps.n4, "Site")

ds.n4 %>% arrange(padj)


library(vegan)
physeq <- ps.n4
dist <- phyloseq::distance(physeq, "bray")
metadata <- as(sample_data(physeq@sam_data), "data.frame")
ad <- adonis(dist ~ Site, data = metadata, permutations = 9999)

ad

```

```{r,  fig.height=10, fig.width=12}

ps.sseno <- prune_samples(sample_data(ps.f2)$Feature %in% "cane_bush", ps.f2)
ps.sseno  <- prune_taxa(taxa_sums(ps.sseno) > 0, ps.sseno)

ps.first <- prune_samples(sample_data(ps.f2)$Soil %in% "Umbric Gleyic", ps.f2)
ps.first <- prune_taxa(taxa_sums(ps.first) > 0, ps.first)

amp.sseno <- phyloseq_to_amp(ps.first)

amp_heatmap(amp.sseno, tax_show = 30, group_by = "Site", tax_aggregate = "Family", tax_class = "Proteobacteria", tax_add = "Phylum") + theme_bw() + theme(text = element_text(size=15), legend.position = "none") + theme(axis.text.x=element_text(angle=45,hjust=1))


```


###Venn

```{r, message=FALSE,echo=TRUE, fig.height=9, fig.width=9}
library(ggVennDiagram)

ps_f_family <- tax_glom(ps_f, taxrank="Family")

get_list_from_ps <- function(physeq, amaz_fac) {
  my_keys <- levels(sample_data(physeq)[[amaz_fac]])
  mylist <- vector(mode="list", length=length(my_keys))
  names(mylist) <- my_keys
  for (i in levels(sample_data(physeq)[[amaz_fac]])) { 
    x <- prune_samples(sample_data(physeq)[[amaz_fac]] %in% i, physeq)
    x <- prune_taxa(taxa_sums(x) > 0, x)
    mylist[[i]] <- taxa_names(x)
  }
  return(mylist)
}

ps_1.3 <- prune_samples(sample_data(ps.f)$Site %in% c("N1", "N3"), ps.f)
ps_1.3  <- prune_taxa(taxa_sums(ps_1.3) > 0, ps_1.3)
physeq <- ps_1.3

ps_5.6.7 <- prune_samples(sample_data(ps.f)$Site %in% c("N5", "N6", "N7"), ps.f)
ps_5.6.7  <- prune_taxa(taxa_sums(ps_1.3) > 0, ps_5.6.7)
physeq <- ps_5.6.7

physeq <- ps.f


library(ggVennDiagram)
?ggVennDiagram
ggVennDiagram(get_list_from_ps(ps_f_family, "Soil")) + 
  ggplot2::scale_fill_viridis_c(option = "D", begin=0.2)

ps.f@sam_data

ggVennDiagram(get_list_from_ps(physeq, "What")) + 
  ggplot2::scale_fill_viridis_c(option = "D", begin=0.2)

sample_data(ps) <- map_m
ps.f@sam_data



```

### some tree

```{r,fig.height=9, fig.width=8, results=FALSE}

# rooted_tree <- read_tree(treefile="rooted.tree")
# ps.f.rooted <- ps.f
# 
# ps.f.rooted@phy_tree <- rooted_tree
# plot_tree(ps.f, nodelabf=nodeplotboot(), ladderize="left", color="Phylum")


ps.merged <- phyloseq::merge_samples(ps.f, "Repeat")

sw <- summarize_all(as.data.frame(ps.merged@otu_table), sum)
head_asv <- as_tibble(t(sw), rownames = "ID") %>% 
  arrange(desc(V1)) %>% top_n(n = 100) %>% 
  pull(ID)
head_asv

ps.top <- subset_taxa(ps.merged, taxa_names(ps.f) %in% head_asv)


plot_tree(ps.top,
          # nodelabf=nodeplotboot(),
          ladderize="left",
          color="What", 
          label.tips="Genus",
          # size="abundance",
          # sizebase=5,
          base.spacing=0.015)

rank_names(ps.top)

```

### philR
#### Progress

```{r, fig.height=8, fig.width=24}

library(philr)
library(glmnet)

GP <- ps.cc
GP.f <-  phyloseq::filter_taxa(GP, function(x) sum(x > 3) > (0.1*length(x)), TRUE)
GP <- GP.f
phy_tree(GP) <- makeNodeLabel(phy_tree(GP), method="number", prefix='n')
GP <- transform_sample_counts(GP, function(x) x+1)

otu.table <- otu_table(GP)
tree <- phy_tree(GP)
metadata <- sample_data(GP)
tax <- tax_table(GP)
# is.binary(tree)
# is.rooted(tree)

gp.philr <- philr(otu.table, tree, 
                  part.weights='enorm.x.gm.counts', 
                  ilr.weights='blw.sqrt')

gp.dist <- dist(gp.philr, method="euclidean")
glmmod <- glmnet(gp.philr, sample_data(GP)$Quarry_or_Benchmark , alpha=1, family="binomial")

cv.res <- cv.glmnet(gp.philr, as.double(sample_data(GP)$Quarry_or_Benchmark), nfolds = 3, family = "binomial", type.measure = "mse")

top.coords <- as.matrix(coefficients(glmmod, cv.res$lambda.min))
top.coords <- rownames(top.coords)[which(top.coords != 0)]
top.coords <- top.coords[2:length(top.coords)]
tc.names <- sapply(top.coords, function(x) name.balance(tree, tax, x))
tc.names

```


```{r}

gp.philr.long <- convert_to_long(gp.philr, get_variable(GP, 'Quarry_or_Benchmark')) %>%
  filter(coord %in% top.coords)

# gp.philr.long$taxa <- paste0("n",str_sub(gp.philr.long$coord, 2)) %>% 
#   as.factor() %>% 
#   recode(!!!tc.names)

balance_taxa <- levels(gp.philr.long$taxa)

library(naturalsort)
ggplot(gp.philr.long, aes(x=labels, y=value)) +
  geom_boxplot(fill='lightgrey') +
  facet_grid(.~coord, scales='free_x') +
  xlab('Quarry or Benchmark') + ylab('Balance Value') +
  theme_bw() 

```
```{r}
top.coords
```

```{r}

tc.nn

```

```{r, fig.height=7, fig.width=7}

plot_philr <- function(){
  
  tc.nn <- name.to.nn(tree, top.coords)
  tc.colors <- scales::viridis_pal()(length(top.coords))
  p <- ggtree(tree, layout='fan')
  top.seq <- seq(1, length(top.coords))
  
  for (i in top.seq){
    p <- p + geom_balance(node=tc.nn[i], fill=tc.colors[i], alpha=0.6)
  }
  
  for (i in top.coords){
    p <- annotate_balance(tree, i, p=p,
                          labels = c(paste0(i, "+"),
                                     paste0(i, "-")),
                   offset.text=0.15,
                   bar=FALSE)
  }
  
  p
  
}   

plot_philr()

```

```{r}

votes <- name.balance(tree, tax, 'n583', return.votes = c('up', 'down'))
message("up.votes")
sort(votes[[c('up.votes', 'Order')]])
message("down.votes")
sort(votes[[c('down.votes', 'Order')]])

```

#### Urvan

```{r, fig.height=8, fig.width=24}

library(philr)
library(glmnet)

GP <- ps.nc
GP.f <-  phyloseq::filter_taxa(GP, function(x) sum(x > 3) > (0.1*length(x)), TRUE)
GP <- GP.f
phy_tree(GP) <- makeNodeLabel(phy_tree(GP), method="number", prefix='n')
GP <- transform_sample_counts(GP, function(x) x+1)

otu.table <- otu_table(GP)
tree <- phy_tree(GP)
metadata <- sample_data(GP)
tax <- tax_table(GP)
# is.binary(tree)
# is.rooted(tree)

gp.philr <- philr(otu.table, tree, 
                  part.weights='enorm.x.gm.counts', 
                  ilr.weights='blw.sqrt')

gp.dist <- dist(gp.philr, method="euclidean")
glmmod <- glmnet(gp.philr, sample_data(GP)$Quarry_or_Benchmark , alpha=1, family="binomial")

cv.res <- cv.glmnet(gp.philr, as.double(sample_data(GP)$Quarry_or_Benchmark), nfolds = 3, family = "binomial", type.measure = "mse")

top.coords <- as.matrix(coefficients(glmmod, cv.res$lambda.min))
top.coords <- rownames(top.coords)[which(top.coords != 0)]
top.coords <- top.coords[2:length(top.coords)]
tc.names <- sapply(top.coords, function(x) name.balance(tree, tax, x))
tc.names

```

```{r}

amp <- phyloseq_to_amp(GP)
amp_heatmap(amp, tax_show = 20, group_by = "Quarry_or_Benchmark", tax_aggregate = "Class", tax_class = "Proteobacteria", tax_add = "Phylum") + theme_bw() + theme(text = element_text(size=15), legend.position = "none") + theme(axis.text.x=element_text(angle=45,hjust=1))

```

```{r}

gp.philr.long <- convert_to_long(gp.philr, get_variable(GP, 'Quarry_or_Benchmark')) %>%
  filter(coord %in% top.coords)

# gp.philr.long$taxa <- paste0("n",str_sub(gp.philr.long$coord, 2)) %>% 
#   as.factor() %>% 
#   recode(!!!tc.names)

balance_taxa <- levels(gp.philr.long$taxa)

library(naturalsort)
ggplot(gp.philr.long, aes(x=labels, y=value)) +
  geom_boxplot(fill='lightgrey') +
  facet_grid(.~coord, scales='free_x') +
  xlab('Quarry or Benchmark') + ylab('Balance Value') +
  theme_bw() 

```


```{r, fig.height=7, fig.width=7}

plot_philr()

```

```{r}

votes <- name.balance(tree, tax, 'n733', return.votes = c('up', 'down'))
message("up.votes")
sort(votes[[c('up.votes', 'Order')]])
message("down.votes")
sort(votes[[c('down.votes', 'Order')]])

```

```{r}

ps.merged <- phyloseq::merge_samples(GP, "Quarry_or_Benchmark")
ps.merged@sam_data$Quarry_or_Benchmark <- as.factor(c("Benchmark", "Quarry"))

colorspalette <- c("#fde725", "#35b779")

part_tree_ph <- function(order){
  ps.strange <- prune_taxa(
    rownames(filter(
      as.data.frame(ps.merged@tax_table@.Data),Class==order)),
    ps.merged)
  p <- plot_tree(ps.strange,
            ladderize="FALSE",
            color="Quarry_or_Benchmark", 
            label.tips="Genus",
            size="abundance",
            sizebase=2,
            plot.margin=0.5,
            base.spacing=0.03) + 
    scale_color_manual(values=c(colorspalette))
  return(p)
}

part_tree_ph("Planctomycetes")

```


```{r}

votes <- name.balance(tree, tax, 'n1005', return.votes = c('up', 'down'))
message("up.votes")
sort(votes[[c('up.votes', 'Phylum')]])
message("down.votes")
sort(votes[[c('down.votes', 'Phylum')]])

```



```{r}

ggarrange(p.philr.urvan, p.philr.progress, ncol=2)

```


```{r}
tc.nn
p.tree <- ggtree(tree) +
  geom_balance(node=tc.nn[6], fill=tc.colors[6], alpha=0.6)

annotate_balance(tree, 'n210', p=p.tree, labels = c('n210+', 'n210-'),
                 offset.text=0.15, bar=FALSE)
```


```{r, fig.height=12, fig.width=9}

plot_tree(GP, ladderize="left", color="Phylum")

```


### distances

```{r}

ps.nc <- prune_samples(sample_data(ps.f)$Soil %in% c("meadow_humus-cryptogley"), ps.f)
ps.nc  <- prune_taxa(taxa_sums(ps.nc) > 0, ps.nc)

unifrac.nc <- unifrac(ps.nc@otu_table@.Data, ps.nc@phy_tree)
metadata.medium <- as(sample_data(ps.nc@sam_data), "data.frame")
adonis2(unifrac.medium ~ Soil , data = metadata.medium)

```


### PERMANOVA

```{r}

dist <- phyloseq::distance(ps.nc, "unifrac")
metadata <- as(sample_data(ps.nc@sam_data), "data.frame")
adonis( dist ~ Quarry_or_Benchmark, data = metadata)

```

```{r}

dist <- phyloseq::distance(ps.cc2, "unifrac")
metadata <- as(sample_data(ps.cc2@sam_data), "data.frame")
adonis( dist ~ Soil, data = metadata)

```

```{r}

dist <- phyloseq::distance(ps.nc, "bray")
names.nc.b <- subset(ps.nc@sam_data, ps.nc@sam_data$Quarry_or_Benchmark %in% "Benchmark") %>%
  rownames()
names.nc.q <- subset(ps.nc@sam_data, ps.nc@sam_data$Quarry_or_Benchmark %in% "Quarry") %>%
  rownames()

usedist::dist_between_ centroids(dist, names.nc.b, names.nc.q, squared = TRUE)

```

```{r}

dist <- phyloseq::distance(ps.cc2, "bray")
names.cc.m <- subset(ps.cc2@sam_data, ps.cc2@sam_data$Soil %in% "meadow_humus-cryptogley") %>%
  rownames()
names.cc.c <- subset(ps.cc2@sam_data, ps.cc2@sam_data$Soil %in% "Carbonate_chernozem") %>%
  rownames()

usedist::dist_between_centroids(dist, names.cc.m, names.cc.c, squared = TRUE)

```


```{r}

dist <- phyloseq::distance(ps.cc, "unifrac")
metadata <- as(sample_data(ps.cc@sam_data), "data.frame")
adonis( dist ~ Quarry_or_Benchmark, data = metadata)


ps.cc@sam_data
```

### more alphass

```{r, message=FALSE,echo=TRUE,fig.height=8, fig.width=12}

library(ggpubr)

alpha.custom <- function(ps.f, arrga = "PD"){
  
  require(picante)
  
  pd <- pd(ps.f@otu_table@.Data, ps.f@phy_tree)
  
  pd1 <- estimate_richness(ps.f, measures=c("Observed", "InvSimpson", "Shannon"))
  pd <- cbind(pd, ps.f@sam_data, pd1 )
  
  bmi <- levels(as.factor(pd$Quarry_or_Benchmark))
  bmi.pairs <- combn(seq_along(bmi), 2, simplify = FALSE, FUN = function(i)bmi[i])
  p1 <- ggviolin(pd, x = "Quarry_or_Benchmark", y = arrga,
                 add = "boxplot", fill = "Quarry_or_Benchmark") + stat_compare_means(comparisons = bmi.pairs, method = "wilcox.test", size=3) +
    scale_x_discrete(limits = c("Quarry", "Benchmark")) +
    theme(axis.title.x = element_blank(),
          legend.title = element_blank(),
          legend.position = "none",
          axis.text.x = element_text(angle = 45, hjust = 1),
          text = element_text(size=9))
  return(p1)
  return(p1)
}

alpha.custom2 <- function(ps.f, arrga = "PD"){
  
  require(picante)
  
  pd <- pd(ps.f@otu_table@.Data, ps.f@phy_tree)
  
  pd1 <- estimate_richness(ps.f, measures=c("Observed", "InvSimpson", "Shannon"))
  pd <- cbind(pd, ps.f@sam_data, pd1 )
  
  bmi <- levels(as.factor(pd$Quarry_or_Benchmark))
  bmi.pairs <- combn(seq_along(bmi), 2, simplify = FALSE, FUN = function(i)bmi[i])
  p1 <- ggviolin(pd, x = "Quarry_or_Benchmark", y = arrga,
                 add = "boxplot", fill = "Quarry_or_Benchmark") +
    stat_compare_means(comparisons = bmi.pairs, method = "wilcox.test", size=3) +
    scale_x_discrete(limits = c("Quarry", "Benchmark")) +
    theme(axis.title.x = element_blank(),
          legend.title = element_blank(),
          legend.position = "none",
          axis.text.x = element_text(angle = 45, hjust = 1),
          text = element_text(size=9))
  return(p1)
}

p.alpha.Cdt.oo <- alpha.custom(ps.cc, arrga = "Observed")
p.alpha.Cdt.sh <- alpha.custom(ps.cc, arrga = "PD")
p.alpha.Cdt.is <- alpha.custom(ps.cc, arrga = "Shannon")
p.alpha.Cdt.pd <- alpha.custom(ps.cc, arrga = "InvSimpson")

p.alpha.Cdt.oo2 <- alpha.custom2(ps.nc, arrga = "Observed")
p.alpha.Cdt.sh2 <- alpha.custom2(ps.nc, arrga = "PD")
p.alpha.Cdt.is2 <- alpha.custom2(ps.nc, arrga = "Shannon")
p.alpha.Cdt.pd2 <- alpha.custom2(ps.nc, arrga = "InvSimpson")

p1 <- ggarrange(p.alpha.Cdt.oo, p.alpha.Cdt.sh,p.alpha.Cdt.is, p.alpha.Cdt.pd , 
                p.alpha.Cdt.oo2, p.alpha.Cdt.sh2, p.alpha.Cdt.is2, p.alpha.Cdt.pd2,
                ncol = 4 ,label.x = 0.3, nrow = 2, font.label = c(size = 10), labels = c("Progress", "", "", "", "Urvan")) 
p1


ggsave(file="plots/alpha.tiff",
        device="tiff",
        plot=p1, 
        width=2200, 
        height=1700,
        units ="px", 
        dpi=300,
        compression = "lzw"
        )


``` 



```{r}


library(ggpubr)

alpha.custom <- function(ps.f, arrga = "PD"){
  
  require(picante)
  
  pd <- pd(ps.f@otu_table@.Data, ps.f@phy_tree)
  
  pd1 <- estimate_richness(ps.f, measures=c("Observed", "InvSimpson", "Shannon"))
  pd <- cbind(pd, ps.f@sam_data, pd1 )
  
  bmi <- levels(as.factor(pd$Quarry_or_Benchmark))
  bmi.pairs <- combn(seq_along(bmi), 2, simplify = FALSE, FUN = function(i)bmi[i])
  p1 <- ggviolin(pd, x = "Quarry_or_Benchmark", y = arrga,
                 add = "boxplot", fill = "Quarry_or_Benchmark") + stat_compare_means(comparisons = bmi.pairs, method = "wilcox.test") +
    scale_x_discrete(limits = c("Quarry", "Benchmark")) +
    theme(axis.title.x = element_blank(),
          legend.title = element_blank(),
          legend.position = "none",
          axis.text.x = element_text(angle = 45, hjust = 1),
          text = element_text(size=14))
  return(p1)
}

alpha.custom2 <- function(ps.f, arrga = "PD"){
  
  require(picante)
  
  pd <- pd(ps.f@otu_table@.Data, ps.f@phy_tree)
  
  pd1 <- estimate_richness(ps.f, measures=c("Observed", "InvSimpson", "Shannon"))
  pd <- cbind(pd, ps.f@sam_data, pd1 )
  
  bmi <- levels(as.factor(pd$Quarry_or_Benchmark))
  bmi.pairs <- combn(seq_along(bmi), 2, simplify = FALSE, FUN = function(i)bmi[i])
  p1 <- ggviolin(pd, x = "Quarry_or_Benchmark", y = arrga,
                 add = "boxplot", fill = "Quarry_or_Benchmark") + stat_compare_means(comparisons = bmi.pairs, method = "wilcox.test") +
    scale_x_discrete(limits = c("Quarry", "Benchmark")) +
    theme(axis.title.x = element_blank(),
          legend.title = element_blank(),
          legend.position = "none",
          axis.text.x = element_text(angle = 45, hjust = 1),
          text = element_text(size=14))
  return(p1)
}

p.alpha.Cdt.oo <- alpha.custom(ps.f, arrga = "Observed")
p.alpha.Cdt.sh <- alpha.custom(ps.f, arrga = "PD")
p.alpha.Cdt.is <- alpha.custom(ps.f, arrga = "Shannon")
p.alpha.Cdt.pd <- alpha.custom(ps.f, arrga = "InvSimpson")


p.alpha.more <- ggarrange(p.alpha.Cdt.oo, p.alpha.Cdt.sh,p.alpha.Cdt.is, p.alpha.Cdt.pd ,
                ncol = 4 ,label.x = 0.105, nrow = 1)
p.alpha.more



```


### NO3

```{r}

ps.no3 <- subset_taxa(ps.varstab.merged, taxa_names(ps.varstab) %in% c("Seq70", "Seq71", "Seq161", "Seq170", "Seq171"))
otus.no3 <- as.data.frame(ps.no3@otu_table)
otus.no3$NO3 <- ps.no3@sam_data$NO3
cor.test(otus.no3$Seq170, otus.no3$NO3, method="spearman")
```

### shit venns

```{r, fig.height=9, fig.width=9}

amp.cc <- phyloseq_to_amp(ps.cc)

p.venn.cc <- amp_venn(amp.cc,
         group_by = "Quarry_or_Benchmark",
         normalise = TRUE,
         cut_a = 0.000001,
         cut_f=0.000001,
            )

?amp_venn

saveRDS(ps.f, file="ps.f2")
```




```{r}
BiocManager::install("ANCOMBC")
library(ANCOMBC)
library(philr)

subset(as.data.frame(ps.f@tax_table@.Data), is.na(Phylum))
subset(as.data.frame(ps.f@tax_table@.Data), Kingdom == "Eukaryota")
levels(as.data.frame(ps.f@tax_table@.Data)$Phylum)
as.data.frame(ps.f@tax_table@.Data)

des.nc <- des_w_simper(ps.nc, "Quarry_or_Benchmark")
des.cc <- des_w_simper(ps.cc, "Quarry_or_Benchmark")
des.so <- des_w_simper(ps.f, "Region")


des.nc.filt <- des.nc %>% 
  filter(baseMean > 10) %>% 
  filter(padj < 0.05) %>% 
  arrange(log2FoldChange)
des.cc.filt <- des.cc %>% 
  filter(baseMean > 10) %>% 
  filter(padj < 0.05) %>% 
  arrange(log2FoldChange)
des.so.filt <- des.so %>% 
  filter(baseMean > 10) %>% 
  filter(padj < 0.05) %>% 
  arrange(log2FoldChange)

des.so.filt

```

```{r}

des.nc %>% arrange(desc(log2FoldChange))
des.nc %>% arrange(log2FoldChange)

des.so %>% arrange(desc(log2FoldChange))
des.so %>% arrange(log2FoldChange)

des.cc %>% arrange(desc(log2FoldChange))
des.cc %>% arrange(log2FoldChange)

```

```{r}

des.so.filt

```

```{r}


```


