---
title: "Nal"
author: "GrGladkov"
date: "19 11 2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/storage/nal/")
```


```{r}
list.files()
```



```{r}
#first run - with controls(its), pool - true, 2.5ee, 220-180
# ls | grep Abacumov-Nal | xargs cp -t  ~/storage/nal/in

setwd("~/storage/nal/")
path = "in/"
path_trein_set = "~/storage/somebases/silva_nr_v138_train_set.fa"
path_trein_set_species = "~/storage/somebases/silva_species_assignment_v138.fa"
name_Brief = "nameBrief.txt"

truncLen = "220,180"
maxEE = "2,5"
mult = TRUE
mlt = NULL
```

### Import libraries

```{r}

require(dada2)
require(Biostrings)
require(DECIPHER)
require(phyloseq)
require(seqinr)
require(data.table)
require(metagMisc)
require(tibble)
require(tidyverse)

```

### quallity plots

```{r}
# docker container run --rm -e AMPLICONLENGTH=[amplicon length] -e FORWARDPRIMERLENGTH=[forward primer length] \
    # -e REVERSEPRIMERLENGTH=[reverse primer length] -v /path/to/fastqs:/data/input \
    # -v / figaro

# docker container run --rm -e AMPLICONLENGTH=260 -e FORWARDPRIMERLENGTH=20 \
# -e REVERSEPRIMERLENGTH=19 -v /home/gladkov/storage/nal/in \
# -v /home/gladkov/storage/nal figaro

# python3 figaro.py -i /home/gladkov/storage/nal/in -o /home/gladkov/storage/nal -a 260 -f 20 -r 19


fnFs <- sort(list.files(path, pattern="_R1_001.fastq", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_001.fastq", full.names = TRUE))
  
plotQualityProfile(fnRs[1:3])

fnFs  
list.files()
```


```{r} 

sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
on.exit()
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(185,185), trimLeft=20, maxN=0, maxEE=c(6,8), rm.phix=TRUE, compress=TRUE, multithread=mult)
errF <- learnErrors(filtFs, multithread=mult)
errR <- learnErrors(filtRs, multithread=mult)
derepFs <- derepFastq(filtFs, verbose=TRUE)
derepRs <- derepFastq(filtRs, verbose=TRUE)
# Name the derep-class objects by the sample names
names(derepFs) <- sample.names
names(derepRs) <- sample.names
dadaFs <- dada(derepFs, err=errF, multithread=mult, pool=FALSE)
dadaRs <- dada(derepRs, err=errR, multithread=mult, pool=FALSE)
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose=TRUE)
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
table(nchar(getSequences(seqtab)))
getN <- function(x) sum(getUniques(x))
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=mult, verbose=TRUE)


briefToSeq <- colnames(seqtab.nochim)
names(briefToSeq) <- paste0("Seq", seq(ncol(seqtab.nochim))) 
st.brief <- seqtab.nochim
colnames(st.brief) <- names(briefToSeq) 

write.table(briefToSeq, file = name_Brief, sep = "\t")

#dna <- DNAStringSet(briefToSeq)
#alignment <- AlignSeqs(DNAStringSet(dna), anchor=NA,verbose=FALSE, processors = mlt)
#writeXStringSet(alignment, file="align.fasta")

taxa.dada2 <- assignTaxonomy(briefToSeq,path_trein_set , multithread=mult)
taxa.dada2.species <- assignSpecies(briefToSeq, path_trein_set_species) # maybe use not seqs but brief 
rownames(taxa.dada2.species) <- rownames(briefToSeq)
briefToSeq.df <- data.frame(briefToSeq)
rownames(taxa.dada2.species) <- rownames(briefToSeq.df)
rownames(taxa.dada2) <- rownames(taxa.dada2.species)
taxa <- cbind2(taxa.dada2, taxa.dada2.species[,2])
colnames(taxa)[7] <- "Species"

getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
write.table(track, file = "track.tsv", sep= "\t", col.names = NA, quote=FALSE)

st.brief.t.df <- data.frame(t(st.brief))
write.table(st.brief.t.df, file = "otu_table.txt", sep= "\t", col.names = NA, quote=FALSE)

briefToSeq.ls <- as.list(briefToSeq.df[,c("briefToSeq")])
briefToSeq.names <- as.list(rownames(briefToSeq.df))
write.fasta( briefToSeq.ls, briefToSeq.names , "rep_seq.fasta", as.string = FALSE)

write.table(taxa, file = "taxa.txt", sep= "\t", col.names = NA, quote=FALSE)
  
```

sed -i '1s/^/#OTU_ID/' otu_table.txt 
biom convert -i otu_table.txt -o otu_table.biom --to-hdf5
qiime tools import \
--input-path rep_seq.fasta \
--output-path rep_seq.qza \
--type 'FeatureData[Sequence]'
qiime fragment-insertion sepp --i-representative-sequences rep_seq.qza --i-reference-database ~/storage/somebases/sepp-refs-silva-128.qza --o-tree insertion-tree.qza  --o-placements insertion-placements.qza --p-threads 20  
qiime tools import \
--input-path otu_table.biom \
--type 'FeatureTable[Frequency]' \
--input-format BIOMV210Format \
--output-path feature-table.qza
qiime fragment-insertion filter-features \
--i-table feature-table.qza \
--i-tree insertion-tree.qza \
--o-filtered-table filtered_table.qza \
--o-removed-table removed_table.qza
unzip -p filtered_table.qza > filtered_table.biom
biom convert -i  filtered_table.biom -o filtered_table.txt --to-tsv
sed -i '1d' filtered_table.txt
unzip -p insertion-tree.qza */data/* > tree.nwk


```{r}

as_tibble(track, rownames = "ID") %>% arrange(nonchim)
reps <- readFasta("rep_seq.fasta")

  
```


```{r}

#some SEPP bash tree magic
map <- read_tsv("mapping_nalchic_2.tsv")
map
map <- column_to_rownames(map, "ID")

map

taxa <- read.csv("taxa.txt" , header=TRUE, sep="\t")
taxa <- column_to_rownames(taxa, 'X')
taxa <- as.matrix(taxa)


filt.otu <- read.csv("otu_table.txt" , header=TRUE, sep="\t")
filt.otu
colnames(filt.otu)[1] <- "ID"
filt.otu <- column_to_rownames(filt.otu, "ID")
filt.otu
library(naturalsort)
filt.otu <- filt.otu[c(naturalsort(colnames(filt.otu)))]
filt.otu
colnames(filt.otu) <- rownames(map)
filt.otu
# class(filt.otu) <- "numeric"

filt.otu <- as.matrix(filt.otu)
filt.otu <- t(filt.otu)
rownames(filt.otu)
rownames(map)

tree <- read_tree(treefile="tree.nwk")

ps <- phyloseq(otu_table(filt.otu, taxa_are_rows=FALSE), 
               sample_data(map),
               tax_table(taxa),
               phy_tree(tree))


length(tree$tip.label)
ps@phy_tree$tip.label
ps

filt.otu[0,] = rownames(map)
rownames(filt.otu) == rownames(map)

rownames(taxa)

intersect(rownames(as.data.frame(filt.otu.matrix)), rownames(map)) %>% length()

colnames(filt.otu)

ps

```


```{r, fig.height=12, fig.width=5}
library(tidyverse)
library(seqinr)

some_rep_fasta <- read.fasta("~/storage/mal/rep_seq.fasta")
d <- density(getLength(some_rep_fasta))
d
sort(getLength(some_rep_fasta), decreasing = TRUE)
(getLength(some_rep_fasta), decreasing = TRUE)
quantile(getLength(some_rep_fasta))

```


```{r}
saveRDS(ps, file = "ps.rds")
packageVersion("DESeq2")
```


### Read some previously generated data

ps, track,etc.
Add refseq data.

```{r}

track <- read_tsv("track.tsv")
ps <- read_rds("ps.rds")
ps <- merge_phyloseq(ps, refseq(readDNAStringSet("rep_seq.fasta")))

pop_taxa <- function(physeq, badTaxa){
  allTaxa = taxa_names(physeq)
  myTaxa <- allTaxa[!(allTaxa %in% badTaxa)]
  return(prune_taxa(myTaxa, physeq))
}

delete_mit_chl_eu <- function(ps){
  badTaxa <- taxa_names(subset_taxa(ps, Order=="Chloroplast"))
  ps <- pop_taxa(ps, badTaxa)
  badTaxa <- taxa_names(subset_taxa(ps, Family=="Mitochondria"))
  ps <- pop_taxa(ps, badTaxa)
  badTaxa <- taxa_names(subset_taxa(ps, Kingdom == "Eukaryota"))
  ps <- pop_taxa(ps, badTaxa)
  badTaxa <- taxa_names(subset_taxa(ps, is.na(Phylum)))
  ps <- pop_taxa(ps, badTaxa)
  return(ps)
}

saveRDS(ps.f, file = "ps_f.rds")
ps
ps.f
ps.f <- delete_mit_chl_eu(ps)
ps.f@tax_table

writeXStringSet(ps.f@refseq, file="ref_filt.fasta")

```

```{r}

ps@sam_data

```




```{r}

pop_taxa <- function(physeq, badTaxa){
  allTaxa = taxa_names(physeq)
  myTaxa <- allTaxa[!(allTaxa %in% badTaxa)]
  return(prune_taxa(myTaxa, physeq))
}

delete_mit_chl <- function(ps){
  badTaxa <- taxa_names(subset_taxa(ps, Order=="Chloroplast"))
  ps <- pop_taxa(ps, badTaxa)
  badTaxa <- taxa_names(subset_taxa(ps, Family=="Mitochondria"))
  ps <- pop_taxa(ps, badTaxa)
  return(ps)
}

ps.f <- delete_mit_chl(ps)
ps
ps.f@sam_data

```
```{r,fig.height=6, fig.width=10}

pssd2veg <- function(physeq) {
  sd <- sample_data(physeq)
  return(as(sd,"data.frame"))
}

sample_data(ps.f2) <- pssd2veg(ps.f2) %>% 
  rownames_to_column("ID") %>% 
    mutate(
    Quarry_or_Benchmark = case_when(
      grepl('Quarry', What) ~ 'Quarry',
      grepl('Benchmark', What) ~ 'Benchmark'
    ) 
    ) %>%
  relocate(Quarry_or_Benchmark, .before = Insolation) %>% 
      column_to_rownames("ID")


```

### Richness/depth correlation.
```{r, message=FALSE,echo=TRUE, fig.height=8, fig.width=12}

library(phyloseq)
library(ggplot2)
plot_rich_reads_samlenames_lm <- function(physeq){
  rish <- estimate_richness(physeq, measures = "Observed")
  reads.sum <- as.data.frame(sample_sums(physeq))
  reads.summary <- cbind(rish, reads.sum)
  colnames(reads.summary) <- c("otus","reads")
  reads.summary["Repeat"] <-unlist(purrr::map(stringr::str_split(rownames(physeq@sam_data), "N_", 2), function(x) x[[2]]))
  reads.summary["Site"] <- physeq@sam_data$Site
  library(ggrepel)
  require(ggforce)
  p1 <- ggplot(data=reads.summary) + 
    geom_point(aes(y=otus, x=log2(reads), color=Site),size=3) + 
    geom_text_repel(aes(y=otus, x=log2(reads), label=paste0(Site, "_", Repeat))) + 
    theme_bw()+
    geom_smooth(aes(y=otus, x=log2(reads), fill=Site, color=Site),method=lm, se=FALSE, ymin = 1) + 
    scale_x_continuous(sec.axis = sec_axis(sec.axis ~ 2**.)) 
  # geom_mark_ellipse(aes(y = otus, x=reads, group = Repeats, label = Repeats, color = Repeats), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
  
  return(p1)
}

plot_rich_reads_samlenames_lm(ps.f)

ps.f2 <- subset_samples(ps.f, sample_names(ps.f) != 'N_95')
ps.f2 <- subset_samples(ps.f2, sample_names(ps.f2) != 'N_96')
ps.f2 <- subset_samples(ps.f2, sample_names(ps.f2) != 'N_89')
ps.f2 <- subset_samples(ps.f2, sample_names(ps.f2) != 'N_31')
# ps.f2 <- subset_samples(ps.f2, sample_names(ps.f2) != 'N_70')
ps.f2 <- subset_samples(ps.f2, Site != 'N8-2')
ps.f2 <- subset_samples(ps.f2, Site != 'N8-3')

plot_rich_reads_samlenames_lm(ps.f2)
sample_names(ps.f2)

ps.test <- subset_samples(ps.f2, Site == 'N3-1')
plot_rich_reads_samlenames_lm(ps.test)

```


```{r}

ps.f2@sam_data$Soil

ps.nc <- prune_samples(! sample_data(ps.f2)$Soil %in% c("Forest_gray"), ps.f2)
ps.nc  <- prune_taxa(taxa_sums(ps.nc) > 0, ps.nc)

ps.nc@sam_data

```

```{r}

amp <- phyloseq_to_amp(ps_f)
amp

```



```{r}
ps.f2@sam_data <- map2 

ps.f2@sam_data

p1 <- plot_ordination(ps.varstab, ordination.b, type="sample", color="Soil", title="Типы почвы", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    theme(legend.position="bottom")

p2 <- plot_ordination(ps.varstab, ordination.b, type="sample", color="Quarry_or_Benchmark", title="Повреждённая почва и контроли", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    theme(legend.position="bottom")

ggarrange(p1, p2, nrow = 1, ncol = 2)

```

#### reexport rs

```{r}

ps.f2@sam_data

ps.nc <- prune_samples(! sample_data(ps_f2)$Soil %in% c("Carbonate_chernozem"), ps_f2)
ps.nc  <- prune_taxa(taxa_sums(ps.nc) > 0, ps.nc)

ps.сc <- prune_samples(sample_data(ps_f2)$Soil %in% c("Carbonate_chernozem"), ps_f2)
ps.сc  <- prune_taxa(taxa_sums(ps.сc) > 0, ps.сc)

ps.nc@sam_data

```


#### vst dataset

```{r}
ps_vst <- function(ps){
  diagdds = phyloseq_to_deseq2(ps, ~ Site)                  
  diagdds = estimateSizeFactors(diagdds, type="poscounts")
  diagdds = estimateDispersions(diagdds, fitType = "local") 
  pst <- varianceStabilizingTransformation(diagdds)
  pst.dimmed <- t(as.matrix(assay(pst))) 
  pst.dimmed[pst.dimmed < 0.0] <- 0.0
  ps.varstab <- ps
  otu_table(ps.varstab) <- otu_table(pst.dimmed, taxa_are_rows = FALSE) 
  return(ps.varstab)
}

library(DESeq2)

ps_var <- ps_vst(ps_f2)

ps.varstab <- ps_vst(ps.f2)
ps.varstab.f <- ps_vst(ps.f)
as.data.frame(ps.varstab.f@otu_table)

```


```{r, message=FALSE,echo=TRUE,fig.height=8, fig.width=20, results=FALSE, fig.keep='all'}

beta_custom_norm_NMDS_elli <- function(ps, seed = 7888, normtype="vst", color="Description"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  require(DESeq2)
  library(ggforce)
  
  
  # beta_NMDS <- function(){
  #normalisation. unifrac - rarefaction; wunifrac,bray - varstab
  
  diagdds = phyloseq_to_deseq2(ps, ~ Site)                  
  diagdds = estimateSizeFactors(diagdds, type="poscounts")
  diagdds = estimateDispersions(diagdds, fitType = "local") 
  if (normtype =="vst")
    pst <- varianceStabilizingTransformation(diagdds)
  if (normtype =="log") 
    pst <- rlogTransformation(diagdds)
  
  pst.dimmed <- t(as.matrix(assay(pst))) 
  pst.dimmed[pst.dimmed < 0.0] <- 0.0
  ps.varstab <- ps
  otu_table(ps.varstab) <- otu_table(pst.dimmed, taxa_are_rows = FALSE) 

  #beta and ordination
  
  ordination.b <- ordinate(ps.varstab, "NMDS", "bray")

  #plotting
  p  <-  plot_ordination(ps, ordination.b, type="sample", color="Site", title="NMDS - Bray-Curtis", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
  
  return(p)
}

p.beta <- beta_custom_norm_NMDS_elli(ps.f2)
p.beta

```



```{r,fig.height=7, fig.width=8}
ordination.b <- ordinate(ps_var, "NMDS", "bray")
ps_var@sam_data

plot_ordination(ps, ordination.b, type="sample", color="Site", title="NMDS - Bray-Curtis", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))

```




```{r}
ps.f2@sam_data
```


### Heatmaps with relative abundance by ampvis2 package
```{r, message=FALSE,echo=TRUE}

map3 <- data.frame(ps.f2@sam_data)
sam_data(ps.f2) <- map3 %>% 
  rownames_to_column("ID") %>% 
    mutate(
    Soil = case_when(
      grepl('meadow_humus', Soil) ~ 'Луговая',
      grepl('Carbonate_chernozem', Soil) ~ 'Чернозём',
      grepl('Forest_gray', Soil) ~ 'Лесная',
    )
  ) %>% 
    mutate(
    Quarry_or_Benchmark = case_when(
      grepl('Quarry', What) ~ 'Quarry',
      grepl('Benchmark', What) ~ 'Benchmark'
    ) 
    )%>% 
      column_to_rownames("ID")


data.frame(map3))
library(tidyverse)
library(phyloseq)

phyloseq_to_amp <- function(ps){
    require(ampvis2)
    require(tibble)
    require(phyloseq)
    colnames(ps@tax_table@.Data) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
    OTU1 = as(otu_table(ps), "matrix")
    OTU1 <- t(OTU1)
    OTUdf = as.data.frame(OTU1)
    taxa.ps <- as(tax_table(ps), "matrix")
    taxa.df = as.data.frame(taxa.ps)
    my_otu_table <- merge(OTUdf, taxa.df, by=0)
    my_otu_table <- column_to_rownames(my_otu_table, var="Row.names")

    my_metadata <- as_tibble(sample_data(ps), rownames=NA)
    my_metadata <- rownames_to_column(my_metadata,var = "SampleID")
    my_tree <- phy_tree(ps)
    amp.ps <- amp_load(otutable = my_otu_table, metadata = my_metadata, tree = my_tree)
    return(amp.ps)
}


amp <- phyloseq_to_amp(ps.f2)
ps.f2@sam_data
```

### Split

```{r}

ps.1 <- prune_samples(sample_data(ps.f2)$Soil %in% c("Луговая"),ps.f2)
ps.1  <- prune_taxa(taxa_sums(ps.1) > 0, ps.1)

ps.2 <- prune_samples(sample_data(ps.f2)$Soil %in% c("Чернозём"),ps.f2)
ps.2  <- prune_taxa(taxa_sums(ps.2) > 0, ps.2)

ps.12 <- prune_samples(! sample_data(ps.f2)$Soil %in% c("Лесная"),ps.f2)
ps.12  <- prune_taxa(taxa_sums(ps.12) > 0, ps.12)

ps.nc <- prune_samples(! sample_data(ps_f2)$Soil %in% c("Calcareous Chernozem"), ps_f2)
ps.nc  <- prune_taxa(taxa_sums(ps.nc) > 0, ps.nc)

ps.cc <- prune_samples(sample_data(ps_f2)$Soil %in% c("Calcareous Chernozem"), ps_f2)
ps.cc  <- prune_taxa(taxa_sums(ps.cc) > 0, ps.cc)

ps_f2@sam_data
```

### heatmappa

```{r, fig.height=10, fig.width=12}

amp_heatmap(amp, tax_show = 40, group_by = "What", tax_aggregate = "Genus", tax_class = "Proteobacteria", tax_add = "Phylum") + theme_bw() + theme(text = element_text(size=15), legend.position = "none") + theme(axis.text.x=element_text(angle=45,hjust=1))

```

```{r, fig.height=7, fig.width=8}

amp_heatmap(amp, tax_show = 15, group_by = "What", tax_aggregate = "Phylum", tax_class = "Proteobacteria") + theme_bw() + theme(text = element_text(size=15), legend.position = "none") + theme(axis.text.x=element_text(angle=45,hjust=1))

```

```{r, fig.height=10, fig.width=12}

amp_heatmap(amp, tax_show = 30, group_by = "What", tax_aggregate = "Family", tax_class = "Proteobacteria", tax_add = "Phylum") + theme_bw() + theme(text = element_text(size=15), legend.position = "none") + theme(axis.text.x=element_text(angle=45,hjust=1))

```



```{r}

library(vegan)
physeq <- ps.nc
dist <- phyloseq::distance(physeq, "bray")
metadata <- as(sample_data(physeq@sam_data), "data.frame")
ad <- adonis2(dist ~ Quarry_or_Benchmark, data = metadata)

ad

```

```{r}

library(vegan)
physeq <- ps.cc
dist <- phyloseq::distance(physeq, "bray")
metadata <- as(sample_data(physeq@sam_data), "data.frame")
ad <- adonis2(dist ~ Quarry_or_Benchmark, data = metadata)

ad

```

```{r,message=FALSE,echo=TRUE, fig.height=5, fig.width=5.8}

dist <- phyloseq::distance(ps.12, method = "bray")
groups <- ps.12@sam_data$Soil
mod <- vegan::betadisper(dist, groups)
mod.box <- boxplot(mod)
TukeyHSD(mod)
?TukeyHSD.betadisper
permutest(mod, permutations = 999)
anova(mod)
plot(TukeyHSD(mod))

```


```{r, message=FALSE,echo=TRUE, fig.height=6, fig.width=9}

amp_heatmap(amp, group_by = "Quarry_or_Benchmark",facet_by = "Soil",tax_show = 30, tax_aggregate = "Genus", tax_add = "Family", tax_class = "Proteobacteria") + theme_bw() + theme(text = element_text(size=15), legend.position = "none")
amp
```


```{r, message=FALSE, echo=FALSE}
des_w_simper <- function(ps, factor){
  require(DESeq2)
  require(vegan)
  require(tibble)
  require(phyloseq)
  
  diagdds = phyloseq_to_deseq2(ps, as.formula(paste( "~", factor)))                  
  diagdds = estimateSizeFactors(diagdds, type="poscounts")
  diagdds = estimateDispersions(diagdds, fitType = "local") 
  diagdds = DESeq(diagdds)
  samp <-sample_data(ps)
  aggdata <- t(aggregate.data.frame(as.data.frame(as.data.frame(t(diagdds@assays@data$mu))), by=list(samp[[factor]]), median))
  colnames(aggdata) <- aggdata[1,]
  aggdata <- aggdata[-1,]
  res = results(diagdds)
  res.df <- as.data.frame(res)
  nice <- cbind(res.df, as.data.frame(ps@tax_table@.Data[rownames(res.df),]), as.data.frame(aggdata)[rownames(res.df),])
} 

```

### Amasing DESeq


```{r, eval=FALSE}

des.nc <- des_w_simper(ps.nc, "Quarry_or_Benchmark")
des.cc <- des_w_simper(ps.cc, "Quarry_or_Benchmark")

des.nc.filt <- des.nc %>% filter(baseMean > 30) %>% filter(padj < 0.05) %>% arrange(log2FoldChange)
des.cc.filt <- des.cc %>% filter(baseMean > 10) %>% filter(padj < 0.05) %>% arrange(log2FoldChange)


write.table(des.nc.filt, file = "des_nc_filt.tsv", sep= "\t", col.names = NA, quote=FALSE)
write.table(des.cc.filt, file = "des_cc_filt.tsv", sep= "\t", col.names = NA, quote=FALSE)

```





```{r,fig.height=8, fig.width=20, results=FALSE}

ps.rand <- rarefy_even_depth(ps.f2)

ordination.b <- ordinate(ps.varstab, "PCoA", "bray")
ordination.u <- ordinate(ps.rand, "PCoA", "unifrac")
ordination.w <- ordinate(ps.varstab, "PCoA", "wunifrac")
  
  #plotting
  p1 <-  plot_ordination(ps, ordination.b, type="sample", color="Site", title="Bray-Curtis", 
                       axes = c(1,2) ) + theme_bw() + theme(text = element_text(size = 10)) + geom_point(size = 3) 
  

  p2 <-  plot_ordination(ps, ordination.u, type="sample", color="Site", title="UniFrac", 
                       axes = c(1,2) ) + theme_bw() + theme(text = element_text(size = 10)) + geom_point(size = 3)
  
  p3 <-  plot_ordination(ps, ordination.w, type="sample", color="Site", title="Weighted UniFrac", 
                       axes = c(1,2) ) + theme_bw() + theme(text = element_text(size = 10)) + geom_point(size = 3) 
  
  #merge by ggpubr
  
  
  
  p.all <- ggarrange(p1, p2, p3, ncol = 3 , nrow = 1, common.legend = TRUE, legend = "bottom", font.label = list(size = 12, face = "bold", color ="black"))
  
p.all
p.beta <- beta_custom_norm_NMDS_elli(ps.f)
ps.varstab@sam_data$Soil
```

```{r, message=FALSE,echo=TRUE,fig.height=8, fig.width=18}

library(ggpubr)

pd1 <- estimate_richness(ps.f2, measures=c("Observed", "InvSimpson", "Shannon"))

pdd  <- pd1 %>% rownames_to_column("Site") %>% mutate(Site = as.factor(Site)) %>% gather(Observed, Shannon, InvSimpson, value = "Index", key = "Type")
pdd <- cbind(pd1, ps.f2@sam_data)
pdd
p1 <- ggplot(data = pdd, aes(y = Observed, x = Quarry_or_Benchmark)) + geom_boxplot() + geom_point(alpha = 0.4) + labs(title = "Observed") + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_grid(~Soil) + theme(axis.title.x = element_blank())
p2 <- ggplot(data = pdd, aes(y = InvSimpson, x = Quarry_or_Benchmark)) + geom_boxplot() + geom_point(alpha = 0.4)  + labs(title = "InvSimpson") + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ facet_grid(~Soil)+ theme(axis.title.x = element_blank())
p3 <- ggplot(data = pdd, aes(y = Shannon, x = Quarry_or_Benchmark)) + geom_boxplot() + geom_point(alpha = 0.4)  + labs(title = "Shannon")+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ facet_grid(~Soil)+ theme(axis.title.x = element_blank())
p1
ggarrange(p1, p2,p3, ncol = 3 ,label.x = 0.105, nrow = 1, common.legend = TRUE) 


p_o
pdd

p.alpha.Cdt.oo <- alpha.custom(ps.f, arrga = "Observed")
p.alpha.Cdt.is <- alpha.custom(ps.f, arrga = "Shannon")
p.alpha.Cdt.pd <- alpha.custom(ps.f, arrga = "InvSimpson")

"Observed","Shannon","InvSimpson"


p1 <- ggarrange(p.alpha.Cdt.oo, p.alpha.Cdt.sh,p.alpha.Cdt.is, p.alpha.Cdt.pd , ncol = 4 ,label.x = 0.105, nrow = 1, common.legend = TRUE) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

p1
```


### Process before tree calculated

```{r}

list.files()



```

### Decontam

Run on all samples
non norm

```{r}

library(decontam)

conc <- read.csv("concentration_nalchic.csv") %>% select(c("ID", "Site", "final_concentration"))
sample_data(ps)$conc <- conc$final_concentration
sample_data(ps)$conc <- sample_data(ps)$conc + 0.000001
sample_data(ps)$conc

```

```{r}

ps.varstab
ps

```



```{r}
sample_sums(ps)

ps.f <- subset_samples(ps, sample_names(ps) != 'N_95')
ps.f <- subset_samples(ps.f, sample_names(ps.f) != 'N_89')
ps.f <- subset_samples(ps.f, sample_names(ps.f) != 'N_96')

ps.vst <- ps_vst(ps.f)

contamdf_ps <- isContaminant(ps.f, method="frequency", conc="conc")
contamdf_ps_vst <- isContaminant(ps.vst, method="frequency", conc="conc")

contamdf_ps["Seq703",]
contamdf_ps_vst["Seq703",]

contamdf_ps %>% arrange(p)
contamdf_ps_vst %>% arrange(p)

```

Seq666
Seq703


```{r}
table(contamdf_ps$contaminant)
cont <- contamdf.freq[contamdf_ps$contaminant == "TRUE",] 
as.data.frame(ps.f@tax_table[rownames(cont),])
```

Seq1123
Seq1461

```{r}
table(contamdf_ps_vst$contaminant)
cont_vst <- contamdf.freq[contamdf_ps_vst$contaminant == "TRUE",] 
as.data.frame(ps.f@tax_table[rownames(cont_vst),])
```



```{r}
plot_frequency(ps.f, "Seq703", conc="conc")

```

```{r}
plot_frequency(ps.vst, "Seq703", conc="conc")

```

### read new mapping

```{r}

map <- read_tsv("mapping_nalchic_new.tsv", 
                col_types = cols(
                  pH = col_character()
                )
              )
map <- column_to_rownames(map, "Sample")
ps <- read_rds("ps_f.rds")
site_column <- ps@sam_data %>% 
  as.data.frame() %>% 
  pull('Site')
map <- map %>% 
  mutate(Type = str_replace(Type, " /", "/")) %>% 
  mutate(`Corg,_%` = str_replace(`Corg,_%`, ",", ".")) %>%
  mutate(`NO3_mg/kg` = str_replace(`NO3_mg/kg` , "," , ".")) %>% 
  mutate(pH = str_replace(pH, ",", ".")) %>% 
  mutate(final_concentration = str_replace(final_concentration, ",", ".")) %>% 
  mutate(pH = as.numeric(pH)) %>% 
  mutate(final_concentration = as.numeric(final_concentration)) %>% 
  mutate(`NO3_mg/kg` = as.numeric(`NO3_mg/kg`)) %>% 
  mutate(`Corg,_%` = as.numeric(`Corg,_%`)) %>% 
  mutate(
  Quarry_or_Benchmark = case_when(
    grepl('Quarry', What) ~ 'Quarry',
    grepl('Benchmark', What) ~ 'Benchmark'
  ) 
  ) %>%
  relocate(Quarry_or_Benchmark, .before = Insolation) %>% 
  mutate(Repeat = site_column) %>% 
  relocate(Repeat, .after = Site)

map <- map %>%  mutate_if(sapply(map, is.character), as.factor)

map_m <- map
colnames(map_m)
colnames(map_m)[16] <- "C"
colnames(map_m)[17] <- "C_org"
colnames(map_m)[18] <- "P"
colnames(map_m)[19] <- "K"
colnames(map_m)[20] <- "NH4"
colnames(map_m)[21] <- "NO3"

colnames(map)

ps.f <- prune_samples(! sample_data(ps.f)$Soil %in% c("Forest_gray"), ps.f)
ps.f  <- prune_taxa(taxa_sums(ps.f) > 0, ps.f)
sample_data(ps.f) <- map_m

ps.nc <- prune_samples(sample_data(ps.f)$Date %in% c("9/23/2020"), ps.f)
map3 <- data.frame(ps.nc@sam_data) %>%  mutate(temp = scale(Temperature))
ps.nc <- merge_phyloseq(ps.nc, sample_data(map3))

ps.nc2 <- prune_samples(! sample_data(ps.f)$Date %in% c("9/24/2020"), ps.f)
map3 <- data.frame(ps.nc2@sam_data) %>%  mutate(temp = scale(Temperature))
ps.nc2 <- merge_phyloseq(ps.nc2, sample_data(map3))

ps.f <- merge_phyloseq(ps.nc, ps.nc2)

ps.f@sam_data


ps.merged <- phyloseq::merge_samples(ps.f, "Repeat")



sample_data(ps.nc)$Quarry_or_Benchmark = factor(sample_data(ps.nc)$Quarry_or_Benchmark)
sample_data(ps.nc)$Feature = factor(sample_data(ps.nc)$Feature)
ps.merged <- phyloseq::merge_samples(ps.nc, "Site")

sample_data(ps.merged)$Soil <- factor(sample_data(ps.merged)$Soil, labels=levels(sample_data(ps.nc)$Soil))
sample_data(ps.merged)$Quarry_or_Benchmark <-  factor(sample_data(ps.merged)$Quarry_or_Benchmark, labels=levels(sample_data(ps.nc)$Quarry_or_Benchmark))
sample_data(ps.merged)$Feature <- factor(sample_data(ps.merged)$Feature, labels=levels(sample_data(ps.nc)$Feature))

sample_data(ps.merged)
factor(sample_data(ps.merged)$Soil, levels = levels(sample_data(ps.nc)$Soil))
sample_data(c)$Soil

```

### CCA again



```{r, message=FALSE,echo=TRUE, fig.height=7, fig.width=7}

library(phyloseq)             
library(vegan)

physeq <- ps.merged

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}

physeq@sam_data

otus.ps.vegan <- veganifyOTU(physeq)

metadata <- as(sample_data(physeq), "data.frame") 

metadata

cca <- vegan::cca(otus.ps.vegan ~  Humidity + pH + C_org + C +  K + NH4 + NO3 + temp, data=metadata)


kate.ggcca.sites <- function(vare.cca){

require(tidyverse)
biplot <- as.data.frame(vare.cca$CCA$biplot)
wa <- as.data.frame(vare.cca$CCA$wa)

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))
wa <- rownames_to_column(wa, "Label") %>% 
  add_column(Score = rep("sites", length(rownames(wa))))
fdat_amazing <- rbind(biplot, wa)

 
p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), aes(x = 0, xend = CCA1, y = 0, yend = CCA2), alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
  geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=4) + 
  theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
  return(p)
}

kate.ggcca.sites(cca)


```


```{r}

anova(cca)
anova(cca, by="terms")
anova(cca, by="mar")
vif.cca(cca)

```


#### Try another CCA models

```{r, message=FALSE,echo=TRUE, fig.height=7, fig.width=7}




cca_w <- vegan::cca(otus.ps.vegan ~  Humidity + pH + C_org + NH4 + temp, data=metadata)


kate.ggcca.sites <- function(vare.cca){

require(tidyverse)
biplot <- as.data.frame(vare.cca$CCA$biplot)
wa <- as.data.frame(vare.cca$CCA$wa)

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))
wa <- rownames_to_column(wa, "Label") %>% 
  add_column(Score = rep("sites", length(rownames(wa))))
fdat_amazing <- rbind(biplot, wa)

 
p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), aes(x = 0, xend = CCA1, y = 0, yend = CCA2), alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
  geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=4) + 
  theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
  # geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
  return(p)
}

kate.ggcca.sites(cca_w)

```

```{r}

anova(cca_w)
anova(cca_w, by="terms")
anova(cca_w, by="mar")
vif.cca(cca_w)

```

#### Same with varstab normalisation

```{r, message=FALSE,echo=TRUE, fig.height=7, fig.width=7}

ps_vst <- function(ps){
  diagdds = phyloseq_to_deseq2(ps, ~ Repeat)                  
  diagdds = estimateSizeFactors(diagdds, type="poscounts")
  diagdds = estimateDispersions(diagdds, fitType = "local") 
  pst <- varianceStabilizingTransformation(diagdds)
  pst.dimmed <- t(as.matrix(assay(pst))) 
  pst.dimmed[pst.dimmed < 0.0] <- 0.0
  ps.varstab <- ps
  otu_table(ps.varstab) <- otu_table(pst.dimmed, taxa_are_rows = FALSE) 
  return(ps.varstab)
}


ps.varstab <- ps_vst(ps.f)
ps.varstab.merged <- phyloseq::merge_samples(ps.varstab, "Repeat")
physeq <- ps.varstab.merged

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}

physeq@sam_data

otus.ps.vegan <- veganifyOTU(physeq)

metadata <- as(sample_data(physeq), "data.frame") 


cca_w_varstab <- vegan::cca(otus.ps.vegan ~  temp + pH + C_org + NH4,  data=metadata)
cca_f_varstab <- vegan::cca(otus.ps.vegan ~  Humidity + pH + C_org + C +  K + NH4 + NO3 + temp + P, data=metadata)


kate.ggcca.sites(cca_w_varstab)

```


```{r}

anova(cca_w_varstab, parallel = 10)
anova(cca_w_varstab, by="terms", parallel = 10)
anova(cca_w_varstab, by="mar", parallel = 10)
vif.cca(cca_w_varstab)

```

#### CCA for taxons - family level

```{r, message=FALSE,echo=TRUE, fig.height=9, fig.width=9}

ps.ff <-  filter_taxa(ps.f, function(x) sum(x > 3) > (0.1*length(x)), TRUE)
ps.varstab <- ps_vst(ps.ff)
ps.varstab.merged <- phyloseq::merge_samples(ps.varstab, "Repeat")
ps.varstab.merged.family <- tax_glom(ps.varstab.merged, taxrank="Family")

physeq <- ps.varstab.merged.family

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}


otus.ps.vegan <- veganifyOTU(physeq)

metadata <- as(sample_data(physeq), "data.frame") 


cca_w_varstab_family <- vegan::cca(otus.ps.vegan ~  temp + pH + C_org + NH4,  data=metadata)

kate.ggcca.species <- function(vare.cca, physeq){

require(tidyverse)
biplot <- as.data.frame(vare.cca$CCA$biplot)

wa <- as.data.frame(vare.cca$CCA$v) %>% 
  rownames_to_column("ID")
seq <- wa
f <- as.data.frame(physeq@tax_table@.Data) %>% 
  select("Family") %>% 
  rownames_to_column("ID")
wa <- full_join(v, f) %>% 
  select(-ID) %>% 
  rename("Family" = "Label") %>% 
  mutate(Score = "sites")

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))
fdat_amazing <- rbind(biplot, wa) %>% 
  mutate(Label = as.factor(Label))

 
p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), aes(x = 0, xend = CCA1, y = 0, yend = CCA2), alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
  geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=4) + 
  theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
  # geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
  return(fdat_amazing)
}


kate.ggcca.species(cca_w_varstab_family, physeq) 

```

### PERMANOVA for all

```{r, echo=TRUE, rows.print = 112}

permanova.forloop.noNestedPlant <- function(factor){
  require(vegan)
  require(phyloseq)
  dist <- phyloseq::distance(physeq, "bray")
  metadata <- as(sample_data(physeq@sam_data), "data.frame")
  ad <- adonis2(as.formula(paste( "dist ~ ", factor)), data = metadata)
  ad <- cbind2(ad[1,][3], ad[1,][5])
  return(ad)
}

physeq <- ps.varstab
permanova.forloop.pos.noPlant <- possibly(permanova.forloop.noNestedPlant, otherwise = NA)
out <- map(colnames(physeq@sam_data[, 3:length(colnames(physeq@sam_data))]), permanova.forloop.pos.noPlant)
out.d.noPlant <- do.call("rbind", out)
out.d.noPlant

```

```{r}

cca_w_varstab_family$CCA$v["Seq10", ]
length(colnames(otus.ps.vegan))
length(rownames(cca_w_varstab_family$CCA$v))
anova(cca_w_varstab_family, parallel = 10)
anova(cca_w_varstab_family, by="terms", parallel = 10)
anova(cca_w_varstab_family, by="mar", parallel = 10)
vif.cca(cca_w_varstab_family)

```


### Import actual phyloseq rds

```{r}

ps <- read_rds("ps_f.rds")
ps@sam_data


ps_f <- prune_samples(! sample_data(ps)$Soil %in% c("Forest_gray"), ps)
ps_f  <- prune_taxa(taxa_sums(ps_f) > 0, ps_f)

ps
ps_f
list.files()

as.data.frame(ps_f@sam_data)

```


```{r, message=FALSE,echo=TRUE, fig.height=8, fig.width=12}

library(phyloseq)
library(ggplot2)

plot_rich_reads_samlenames_lm <- function(physeq){
  rish <- estimate_richness(physeq, measures = "Observed")
  reads.sum <- as.data.frame(sample_sums(physeq))
  reads.summary <- cbind(rish, reads.sum)
  colnames(reads.summary) <- c("otus","reads")
  reads.summary["Repeat"] <-unlist(purrr::map(stringr::str_split(rownames(physeq@sam_data), "N_", 2), function(x) x[[2]]))
  reads.summary["Site"] <- physeq@sam_data$Site
  library(ggrepel)
  require(ggforce)
  p1 <- ggplot(data=reads.summary) + geom_point(aes(y=otus, x=log2(reads), color=Site),size=3) + geom_text_repel(aes(y=otus, x=log2(reads), label=paste0(Site, "_", Repeat))) + theme_bw()+
    geom_smooth(aes(y=otus, x=log2(reads), fill=Site, color=Site),method=lm, se=FALSE, ymin = 1) + scale_x_continuous(sec.axis = sec_axis(sec.axis ~ 2**.)) 
  # geom_mark_ellipse(aes(y = otus, x=reads, group = Repeats, label = Repeats, color = Repeats), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
  
  return(p1)
}

plot_rich_reads_samlenames_lm(ps_f)

ps_f2 <- ps_f
# ps.f2 <- subset_samples(ps.f, sample_names(ps.f) != 'N_95')
# ps.f2 <- subset_samples(ps.f2, sample_names(ps.f2) != 'N_96')
# ps.f2 <- subset_samples(ps.f2, sample_names(ps.f2) != 'N_89')
ps_f2 <- subset_samples(ps_f2, sample_names(ps_f2) != 'N_31')
ps_f2 <- subset_samples(ps_f2, sample_names(ps_f2) != 'N_70')
# ps.f2 <- subset_samples(ps.f2, Site != 'N8-2')
# ps.f2 <- subset_samples(ps.f2, Site != 'N8-3')
ps_f2  <- prune_taxa(taxa_sums(ps_f2) > 0, ps_f2)
plot_rich_reads_samlenames_lm(ps_f2)

# sample_names(ps_f2)
ps_f2
ps_f
# ps.test <- subset_samples(ps.f2, Site == 'N3-1')
plot_rich_reads_samlenames_lm(ps_f2_2)
```



```{r}
map3 <- data.frame(ps_f2@sam_data)
sam_data(ps_f2) <- map3 %>% 
  rownames_to_column("ID") %>% 
    mutate(
    Soil = case_when(
      grepl('meadow_humus', Soil) ~ 'Umbric Gleyic',
      grepl('Carbonate_chernozem', Soil) ~ 'Calcareous Chernozem'
    )
  ) %>% 
    mutate(
    Quarry_or_Benchmark = case_when(
      grepl('Quarry', What) ~ 'Disturbed_soil',
      grepl('Benchmark', What) ~ 'Control'
    ) 
    )%>% 
    mutate(
    TH = Humidity/Temperature
    )  %>% 
      column_to_rownames("ID")


ps_f2@sam_data
ps.varstab <- ps_vst(ps_f2)
ordination.b <- ordinate(ps.varstab, "NMDS", "bray")

p1 <- plot_ordination(ps.varstab, ordination.b, type="sample", color="Soil", title="Soil types", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    theme(legend.position="bottom", legend.title = element_blank(), legend.text=element_text(size=9))

p2 <- plot_ordination(ps.varstab, ordination.b, type="sample", color="Quarry_or_Benchmark", title="Disturbed soil or control", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    theme(legend.position="bottom", legend.title = element_blank(), legend.text=element_text(size=9))

ggarrange(p1, p2, nrow = 1, ncol = 2)

pH + Temperature + Humidity + P_mg.kg + K_mg.kg + NH4_mg.kg + NO3_mg.kg

```

### Plot of geographic map

```{r}

library(rnaturalearth)
library(rnaturalearthdata)
library(tidyverse)
library(sf)
library(rgeos)
library(ggrepel)

world <- ne_countries(scale = "large", returnclass = "sf")
spdf_france_geounit <- ne_states(geounit = 'russia', returnclass = "sf")
probdata = tibble(site = c("Quarry on Chernozem", "Quarry on Gleyic"),
                  longitude = c(43.51137, 43.77252), 
                  latitude =  c(43.82965, 43.35022), 
                  xlabel = c(40, 47), 
                  ylabel = c(44.3, 43))

# rivers <- ne_download(scale = 10, type = 'rivers_lake_centerlines', category = 'physical', returnclass = "sf")
# lakes <- ne_download(scale = 50, type = 'lakes', category = 'physical', returnclass = "sf")



 ggplot(data = world) + 
  geom_sf() +
  geom_sf(data = spdf_france_geounit, fill="green",  alpha = 0.1 ) +
  # geom_sf(data = lakes ) +
  coord_sf(xlim = c(35, 50), ylim = c(40, 50), expand = FALSE) +
  geom_point(data = probdata, aes(x = longitude, y = latitude), size = 3, 
  shape = 20, colour = "red") +
  geom_text(data = probdata, aes(x = xlabel, y = ylabel, label = site), size = 5, colour="red") + 
  theme_bw() 


```


```{r,message=FALSE,echo=TRUE, fig.height=5, fig.width=5.8}

dist <- phyloseq::distance(ps_f2, method = "bray")
groups <- ps_f2@sam_data$Soil
mod <- vegan::betadisper(dist, groups)
mod.box <- boxplot(mod)
TukeyHSD(mod)
?TukeyHSD.betadisper
permutest(mod, permutations = 999)
anova(mod)
plot(TukeyHSD(mod))

```


```{r, message=FALSE,echo=TRUE, fig.height=7, fig.width=7}

library(phyloseq)             
library(vegan)

ps.f <- ps_f2_2
ps.merged <- merge_samples(ps.f, "Site")
physeq <- ps.merged

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}


otus.ps.vegan <- veganifyOTU(physeq)

metadata <- as(sample_data(physeq), "data.frame") 


vare.cca <- vegan::cca(otus.ps.vegan ~  pH + scaled_temp + P_mg.kg + K_mg.kg + NH4_mg.kg + NO3_mg.kg, data=metadata)


kate.ggcca.sites <- function(vare.cca){

require(tidyverse)
biplot <- as.data.frame(vare.cca$CCA$biplot)
wa <- as.data.frame(vare.cca$CCA$wa)

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))
wa <- rownames_to_column(wa, "Label") %>% 
  add_column(Score = rep("sites", length(rownames(wa))))
fdat_amazing <- rbind(biplot, wa)

 
p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), aes(x = 0, xend = CCA1, y = 0, yend = CCA2), alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
  geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=4) + 
  theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
  return(p)
}

kate.ggcca.sites(vare.cca)


```

```{r}

library(re)

M <- cor(met)
corrplot(met, method = "circle")

metadata %>% mutate(So)

met <- ps_f2_2@sam_data
dummy <- as.numeric(year == 1957)
dummy <- as.numeric(year == 1957)
map3 <- data.frame(ps_f2@sam_data)
met <- map3 %>% mutate(Soil_d = ifelse(Soil == "Umbric Gleyic", 1, 0)) %>%  mutate(Soil_d = ifelse(Soil == "Umbric Gleyic", 1, 0))
met
cor.test(met$Temperature, met$Soil_d)
cor.test(met$pH, met$Soil_d)
cor.test(met$scaled_temp, met$Humidity)
cor.test(met$Temperature, met$Humidity)

ps.nc <- prune_samples(sample_data(ps.varstab)$Soil %in% c("Umbric Gleyic"), ps.varstab)
map3 <- data.frame(ps.nc@sam_data) %>%  mutate(temp = scale(Temperature))
ps.nc <- merge_phyloseq(ps.nc, sample_data(map3))


ps.nc2 <- prune_samples(! sample_data(ps.varstab)$Soil %in% c("Umbric Gleyic"), ps.varstab)
map3 <- data.frame(ps.nc2@sam_data) %>%  mutate(temp = scale(Temperature))
ps.nc2 <- merge_phyloseq(ps.nc2, sample_data(map3))

ps_f2_2 <- merge_phyloseq(ps.nc, ps.nc2)

ps_f2_2@sam_data

ps.nc <- prune_samples(! sample_data(ps.f2)$Soil %in% c("Calcareous Chernozem"), ps_f2)
ps.nc  <- prune_taxa(taxa_sums(ps.nc) > 0, ps.nc)


filter()

```



```{r}

map_f3 <- data.frame(ps_f2_2@sam_data)
map_f3 <- map_f3 %>%  mutate_if(sapply(map_f3, is.character), as.factor)
ps.nc <- merge_phyloseq(ps_f2_2, sample_data(map_f3))
ps.merged <- phyloseq::merge_samples(ps.nc, "Site")


mutate_if(sapply(iris_char, is.character), as.factor)
sample_data(ps.nc)$Soil = factor(sample_data(ps.nc)$Soil)
sample_data(ps.nc)$Quarry_or_Benchmark = factor(sample_data(ps.nc)$Quarry_or_Benchmark)
sample_data(ps.nc)$Feature = factor(sample_data(ps.nc)$Feature)
ps.merged <- phyloseq::merge_samples(ps.nc, "Site")

sample_data(ps.merged)$Soil <- factor(sample_data(ps.merged)$Soil, labels=levels(sample_data(ps.nc)$Soil))
sample_data(ps.merged)$Quarry_or_Benchmark <-  factor(sample_data(ps.merged)$Quarry_or_Benchmark, labels=levels(sample_data(ps.nc)$Quarry_or_Benchmark))
sample_data(ps.merged)$Feature <- factor(sample_data(ps.merged)$Feature, labels=levels(sample_data(ps.nc)$Feature))

sample_data(ps.merged)
factor(sample_data(ps.merged)$Soil, levels = levels(sample_data(ps.nc)$Soil))
sample_data(c)$Soil

map_m <- data.frame(ps.merged@sam_data)


ggplot(data=map_m, aes(Soil, temp)) + geom_boxplot(alpha=0.5) + geom_point(alpha=0.5) + theme_bw()
ggplot(data=map_m, aes(Soil, Temperature)) + geom_boxplot(alpha=0.5) + geom_point(alpha=0.5) + theme_bw()
ggplot(data=map_m, aes(Soil, scale(Temperature))) + geom_bo
xplot(alpha=0.5) + geom_point(alpha=0.5) + theme_bw()

```




### CCA statistics

```{r}

anova(vare.cca)

```

#### axis

```{r}

anova(vare.cca, by="terms")

```

#### multicolleniarity 

```{r}

vif.cca(vare.cca)

```



#### margins
```{r}

anova(vare.cca, by="mar")

```

```{r, message=FALSE,echo=TRUE, fig.height=7, fig.width=7}

library(phyloseq)             
library(vegan)

ps.f <- ps_f2_2
ps.merged <- merge_samples(ps.f, "Site")
physeq <- ps.merged

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}


otus.ps.vegan <- veganifyOTU(physeq)

metadata <- as(sample_data(physeq), "data.frame") 


rda <- vegan::rda(otus.ps.vegan ~  pH + scaled_temp + P_mg.kg + K_mg.kg + NH4_mg.kg + NO3_mg.kg, data=metadata)


kate.ggcca.sites <- function(vare.cca){

require(tidyverse)
biplot <- as.data.frame(vare.cca$CCA$biplot)
wa <- as.data.frame(vare.cca$CCA$wa)

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))
wa <- rownames_to_column(wa, "Label") %>% 
  add_column(Score = rep("sites", length(rownames(wa))))
fdat_amazing <- rbind(biplot, wa)

 
p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), aes(x = 0, xend = CCA1, y = 0, yend = CCA2), alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
  geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=4) + 
  theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
  return(p)
}

kate.ggcca.sites(rda)

metadata

```

### NMDS of filtered phyloseq

```{r, fig.height=10, fig.width=12}

ordination.b <- ordinate(ps_f2_2, "NMDS", "bray")

  #plotting
plot_ordination(ps_f2_2, ordination.b, type="sample", color="NH4_mg.kg", title="NMDS - Bray-Curtis", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))

```
```{r, fig.height=6, fig.width=7}
plot_ordination(ps_f2_2, ordination.b, type="sample", color="temp", title="NMDS - Bray-Curtis", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
```


```{r, fig.height=6, fig.width=7}
plot_ordination(ps_f2_2, ordination.b, type="sample", color="K_mg.kg", title="NMDS - Bray-Curtis", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
```

```{r, fig.height=6, fig.width=7}
plot_ordination(ps_f2_2, ordination.b, type="sample", color="pH", title="NMDS - Bray-Curtis", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
```

```{r, fig.height=6, fig.width=7}
plot_ordination(ps_f2, ordination.b, type="sample", color="What", title="NMDS - Bray-Curtis", axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes(group = Site, label = Site), label.fontsize = 10, label.buffer = unit(2, "mm"), label.minwidth = unit(5, "mm"),con.cap = unit(0.1, "mm"))
```

```{r}

ps_f2_2@sam_data

```
```{r}
ps.merged@sam_data
```

```{r, fig.height=6, fig.width=8}
ps.merged@sam_data

phyloseq_to_amp <- function(ps){
    require(ampvis2)
    require(tibble)
    require(phyloseq)
    colnames(ps@tax_table@.Data) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
    OTU1 = as(otu_table(ps), "matrix")
    OTU1 <- t(OTU1)
    OTUdf = as.data.frame(OTU1)
    taxa.ps <- as(tax_table(ps), "matrix")
    taxa.df = as.data.frame(taxa.ps)
    my_otu_table <- merge(OTUdf, taxa.df, by=0)
    my_otu_table <- column_to_rownames(my_otu_table, var="Row.names")

    my_metadata <- as_tibble(sample_data(ps), rownames=NA)
    my_metadata <- rownames_to_column(my_metadata,var = "SampleID")
    my_tree <- phy_tree(ps)
    amp.ps <- amp_load(otutable = my_otu_table, metadata = my_metadata, tree = my_tree)
    return(amp.ps)
}

ps.n4 <- prune_samples(sample_data(ps_f2_2)$Site %in% c("N4-1", 	"N4-2"), ps_f2_2)
ps.n4  <- prune_taxa(taxa_sums(ps.n4) > 0, ps.n4)

amp.n4 <- phyloseq_to_amp(ps.n4)
ps.n4@otu_table[,1]
amp_heatmap(amp.n4, tax_show = 40, group_by = "Site", tax_aggregate = "Genus", tax_class = "Proteobacteria") + theme_bw() + theme(text = element_text(size=15), legend.position = "none") + theme(axis.text.x=element_text(angle=45,hjust=1))


amp.f <- phyloseq_to_amp(ps.f)

amp_heatmap(amp.f, tax_show = 15, group_by = "Site", tax_aggregate = "Phylum", tax_class = "Proteobacteria") + theme_bw() + theme(text = element_text(size=15), legend.position = "none") + theme(axis.text.x=element_text(angle=45,hjust=1))

```



```{r}

ps.n4 <- prune_samples(sample_data(ps_f2_2)$Site %in% c("N4-1", 	"N4-2"), ps.f2)
ps.n4  <- prune_taxa(taxa_sums(ps.n4) > 0, ps.n4)

ps.f2

```


```{r}

des_w_simper <- function(ps, factor){
  require(DESeq2)
  require(vegan)
  require(tibble)
  require(phyloseq)
  
  diagdds = phyloseq_to_deseq2(ps, as.formula(paste( "~", factor)))                  
  diagdds = estimateSizeFactors(diagdds, type="poscounts")
  diagdds = estimateDispersions(diagdds, fitType = "local") 
  diagdds = DESeq(diagdds)
  samp <-sample_data(ps)
  aggdata <- t(aggregate.data.frame(as.data.frame(as.data.frame(t(diagdds@assays@data$mu))), by=list(samp[[factor]]), median))
  colnames(aggdata) <- aggdata[1,]
  aggdata <- aggdata[-1,]
  res = results(diagdds)
  res.df <- as.data.frame(res)
  nice <- cbind(res.df, as.data.frame(ps@tax_table@.Data[rownames(res.df),]), as.data.frame(aggdata)[rownames(res.df),])
} 


ds.n4 <- des_w_simper(ps.n4, "Site")

ds.n4 %>% arrange(padj)


library(vegan)
physeq <- ps.n4
dist <- phyloseq::distance(physeq, "bray")
metadata <- as(sample_data(physeq@sam_data), "data.frame")
ad <- adonis(dist ~ Site, data = metadata, permutations = 9999)

ad

```

```{r,  fig.height=10, fig.width=12}

ps.sseno <- prune_samples(sample_data(ps.f2)$Feature %in% "cane_bush", ps.f2)
ps.sseno  <- prune_taxa(taxa_sums(ps.sseno) > 0, ps.sseno)

ps.first <- prune_samples(sample_data(ps.f2)$Soil %in% "Umbric Gleyic", ps.f2)
ps.first <- prune_taxa(taxa_sums(ps.first) > 0, ps.first)

amp.sseno <- phyloseq_to_amp(ps.first)

amp_heatmap(amp.sseno, tax_show = 30, group_by = "Site", tax_aggregate = "Family", tax_class = "Proteobacteria", tax_add = "Phylum") + theme_bw() + theme(text = element_text(size=15), legend.position = "none") + theme(axis.text.x=element_text(angle=45,hjust=1))


```

