---
title: "RNF_grain_2023"
author: "GrGladkov"
date: "2023-11-25"
output: html_document
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/storage/ov/")
rmarkdown::find_pandoc(dir = "/home/gladkov/.conda/envs/bi/bin")

```

## dada2 pipeline


```{r}

require(dada2)
require(Biostrings)
require(DECIPHER)
require(phyloseq)
require(seqinr)
require(data.table)
require(metagMisc)
require(tibble)
library(phyloseq)
library(tidyverse)
library(vegan)


```

```{r}

mult = 10
mlt = 10
path_trein_set = "~/storage/somebases/silva_nr99_v138.1_train_set.fa.gz"
path_trein_set_species = "~/storage/somebases/silva_species_assignment_v138.1.fa.gz"
name_Brief = "nameBrief.txt"
path <- "/home/gladkov/storage/ov/raw"
active_dir <- "/home/gladkov/storage/ov"
fnFs <- sort(list.files(path, pattern="_R1_001.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_001.fastq.gz", full.names = TRUE))

```


```{r}

plotQualityProfile(fnFs, aggregate = TRUE)

```



```{r}

plotQualityProfile(fnRs, aggregate = TRUE)

```
```{r}

sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
on.exit()
filtFs <- file.path(active_dir, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(active_dir, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(180,170), trimLeft=c(19,20), maxN=0, truncQ=2, maxEE=c(2,5), rm.phix=TRUE, compress=TRUE, multithread=mult)
errF <- learnErrors(filtFs, multithread=mult)
errR <- learnErrors(filtRs, multithread=mult)
derepFs <- derepFastq(filtFs, verbose=TRUE)
derepRs <- derepFastq(filtRs, verbose=TRUE)

# Name the derep-class objects by the sample names
names(derepFs) <- sample.names
names(derepRs) <- sample.names
dadaFs <- dada(derepFs, err=errF, multithread=mult, pool='pseudo')
dadaRs <- dada(derepRs, err=errR, multithread=mult, pool='pseudo')
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose=TRUE)
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
table(nchar(getSequences(seqtab)))
getN <- function(x) sum(getUniques(x))
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=mult, verbose=TRUE)

briefToSeq <- colnames(seqtab.nochim)
names(briefToSeq) <- paste0("Seq", seq(ncol(seqtab.nochim))) 
st.brief <- seqtab.nochim
colnames(st.brief) <- names(briefToSeq)

write.table(briefToSeq, file = name_Brief, sep = "\t")

taxa.dada2 <- assignTaxonomy(briefToSeq, path_trein_set , multithread=mult)
taxa.dada2.species <- assignSpecies(briefToSeq, path_trein_set_species) # maybe use not seqs but brief 
rownames(taxa.dada2.species) <- rownames(briefToSeq)
briefToSeq.df <- data.frame(briefToSeq)
rownames(taxa.dada2.species) <- rownames(briefToSeq.df)
rownames(taxa.dada2) <- rownames(taxa.dada2.species)
taxa <- cbind2(taxa.dada2, taxa.dada2.species[,2])
colnames(taxa)[7] <- "Species"

getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
write.table(track, file = "track.tsv", sep= "\t", col.names = NA, quote=FALSE)

st.brief.t.df <- data.frame(t(st.brief))
write.table(st.brief.t.df, file = "otu_table.txt", sep= "\t", col.names = NA, quote=FALSE)

briefToSeq.ls <- as.list(briefToSeq.df[,c("briefToSeq")])
briefToSeq.names <- as.list(rownames(briefToSeq.df))
seqinr::write.fasta(briefToSeq.ls, briefToSeq.names , "rep_seq.fasta", as.string = FALSE)

write.table(taxa, file = "taxa.txt", sep= "\t", col.names = NA, quote=FALSE)
track

```


```{r}

list.files()

```


## import 

```{r}

mad

```



```{r}

require(dada2)
require(Biostrings)
require(DECIPHER)
require(phyloseq)
require(seqinr)
require(data.table)
require(metagMisc)
require(tibble)
library(phyloseq)
library(tidyverse)
library(vegan)
library(ampvis2)
library(vegan)
library(metagMisc)

```


```{r}

setwd("~/storage/ov/")

library(readxl)
library(tidyverse)
library(phyloseq)


# 
# list.files()
# read_excel("fallow2023_meta.xlsx")
# path <- "fallow2023_meta.xlsx"


otu_table <- read_tsv("16S/otu_table.txt") %>% 
  column_to_rownames("...1")

hue <- otu_table %>% 
  colnames()

colnames(otu_table) <- unlist(purrr::map(stringr::str_split(hue, "\\.", 3), function(x) paste0(x[[2]], '_' ,x[[3]]))) 
hue_mod <- colnames(otu_table)

map <- data.frame(
  ID = hue_mod,
  Substrate = unlist(purrr::map(stringr::str_split(hue_mod, "\\_", 3), function(x) x[[2]] %>% substring(1, 1))),
  Repeat = unlist(purrr::map(stringr::str_split(hue_mod, "\\_", 3), function(x) x[[2]] %>% substring(2, 3))),
  Time = unlist(purrr::map(stringr::str_split(hue_mod, "\\_", 3), function(x) x[[2]] %>% substring(5, 10)))
)

map <- map %>% 
  mutate(Time = as.factor(Time) %>% forcats::fct_recode("30" = "0", "68" = "I", "124" = "II") %>% as.character() %>% as.double()) %>%
  mutate(Substrate = as.factor(Substrate) %>% forcats::fct_recode("CN" = "1", "CN_БАГС" = "2", "CN_ЦА" = "3")) %>% 
  mutate(Group = paste0(Substrate, '_' ,Time) %>% as.factor() %>% forcats::fct_relevel(gtools::mixedsort)) %>% 
  column_to_rownames("ID")

otus <- otu_table %>% 
  as.matrix() %>% 
  t()

taxa <- read_tsv("16S/taxa.txt") %>% 
  column_to_rownames("...1") %>% 
  as.matrix()
taxa[taxa == "Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium"] <- "Allorhizobium"
ps <- phyloseq(otu_table(otus, taxa_are_rows=FALSE), 
               sample_data(map),
               tax_table(taxa))


ps

map$Group

```


### functions

```{r}

plot_rich_reads_samlenames_lm <- function(physeq, group = "Site", label = "Repeat"){
  rish <- estimate_richness(physeq, measures = "Observed")
  reads.sum <- as.data.frame(sample_sums(physeq))
  reads.summary <- cbind(rish, reads.sum)
  colnames(reads.summary) <- c("otus","reads")
  reads.summary["Repeat"] <-unlist(purrr::map(stringr::str_split(rownames(physeq@sam_data), "\\_", 2), function(x) x[[2]]))
  reads.summary["Site"] <- physeq@sam_data[[group]]
  library(ggrepel)
  require(ggforce)
  p1 <- ggplot(data=reads.summary) + 
    geom_point(aes(y=otus, x=log2(reads), color=Site),size=3) + 
    geom_text_repel(aes(y=otus, x=log2(reads), label=paste0(Repeat))) + 
    theme_bw() +
    geom_smooth(aes(y=otus, x=log2(reads), fill=Site, color=Site),method=lm, se=FALSE, ymin = 1) + 
    scale_x_continuous(sec.axis = sec_axis(sec.axis ~ 2**.)) 

  return(p1)
}


beta_custom_norm_NMDS_elli_w <- function(ps, seed = 7888, normtype="vst", Color="What", Group="Repeat"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  library(ggforce)
  
  ps@otu_table[ps@otu_table < 0] <- 0
  ordination.b <- ordinate(ps, "NMDS", "bray")
  mds <- as.data.frame(ordination.b$points)
  p  <-  plot_ordination(ps,
                         ordination.b,
                         type="sample",
                         color = Color,
                         title="NMDS - Bray-Curtis",
                         # title=NULL,
                         axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    annotate("text",
    x=min(mds$MDS1) + abs(min(mds$MDS1))/4,
    y=max(mds$MDS2),
    label=paste0("Stress -- ", round(ordination.b$stress, 3))) +
    geom_mark_ellipse(aes_string(group = Group, label = Group),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                      con.colour='gray') +
    theme(legend.position = "none") +
    scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8)
  
  return(p)
}

plot_alpha_w_toc <- function(ps, group, metric) {
  
  require(phyloseq)
  require(ggplot2)
  
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  
  er <- estimate_richness(ps_a)
  df_er <- cbind(ps_a@sam_data, er)
  df_er <- df_er %>% select(c(group, metric))
  stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
    rstatix::tukey_hsd()
  y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))

  plot_richness(ps_a, x=group, measures=metric) + 
    geom_boxplot() +
    geom_point(size=1.2, alpha=0.3) +
    ggpubr::stat_pvalue_manual(
      stat.test, 
      label = "p.adj.signif", 
      y.position = y) +
    theme_light() + 
    scale_color_brewer(palette="Dark2") +
    theme(axis.text.x = element_text(angle = 90),
          axis.title.x=element_blank()) +
    labs(y=paste(metric, "index")) 
}

phyloseq_to_ampvis2 <- function(physeq) {
  #check object for class
  if(!any(class(physeq) %in% "phyloseq"))
    stop("physeq object must be of class \"phyloseq\"", call. = FALSE)
  
  #ampvis2 requires taxonomy and abundance table, phyloseq checks for the latter
  if(is.null(physeq@tax_table))
    stop("No taxonomy found in the phyloseq object and is required for ampvis2", call. = FALSE)
  
  #OTUs must be in rows, not columns
  if(phyloseq::taxa_are_rows(physeq))
    abund <- as.data.frame(phyloseq::otu_table(physeq)@.Data)
  else
    abund <- as.data.frame(t(phyloseq::otu_table(physeq)@.Data))
  
  #tax_table is assumed to have OTUs in rows too
  tax <- phyloseq::tax_table(physeq)@.Data
  
  #merge by rownames (OTUs)
  otutable <- merge(
    abund,
    tax,
    by = 0,
    all.x = TRUE,
    all.y = FALSE,
    sort = FALSE
  )
  colnames(otutable)[1] <- "OTU"
  
  #extract sample_data (metadata)
  if(!is.null(physeq@sam_data)) {
    metadata <- data.frame(
      phyloseq::sample_data(physeq),
      row.names = phyloseq::sample_names(physeq), 
      stringsAsFactors = FALSE, 
      check.names = FALSE
    )
    
    #check if any columns match exactly with rownames
    #if none matched assume row names are sample identifiers
    samplesCol <- unlist(lapply(metadata, function(x) {
      identical(x, rownames(metadata))}))
    
    if(any(samplesCol)) {
      #error if a column matched and it's not the first
      if(!samplesCol[[1]])
        stop("Sample ID's must be in the first column in the sample metadata, please reorder", call. = FALSE)
    } else {
      #assume rownames are sample identifiers, merge at the end with name "SampleID"
      if(any(colnames(metadata) %in% "SampleID"))
        stop("A column in the sample metadata is already named \"SampleID\" but does not seem to contain sample ID's", call. = FALSE)
      metadata$SampleID <- rownames(metadata)
      
      #reorder columns so SampleID is the first
      metadata <- metadata[, c(which(colnames(metadata) %in% "SampleID"), 1:(ncol(metadata)-1L)), drop = FALSE]
    }
  } else
    metadata <- NULL
  
  #extract phylogenetic tree, assumed to be of class "phylo"
  if(!is.null(physeq@phy_tree)) {
    tree <- phyloseq::phy_tree(physeq)
  } else
    tree <- NULL
  
  #extract OTU DNA sequences, assumed to be of class "XStringSet"
  if(!is.null(physeq@refseq)) {
    #convert XStringSet to DNAbin using a temporary file (easiest)
    fastaTempFile <- tempfile(pattern = "ampvis2_", fileext = ".fa")
    Biostrings::writeXStringSet(physeq@refseq, filepath = fastaTempFile)
  } else
    fastaTempFile <- NULL
  
  #load as normally with amp_load
  ampvis2::amp_load(
    otutable = otutable,
    metadata = metadata,
    tree = tree,
    fasta = fastaTempFile
  )
}

detachAllPackages <- function() {

  basic.packages <- c("package:stats","package:graphics","package:grDevices","package:utils","package:datasets","package:methods","package:base")

  package.list <- search()[ifelse(unlist(gregexpr("package:",search()))==1,TRUE,FALSE)]

  package.list <- setdiff(package.list,basic.packages)

  if (length(package.list)>0)  for (package in package.list) detach(package, character.only=TRUE, force = TRUE)

}

ps_vst <- function(ps, factor){
  diagdds = phyloseq::phyloseq_to_deseq2(ps, as.formula(paste( "~", factor)))              
  diagdds = DESeq2::estimateSizeFactors(diagdds, type="poscounts")
  diagdds = DESeq2::estimateDispersions(diagdds, fitType = "local") 
  pst <- DESeq2::varianceStabilizingTransformation(diagdds)
  pst.dimmed <- t(as.matrix(SummarizedExperiment::assay(pst))) 
  # pst.dimmed[pst.dimmed < 0.0] <- 0.0
  ps.varstab <- ps
  otu_table(ps.varstab) <- otu_table(pst.dimmed, taxa_are_rows = FALSE) 
  return(ps.varstab)
} 

filter_chrmitnas <- function(physeq){
  ps_object <- physeq 
  ps_object <- subset_taxa(ps_object, Phylum != "NA")
  
  ps_object@tax_table[is.na(ps_object@tax_table)] <- ps@tax_table@.Data
  ps_object <- subset_taxa(ps_object,
                           !(Family  == "Mitochondria" |
                               Class   == "Chloroplast" |
                               Order   == "Chloroplast"))
  ps_object@tax_table <- dplyr::na_if(ps_object@tax_table, TRUE)
  ps.f <- ps_object
  
  out_old <- capture.output(physeq)[4] %>% 
    str_extract(regex("([1-9]).*(?= by)"))
  out_new <- capture.output(ps.f)[4] %>% 
    str_extract(regex("([1-9]).*(?= by)"))
  message(paste0("old = ", out_old))
  message(paste0("filtered = ", out_new))
  
  return(ps.f)
  
}

beta_custom_norm_PCA_elli_w <- function(ps, seed = 7888, normtype="vst", Color="What", Group="Repeat"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  library(ggforce)
  
  ps@otu_table[ps@otu_table < 0] <- 0
  ordination.b <- ordinate(ps, "PCoA", "euclidean")
  mds <- as.data.frame(ordination.b$points)
  p  <-  plot_ordination(ps,
                         ordination.b,
                         type="sample",
                         color = Color,
                         title="PCA - clr-transformed",
                         # title=NULL,
                         axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes_string(group = Group, label = Group),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                      con.colour='gray') +
    theme(legend.position = "none") +
    scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8)
  
  return(p)
}

filter_chrmitnas <- function(physeq){
  ps_object <- physeq 
  ps_object <- subset_taxa(ps_object, Phylum != "NA")
  
  ps_object@tax_table <- ps_object@tax_table %>% 
    as.data.frame() %>% 
    mutate_all(~ replace_na(., "unknown")) %>% 
    as.matrix() %>% 
    tax_table()
  
  ps_object <- subset_taxa(ps_object,
                           !(Family  == "Mitochondria" | 
                               Class   == "Chloroplast" | 
                               Order   == "Chloroplast"))

  ps_object@tax_table <- ps_object@tax_table %>% 
    as.data.frame() %>% 
    mutate_all(~ na_if(., "unknown")) %>% 
    as.matrix() %>% 
    tax_table()
  
  ps.f <- ps_object
  
  out_old <- capture.output(physeq)[4] %>%
    str_extract(regex("([1-9]).*(?= by)"))
  out_new <- capture.output(ps.f)[4] %>%
    str_extract(regex("([1-9]).*(?= by)"))
  message(paste0("old = ", out_old))
  message(paste0("filtered = ", out_new))

  return(ps.f)
  
}

get_list_from_ps <- function(physeq, amaz_fac) {
  my_keys <- levels(sample_data(physeq)[[amaz_fac]])
  mylist <- vector(mode="list", length=length(my_keys))
  names(mylist) <- my_keys
  for (i in levels(sample_data(physeq)[[amaz_fac]])) { 
    x <- prune_samples(sample_data(physeq)[[amaz_fac]] %in% i, physeq)
    x <- prune_taxa(taxa_sums(x) > 0, x)
    mylist[[i]] <- taxa_names(x)
  }
  return(mylist)
}


beta_custom_norm_PCoA_elli_w <- function(ps, seed = 7888, normtype="vst", Color="What", Group="Repeat"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  library(ggforce)
  
  ps@otu_table[ps@otu_table < 0] <- 0
  ordination.b <- ordinate(ps, "PCoA", "bray")
  mds <- as.data.frame(ordination.b$points)
  p  <-  plot_ordination(ps,
                         ordination.b,
                         type="sample",
                         color = Color,
                         title="PCoA - bray - rarefied",
                         # title=NULL,
                         axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes_string(group = Group, label = Group),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                      con.colour='gray') +
    theme(legend.position = "none") +
    scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8)
  
  return(p)
}

plot_alpha_w_toc <- function(ps, group, metric) {
  
  require(phyloseq)
  require(ggplot2)
  
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  
  er <- estimate_richness(ps_a)
  df_er <- cbind(ps_a@sam_data, er)
  df_er <- df_er %>% select(c(group, metric))
  stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
    rstatix::tukey_hsd()
  y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))

  plot_richness(ps_a, x=group, measures=metric) + 
    geom_boxplot() +
    geom_point(size=1.2, alpha=0.3) +
    ggpubr::stat_pvalue_manual(
      stat.test, 
      label = "p.adj.signif", 
      y.position = y) +
    theme_light() + 
    scale_color_brewer(palette="Dark2") +
    theme(axis.text.x = element_text(angle = 90),
          axis.title.x=element_blank()) +
    labs(y=paste(metric, "index")) 
}


lsf.str()

```


```{r, fig.height=6, fig.width=6}

ps.f <- filter_chrmitnas(ps)


ps.f@tax_table <- ps.f@tax_table@.Data %>% 
  data.frame() %>% 
  mutate(Phylum = str_replace_all(Phylum, c("Firmicutes" = "Bacillota",
                                            "Chloroflexi" = "Chloroflexota",
                                            "Proteobacteria" = "Pseudomonadota",
                                            "Desulfobacterota" = "Thermodesulfobacteriota",
                                            "Crenarchaeota" = "Thermoproteota",
                                            "Cyanobacteria" = "Cyanobacteriota"))) %>% 
  as.matrix() %>% 
  tax_table()

# ps.gra <- prune_samples(sampleNames(ps.gra) != 'Abakumov.SEQ283.38', ps.gra)
# ps.gra <- prune_samples(sampleNames(ps.gra) != 'Abakumov.SEQ283.17', ps.gra)
# ps.gra <- prune_samples(sampleNames(ps.gra) != 'Abakumov.SEQ283.19', ps.gra)
# ps.gra@sam_data

ps.f@sam_data$Group

```


```{r, fig.width=9, fig.height=7}

plot_rich_reads_samlenames_lm(ps.f, group = "Group", label = "Group")  

ps.f <- prune_samples(sampleNames(ps.f) != 'sol_1a2.I', ps.f)
ps.f <- prune_samples(sampleNames(ps.f) != 'sol_3a2.0', ps.f)

plot_rich_reads_samlenames_lm(ps.f, group = "Group", label = "Group")  

```

```{r, fig.width=7, fig.height=6}

library(dplyr)


# ps.gra.clr <- ps.gra
# ps.otu.clr <- as.data.frame(t(ps.gra@otu_table)) %>% 
#   compositions::clr() %>% 
#   t()
# ps.gra.r <- rarefy_even_depth(ps.gra)
# 
# ps.gra.clr@otu_table <-  otu_table(ps.otu.clr, taxa_are_rows = FALSE)
# 
# beta_custom_norm_PCA_elli_w(ps.gra.clr, Group = 'Group', Color = 'wheat')
# beta_custom_norm_PCoA_elli_w(ps.gra.r, Group = 'Group', Color = 'wheat')
# beta_custom_norm_NMDS_elli_w(ps.gra.clr, Group = 'Group', Color = 'wheat')

beta_custom_norm_NMDS_elli_w(ps.f, Group = 'Group', Color = 'Substrate')

```




```{r, fig.width=12, fig.height=6}

plot_alpha_w_toc_mod <- function(ps, group, metric) {
  
  require(phyloseq)
  require(ggplot2)
  
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  
  er <- estimate_richness(ps_a)
  df_er <- cbind(ps_a@sam_data, er)
  df_er <- df_er %>% select(c(group, metric))
  stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
    rstatix::tukey_hsd() %>%
    filter(p.adj.signif != "ns")
  y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))

  plot_richness(ps_a, x=group, measures=metric, color="Substrate") + 
    geom_boxplot() +
    geom_point(size=1.2, alpha=0.3) +
    # geom_boxplot(stat="identity", position=position_dodge()) +
    # ggpubr::stat_pvalue_manual(
    #   stat.test,
    #   label = "p.adj.signif",
    #   y.position = y,
    #   tip.length = 0.005,
    #   bracket.nudge.y = 1.5,
    #   vjust = 0.6,
    #   size = 3) +
    theme_light() + 
        scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8) +
    theme(axis.text.x = element_text(size=12, angle = 45, hjust=1),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),
          legend.position = "None") +
    labs(y=paste(metric, "index")) 
}


p_a1 <-  plot_alpha_w_toc_mod(ps.f, group="Group",  metric="Observed") 
p_a2 <- plot_alpha_w_toc_mod(ps.f, group="Group", metric="Shannon") 
p_a3 <- plot_alpha_w_toc_mod(ps.f, group="Group", metric="InvSimpson") 

p_alpha <- ggpubr::ggarrange(p_a1, p_a2, p_a3, nrow = 1)
p_alpha
```


```{r}

ps.anc.liv <- ps.gra.fie

ps.anc.liv <- prune_samples(sample_data(ps.anc.liv)$wheat == 'true', ps.anc.liv)
ps.anc.liv  <- prune_taxa(taxa_sums(ps.anc.liv) > 0, ps.anc.liv)


dist <- phyloseq::distance(ps.anc.liv, "euclidean")
metadata <- as(sample_data(ps.anc.liv@sam_data), "data.frame")
vegan::adonis2(dist ~ preparation, data = metadata)

```



```{r,fig.height=7, fig.width=9}

amp <- phyloseq_to_ampvis2(ps.f)
ampvis2::amp_heatmap(amp,
            tax_show = 11,
            tax_aggregate = "Phylum",
            tax_class = "Pseudomonadota",
            group_by = "Group",
            plot_values_size = 4, 
            facet_by = "Substrate",
            showRemainingTaxa = TRUE)

```

```{r,fig.height=7, fig.width=9}

ampvis2::amp_heatmap(amp,
            tax_show = 30,
            tax_aggregate = "OTU",
            tax_add = "Genus",
            group_by = "Group",
            plot_values_size = 4, 
            facet_by = "Substrate",
            showRemainingTaxa = TRUE)

```
```{r}

ps.m@sam_data %>% 
  data.frame() %>% 
  colnames()

```


### cca

```{r,fig.height=7, fig.width=9}

ps.m <- merge_samples(ps.f, "Group")

met <- read_tsv("meta.tsv") %>% 
  column_to_rownames('ID') %>% 
  mutate_all(as.double) %>% 
  rownames_to_column('ID')

ps.m@sam_data <- ps.m@sam_data %>% 
  data.frame() %>% 
  mutate(Substrate = as.factor(Substrate) %>% forcats::fct_recode("CN" = "1", "CN_БАГС" = "2", "CN_ЦА" = "3")) %>% 
  mutate(Group = paste0(Substrate, '_' ,Time)) %>% 
  select(-Repeat) %>%
  rownames_to_column('ID') %>% 
  left_join(met) %>%
  column_to_rownames('ID') %>% 
  phyloseq::sam_data()

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}

physeq <- ps.m

otus.ps.vegan <- veganifyOTU(physeq)
rownames(otus.ps.vegan) <- physeq@sam_data$Group
metadata <- as(sample_data(physeq), "data.frame") 

otus.ps.vegan %>% 
  data.frame()
metadata

cca <- vegan::cca(otus.ps.vegan ~ Зольность + Потери + Влажность + Нитраты + Дыхание + Азот + Углерод + Гумус, data=metadata)

biplot <- as.data.frame(cca$CCA$biplot)
wa <- as.data.frame(cca$CCA$wa)

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot)))) %>% 
  mutate(
  Type = rep("biplot", length(rownames(biplot))),
)

wa <- rownames_to_column(wa, "Label") %>% 
  add_column(Score = rep("sites", length(rownames(wa)))) %>% mutate(
  Type = metadata$Group,
)

fdat_amazing <- rbind(biplot, wa) 

fdat_amazing <- fdat_amazing %>%
  mutate(label_size = ifelse(Score == "biplot", 4.5, 3)) %>% 
  mutate_if(is.character, as.factor)

summary.cca <- summary(cca)

list.pe <- summary.cca$cont$importance %>% 
  as.data.frame() %>% 
  select(c("CCA1", "CCA2")) %>% 
  dplyr::slice(2L) %>% 
  t() %>% 
  as.vector()



# kate.ggcca.sites <- function(vare.cca){
# 
#   require(tidyverse)
#   biplot <- as.data.frame(vare.cca$CCA$biplot)
#   wa <- as.data.frame(vare.cca$CCA$wa)
# 
#   biplot <- rownames_to_column(biplot, "Label") %>% 
#     add_column(Score = rep("biplot", length(rownames(biplot))))
#   wa <- rownames_to_column(wa, "Label") %>% 
#     add_column(Score = rep("sites", length(rownames(wa))))
#   fdat_amazing <- rbind(biplot, wa) 
#   fdat_amazing <- fdat_amazing %>%
#     mutate(label_size = ifelse(Score == "biplot", 4.5, 3))
#  
#   p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
#     # geom_abline(intercept=seq(-100, 100, 25), slope=1, colour="grey", alpha=0.5) +
#     # geom_abline(intercept=seq(-100, 100, 25), slope=-1, colour="grey", alpha=0.5) +
#     geom_vline(xintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
#     geom_hline(yintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
#     geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), 
#                mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
#     geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), 
#                  aes(x = 0, xend = CCA1, y = 0, yend = CCA2),
#                   size = 0.6, 
#                  alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
#   geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=fdat_amazing$label_size) + 
#   xlab(paste0(round(cca$CCA$eig[1], 3)*100, "% CCA1")) +
#   ylab(paste0(round(cca$CCA$eig[2], 3)*100, "% CCA2")) +
#   grids(linetype = "dashed") +
#   theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
#   return(p)
# }
# 
# kate.ggcca.sites(cca)
fdat_amazing

fdat_amazing <- fdat_amazing %>%  
  separate(Label, c("Label", NA), sep = '\\.', convert=TRUE)



cca1 <- ggplot(fdat_amazing) + 
  geom_vline(xintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  geom_hline(yintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  geom_point(data = fdat_amazing %>% 
               dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score)))   + 
  ggforce::geom_mark_ellipse(data = fdat_amazing %>%
                                      dplyr::filter(Score == "sites"),
                             aes(x=CCA1, y=CCA2, group = Label, label = Label, col = 'grey'),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                  con.colour='salmon',
                  label.colour='salmon'
                  ) +
  geom_segment(data = fdat_amazing %>% 
                 dplyr::filter(Score == "biplot"), 
               aes(x = 0, xend = CCA1, y = 0, yend = CCA2),
               size = 0.6, 
               alpha=0.8, 
               color = "red",
               arrow = arrow(angle = 3)) +
  ggrepel::geom_text_repel(data = fdat_amazing %>% 
                 dplyr::filter(Score == "biplot"), aes(x=CCA1, y=CCA2, label= Label), size=5) +
  xlab(paste0(round(list.pe[1], 3)*100, "% CCA1")) +
  ylab(paste0(round(list.pe[2], 3)*100, "% CCA2")) +
  grids(linetype = "dashed") +
  theme(legend.position = "none", 
        panel.background = element_rect(fill = "white", colour = "grey50"))

cca1

```

```{r}

out <- ampvis2::amp_heatmap(amp,
            tax_show = 200,
            tax_aggregate = "OTU",
            # tax_add = "Genus",
            group_by = "Group",
            plot_values_size = 4, 
            showRemainingTaxa = TRUE, textmap = TRUE)


out <- out %>% 
  mutate_all(~round(., 2)) %>% 
  rownames_to_column('ID') %>% 
  left_join(ps.f@tax_table %>% data.frame() %>% rownames_to_column('ID'))

  

write_tsv(out, 'out_16S.tsv')


out <- ampvis2::amp_heatmap(ampits,
            tax_show = 200,
            tax_aggregate = "OTU",
            # tax_add = "Genus",
            group_by = "Group",
            plot_values_size = 4, 
            showRemainingTaxa = TRUE, textmap = TRUE)


out <- out %>% 
  mutate_all(~round(., 2)) %>% 
  rownames_to_column('ID') %>% 
  left_join(psits@tax_table %>% data.frame() %>% rownames_to_column('ID'))

write_tsv(out, 'out_its2.tsv')

```


```{r, fig.width=12, fig.height=12}

library(ggrepel)

library(ggrepel)

# physeq <- ps.clr
physeq <- ps.m

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}

otus.ps.vegan <- veganifyOTU(physeq)
rownames(otus.ps.vegan) <- physeq@sam_data$Group
metadata <- as(sample_data(physeq), "data.frame") 

otus.ps.vegan %>% 
  data.frame()
metadata

cca <- vegan::cca(otus.ps.vegan ~  Зольность + Потери + Влажность + Нитраты + Дыхание + Азот + Углерод + Гумус, data=metadata)
wa <- as.data.frame(cca$CCA$v) %>% 
  rownames_to_column("ID")

taxa.pruned <- as.data.frame(physeq@tax_table@.Data) %>% 
  rownames_to_column("ID")

taxa.pruned <- taxa.pruned %>%  
  mutate_all(as.character)

# old.tips <- tree$tip.label
# matches <- regmatches(tree$tip.label, gregexpr("[[:digit:]]+", tree$tip.label))
# taxa.pruned$number <- as.numeric(unlist(matches))
# Shitty taxa formating part

taxa.pruned$taxa <- ifelse(is.na(taxa.pruned$Genus),
                            ifelse(is.na(taxa.pruned$Family),
                            ifelse(is.na(taxa.pruned$Order), 
                            ifelse(is.na(taxa.pruned$Class), taxa.pruned$Phylum, taxa.pruned$Class) , taxa.pruned$Order), taxa.pruned$Family), taxa.pruned$Genus)
taxa.pruned[taxa.pruned == "Burkholderia-Caballeronia-Paraburkholderia"] <- "Burkholderia"
taxa.pruned[taxa.pruned == "Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium"] <- "Pararhizobium"

taxa.pruned$taxa2 <- ifelse(is.na(taxa.pruned$Species),
                            with(taxa.pruned, paste0(taxa)),
                            with(taxa.pruned, paste0(taxa, " ", Species )))
taxa.pruned$phylum <- ifelse(taxa.pruned$Phylum == "Proteobacteria", with(taxa.pruned, paste0(taxa.pruned$Class)), with(taxa.pruned, paste0(taxa.pruned$Phylum)))
taxa.pruned$Label <- paste0(taxa.pruned$ID, "_", taxa.pruned$taxa2)

wa <- full_join(wa, taxa.pruned, by="ID") %>% 
  mutate(Score = "sites")

#For the top 50 most abundant taxa, skip if use des res
head_asv <- physeq@otu_table@.Data %>% 
  data.frame %>% 
  colSums() %>% 
  sort(decreasing = TRUE) %>% 
  head(n=100) %>% 
  names()
  

wa <- filter(wa, ID %in% head_asv)
# maybe only different by des? Picture will be nice.
# wa <- filter(wa, ID %in% row.des)
# wa <- filter(wa, ID %in% row.des.so)

biplot <- as.data.frame(cca$CCA$biplot)
biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))

fdat_amazing <- plyr::rbind.fill(biplot, wa) %>% 
  
  mutate(Label = as.factor(Label))
fdat_amazing <- fdat_amazing %>%
  mutate(label_size = ifelse(Score == "biplot", 4, 3))

p.cca.species <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2)) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), aes(x = 0, xend = CCA1, y = 0, yend = CCA2), alpha=0.8, color = "red", arrow = arrow(angle = 3)) +
  geom_text_repel(aes(x=CCA1, y=CCA2, label= Label, colour = phylum),force=15, size=fdat_amazing$label_size) + 
  xlab(paste0(round(list.pe[1], 3)*100, "% CCA1")) +
  ylab(paste0(round(list.pe[2], 3)*100, "% CCA2")) +
  grids(linetype = "dashed") +
  geom_vline(xintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  geom_hline(yintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  theme(legend.position = "top", panel.background = element_rect(fill = "white", colour = "grey50"))

p.cca.species


```

### its


```{r}

setwd("~/storage/ov/")

library(readxl)
library(tidyverse)
library(phyloseq)


# 
# list.files()
# read_excel("fallow2023_meta.xlsx")
# path <- "fallow2023_meta.xlsx"


otu_table <- read_tsv("its2/otu_table.txt") %>% 
  column_to_rownames("...1")

hue <- otu_table %>% 
  colnames()

colnames(otu_table) <- unlist(purrr::map(stringr::str_split(hue, "\\.", 3), function(x) paste0(x[[2]], '_' ,x[[3]]))) 
hue_mod <- colnames(otu_table)

map <- data.frame(
  ID = hue_mod,
  Substrate = unlist(purrr::map(stringr::str_split(hue_mod, "\\_", 3), function(x) x[[2]] %>% substring(1, 1))),
  Repeat = unlist(purrr::map(stringr::str_split(hue_mod, "\\_", 3), function(x) x[[2]] %>% substring(2, 3))),
  Time = unlist(purrr::map(stringr::str_split(hue_mod, "\\_", 3), function(x) x[[2]] %>% substring(5, 10)))
)

map <- map %>% 
  mutate(Time = as.factor(Time) %>% forcats::fct_recode("30" = "0", "68" = "I", "124" = "II") %>% as.character() %>% as.double()) %>%
  mutate(Substrate = as.factor(Substrate) %>% forcats::fct_recode("CN" = "1", "CN_БАГС" = "2", "CN_ЦА" = "3")) %>% 
  mutate(Group = paste0(Substrate, '_' ,Time) %>% as.factor() %>% forcats::fct_relevel(gtools::mixedsort)) %>% 
  column_to_rownames("ID")

otus <- otu_table %>% 
  as.matrix() %>% 
  t()

taxa <- read_tsv("its2/taxa.txt") %>% 
  column_to_rownames("...1") %>% 
  as.matrix()
taxa[taxa == "Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium"] <- "Allorhizobium"
psits <- phyloseq(otu_table(otus, taxa_are_rows=FALSE), 
               sample_data(map),
               tax_table(taxa))


psits@tax_table <- psits@tax_table %>% 
  data.frame() %>% 
  select(-`...9`) %>% 
  mutate_all(~ str_sub(. , 4, -1)) %>% 
  as.matrix() %>% 
  phyloseq::tax_table()

psits@tax_table

```

```{r, fig.width=9, fig.height=7}

plot_rich_reads_samlenames_lm(psits, group = "Group", label = "Group")  

ps.f <- prune_samples(sampleNames(ps.f) != 'sol_1a2.I', ps.f)
ps.f <- prune_samples(sampleNames(ps.f) != 'sol_3a2.0', ps.f)

plot_rich_reads_samlenames_lm(ps.f, group = "Group", label = "Group")  

```

```{r, fig.width=9, fig.height=7}

beta_custom_norm_NMDS_elli_w(psits, Group = 'Group', Color = 'Substrate')

```



```{r, fig.width=12, fig.height=6}

plot_alpha_w_toc_mod <- function(ps, group, metric) {
  
  require(phyloseq)
  require(ggplot2)
  
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  
  er <- estimate_richness(ps_a)
  df_er <- cbind(ps_a@sam_data, er)
  df_er <- df_er %>% select(c(group, metric))
  stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
    rstatix::tukey_hsd() %>%
    filter(p.adj.signif != "ns")
  y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))

  plot_richness(ps_a, x=group, measures=metric, color="Substrate") + 
    geom_boxplot() +
    geom_point(size=1.2, alpha=0.3) +
    # geom_boxplot(stat="identity", position=position_dodge()) +
    # ggpubr::stat_pvalue_manual(
    #   stat.test,
    #   label = "p.adj.signif",
    #   y.position = y,
    #   tip.length = 0.005,
    #   bracket.nudge.y = 1.5,
    #   vjust = 0.6,
    #   size = 3) +
    theme_light() + 
        scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8) +
    theme(axis.text.x = element_text(size=12, angle = 45, hjust=1),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),
          legend.position = "None") +
    labs(y=paste(metric, "index")) 
}


p_a1 <-  plot_alpha_w_toc_mod(psits, group="Group",  metric="Observed") 
p_a2 <- plot_alpha_w_toc_mod(psits, group="Group", metric="Shannon") 
p_a3 <- plot_alpha_w_toc_mod(psits, group="Group", metric="InvSimpson") 

p_alpha <- ggpubr::ggarrange(p_a1, p_a2, p_a3, nrow = 1)
p_alpha
```


```{r}

ps.anc.liv <- ps.gra.fie

ps.anc.liv <- prune_samples(sample_data(ps.anc.liv)$wheat == 'true', ps.anc.liv)
ps.anc.liv  <- prune_taxa(taxa_sums(ps.anc.liv) > 0, ps.anc.liv)


dist <- phyloseq::distance(ps.anc.liv, "euclidean")
metadata <- as(sample_data(ps.anc.liv@sam_data), "data.frame")
vegan::adonis2(dist ~ preparation, data = metadata)

```



```{r,fig.height=4, fig.width=9}

ampits <- phyloseq_to_ampvis2(psits)
ampvis2::amp_heatmap(ampits,
            tax_show = 4,
            tax_aggregate = "Phylum",
            group_by = "Group",
            plot_values_size = 4, 
            facet_by = "Substrate",
            showRemainingTaxa = TRUE)

```

```{r,fig.height=7, fig.width=9}

ampvis2::amp_heatmap(ampits,
            tax_show = 30,
            tax_aggregate = "OTU",
            tax_add = "Genus",
            group_by = "Group",
            plot_values_size = 4, 
            facet_by = "Substrate",
            showRemainingTaxa = TRUE)

```


```{r, , fig.width=8, fig.height=8}

library(ggrepel)

# physeq <- ps.clr
physeq <- ps.gra.r

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}

otus.ps.vegan <- veganifyOTU(physeq)
rownames(otus.ps.vegan) <- physeq@sam_data$Group
metadata <- as(sample_data(physeq), "data.frame") 

otus.ps.vegan %>% 
  data.frame()
metadata

cca <- vegan::cca(otus.ps.vegan ~  рН + NH4 + NO3 + P + K + ear_count + ear_mass + sheaf_mass, data=metadata)

biplot <- as.data.frame(cca$CCA$biplot)
wa <- as.data.frame(cca$CCA$wa)

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot)))) %>% 
  mutate(
  Type = rep("biplot", length(rownames(biplot))),
)

wa <- rownames_to_column(wa, "Label") %>% 
  add_column(Score = rep("sites", length(rownames(wa)))) %>% mutate(
  Type = metadata$Group,
)

fdat_amazing <- rbind(biplot, wa) 

fdat_amazing <- fdat_amazing %>%
  mutate(label_size = ifelse(Score == "biplot", 4.5, 3)) %>% 
  mutate_if(is.character, as.factor)

summary.cca <- summary(cca)

list.pe <- summary.cca$cont$importance %>% 
  as.data.frame() %>% 
  select(c("CCA1", "CCA2")) %>% 
  dplyr::slice(2L) %>% 
  t() %>% 
  as.vector()



# kate.ggcca.sites <- function(vare.cca){
# 
#   require(tidyverse)
#   biplot <- as.data.frame(vare.cca$CCA$biplot)
#   wa <- as.data.frame(vare.cca$CCA$wa)
# 
#   biplot <- rownames_to_column(biplot, "Label") %>% 
#     add_column(Score = rep("biplot", length(rownames(biplot))))
#   wa <- rownames_to_column(wa, "Label") %>% 
#     add_column(Score = rep("sites", length(rownames(wa))))
#   fdat_amazing <- rbind(biplot, wa) 
#   fdat_amazing <- fdat_amazing %>%
#     mutate(label_size = ifelse(Score == "biplot", 4.5, 3))
#  
#   p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
#     # geom_abline(intercept=seq(-100, 100, 25), slope=1, colour="grey", alpha=0.5) +
#     # geom_abline(intercept=seq(-100, 100, 25), slope=-1, colour="grey", alpha=0.5) +
#     geom_vline(xintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
#     geom_hline(yintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
#     geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), 
#                mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
#     geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), 
#                  aes(x = 0, xend = CCA1, y = 0, yend = CCA2),
#                   size = 0.6, 
#                  alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
#   geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=fdat_amazing$label_size) + 
#   xlab(paste0(round(cca$CCA$eig[1], 3)*100, "% CCA1")) +
#   ylab(paste0(round(cca$CCA$eig[2], 3)*100, "% CCA2")) +
#   grids(linetype = "dashed") +
#   theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
#   return(p)
# }
# 
# kate.ggcca.sites(cca)
fdat_amazing

fdat_amazing <- fdat_amazing %>%  
  separate(Label, c("Label", NA), sep = '\\.', convert=TRUE)



cca1 <- ggplot(fdat_amazing) + 
  geom_vline(xintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  geom_hline(yintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  geom_point(data = fdat_amazing %>% 
               dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score)))   + 
  ggforce::geom_mark_ellipse(data = fdat_amazing %>%
                                      dplyr::filter(Score == "sites"),
                             aes(x=CCA1, y=CCA2, group = Label, label = Label, col = 'grey'),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                  con.colour='salmon',
                  label.colour='salmon'
                  ) +
  geom_segment(data = fdat_amazing %>% 
                 dplyr::filter(Score == "biplot"), 
               aes(x = 0, xend = CCA1, y = 0, yend = CCA2),
               size = 0.6, 
               alpha=0.8, 
               color = "red",
               arrow = arrow(angle = 3)) +
  ggrepel::geom_text_repel(data = fdat_amazing %>% 
                 dplyr::filter(Score == "biplot"), aes(x=CCA1, y=CCA2, label= Label), size=5) +
  xlab(paste0(round(list.pe[1], 3)*100, "% CCA1")) +
  ylab(paste0(round(list.pe[2], 3)*100, "% CCA2")) +
  grids(linetype = "dashed") +
  theme(legend.position = "none", 
        panel.background = element_rect(fill = "white", colour = "grey50"))

cca1

```



```{r, fig.width=12, fig.height=12}

library(ggrepel)

library(ggrepel)

# physeq <- ps.clr
physeq <- ps.gra.r

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}

otus.ps.vegan <- veganifyOTU(physeq)
rownames(otus.ps.vegan) <- physeq@sam_data$Group
metadata <- as(sample_data(physeq), "data.frame") 

otus.ps.vegan %>% 
  data.frame()
metadata

cca <- vegan::cca(otus.ps.vegan ~  рН + NH4 + NO3 + P + K + ear_count + ear_mass + sheaf_mass, data=metadata)
wa <- as.data.frame(cca$CCA$v) %>% 
  rownames_to_column("ID")

taxa.pruned <- as.data.frame(physeq@tax_table@.Data) %>% 
  rownames_to_column("ID")

taxa.pruned <- taxa.pruned %>%  
  mutate_all(as.character)

# old.tips <- tree$tip.label
# matches <- regmatches(tree$tip.label, gregexpr("[[:digit:]]+", tree$tip.label))
# taxa.pruned$number <- as.numeric(unlist(matches))
# Shitty taxa formating part

taxa.pruned$taxa <- ifelse(is.na(taxa.pruned$Genus),
                            ifelse(is.na(taxa.pruned$Family),
                            ifelse(is.na(taxa.pruned$Order), 
                            ifelse(is.na(taxa.pruned$Class), taxa.pruned$Phylum, taxa.pruned$Class) , taxa.pruned$Order), taxa.pruned$Family), taxa.pruned$Genus)
taxa.pruned[taxa.pruned == "Burkholderia-Caballeronia-Paraburkholderia"] <- "Burkholderia"
taxa.pruned[taxa.pruned == "Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium"] <- "Pararhizobium"

taxa.pruned$taxa2 <- ifelse(is.na(taxa.pruned$Species),
                            with(taxa.pruned, paste0(taxa)),
                            with(taxa.pruned, paste0(taxa, " ", Species )))
taxa.pruned$phylum <- ifelse(taxa.pruned$Phylum == "Proteobacteria", with(taxa.pruned, paste0(taxa.pruned$Class)), with(taxa.pruned, paste0(taxa.pruned$Phylum)))
taxa.pruned$Label <- paste0(taxa.pruned$ID, "_", taxa.pruned$taxa2)

wa <- full_join(wa, taxa.pruned, by="ID") %>% 
  mutate(Score = "sites")

#For the top 50 most abundant taxa, skip if use des res
head_asv <- physeq@otu_table@.Data %>% 
  data.frame %>% 
  colSums() %>% 
  sort(decreasing = TRUE) %>% 
  head(n=100) %>% 
  names()
  

wa <- filter(wa, ID %in% head_asv)
# maybe only different by des? Picture will be nice.
# wa <- filter(wa, ID %in% row.des)
# wa <- filter(wa, ID %in% row.des.so)

biplot <- as.data.frame(cca$CCA$biplot)
biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot))))

fdat_amazing <- plyr::rbind.fill(biplot, wa) %>% 
  
  mutate(Label = as.factor(Label))
fdat_amazing <- fdat_amazing %>%
  mutate(label_size = ifelse(Score == "biplot", 4, 3))

p.cca.species <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
  geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2)) + 
  geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), aes(x = 0, xend = CCA1, y = 0, yend = CCA2), alpha=0.8, color = "red", arrow = arrow(angle = 3)) +
  geom_text_repel(aes(x=CCA1, y=CCA2, label= Label, colour = phylum),force=15, size=fdat_amazing$label_size) + 
  xlab(paste0(round(list.pe[1], 3)*100, "% CCA1")) +
  ylab(paste0(round(list.pe[2], 3)*100, "% CCA2")) +
  grids(linetype = "dashed") +
  geom_vline(xintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  geom_hline(yintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  theme(legend.position = "top", panel.background = element_rect(fill = "white", colour = "grey50"))




```
```{r, fig.width=12, fig.height=6}

ggpubr::ggarrange(cca1, p.cca.species, common.legend = TRUE)

```
#here

```{r, fig.width=12, fig.height=12}


read_excel("fallow2023_meta.xlsx")
path <- "fallow2023_meta.xlsx"

mad <- path %>%
        excel_sheets() %>%
        set_names() %>%
       purrr::map_df(read_excel,
           path = path,
           .id = "type")


mad <- mad %>% 
  select(-c("type"))

mad_fal <- mad %>% group_by(sample) %>%  mutate(across(where(is.numeric), ~replace_na(., median(., na.rm=TRUE)))) %>% ungroup()
mad_fal


ps.fal <- prune_samples(sample_data(ps.f)$fallow == TRUE, ps.f)
ps.fal  <- prune_taxa(taxa_sums(ps.fal) > 0, ps.fal)
ps.fal


ps.fal@sam_data <- ps.fal@sam_data %>%
  data.frame() %>% 
  rownames_to_column('ID') %>% 
  mutate(id = as.double(str_match(ID, "(?<=4\\.).+$"))) %>% 
  arrange(id) %>% 
  mutate(id = seq(1, 28)) %>% 
  left_join(mad_fal, by = 'id')%>% 
  column_to_rownames('ID') %>% 
  select(-c('id', 'grain' , 'fallow')) %>% 
  mutate_if(is.character, as.factor) %>% 
  mutate(age = fct_relevel(age, c('benchmark', '120', '80', '40'))) %>% 
  mutate(sample = fct_relevel(sample, c('BA1_O', 'BA1_e', 'BA2_PY', 'BA2_O', 'BA3_PY', 'BA4_PY', 'BA4_BF'))) %>% 
  rename('Group' = 'sample') %>% 
  sam_data()


ps.fal@sam_data

plot_rich_reads_samlenames_lm(ps.fal, group = "Group")
ps.fal <- prune_samples(sampleNames(ps.fal) != 'Abakumov.SEQ284.44', ps.fal)
ps.fal <- prune_samples(sampleNames(ps.fal) != 'Abakumov.SEQ284.48', ps.fal)
# ps.fal <- prune_samples(sampleNames(ps.gra) != 'Abakumov.SEQ283.19', ps.gra)
plot_rich_reads_samlenames_lm(ps.fal, group = "Group")
ps.fal.r <- rarefy_even_depth(ps.fal)
plot_rich_reads_samlenames_lm(ps.fal.r, group = "Group")

```

```{r, fig.width=11, fig.height=10}
plot_rich_reads_samlenames_lm(ps.fallen, group = "Group")
ps.fallen <- prune_samples(sample_data(ps.fal)$Group == 'BA2_PY', ps.fal)
ps.fallen  <- prune_taxa(taxa_sums(ps.fallen) > 0, ps.fallen)

ps.fal@sam_data
ps.fallen@sam_data
amp.fallen <- phyloseq_to_ampvis2(ps.fallen)

amp_heatmap(amp.fallen,
            tax_show = 50,
            tax_aggregate = "Genus",
            tax_add = "Phylum",
            plot_values_size = 4, facet_by = "age",
            showRemainingTaxa = TRUE)

amp_heatmap(amp.fallen,
            tax_show = 50,
            tax_aggregate = "Phylum",
            plot_values_size = 4, facet_by = "age",
            showRemainingTaxa = TRUE)



amp_rarecurve(amp.fallen)

```

```{r}

amp.fallen$abund %>% 
  rownames_to_column('id') %>% 
  dplyr::left_join(amp.fallen$tax %>% rownames_to_column('id')) %>% 
  select(-c('Abakumov.SEQ284.45', 'Abakumov.SEQ284.46', 'Abakumov.SEQ284.47')) %>% 
  arrange(Abakumov.SEQ284.48) %>% 
  filter(Abakumov.SEQ284.48 > 0)

amp.fallen$abund 
amp.fallen$tax

```



```{r, fig.width=7, fig.height=7}

beta_custom_norm_NMDS_elli_w(ps.fal, Color = "age", Group = "Group")

```


```{r, fig.width=12, fig.height=6}

plot_alpha_w_toc_mod <- function(ps, group, metric) {
  
  require(phyloseq)
  require(ggplot2)
  
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  
  er <- estimate_richness(ps_a)
  df_er <- cbind(ps_a@sam_data, er)
  df_er <- df_er %>% select(c(group, metric))
  stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
    rstatix::tukey_hsd() %>%
    filter(p.adj.signif != "ns")
  y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))

  plot_richness(ps_a, x=group, measures=metric, color="Group") + 
    geom_boxplot() +
    geom_point(size=1.2, alpha=0.3) +
    # ggpubr::stat_pvalue_manual(
    #   stat.test, 
    #   label = "p.adj.signif", 
    #   y.position = y,
    #   tip.length = 0.005, 
    #   bracket.nudge.y = 1.5,
    #   vjust = 0.6, 
    #   size = 3) +
    theme_light() + 
        scale_colour_viridis_d(option = "magma", 
                       aesthetics = "color", 
                       begin = 0, 
                       end = 0.8) +
    theme(axis.text.x = element_text(size=12, angle = 45, hjust=1),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),
          legend.position = "None") +
    labs(y=paste(metric, "index")) 
}


p_a1 <-  plot_alpha_w_toc_mod(ps.fal, group="Group",  metric="Observed") 
p_a2 <- plot_alpha_w_toc_mod(ps.fal, group="Group", metric="Shannon") 
p_a3 <- plot_alpha_w_toc_mod(ps.fal, group="Group", metric="InvSimpson") 

p_alpha <- ggpubr::ggarrange(p_a1, p_a2, p_a3, nrow = 1)
p_alpha

```



```{r}

ps.fal@sam_data

```

```{r, fig.width=15}


amp.fal <- phyloseq_to_ampvis2(ps.fal)

amp_heatmap(amp.fal,
            tax_show = 30,
            tax_aggregate = "Genus",
            tax_add = "Phylum",
            group_by = "Group",
            plot_values_size = 7, facet_by = "age",
            showRemainingTaxa = TRUE)


ps.gra@tax_table
amp.gra$metadata %>% 
  as.data.frame() %>% 
  filter(soil_type == 'field') 

```


```{r,  fig.height=10, fig.width=12}

# amp.gra.f <- ampvis2::amp_filter_samples(amp.gra, wheat=='true', normalise = TRUE)
amp_heatmap(amp.fal,
            tax_show = 50,
            tax_aggregate = "OTU",
            tax_add = "Genus",
            group_by = "Group",
            plot_values_size = 4, facet_by = "soil_type",
            showRemainingTaxa = TRUE)

```



```{r}



```

```{r,fig.height=7, fig.width=9}

amp_heatmap(amp.fal,
            tax_show = 13,
            tax_aggregate = "Phylum",tax_class = "Pseudomonadota",
            group_by = "Group",
            plot_values_size = 4, facet_by = "age",
            showRemainingTaxa = TRUE)


```

```{r}




```

```{r, , fig.width=8, fig.height=8}

library(ggrepel)

# physeq <- ps.clr
physeq <- ps.fal.r

veganifyOTU <- function(physeq){
  require(phyloseq)
  if(taxa_are_rows(physeq)){physeq <- t(physeq)}
  return(as(otu_table(physeq), "matrix"))
}

otus.ps.vegan <- veganifyOTU(physeq)
rownames(otus.ps.vegan) <- physeq@sam_data$Group
metadata <- as(sample_data(physeq), "data.frame") 

cca <- vegan::cca(otus.ps.vegan ~  С_per+рН + NH4 + NO3 + P + K, data=metadata)
metadata


biplot <- as.data.frame(cca$CCA$biplot)
wa <- as.data.frame(cca$CCA$wa)

biplot <- rownames_to_column(biplot, "Label") %>% 
  add_column(Score = rep("biplot", length(rownames(biplot)))) %>% 
  mutate(
  Type = rep("biplot", length(rownames(biplot))),
)

wa <- rownames_to_column(wa, "Label") %>% 
  add_column(Score = rep("sites", length(rownames(wa)))) %>% mutate(
  Type = metadata$Group,
)

fdat_amazing <- rbind(biplot, wa) 

fdat_amazing <- fdat_amazing %>%
  mutate(label_size = ifelse(Score == "biplot", 4.5, 3)) %>% 
  mutate_if(is.character, as.factor)

summary.cca <- summary(cca)

list.pe <- summary.cca$cont$importance %>% 
  as.data.frame() %>% 
  select(c("CCA1", "CCA2")) %>% 
  dplyr::slice(2L) %>% 
  t() %>% 
  as.vector()



# kate.ggcca.sites <- function(vare.cca){
# 
#   require(tidyverse)
#   biplot <- as.data.frame(vare.cca$CCA$biplot)
#   wa <- as.data.frame(vare.cca$CCA$wa)
# 
#   biplot <- rownames_to_column(biplot, "Label") %>% 
#     add_column(Score = rep("biplot", length(rownames(biplot))))
#   wa <- rownames_to_column(wa, "Label") %>% 
#     add_column(Score = rep("sites", length(rownames(wa))))
#   fdat_amazing <- rbind(biplot, wa) 
#   fdat_amazing <- fdat_amazing %>%
#     mutate(label_size = ifelse(Score == "biplot", 4.5, 3))
#  
#   p <- ggplot(fdat_amazing %>% filter(Score %in% c("sites","biplot"))) + 
#     # geom_abline(intercept=seq(-100, 100, 25), slope=1, colour="grey", alpha=0.5) +
#     # geom_abline(intercept=seq(-100, 100, 25), slope=-1, colour="grey", alpha=0.5) +
#     geom_vline(xintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
#     geom_hline(yintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
#     geom_point(data = fdat_amazing %>% dplyr::filter(Score == "sites"), 
#                mapping = aes(x=CCA1, y=CCA2, colour = factor(Score))) + 
#     geom_segment(data = fdat_amazing %>% dplyr::filter(Score == "biplot"), 
#                  aes(x = 0, xend = CCA1, y = 0, yend = CCA2),
#                   size = 0.6, 
#                  alpha=0.8, color = "red",arrow = arrow(angle = 3)) +
#   geom_text_repel(aes(x=CCA1, y=CCA2, label= Label),size=fdat_amazing$label_size) + 
#   xlab(paste0(round(cca$CCA$eig[1], 3)*100, "% CCA1")) +
#   ylab(paste0(round(cca$CCA$eig[2], 3)*100, "% CCA2")) +
#   grids(linetype = "dashed") +
#   theme(legend.position = "none", panel.background = element_rect(fill = "white", colour = "grey50"))
#   return(p)
# }
# 
# kate.ggcca.sites(cca)
fdat_amazing

fdat_amazing <- fdat_amazing %>%  
  separate(Label, c("Label", NA), sep = '\\.', convert=TRUE)



ggplot(fdat_amazing) + 
  geom_vline(xintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  geom_hline(yintercept = 0, size = 0.75, color = "#737373", alpha=0.5) +
  geom_point(data = fdat_amazing %>% 
               dplyr::filter(Score == "sites"), mapping = aes(x=CCA1, y=CCA2, colour = factor(Score)))   + 
  ggforce::geom_mark_ellipse(data = fdat_amazing %>%
                                      dplyr::filter(Score == "sites"),
                             aes(x=CCA1, y=CCA2, group = Label, label = Label, col = 'grey'),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                  con.colour='salmon',
                  label.colour='salmon'
                  ) +
  geom_segment(data = fdat_amazing %>% 
                 dplyr::filter(Score == "biplot"), 
               aes(x = 0, xend = CCA1, y = 0, yend = CCA2),
               size = 0.6, 
               alpha=0.8, 
               color = "red",
               arrow = arrow(angle = 3)) +
  ggrepel::geom_text_repel(data = fdat_amazing %>% 
                 dplyr::filter(Score == "biplot"), aes(x=CCA1, y=CCA2, label= Label), size=5) +
  xlab(paste0(round(list.pe[1], 3)*100, "% CCA1")) +
  ylab(paste0(round(list.pe[2], 3)*100, "% CCA2")) +
  grids(linetype = "dashed") +
  theme(legend.position = "none", 
        panel.background = element_rect(fill = "white", colour = "grey50"))

metadata %>%  colnames()
```




```{r, fig.width=9, fig.height=9}

library(ggVennDiagram)

ps.fal.e <- prune_samples(sample_data(ps.fal)$horizon %in% c('PY', 'E1', 'PYe'), ps.fal)
ps.fal.e  <- prune_taxa(taxa_sums(ps.fal.e) > 0, ps.fal.e)

plot_vienn <- function(ps, group){
  physeq <- prune_taxa(taxa_sums(ps) > 0, ps)
  groups <- levels(sample_data(physeq)[[group]] %>% as.factor())
  data <- merge_samples(physeq, group) %>% 
    psmelt() %>% 
    group_by(Sample, OTU) %>% 
    summarise(ASVs_abund = list(paste(OTU, 1:sum(Abundance))), Abund = sum(Abundance), .groups='keep') %>% 
    filter(Abund > 0)
  
  asvs <- data %>% select(Sample, OTU) %>% group_by(Sample) %>% summarise(ASVs = list(OTU)) %>% as.list()
  d1 <- asvs[[2]]
  names(d1) <- asvs[[1]]
  d1
  
  weighted.asvs <- data %>% select(Sample, ASVs_abund) %>% group_by(Sample) %>% summarise(ASVs = list(unlist(ASVs_abund)))
  d2 <- weighted.asvs[[2]]
  names(d2) <- weighted.asvs[[1]]
  d2
  
  
  list(ggVennDiagram(d1) + ggtitle("ASVs") + scale_fill_distiller(palette = "OrRd", trans = "reverse"),
       ggVennDiagram(d2) + ggtitle("Reads") + scale_fill_distiller(palette = "OrRd", trans = "reverse"))
}

plot_vienn(ps.fal.e, "Group")

```


```{r}

ps.syk.f2 <- phyloseq::filter_taxa(ps.fal.e, function(x) sum(x > 10) > (0.1*length(x)), TRUE)
ps.syk.f2

out.f <-  ANCOMBC::ancombc2(data=ps.syk.f2, 
  fix_formula = "age",
  prv_cut = 0,
  p_adj_method = "BH",
  tax_level = 'Phylum',
  rand_formula= NULL,
  pseudo = 0, 
  pseudo_sens = TRUE,
  s0_perc = 0.05,
  group = "age",
  struc_zero = TRUE,
  neg_lb = TRUE,
  alpha = 0.05,
  n_cl = 20, 
  verbose = TRUE,
  global = TRUE, 
  pairwise = FALSE, 
  dunnet = FALSE, 
  trend = FALSE
  )
  
out.f

unregister <- function() {
  env <- foreach:::.foreachGlobals
  rm(list=ls(name=env), pos=env)
}

unregister()
out.f

out.f$res_global %>% 
  filter(diff_abn == TRUE) %>% 
  left_join(ps.syk.f2@tax_table %>% as.data.frame() %>% rownames_to_column('taxon'))

```

