
---
title: "cellulo_com"
author: "GrGladkov"
date: "5/19/2021"
output: html_document
---
```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/storage/yam2")

```


```{r}

# 
# find /mnt/nas/runs/2017_illumina_1/20170818_MiSeq -regextype posix-extended -regex '^.*S((1[9][6-9])|(2[0-5][0-9])|26[0-5])_.*$' | while read files; do
#     filename=$(basename $files)
#     ln -s $files /home/gladkov/storage/mal/raw/$filename
#     echo Linked $files to /home/gladkov/storage/mal/raw/$filename
# done

setwd("~/storage/yam2")
path = "raw/"
path_trein_set = "~/storage/somebases/silva_nr99_v138.1_train_set.fa.gz"
path_trein_set_species = "~/storage/somebases/silva_species_assignment_v138.1.fa.gz"
name_Brief = "nameBrief.txt"

truncLen = "210,160"
maxEE = "2,5"
mult = 10
mlt = 10

require(dada2)
require(Biostrings)
require(DECIPHER)
require(phyloseq)
require(seqinr)
require(data.table)
require(metagMisc)
require(tibble)
library(phyloseq)
library(tidyverse)
library(vegan)

fnFs <- sort(list.files(path, pattern="_R1_001.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_001.fastq.gz", full.names = TRUE))

length(fnFs)
fnRs
list.files()
```



```{r}

plotQualityProfile(fnFs, aggregate = TRUE)

```

```{r}

plotQualityProfile(fnRs, aggregate = TRUE)

```

```{r}

sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
on.exit()
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(200,180), trimLeft=c(19,20), maxN=0, truncQ=2, maxEE=c(2,4), rm.phix=TRUE, compress=TRUE, multithread=mult)
errF <- learnErrors(filtFs, multithread=mult)
errR <- learnErrors(filtRs, multithread=mult)
derepFs <- derepFastq(filtFs, verbose=TRUE)
derepRs <- derepFastq(filtRs, verbose=TRUE)

# Name the derep-class objects by the sample names
names(derepFs) <- sample.names
names(derepRs) <- sample.names
dadaFs <- dada(derepFs, err=errF, multithread=mult)
dadaRs <- dada(derepRs, err=errR, multithread=mult)
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose=TRUE)
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
table(nchar(getSequences(seqtab)))
getN <- function(x) sum(getUniques(x))
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=mult, verbose=TRUE)

briefToSeq <- colnames(seqtab.nochim)
names(briefToSeq) <- paste0("Seq", seq(ncol(seqtab.nochim))) 
st.brief <- seqtab.nochim
colnames(st.brief) <- names(briefToSeq)

write.table(briefToSeq, file = name_Brief, sep = "\t")

#dna <- DNAStringSet(briefToSeq)
#alignment <- AlignSeqs(DNAStringSet(dna), anchor=NA,verbose=FALSE, processors = mlt)
#writeXStringSet(alignment, file="align.fasta")

taxa.dada2 <- assignTaxonomy(briefToSeq, path_trein_set , multithread=mult)
taxa.dada2.species <- assignSpecies(briefToSeq, path_trein_set_species) # maybe use not seqs but brief 
rownames(taxa.dada2.species) <- rownames(briefToSeq)
briefToSeq.df <- data.frame(briefToSeq)
rownames(taxa.dada2.species) <- rownames(briefToSeq.df)
rownames(taxa.dada2) <- rownames(taxa.dada2.species)
taxa <- cbind2(taxa.dada2, taxa.dada2.species[,2])
colnames(taxa)[7] <- "Species"

getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
write.table(track, file = "track.tsv", sep= "\t", col.names = NA, quote=FALSE)

st.brief.t.df <- data.frame(t(st.brief))
write.table(st.brief.t.df, file = "otu_table.txt", sep= "\t", col.names = NA, quote=FALSE)

briefToSeq.ls <- as.list(briefToSeq.df[,c("briefToSeq")])
briefToSeq.names <- as.list(rownames(briefToSeq.df))
write.fasta(briefToSeq.ls, briefToSeq.names , "rep_seq.fasta", as.string = FALSE)

write.table(taxa, file = "taxa.txt", sep= "\t", col.names = NA, quote=FALSE)

```




```{r}

read_csv2("otu_ta")
  
```

```{r}

as.data.frame(track) %>% 
  pivot_longer(
    names_to = "type",
    cols = c("input", "nonchim"),
    values_to = "reads"
) %>% 
  ggplot(aes(x=reads, color=type, fill=type)) +
  geom_histogram(alpha=0.3, position="identity", bins = 30) +
  scale_x_continuous(breaks=seq(0, 100000, 10000), limits=c(0,100000)) +
  theme_minimal()


```



```{r}

list.files()

```



```{r}

map <- read_tsv("map.cel.16.csv")
map <- column_to_rownames(map, "ID")
map

#read taxa file
taxa <- read.csv("taxa.txt" , header=TRUE, sep="\t")
taxa <- column_to_rownames(taxa, 'X')
taxa <- as.matrix(taxa)


#read otu file
filt.otu <- read.csv("otu_table.txt" , header=TRUE, sep="\t")
colnames(filt.otu)[1] <- "ID"
filt.otu <- column_to_rownames(filt.otu, "ID")

st.brief.t.df %>% 
  head()

naturalsort::naturalsort(colnames(filt.otu))

filt.otu <- filt.otu[,naturalsort::naturalsort(colnames(filt.otu))] 

colnames(filt.otu)

names_amazing <- colnames(filt.otu) %>% 
  str_split("Abakumov.") %>% 
  sapply(function(x) paste0("y", x[2]))

map <- read_delim("yam2_map.csv") %>% 
  mutate(Arina_ID = names_amazing) %>% 
  column_to_rownames("Arina_ID")

colnames(filt.otu) <- names_amazing
ps <- phyloseq(otu_table(t(filt.otu), taxa_are_rows=FALSE), 
               sample_data(map),
               tax_table(taxa))

as.data.frame(filt.otu)
map

ps

```


```{r}

st.brief.t.df %>% 
  head()

naturalsort::naturalsort(colnames(st.brief.t.df))

otus_d <- st.brief.t.df[,naturalsort::naturalsort(colnames(st.brief.t.df))] 

colnames(otus_d)

names_amazing <- colnames(otus_d) %>% 
  str_split("Abakumov.") %>% 
  sapply(function(x) paste0("y", x[2]))

names_amazing

colnames(otus_d) <- names_amazing

tree <- read_tree(treefile="tree")
ps.f2@phy_tree <- tree

```



```{r}

x <- rownames(track)

map <- data.frame(row.names = x,
           Day = as.factor(day),
           Description = as.factor(Decription)) 
map <- map %>% 
  column_to_rownames("Arina_ID")
map

ps <- phyloseq(otu_table(t(otus_d), taxa_are_rows=FALSE), 
               sample_data(map),
               tax_table(taxa))

ps


```



```{r}

ps_object <- ps 
ps_object <- subset_taxa(ps_object, Phylum != "NA")

ps_object@tax_table[is.na(ps_object@tax_table)] <- TRUE
ps_object <- subset_taxa(ps_object,
                         !(Family  == "Mitochondria" |
                             Class   == "Chloroplast" |
                             Order   == "Chloroplast"))
ps_object@tax_table <- dplyr::na_if(ps_object@tax_table, TRUE)
ps.f <- ps_object

```


```{r}

ps.f
saveRDS(ps.f, "ps.f")

```


```{r, fig.width=10, fig.height=8}

plot_rich_reads_samlenames_lm <- function(physeq){
  rish <- estimate_richness(physeq, measures = "Observed")
  reads.sum <- as.data.frame(sample_sums(physeq))
  reads.summary <- cbind(rish, reads.sum)
  colnames(reads.summary) <- c("otus","reads")
  reads.summary["Repeat"] <-unlist(purrr::map(stringr::str_split(rownames(physeq@sam_data), "y", 2), function(x) x[[2]]))
  reads.summary["Site"] <- physeq@sam_data$Site
  library(ggrepel)
  require(ggforce)
  p1 <- ggplot(data=reads.summary) + 
    geom_point(aes(y=otus, x=log2(reads), color=Site),size=3) + 
    geom_text_repel(aes(y=otus, x=log2(reads), label=paste0(Site, "_", Repeat))) + 
    theme_bw() +
    geom_smooth(aes(y=otus, x=log2(reads), fill=Site, color=Site),method=lm, se=FALSE, ymin = 1) + 
    scale_x_continuous(sec.axis = sec_axis(sec.axis ~ 2**.)) 

  return(p1)
}

plot_rich_reads_samlenames_lm(ps.f)
ps.f2 <- subset_samples(ps.f, sample_names(ps.f) != 'y8')

plot_rich_reads_samlenames_lm(ps.f2)

plot_rich_reads_samlenames_lm(ps)
```


```{r}

ps.f2 <- subset_samples(ps.f, sample_names(ps.f) != 'straw-16s-D07-1')

```



```{r, fig.width=10, fig.height=8}

beta_custom_norm_NMDS_elli_w <- function(ps, seed = 7888, normtype="vst", Color="What", Group="Repeat"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  library(ggforce)

  # попробуй заменить NMDS на PCoA
  # https://joey711.github.io/phyloseq/plot_ordination-examples.html и вот здесь куча вариантов ординаций в phyloseq - попробуй нарисовать отдельно для таксонов, а не образцов(не моей функцией, у меня зависнет, и лучше PCoA - так быстрее))
  ordination.b <- ordinate(ps, "NMDS", "bray")
  mds <- as.data.frame(ordination.b$points)
  p  <-  plot_ordination(ps,
                         ordination.b,
                         type="sample",
                         color = Color,
                         # title="NMDS - Bray-Curtis",
                         title=NULL,
                         axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    annotate("text",
             x=min(mds$MDS1) + abs(min(mds$MDS1))/10,
             y=max(mds$MDS2),
             label=paste0("Stress -- ", round(ordination.b$stress, 3))) +
    geom_mark_ellipse(aes_string(group = Group, label = Group),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                      con.colour='gray') +
    theme(legend.position = "none") 
  
  return(p)
}

beta_custom_norm_NMDS_elli_w(ps.ym, Color="Site", Group="Repeat")

```


```{r, fig.width=6, fig.height=5}


ps.ym <- prune_samples(sample_data(ps.f2)$Region %in% c("YAMAO"), ps.f2)
ps.ym  <- prune_taxa(taxa_sums(ps.ym) > 0, ps.ym)

ps.ya <- prune_samples(sample_data(ps.f2)$Region %in% c("Yakutsk"), ps.f2)
ps.ya  <- prune_taxa(taxa_sums(ps.ya) > 0, ps.ya)

p.wt <- beta_custom_norm_NMDS_elli_w(ps.ym, Color="Фон", Group="Site")
p.za <- beta_custom_norm_NMDS_elli_w(ps.ym, Color="Залежь", Group="Site")

p.za
p.wt
ps.f2@sam_data$Region

beta_custom_norm_NMDS_elli_w(ps.ya, Color="Site", Group="Repeat")

```



```{r}

phyloseq_to_amp <- function(ps){
  require(ampvis2)
  require(tibble)
  require(phyloseq)
  colnames(ps@tax_table@.Data) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
  OTU1 = as(otu_table(ps), "matrix")
  OTU1 <- t(OTU1)
  OTUdf = as.data.frame(OTU1)
  taxa.ps <- as(tax_table(ps), "matrix")
  taxa.df = as.data.frame(taxa.ps)
  my_otu_table <- merge(OTUdf, taxa.df, by=0)
  my_otu_table <- column_to_rownames(my_otu_table, var="Row.names")
  
  my_metadata <- as_tibble(sample_data(ps), rownames=NA)
  my_metadata <- rownames_to_column(my_metadata,var = "SampleID")
  # my_tree <- phy_tree(ps)
  amp.ps <- amp_load(otutable = my_otu_table, metadata = my_metadata)
                     # , tree = my_tree)
  return(amp.ps)
}

amp.ym <- phyloseq_to_amp(ps.ya)
amp.ym

amp_heatmap(amp.ym, tax_show = 15, group_by = "Repeat",tax_aggregate = "Phylum", tax_class = "Proteobacteria") +
  theme_bw() + 
  theme(text = element_text(size=15), legend.position = "none") + 
  theme(axis.text.x=element_text(angle=45,hjust=1))

amp_venn(amp.ym, group_by = "Feature", cut_a = 0, cut_f = 0.001)

amp.ym

```
```{r}


ancom_da_glob_2 <- ancombc(phyloseq = ps.ym,
                         formula = "Feature",
                         p_adj_method = "fdr",
                         zero_cut = 0.90,
                         lib_cut = 1000,
                         group = "Фон",
                         struc_zero = TRUE,
                         neg_lb = TRUE,
                         tol = 1e-5,
                         max_iter = 100,
                         conserve = TRUE,
                         alpha = 0.05,
                         global = TRUE)


rt <- ancom_da_glob_2$res$diff_abn %>% 
  filter(ФонTRUE == TRUE) %>% 
  rownames()

ancom_da_glob_2$res$beta %>% 
  rownames_to_column("ID") %>% 
  left_join(taxa) %>% 
  filter(ID %in% rt) %>% 
  arrange(desc(ФонTRUE))

ancom_da_glob_2

```

```{r}

ps.ya@sam_data

ps.rel  <-  phyloseq::transform_sample_counts(ps.ya, function(x) x / sum(x))

ps.inall <- phyloseq::filter_taxa(ps.ya, function(x) sum(x > 10) > (0.1*length(x)), TRUE)
ps.inall


ps.exl  <- phyloseq::filter_taxa(ps.ya, function(x) sum(x > 10) < (0.1*length(x)), TRUE)
ps.exl  <- prune_taxa(taxa_sums(ps.exl) > 100, ps.exl)

ps.ya@sam_data

```

## functions

```{r}

#phyloseq object to ampvis2 object
#https://gist.github.com/KasperSkytte/8d0ca4206a66be7ff6d76fc4ab8e66c6

phyloseq_to_ampvis2 <- function(physeq) {
  #check object for class
  if(!any(class(physeq) %in% "phyloseq"))
    stop("physeq object must be of class \"phyloseq\"", call. = FALSE)
  
  #ampvis2 requires taxonomy and abundance table, phyloseq checks for the latter
  if(is.null(physeq@tax_table))
    stop("No taxonomy found in the phyloseq object and is required for ampvis2", call. = FALSE)
  
  #OTUs must be in rows, not columns
  if(phyloseq::taxa_are_rows(physeq))
    abund <- as.data.frame(phyloseq::otu_table(physeq)@.Data)
  else
    abund <- as.data.frame(t(phyloseq::otu_table(physeq)@.Data))
  
  #tax_table is assumed to have OTUs in rows too
  tax <- phyloseq::tax_table(physeq)@.Data
  
  #merge by rownames (OTUs)
  otutable <- merge(
    abund,
    tax,
    by = 0,
    all.x = TRUE,
    all.y = FALSE,
    sort = FALSE
  )
  colnames(otutable)[1] <- "OTU"
  
  #extract sample_data (metadata)
  if(!is.null(physeq@sam_data)) {
    metadata <- data.frame(
      phyloseq::sample_data(physeq),
      row.names = phyloseq::sample_names(physeq), 
      stringsAsFactors = FALSE, 
      check.names = FALSE
    )
    
    #check if any columns match exactly with rownames
    #if none matched assume row names are sample identifiers
    samplesCol <- unlist(lapply(metadata, function(x) {
      identical(x, rownames(metadata))}))
    
    if(any(samplesCol)) {
      #error if a column matched and it's not the first
      if(!samplesCol[[1]])
        stop("Sample ID's must be in the first column in the sample metadata, please reorder", call. = FALSE)
    } else {
      #assume rownames are sample identifiers, merge at the end with name "SampleID"
      if(any(colnames(metadata) %in% "SampleID"))
        stop("A column in the sample metadata is already named \"SampleID\" but does not seem to contain sample ID's", call. = FALSE)
      metadata$SampleID <- rownames(metadata)
      
      #reorder columns so SampleID is the first
      metadata <- metadata[, c(which(colnames(metadata) %in% "SampleID"), 1:(ncol(metadata)-1L)), drop = FALSE]
    }
  } else
    metadata <- NULL
  
  #extract phylogenetic tree, assumed to be of class "phylo"
  if(!is.null(physeq@phy_tree)) {
    tree <- phyloseq::phy_tree(physeq)
  } else
    tree <- NULL
  
  #extract OTU DNA sequences, assumed to be of class "XStringSet"
  if(!is.null(physeq@refseq)) {
    #convert XStringSet to DNAbin using a temporary file (easiest)
    fastaTempFile <- tempfile(pattern = "ampvis2_", fileext = ".fa")
    Biostrings::writeXStringSet(physeq@refseq, filepath = fastaTempFile)
  } else
    fastaTempFile <- NULL
  
  #load as normally with amp_load
  ampvis2::amp_load(
    otutable = otutable,
    metadata = metadata,
    tree = tree,
    fasta = fastaTempFile
  )
}

#variance stabilisation from DESeq2

ps_vst <- function(ps, factor){
  

  
  diagdds = phyloseq_to_deseq2(ps, as.formula(paste( "~", factor)))              
  diagdds = estimateSizeFactors(diagdds, type="poscounts")
  diagdds = estimateDispersions(diagdds, fitType = "local") 
  pst <- varianceStabilizingTransformation(diagdds)
  pst.dimmed <- t(as.matrix(assay(pst))) 
  # pst.dimmed[pst.dimmed < 0.0] <- 0.0
  ps.varstab <- ps
  otu_table(ps.varstab) <- otu_table(pst.dimmed, taxa_are_rows = FALSE) 
  return(ps.varstab)
}

#WGCNA visualisation
#result - list class object with attributes:
# ps - phyloseq object
# amp - ampwis2 object
# heat - heatmap with absalute read numbers
# heat_rel - hetmap with relative abundances
# tree - phylogenetic tree with taxonomy

color_filt <- function(ps, df){
  
      library(tidyverse)
      library(reshape2)
      library(gridExtra)

  
  l = list()
  for (i in levels(df$module)){
    message(i)
    color_name <-  df %>% 
      filter(module == i) %>% 
      pull(asv) %>% 
      unique()
    ps.col <- prune_taxa(color_name, ps)
    amp.col <- phyloseq_to_ampvis2(ps.col)
    heat <- amp_heatmap(amp.col, tax_show = 60, 
              group_by = "Day", 
              tax_aggregate = "OTU",
              tax_add = "Genus", 
              normalise=FALSE, 
              showRemainingTaxa = TRUE)
    
    ps.rel  <-  phyloseq::transform_sample_counts(ps.col, function(x) x / sum(x) * 100)
    amp.r <- phyloseq_to_ampvis2(ps.rel)
    heat.rel <- amp_heatmap(amp.r, tax_show = 60, 
          group_by = "Day", 
          tax_aggregate = "OTU",
          tax_add = "Genus", 
          normalise=FALSE, 
          showRemainingTaxa = TRUE)
    
    tree <- ps.col@phy_tree
    taxa <- as.data.frame(ps.col@tax_table@.Data)
    p1 <- ggtree(tree) + 
              geom_tiplab(size=2, align=TRUE, linesize=.5) +
              theme_tree2()
    
    taxa[taxa == "Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium"] <- "Allorhizobium"
    taxa[taxa == "Burkholderia-Caballeronia-Paraburkholderia"] <- "Burkholderia"
  
    tx <- taxa %>% 
      rownames_to_column("id") %>% 
      mutate(id = factor(id, levels = rev(get_taxa_name(p1)))) %>% 
      dplyr::select(-c(Kingdom, Species, Order)) %>% 
      melt(id.var = 'id')

    p2 <- ggplot(tx, aes(variable, id)) + 
      geom_tile(aes(fill = value), alpha = 0.4) +
      geom_text(aes(label = value), size = 3) +
      theme_bw() + 
      theme(legend.position = "none",
             axis.ticks.x=element_blank(),
             axis.text.y=element_blank(),
             axis.ticks.y=element_blank(),
             axis.title.x = element_blank(),
            axis.title.y = element_blank()) 

    p <- ggpubr::ggarrange(p1, p2, widths = c(0.9, 1))
    l[[i]] <- list("ps" = ps.col, 
                   "amp" = amp.col,
                   "heat" = heat,
                   "heat_rel" = heat.rel,
                   "tree" = p,
                   "taxa" = knitr::kable(taxa))
                    
  }
  return(l)
}

detachAllPackages <- function() {

  basic.packages <- c("package:stats","package:graphics","package:grDevices","package:utils","package:datasets","package:methods","package:base")

  package.list <- search()[ifelse(unlist(gregexpr("package:",search()))==1,TRUE,FALSE)]

  package.list <- setdiff(package.list,basic.packages)

  if (length(package.list)>0)  for (package in package.list) detach(package, character.only=TRUE, force = TRUE)

}

plot_alpha_w_toc <- function(ps, group, metric) {
  
  require(phyloseq)
  require(ggplot2)
  
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  
  er <- estimate_richness(ps_a)
  df_er <- cbind(ps_a@sam_data, er)
  df_er <- df_er %>% select(c(group, metric))
  stat.test <- aov(as.formula(paste0(metric, "~", group)), data = df_er) %>%
    rstatix::tukey_hsd()
  y <-  seq(max(er[[metric]]), length=length(stat.test$p.adj), by=max(er[[metric]]/20))

  plot_richness(ps_a, x=group, measures=metric) + 
    geom_boxplot() +
    geom_point(size=1.2, alpha=0.3) +
    ggpubr::stat_pvalue_manual(
      stat.test, 
      label = "p.adj.signif", 
      y.position = y) +
    theme_light() + 
    scale_color_brewer(palette="Dark2") +
    theme(axis.text.x = element_text(angle = 90),
          axis.title.x=element_blank()) +
    labs(y=paste(metric, "index")) 
}

beta_custom_norm_NMDS_elli_w <- function(ps, seed = 7888, normtype="vst", Color="What", Group="Repeat"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  library(ggforce)
  
  
  ordination.b <- ordinate(ps, "NMDS", "bray")
  mds <- as.data.frame(ordination.b$points)
  p  <-  plot_ordination(ps,
                         ordination.b,
                         type="sample",
                         color = Color,
                         title="NMDS - Bray-Curtis",
                         # title=NULL,
                         axes = c(2,3) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    annotate("text",
    x=min(mds$MDS1) + abs(min(mds$MDS1))/7,
    y=max(mds$MDS2),
    label=paste0("Stress -- ", round(ordination.b$stress, 3))) +
    geom_mark_ellipse(aes_string(group = Group, label = Group),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                      con.colour='gray') +
    theme(legend.position = "none") +
    ggplot2::scale_fill_viridis_c(option = "H")
  
  return(p)
}


```

```{r, message=FALSE}

library(ampvis2)

amp.exl <- phyloseq_to_ampvis2(ps.exl)
amp_heatmap(amp.exl, tax_show = 40, 
            group_by = "Feature", 
            tax_aggregate = "OTU",
            tax_add = "Genus", 
            normalise=FALSE)

amp.inall <- phyloseq_to_ampvis2(ps.inall)
amp_heatmap(amp.inall, 
            tax_show = 40, 
            group_by = "Feature", 
            tax_aggregate = "OTU",
            tax_add = "Genus", 
            normalise=FALSE)



```

```{r}

phyloseq_to_ampvis2 <- function(physeq) {
  #check object for class
  if(!any(class(physeq) %in% "phyloseq"))
    stop("physeq object must be of class \"phyloseq\"", call. = FALSE)
  
  #ampvis2 requires taxonomy and abundance table, phyloseq checks for the latter
  if(is.null(physeq@tax_table))
    stop("No taxonomy found in the phyloseq object and is required for ampvis2", call. = FALSE)
  
  #OTUs must be in rows, not columns
  if(phyloseq::taxa_are_rows(physeq))
    abund <- as.data.frame(phyloseq::otu_table(physeq)@.Data)
  else
    abund <- as.data.frame(t(phyloseq::otu_table(physeq)@.Data))
  
  #tax_table is assumed to have OTUs in rows too
  tax <- phyloseq::tax_table(physeq)@.Data
  
  #merge by rownames (OTUs)
  otutable <- merge(
    abund,
    tax,
    by = 0,
    all.x = TRUE,
    all.y = FALSE,
    sort = FALSE
  )
  colnames(otutable)[1] <- "OTU"
  
  #extract sample_data (metadata)
  if(!is.null(physeq@sam_data)) {
    metadata <- data.frame(
      phyloseq::sample_data(physeq),
      row.names = phyloseq::sample_names(physeq), 
      stringsAsFactors = FALSE, 
      check.names = FALSE
    )
    
    #check if any columns match exactly with rownames
    #if none matched assume row names are sample identifiers
    samplesCol <- unlist(lapply(metadata, function(x) {
      identical(x, rownames(metadata))}))
    
    if(any(samplesCol)) {
      #error if a column matched and it's not the first
      if(!samplesCol[[1]])
        stop("Sample ID's must be in the first column in the sample metadata, please reorder", call. = FALSE)
    } else {
      #assume rownames are sample identifiers, merge at the end with name "SampleID"
      if(any(colnames(metadata) %in% "SampleID"))
        stop("A column in the sample metadata is already named \"SampleID\" but does not seem to contain sample ID's", call. = FALSE)
      metadata$SampleID <- rownames(metadata)
      
      #reorder columns so SampleID is the first
      metadata <- metadata[, c(which(colnames(metadata) %in% "SampleID"), 1:(ncol(metadata)-1L)), drop = FALSE]
    }
  } else
    metadata <- NULL
  
  #extract phylogenetic tree, assumed to be of class "phylo"
  if(!is.null(physeq@phy_tree)) {
    tree <- phyloseq::phy_tree(physeq)
  } else
    tree <- NULL
  
  #extract OTU DNA sequences, assumed to be of class "XStringSet"
  if(!is.null(physeq@refseq)) {
    #convert XStringSet to DNAbin using a temporary file (easiest)
    fastaTempFile <- tempfile(pattern = "ampvis2_", fileext = ".fa")
    Biostrings::writeXStringSet(physeq@refseq, filepath = fastaTempFile)
  } else
    fastaTempFile <- NULL
  
  #load as normally with amp_load
  ampvis2::amp_load(
    otutable = otutable,
    metadata = metadata,
    tree = tree,
    fasta = fastaTempFile
  )
}

```

```{r}

library("ANCOMBC")

ancom_da_glob <- ancombc(phyloseq = ps.inall,
                         formula = "Feature",
                         p_adj_method = "BH", 
                         prv_cut = 0, 
                         group = "Feature", 
                         struc_zero = TRUE, 
                         neg_lb = FALSE, 
                         tol = 1e-5,
                         max_iter = 100, 
                         conserve = TRUE, 
                         alpha = 0.05, 
                         global = TRUE
                         )

```


```{r}

ps.inall@sam_data$Repeat

```


```{r}

ps_vst <- function(ps, factor){
  
  require(DESeq2)
  require(phyloseq)
  
  diagdds = phyloseq_to_deseq2(ps, as.formula(paste( "~", factor)))              
  diagdds = estimateSizeFactors(diagdds, type="poscounts")
  diagdds = estimateDispersions(diagdds, fitType = "local") 
  pst <- varianceStabilizingTransformation(diagdds)
  pst.dimmed <- t(as.matrix(assay(pst))) 
  # pst.dimmed[pst.dimmed < 0.0] <- 0.0
  ps.varstab <- ps
  otu_table(ps.varstab) <- otu_table(pst.dimmed, taxa_are_rows = FALSE) 
  return(ps.varstab)
}

ps.varstab <- ps_vst(ps.inall, "Repeat")


```


```{r}

data <- ps.varstab@otu_table@.Data %>% 
  as.data.frame()

data
```


```{r}

powers <-  c(c(1:10), seq(from = 12, to=30, by=1))
powers
sft <-  pickSoftThreshold(data, powerVector = powers, verbose = 5, networkType = "unsigned")
sft 

```


```{r}

plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2], xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n", main = paste("Scale independence"))
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2], labels=powers,cex=0.9,col="red")
abline(h=0.9,col="salmon")

```

```{r}

detachAllPackages()
library(WGCNA)


net2 <-   blockwiseModules(data, power=4, TOMType="unsigned", networkType="unsigned")
mergedColors2 <-  labels2colors(net2$colors, colorSeq = c("salmon", "darkgreen", "cyan", "red", "black", "darkred", "palegreen", "gray", "blue", "yellow"))

plotDendroAndColors(
  net2$dendrograms[[1]],
  mergedColors2[net2$blockGenes[[1]]],
  "Module colors",
  dendroLabels = FALSE,
  hang = 0.03,
  addGuide = TRUE,
  guideHang = 0.05)


```


##### Импорт 2

```{r, warning=FALSE, message=FALSE}

library(phyloseq)
library(tidyverse)
library(ggpubr)
library(ampvis2)
library(heatmaply)
library(WGCNA)
library(phyloseq)
library(ggtree)
library(tidyverse)

```
```{r, fig.width=8, fig.height=6}
# define numbers of genes and samples
nOTUs <- ncol(data)
nSamples <- nrow(data)

# Recalculate MEs with color labels
MEs0 <- moduleEigengenes(data, mergedColors2)$eigengenes
MEs <- orderMEs(MEs0)

names(MEs) <- substring(names(MEs), 3)


moduleTraitCor <- cor(MEs, agra_data, use = "p")
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples)

# PLOT
sizeGrWindow(10,6)
textMatrix <- paste(signif(moduleTraitCor, 2), "\n(", signif(moduleTraitPvalue, 1), ")", sep = "")
dim(textMatrix) <- dim(moduleTraitCor)
par(mar = c(222, 8.5, 11, 3))

# Display the correlation values within a heatmap
labeledHeatmap(Matrix = moduleTraitCor, 
               xLabels = names(agra_data),
               yLabels = names(MEs), 
               ySymbols = names(MEs), 
               # yLabelsPosition = "right",
               cex.lab.y = 0.85,
               colorLabels = FALSE, 
               colors = blueWhiteRed(50),
               textMatrix = textMatrix, 
               setStdMargins = FALSE,
               cex.text = 0.8,
               zlim = c(-1,1),
               main = paste("Корреляция кластеров с параметрами почвы"))

```


```{r}

modules_of_interest = mergedColors2 %>% unique()

module_df <- data.frame(
  asv = names(net2$colors),
  colors = mergedColors2
)

# module_df[module_df == "yellow"] <- "salmon"

submod <-  module_df %>%
  subset(colors %in% modules_of_interest)

row.names(module_df) = module_df$asv

subexpr = as.data.frame(t(data))[submod$asv,]

submod_df <- data.frame(subexpr) %>%
  mutate(
    asv = row.names(.)
  ) %>%
  pivot_longer(-asv) %>%
  mutate(
    module = module_df[asv,]$colors
  )


submod_df <- submod_df %>% 
  mutate(name = gsub("\\_.*","",submod_df$name)) %>% 
  group_by(name, asv) %>% 
  summarise(value = mean(value), asv = asv, module = module) %>% 
  relocate(c(asv, name, value, module)) %>% 
  ungroup() %>% 
  mutate(module = as.factor(module))


p <- submod_df %>% 
  ggplot(., aes(x=name, y=value, group=asv)) +
  geom_line(aes(color = module),
            alpha = 0.2) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90),
    legend.position = "none"
  ) +
  facet_grid(rows = vars(module)) +
  labs(x = "treatment",
       y = "normalized expression")


p + scale_color_manual(values = levels(submod_df$module))


```

```{r, warning=FALSE}

color_filt <- function(ps, df){
  
      library(tidyverse)
      library(reshape2)
      library(gridExtra)

  
  l = list()
  for (i in levels(df$module)){
    message(i)
    color_name <-  df %>% 
      filter(module == i) %>% 
      pull(asv) %>% 
      unique()
    ps.col <- prune_taxa(color_name, ps)
    amp.col <- phyloseq_to_ampvis2(ps.col)
    heat <- amp_heatmap(amp.col, tax_show = 50, 
              group_by = "Repeat", 
              tax_aggregate = "OTU",
              tax_add = "Genus", 
              normalise=FALSE, 
              showRemainingTaxa = TRUE)
    
    ps.rel  <-  phyloseq::transform_sample_counts(ps.col, function(x) x / sum(x))
    amp.r <- phyloseq_to_ampvis2(ps.rel)
    heat.rel <- amp_heatmap(amp.r, tax_show = 50, 
          group_by = "Repeat", 
          tax_aggregate = "OTU",
          tax_add = "Genus", 
          normalise=FALSE, 
          showRemainingTaxa = TRUE)
    otus <- heat$data$Display %>% 
        unique() %>% 
        str_extract("Seq[1-9]+")
    ps.col <- prune_taxa(taxa = otus, ps.col)
    tree <- ps.col@phy_tree
    taxa <- as.data.frame(ps.col@tax_table@.Data)
    p1 <- ggtree(tree) +
          geom_tiplab(size=2, align=TRUE, linesize=.5) +
              theme_tree2()

    taxa[taxa == "Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium"] <- "Allorhizobium"
    taxa[taxa == "Burkholderia-Caballeronia-Paraburkholderia"] <- "Burkholderia"

    tx <- taxa %>%
      rownames_to_column("id") %>%
      mutate(id = factor(id, levels = rev(get_taxa_name(p1)))) %>%
      dplyr::select(-c(Kingdom, Species, Order)) %>%
      melt(id.var = 'id')

    p2 <- ggplot(tx, aes(variable, id)) +
      geom_tile(aes(fill = value), alpha = 0.4) +
      geom_text(aes(label = value), size = 2) +
      theme_bw() +
      theme(legend.position = "none",
             axis.ticks.x=element_blank(),
             axis.text.y=element_blank(),
             axis.ticks.y=element_blank(),
             axis.title.x = element_blank(),
            axis.title.y = element_blank())

    p <- ggpubr::ggarrange(p1, p2, widths = c(0.9, 1))
    l[[i]] <- list("ps" = ps.col, 
                   "amp" = amp.col,
                   "heat" = heat,
                   "heat_rel" = heat.rel,
                   "tree" = p)

  }
  return(l)
}


```


```{r, warning=FALSE}

l_filt <- color_filt(ps.inall, submod_df)

l_filt$gray$ps


l_filt$gray$tree

```

```{r, warning=FALSE}

l_filt$gray$tree
l_filt$gray$ps

amp_heatmap(l_filt$gray$amp, tax_show = 40, 
      group_by = "Repeat", 
      tax_aggregate = "Genus",
      tax_add = "Phylum", 
      normalise=TRUE, 
      showRemainingTaxa = TRUE)

```


```{r}

ps.ym.class <- tax_glom(ps.ym, taxrank = "Class") 




ancom_da_glob <- ancombc(phyloseq = ps.ym.class,
                         formula = "Фон",
                         p_adj_method = "fdr",
                         zero_cut = 0.90,
                         lib_cut = 1000,
                         group = "Фон",
                         struc_zero = TRUE,
                         neg_lb = TRUE,
                         tol = 1e-5,
                         max_iter = 100,
                         conserve = TRUE,
                         alpha = 0.05,
                         global = TRUE)

rt <- ancom_da_glob$res$diff_abn %>% 
  filter(ФонTRUE == TRUE) %>% 
  rownames()
rt

ancom_da_glob$res$diff_abn 
ancom_da_glob$res$diff_abn

ancom_da_glob$res$diff_abn %>% 
  rownames_to_column("ID") %>% 
  left_join(taxas)


ps.fon <- prune_taxa(rt, ps.ym.class)
taxas <- as.data.frame(ps.ym.class@tax_table@.Data) %>% 
  rownames_to_column("ID")

?prune_taxa
```

```{r}

ancom_da_glob$res$beta %>% 
  rownames_to_column("ID") %>% 
  left_join(taxas) %>% 
  filter(ID %in% rt) %>% 
  arrange(desc(ФонTRUE))

```


```{r}

as.data.frame(t(ps.ym@otu_table)) %>% head()

```


```{r, fig.width=11, fig.height=9}

ps.ym.clr <- ps.ym
clr.otu <- as.data.frame(t(ps.ym@otu_table)) %>% 
  compositions::clr() %>% 
  t()
ps.ym.clr <- merge_phyloseq(ps.ym.clr, otu_table(clr.otu, taxa_are_rows = FALSE))


beta_custom_norm_pca_elli_w <- function(ps, seed = 7888, normtype="vst", Color="What", Group="Repeat"){
  require(phyloseq)
  require(ggplot2)
  require(ggpubr)
  library(ggforce)

  # попробуй заменить NMDS на PCoA
  # https://joey711.github.io/phyloseq/plot_ordination-examples.html и вот здесь куча вариантов ординаций в phyloseq - попробуй нарисовать отдельно для таксонов, а не образцов(не моей функцией, у меня зависнет, и лучше PCoA - так быстрее))
  ordination.b <- ordinate(ps, "MDS", "canberra")
  mds <- as.data.frame(ordination.b$points)
  p  <-  plot_ordination(ps,
                         ordination.b,
                         type="sample",
                         color = Color,
                         # title="NMDS - Bray-Curtis",
                         title=NULL,
                         axes = c(1,2) ) + 
    theme_bw() + 
    theme(text = element_text(size = 10)) + 
    geom_point(size = 3) +
    geom_mark_ellipse(aes_string(group = Group, label = Group),
                      label.fontsize = 10,
                      label.buffer = unit(2, "mm"),
                      label.minwidth = unit(5, "mm"),
                      con.cap = unit(0.1, "mm"),
                      con.colour='gray') +
    theme(legend.position = "none") 
  
  return(list(p, ordination.b))
}

beta_custom_norm_pca_elli_w(ps.ym.clr, Color="Site", Group="Repeat")[[1]]

pca <- beta_custom_norm_pca_elli_w(ps.ym.clr, Color="Site", Group="Repeat")[[2]]
pca

plot_scree(pca, title = NULL)

clr.otu %>% 
  t() %>% 
  as.data.frame() %>% 
  head()

```



```{r}

ps.ym.class@sam_data


ancom_da_glob_2 <- ancombc(phyloseq = ps.ym.class,
                         formula = "Фон + Залежь",
                         p_adj_method = "fdr",
                         zero_cut = 0.90,
                         lib_cut = 1000,
                         group = "Фон",
                         struc_zero = TRUE,
                         neg_lb = TRUE,
                         tol = 1e-5,
                         max_iter = 100,
                         conserve = TRUE,
                         alpha = 0.05,
                         global = TRUE)


rt <- ancom_da_glob_2$res$diff_abn %>% 
  filter(ФонTRUE == TRUE) %>% 
  rownames()

ancom_da_glob_2$res$beta %>% 
  rownames_to_column("ID") %>% 
  left_join(taxa) %>% 
  filter(ID %in% rt) %>% 
  arrange(desc(ФонTRUE))

ancom_da_glob_2

```


```{r}



```



```{r}

library(vegan)
physeq <- ps.ym
dist <- phyloseq::distance(physeq, "bray")
metadata <- as(sample_data(physeq@sam_data), "data.frame")
ad <- adonis2(dist ~ Фон, data = metadata)

ad

```

```{r}

library(vegan)
physeq <- ps.ym
dist <- phyloseq::distance(physeq, "bray")
metadata <- as(sample_data(physeq@sam_data), "data.frame")
ad <- adonis2(dist ~ Залежь, data = metadata)

ad


```




```{r}

library(vegan)
physeq <- ps.ym
dist <- phyloseq::distance(physeq, "bray")
metadata <- as(sample_data(physeq@sam_data), "data.frame")
ad <- adonis2(dist ~ Site, data = metadata)

ad


```

```{r}


ps.ym.ph <- prune_samples(sample_data(ps.ym)$pH > 0, ps.ym)
ps.ym.ph  <- prune_taxa(taxa_sums(ps.ym.ph) > 0, ps.ym.ph)

ps.ym.ag <- prune_samples(sample_data(ps.ym)$Фон != TRUE, ps.ym)
ps.ym.ag  <- prune_taxa(taxa_sums(ps.ym.ph) > 0, ps.ym.ph)
ps.ym.ag.f <- phyloseq::filter_taxa(ps.ym.ag, function(x) sum(x > 10) > 0, TRUE)


```

```{r}

ancom_da_glob_ag <- ancombc(phyloseq = ps.ym.ag.f,
                         formula = "Залежь",
                         p_adj_method = "fdr",
                         zero_cut = 0.90,
                         lib_cut = 1000,
                         group = "Залежь",
                         struc_zero = TRUE,
                         neg_lb = TRUE,
                         tol = 1e-5,
                         max_iter = 100,
                         conserve = TRUE,
                         alpha = 0.05,
                         global = TRUE)


ancom_da_glob_ag


```

```{r}

rt <- ancom_da_glob_ag$res$diff_abn %>% 
  filter(ЗалежьTRUE == TRUE) %>% 
  rownames()

ancom_da_glob_ag$res$beta %>% 
  rownames_to_column("ID") %>% 
  left_join(taxa) %>% 
  filter(ID %in% rt) %>% 
  arrange(desc(ЗалежьTRUE)) %>% 
  filter(Phylum %in% c("Proteobacteria", "Acidobacteriota", "Actinobacteriota","Chloroflexi", "Planctomycetota", "Bacteroidota", "Verrucomicrobiota", "Firmicutes")) %>% 
  mutate(Rank = ifelse(Phylum == "Proteobacteria", Class, Phylum)) %>%
  mutate(binary = ifelse(ЗалежьTRUE > 0, "залежь", "действующий агрозём")) %>% 
  dplyr::group_by(тип, Rank) %>% 
  dplyr::count() %>% 
  mutate(N = ifelse(тип == "залежь", n, 0 - n)) %>% 
  ggplot(aes(x = Rank, y = N, fill = тип)) +
  geom_bar(stat="identity") +
  theme_minimal() + 
  coord_flip() +
  theme(legend.position = "top")


```


```{r}

library(vegan)
physeq <- ps.ym.ph
dist <- phyloseq::distance(physeq, "bray")
metadata <- as(sample_data(physeq@sam_data), "data.frame")
ad <- adonis2(dist ~ pH, data = metadata)

ad


```




```{r}

ancom_da_glob_all <- ancombc(phyloseq = ps.ym.f,
                         formula = "Фон",
                         p_adj_method = "fdr",
                         zero_cut = 0.90,
                         lib_cut = 1000,
                         group = "Фон",
                         struc_zero = TRUE,
                         neg_lb = TRUE,
                         tol = 1e-5,
                         max_iter = 100,
                         conserve = TRUE,
                         alpha = 0.05,
                         global = TRUE)


ancom_da_glob_all

```


```{r}

ps.ym.f <-  phyloseq::filter_taxa(ps.ym, function(x) sum(x > 3) > (0.1*length(x)), TRUE)
ps.ym
ps.ym.f
p

p <- phyloseq::filter_taxa(ps.ym, function(x) !sum(x > 3) > (0.1*length(x)), TRUE)
ps.ym.f <- phyloseq::filter_taxa(ps.ym, function(x) sum(x > 10) > 0, TRUE)
p <- phyloseq::filter_taxa(p, function(x) sum(x) > 100, TRUE) 

p@otu_table %>% 
  t() %>% 
  as.data.frame() 

ps.ym@tax_table@.Data %>% 
  as.data.frame() %>% 
  rownames_to_column("id") %>% 
  filter(id %in% list_strange)

ps.ym.ph.f

```

```{r}


ps.ym.ph.f <-  phyloseq::filter_taxa(ps.ym.ph, function(x) sum(x > 10) > 0, TRUE)
ps.ym.ph.merged <- phyloseq::merge_samples(ps.ym.ph.f, "Repeat")

ps.ym.ph.merged.clr <- ps.ym.ph.merged
clr.2.otu <- as.data.frame(t(ps.ym.ph.merged.clr@otu_table)) %>% 
  compositions::clr() %>% 
  t()
ps.ym.ph.merged.clr <- merge_phyloseq(ps.ym.ph.merged.clr, otu_table(clr.2.otu, taxa_are_rows = FALSE))

otus_cor <- ps.ym.ph.merged.clr@otu_table %>% 
  as.data.frame()

s1 <- sapply(colnames(otus_cor), function(x) {
  cor <- cor.test(otus_cor[[x]], meta_ph, method="spearman", exact = FALSE)
  c1 <- cor$p.value
  c2 <- cor$estimate %>% unname()
  return(list("p.value"=c1, "rho"=c2))})

s1 <- s1 %>% 
  t() %>% 
  as.data.frame() %>% 
  unnest(cols = c(p.value, rho), .id = "id") %>% 
  column_to_rownames("id") %>% 
  mutate(p.adj = p.adjust(p.value, method = "BH", n = length(p.value)))


s1
taxas
taxa <- as.data.frame(ps.ym@tax_table@.Data) %>% 
  rownames_to_column("ID")

s1 %>%
  rownames_to_column("ID") %>% 
  left_join(taxa) %>% 
  arrange(rho) %>%
  filter(Phylum %in% c("Proteobacteria", "Acidobacteriota", "Actinobacteriota","Chloroflexi", "Planctomycetota", "Bacteroidota", "Verrucomicrobiota", "Firmicutes")) %>% 
  mutate(Rank = ifelse(Phylum == "Proteobacteria", Class, Phylum)) %>%
  ggplot(aes(x=rho, color=Rank, fill=Rank)) +
  geom_histogram(alpha=0.3, bins = 30) +
  theme_minimal() +
  facet_wrap(~ Rank) +
  theme(legend.position="none")

```



```{r, fig.width=12, fig.height=8}

library(ggpubr)

alpha.custom <- function(ps.f, arrga = "PD"){
  

  
  pd1 <- estimate_richness(ps.f, measures=c("Observed", "InvSimpson", "Shannon"))
  pd <- cbind(pd1, ps.f@sam_data)
  
  bmi <- levels(as.factor(pd$Залежь))
  bmi.pairs <- combn(seq_along(bmi), 2, simplify = FALSE, FUN = function(i)bmi[i])
  p1 <- ggviolin(pd, x = "Залежь", y = arrga,
                 add = "boxplot", fill = "Залежь") + 
    stat_compare_means(comparisons = bmi.pairs, method = "wilcox.test", size=3) +
    scale_x_discrete(limits = c("True", "False")) +
    theme(axis.title.x = element_blank(),
          legend.title = element_blank(),
          legend.position = "none",
          axis.text.x = element_text(angle = 45, hjust = 1),
          text = element_text(size=9))
  return(p1)
}

alpha.custom2 <- function(ps.f, arrga = "PD"){
  
  require(picante)
  
  pd <- pd(ps.f@otu_table@.Data, ps.f@phy_tree)
  
  pd1 <- estimate_richness(ps.f, measures=c("Observed", "InvSimpson", "Shannon"))
  pd <- cbind(pd, ps.f@sam_data, pd1 )
  
  bmi <- levels(as.factor(pd$Quarry_or_Benchmark))
  bmi.pairs <- combn(seq_along(bmi), 2, simplify = FALSE, FUN = function(i)bmi[i])
  p1 <- ggviolin(pd, x = "Quarry_or_Benchmark", y = arrga,
                 add = "boxplot", fill = "Quarry_or_Benchmark") +
    stat_compare_means(comparisons = bmi.pairs, method = "wilcox.test", size=3) +
    scale_x_discrete(limits = c("Quarry", "Benchmark")) +
    theme(axis.title.x = element_blank(),
          legend.title = element_blank(),
          legend.position = "none",
          axis.text.x = element_text(angle = 45, hjust = 1),
          text = element_text(size=9))
  return(p1)
}

p.alpha.Cdt.oo <- alpha.custom(ps.cc, arrga = "Observed")
p.alpha.Cdt.sh <- alpha.custom(ps.cc, arrga = "PD")
p.alpha.Cdt.is <- alpha.custom(ps.cc, arrga = "Shannon")
p.alpha.Cdt.pd <- alpha.custom(ps.cc, arrga = "InvSimpson")

p.alpha.Cdt.oo2 <- alpha.custom2(ps.nc, arrga = "Observed")
p.alpha.Cdt.sh2 <- alpha.custom2(ps.nc, arrga = "PD")
p.alpha.Cdt.is2 <- alpha.custom2(ps.nc, arrga = "Shannon")
p.alpha.Cdt.pd2 <- alpha.custom2(ps.nc, arrga = "InvSimpson")

p1 <- ggarrange(p.alpha.Cdt.oo, p.alpha.Cdt.sh,p.alpha.Cdt.is, p.alpha.Cdt.pd , 
                p.alpha.Cdt.oo2, p.alpha.Cdt.sh2, p.alpha.Cdt.is2, p.alpha.Cdt.pd2,
                ncol = 4 ,label.x = 0.3, nrow = 2, font.label = c(size = 10), labels = c("Progress", "", "", "", "Urvan")) 
p1

plot_alpha <- function(ps, metric, group) {
  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  plot_richness(ps_a, x=group, measures=metric) + 
    geom_boxplot(, fill="#4DBBD5B2", alpha=0.4) +
    geom_point(size=1.2, alpha=0.3) + 
    theme_light() + scale_color_brewer(palette="Dark2") +
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
          axis.title.x=element_blank(),
          text = element_text(size=20)) +
    labs(y=paste("alpha-diversity index")) 
}

plot_alpha(ps.ya, c("Observed","Shannon","InvSimpson"), "Repeat") 
```






```{r, fig.height= 11, fig.width=6}

plot_alpha <- function(ps, group, metric) {
  require(phyloseq)
  require(ggplot2)
  

  ps_a <- prune_taxa(taxa_sums(ps) > 0, ps)
  plot_richness(ps_a, x=group, measures=metric) + 
    geom_boxplot() +
    geom_point(size=1.2, alpha=0.3) + 
    theme_light() + scale_color_brewer(palette="Dark2") +
    theme(axis.text.x = element_text(angle = 90),
          axis.title.x=element_blank()) +
    labs(y=paste(metric, "index")) 
}


p1 <- plot_alpha(ps.ya, 'Фон' ,'Observed')   
p2 <- plot_alpha(ps.ya, 'Залежь' ,'Observed') 

p3 <- plot_alpha(ps.ya, 'Фон' ,'Shannon')   
p4 <- plot_alpha(ps.ya, 'Залежь' ,'Shannon') 

p5 <- plot_alpha(ps.ya, 'Фон' ,'InvSimpson')   
p6 <- plot_alpha(ps.ya, 'Залежь' ,'InvSimpson') 

ggarrange(p1, p2, p3, p4, p5, p6, labels = c("Фон", "Залежь", "", "", "", "", ""), ncol = 2, nrow = 3)
                  
```




```{r}

ps.f <- readRDS("ps.f")
mp$AEV_site

```

```{r}

ps.ya <- prune_samples(sample_data(ps.f)$Region %in% c("Yakutsk"), ps.f)
ps.ya  <- prune_taxa(taxa_sums(ps.ya) > 0, ps.ya)

```

```{r}
ids <- ps.f@sam_data %>% 
  data.frame() %>%
  rownames_to_column("ID") %>% 
  filter(Region == "Yakutsk") %>%
  pull(ID)

agra <- data_frame(
  AEV_site = c(rep("Y1", 3), rep("Y2", 3), rep("Y3", 3), rep("Y7", 3), rep("Y9", 3), rep("Y10", 3), rep("Y11", 3)),
  pH = c(rep(5.6, 3), rep(5.16, 3), rep(5.9, 3), rep(6.95, 3), rep(5.23, 3), rep(5.47, 3), rep(5.72, 3)),
  P = c(rep(40, 3), rep(29, 3), rep(346, 3), rep(248, 3), rep(34, 3), rep(120, 3), rep(125, 3)),
  K = c(rep(39, 3), rep(174, 3), rep(43, 3), rep(30, 3), rep(38, 3), rep(114, 3), rep(41, 3)),
  NH3 = c(rep(8.07, 3), rep(18.3, 3), rep(6.95, 3), rep(20.6, 3), rep(10.8, 3), rep(34.5, 3), rep(23.0, 3)),
  NO3 = c(rep(0.8, 3), rep(2.74, 3), rep(2.73, 3), rep(1.6, 3), rep(7.54, 3), rep(1.46, 3), rep(0.94, 3)),)

ps.ya@sam_data <- ps.f@sam_data %>% 
  data.frame() %>%
  filter(Region == "Yakutsk") %>% 
  mutate(AEV_site = as.factor(AEV_site)) %>% 
  select_if(~ !any(is.na(.))) %>% 
  cbind(agra) %>% 
  sample_data()


tree <- read_tree(treefile="tree")
ps.ya@phy_tree <- tree
ps.ya

agra_data <- agra %>% select(-AEV_site)
agra_data

```




```{r}


beta_custom_norm_NMDS_elli_w(ps.ya, Color = "AEV_site", Group = "AEV_site")


```







